{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib\n",
    "import dataframe_image as dfi\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "#import angeli file\n",
    "angeli = pd.read_csv('/Users/bencekover/Library/CloudStorage/OneDrive-UniversityCollegeLondon/MSci Bahler lab/S.-Pombe-biofilm/Bence folder/Angeli/angeli/bioneer_angeli_results.txt', sep = '\\t', header = 0, index_col = None)\n",
    "\n",
    "#leave top 10 gene expression\n",
    "\n",
    "angeli_fypo= angeli[angeli['Category_Name'] == 'Phenotypes (FYPO)'].iloc[0:5]\n",
    "angeli = angeli[angeli['Category_Name'] != 'Phenotypes (FYPO)']\n",
    "angeli = pd.concat([angeli_fypo, angeli])\n",
    "\n",
    "\n",
    "angeli_gene = angeli[angeli['Category_Name'] == 'Gene Expression'].iloc[0:10]\n",
    "angeli = angeli[angeli['Category_Name'] != 'Gene Expression']\n",
    "angeli = pd.concat([angeli_gene, angeli])\n",
    "\n",
    "\n",
    "angeli_mf = angeli[angeli['Category_Name'] == 'GO Molecular Function'].iloc[0:3]\n",
    "angeli = angeli[angeli['Category_Name'] != 'GO Molecular Function']\n",
    "angeli = pd.concat([angeli_mf, angeli])\n",
    "angeli\n",
    "\n",
    "angeli_cc = angeli[angeli['Category_Name'] == 'GO Cellular Component'].iloc[0:3]\n",
    "angeli = angeli[angeli['Category_Name'] != 'GO Cellular Component']\n",
    "angeli = pd.concat([angeli_cc, angeli])\n",
    "\n",
    "angeli_bp = angeli[angeli['Category_Name'] == 'GO Biological Process'].iloc[0:3]\n",
    "#remove all biological process and add back only the first 3 rows to the top\n",
    "angeli = angeli[angeli['Category_Name'] != 'GO Biological Process']\n",
    "#add back to top of angeli\n",
    "angeli = pd.concat([angeli_bp, angeli])\n",
    "\n",
    "#reset index\n",
    "angeli = angeli.reset_index(drop = True)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Category_Name</th>\n",
       "      <th>External_ID</th>\n",
       "      <th>GeneSet_Name</th>\n",
       "      <th>over_represented/under_represented</th>\n",
       "      <th>List_Frequency</th>\n",
       "      <th>Background_Frequency</th>\n",
       "      <th>Corrected_pvalue</th>\n",
       "      <th>Genes_In_Common/Interacting_Gene_Pair</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GO Biological Process</td>\n",
       "      <td>GO:0009058;</td>\n",
       "      <td>biosynthetic process</td>\n",
       "      <td>Enriched</td>\n",
       "      <td>52.317880794702 (79/151)</td>\n",
       "      <td>27.456839309429 (827/3012)</td>\n",
       "      <td>1.933990e-07</td>\n",
       "      <td>SPAC24H6.07;SPAC22F3.09c;SPAC13D6.02c;SPAC1783...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GO Biological Process</td>\n",
       "      <td>GO:0044249;</td>\n",
       "      <td>cellular biosynthetic process</td>\n",
       "      <td>Enriched</td>\n",
       "      <td>51.6556291390728 (78/151)</td>\n",
       "      <td>26.8260292164675 (808/3012)</td>\n",
       "      <td>1.933990e-07</td>\n",
       "      <td>SPAC24H6.07;SPAC22F3.09c;SPAC13D6.02c;SPAC1783...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GO Biological Process</td>\n",
       "      <td>GO:1901576;</td>\n",
       "      <td>organic substance biosynthetic process</td>\n",
       "      <td>Enriched</td>\n",
       "      <td>51.6556291390728 (78/151)</td>\n",
       "      <td>27.0252324037185 (814/3012)</td>\n",
       "      <td>1.933990e-07</td>\n",
       "      <td>SPAC24H6.07;SPAC22F3.09c;SPAC13D6.02c;SPAC1783...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GO Cellular Component</td>\n",
       "      <td>GO:0022626;</td>\n",
       "      <td>cytosolic ribosome</td>\n",
       "      <td>Enriched</td>\n",
       "      <td>12.5827814569536 (19/151)</td>\n",
       "      <td>2.75564409030544 (83/3012)</td>\n",
       "      <td>2.302700e-05</td>\n",
       "      <td>SPAC24H6.07;SPAC959.08;SPCC74.05;SPCC364.03;SP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GO Cellular Component</td>\n",
       "      <td>GO:0044445;</td>\n",
       "      <td>cytosolic part</td>\n",
       "      <td>Enriched</td>\n",
       "      <td>13.2450331125828 (20/151)</td>\n",
       "      <td>3.15405046480744 (95/3012)</td>\n",
       "      <td>2.851560e-05</td>\n",
       "      <td>SPAC24H6.07;SPAC959.08;SPCC74.05;SPCC364.03;SP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GO Cellular Component</td>\n",
       "      <td>GO:0005840;</td>\n",
       "      <td>ribosome</td>\n",
       "      <td>Enriched</td>\n",
       "      <td>12.5827814569536 (19/151)</td>\n",
       "      <td>3.22045152722444 (97/3012)</td>\n",
       "      <td>1.322570e-04</td>\n",
       "      <td>SPAC24H6.07;SPAC959.08;SPCC74.05;SPCC364.03;SP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GO Molecular Function</td>\n",
       "      <td>GO:0003735;</td>\n",
       "      <td>structural constituent of ribosome</td>\n",
       "      <td>Enriched</td>\n",
       "      <td>11.9205298013245 (18/151)</td>\n",
       "      <td>2.75564409030544 (83/3012)</td>\n",
       "      <td>5.963010e-05</td>\n",
       "      <td>SPAC24H6.07;SPAC959.08;SPCC74.05;SPCC364.03;SP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>GO Molecular Function</td>\n",
       "      <td>GO:0005198;</td>\n",
       "      <td>structural molecule activity</td>\n",
       "      <td>Enriched</td>\n",
       "      <td>11.9205298013245 (18/151)</td>\n",
       "      <td>3.51925630810093 (106/3012)</td>\n",
       "      <td>1.100610e-03</td>\n",
       "      <td>SPAC24H6.07;SPAC959.08;SPCC74.05;SPCC364.03;SP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>GO Molecular Function</td>\n",
       "      <td>GO:0019843;</td>\n",
       "      <td>rRNA binding</td>\n",
       "      <td>Enriched</td>\n",
       "      <td>4.63576158940397 (7/151)</td>\n",
       "      <td>0.597609561752988 (18/3012)</td>\n",
       "      <td>3.465800e-03</td>\n",
       "      <td>SPAC24H6.07;SPBC16D10.11c;SPAC22F8.09;SPAC3G9....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Ribosomal protein module Homol D E</td>\n",
       "      <td>Enriched</td>\n",
       "      <td>14.5695364238411 (22/151)</td>\n",
       "      <td>3.68525896414343 (111/3012)</td>\n",
       "      <td>2.302700e-05</td>\n",
       "      <td>SPAC24H6.07;SPAC959.08;SPCC74.05;SPAC23H4.10c;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Ribosomal protein module 2</td>\n",
       "      <td>Enriched</td>\n",
       "      <td>15.8940397350993 (24/151)</td>\n",
       "      <td>4.44887118193891 (134/3012)</td>\n",
       "      <td>2.851560e-05</td>\n",
       "      <td>SPAC24H6.07;SPBPB10D8.04c;SPBC11B10.02c;SPAC95...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Highly expressed genes</td>\n",
       "      <td>Enriched</td>\n",
       "      <td>19.205298013245 (29/151)</td>\n",
       "      <td>6.90571049136786 (208/3012)</td>\n",
       "      <td>1.514130e-04</td>\n",
       "      <td>SPAC24H6.07;SPAC13D6.02c;SPBPB10D8.04c;SPAC959...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Oxidative Stess Cluster 11</td>\n",
       "      <td>Enriched</td>\n",
       "      <td>8.60927152317881 (13/151)</td>\n",
       "      <td>1.85922974767596 (56/3012)</td>\n",
       "      <td>8.621080e-04</td>\n",
       "      <td>SPAC24H6.07;SPAC959.08;SPCC74.05;SPCC364.03;SP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Oxidative Stess Cluster 9</td>\n",
       "      <td>Enriched</td>\n",
       "      <td>9.93377483443709 (15/151)</td>\n",
       "      <td>2.52324037184595 (76/3012)</td>\n",
       "      <td>1.100610e-03</td>\n",
       "      <td>SPAC24H6.07;SPAC959.08;SPCC74.05;SPCC364.03;SP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Caffeine and Rapamycin repressed</td>\n",
       "      <td>Enriched</td>\n",
       "      <td>13.2450331125828 (20/151)</td>\n",
       "      <td>4.58167330677291 (138/3012)</td>\n",
       "      <td>2.881290e-03</td>\n",
       "      <td>SPAC24H6.07;SPAC13D6.02c;SPAC959.08;SPCC74.05;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Core Environmental Stress Response repressed</td>\n",
       "      <td>Enriched</td>\n",
       "      <td>15.8940397350993 (24/151)</td>\n",
       "      <td>6.17529880478088 (186/3012)</td>\n",
       "      <td>2.881290e-03</td>\n",
       "      <td>SPAC24H6.07;SPAC13D6.02c;SPAC959.08;SPCC74.05;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Oxidative Stess Cluster 6</td>\n",
       "      <td>Enriched</td>\n",
       "      <td>9.93377483443709 (15/151)</td>\n",
       "      <td>2.78884462151394 (84/3012)</td>\n",
       "      <td>2.988380e-03</td>\n",
       "      <td>SPAC24H6.07;SPAC959.08;SPCC74.05;SPCC364.03;SP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Phenotypes (FYPO)</td>\n",
       "      <td>FYPO:0000046;</td>\n",
       "      <td>decreased cell population growth</td>\n",
       "      <td>Enriched</td>\n",
       "      <td>41.0596026490066 (62/151)</td>\n",
       "      <td>21.4807436918991 (647/3012)</td>\n",
       "      <td>2.851560e-05</td>\n",
       "      <td>SPAC22F3.09c;SPBC543.09;SPBC1215.01;SPAC1783.0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Phenotypes (FYPO)</td>\n",
       "      <td>FYPO:0001234;</td>\n",
       "      <td>slow vegetative cell population growth</td>\n",
       "      <td>Enriched</td>\n",
       "      <td>23.841059602649 (36/151)</td>\n",
       "      <td>9.42895086321381 (284/3012)</td>\n",
       "      <td>5.892290e-05</td>\n",
       "      <td>SPAC13D6.02c;SPAC1783.07c;SPCC188.07;SPCP1E11....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Phenotypes (FYPO)</td>\n",
       "      <td>FYPO:0001355;</td>\n",
       "      <td>decreased vegetative cell population growth</td>\n",
       "      <td>Enriched</td>\n",
       "      <td>40.3973509933775 (61/151)</td>\n",
       "      <td>21.2483399734396 (640/3012)</td>\n",
       "      <td>5.892290e-05</td>\n",
       "      <td>SPAC22F3.09c;SPBC543.09;SPBC1215.01;SPAC1783.0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Phenotypes (FYPO)</td>\n",
       "      <td>FYPO:0000684;</td>\n",
       "      <td>decreased cell population growth on glycerol c...</td>\n",
       "      <td>Enriched</td>\n",
       "      <td>23.841059602649 (36/151)</td>\n",
       "      <td>10.0265604249668 (302/3012)</td>\n",
       "      <td>2.219750e-04</td>\n",
       "      <td>SPBC543.09;SPBC1215.01;SPAC1783.07c;SPAC3A12.1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Phenotypes (FYPO)</td>\n",
       "      <td>FYPO:0002177;</td>\n",
       "      <td>viable vegetative cell with normal cell morpho...</td>\n",
       "      <td>Underrepresented</td>\n",
       "      <td>68.8741721854305 (104/151)</td>\n",
       "      <td>84.6945551128818 (2551/3012)</td>\n",
       "      <td>3.117270e-04</td>\n",
       "      <td>SPAC24H6.07;SPAPB18E9.01;SPAC22F3.09c;SPAC13D6...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Phenotypic Profiles</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Slowly proliferating mutants</td>\n",
       "      <td>Enriched</td>\n",
       "      <td>17.8807947019868 (27/151)</td>\n",
       "      <td>6.20849933598938 (187/3012)</td>\n",
       "      <td>1.837160e-04</td>\n",
       "      <td>SPAC13D6.02c;SPCC188.07;SPAC23H4.10c;SPCP1E11....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Protein Features</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Arginine</td>\n",
       "      <td>Higher</td>\n",
       "      <td>0.0578795262492446</td>\n",
       "      <td>0.0487964770423139</td>\n",
       "      <td>1.463610e-03</td>\n",
       "      <td>SPAC24H6.07;SPAC22A12.04c;SPAC3A12.12;SPAC23H4...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Protein Features</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Protein copies per proliferating cell</td>\n",
       "      <td>Higher</td>\n",
       "      <td>31596.0256701031</td>\n",
       "      <td>13684.6208433048</td>\n",
       "      <td>1.472360e-03</td>\n",
       "      <td>SPAC24H6.07;SPAC3A12.12;SPAC23H4.10c;SPAC343.1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Protein Features</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Isoelectric point (predicted pH)</td>\n",
       "      <td>Higher</td>\n",
       "      <td>8.10106357615894</td>\n",
       "      <td>7.31765260398461</td>\n",
       "      <td>2.920790e-03</td>\n",
       "      <td>SPAC24H6.07;SPAC22A12.04c;SPAC3A12.12;SPAC23H4...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Transcript Features</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mRNA level (WT)</td>\n",
       "      <td>Higher</td>\n",
       "      <td>4020.90520833333</td>\n",
       "      <td>2387.36666641874</td>\n",
       "      <td>2.211010e-04</td>\n",
       "      <td>SPAC24H6.07;SPAC22A12.04c;SPAC3A12.12;SPAC23H4...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Transcript Features</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mRNA copies per proliferating cell</td>\n",
       "      <td>Higher</td>\n",
       "      <td>18.5992516556291</td>\n",
       "      <td>5.82367806520199</td>\n",
       "      <td>4.121010e-04</td>\n",
       "      <td>SPAC24H6.07;SPAC22A12.04c;SPAC3A12.12;SPAC23H4...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            Category_Name    External_ID  \\\n",
       "0   GO Biological Process    GO:0009058;   \n",
       "1   GO Biological Process    GO:0044249;   \n",
       "2   GO Biological Process    GO:1901576;   \n",
       "3   GO Cellular Component    GO:0022626;   \n",
       "4   GO Cellular Component    GO:0044445;   \n",
       "5   GO Cellular Component    GO:0005840;   \n",
       "6   GO Molecular Function    GO:0003735;   \n",
       "7   GO Molecular Function    GO:0005198;   \n",
       "8   GO Molecular Function    GO:0019843;   \n",
       "9         Gene Expression            NaN   \n",
       "10        Gene Expression            NaN   \n",
       "11        Gene Expression            NaN   \n",
       "12        Gene Expression            NaN   \n",
       "13        Gene Expression            NaN   \n",
       "14        Gene Expression            NaN   \n",
       "15        Gene Expression            NaN   \n",
       "16        Gene Expression            NaN   \n",
       "17      Phenotypes (FYPO)  FYPO:0000046;   \n",
       "18      Phenotypes (FYPO)  FYPO:0001234;   \n",
       "19      Phenotypes (FYPO)  FYPO:0001355;   \n",
       "20      Phenotypes (FYPO)  FYPO:0000684;   \n",
       "21      Phenotypes (FYPO)  FYPO:0002177;   \n",
       "22    Phenotypic Profiles            NaN   \n",
       "23       Protein Features            NaN   \n",
       "24       Protein Features            NaN   \n",
       "25       Protein Features            NaN   \n",
       "26    Transcript Features            NaN   \n",
       "27    Transcript Features            NaN   \n",
       "\n",
       "                                         GeneSet_Name  \\\n",
       "0                                biosynthetic process   \n",
       "1                       cellular biosynthetic process   \n",
       "2              organic substance biosynthetic process   \n",
       "3                                  cytosolic ribosome   \n",
       "4                                      cytosolic part   \n",
       "5                                            ribosome   \n",
       "6                  structural constituent of ribosome   \n",
       "7                        structural molecule activity   \n",
       "8                                        rRNA binding   \n",
       "9                  Ribosomal protein module Homol D E   \n",
       "10                         Ribosomal protein module 2   \n",
       "11                             Highly expressed genes   \n",
       "12                         Oxidative Stess Cluster 11   \n",
       "13                          Oxidative Stess Cluster 9   \n",
       "14                   Caffeine and Rapamycin repressed   \n",
       "15       Core Environmental Stress Response repressed   \n",
       "16                          Oxidative Stess Cluster 6   \n",
       "17                   decreased cell population growth   \n",
       "18             slow vegetative cell population growth   \n",
       "19        decreased vegetative cell population growth   \n",
       "20  decreased cell population growth on glycerol c...   \n",
       "21  viable vegetative cell with normal cell morpho...   \n",
       "22                       Slowly proliferating mutants   \n",
       "23                                           Arginine   \n",
       "24              Protein copies per proliferating cell   \n",
       "25                   Isoelectric point (predicted pH)   \n",
       "26                                    mRNA level (WT)   \n",
       "27                 mRNA copies per proliferating cell   \n",
       "\n",
       "   over_represented/under_represented              List_Frequency  \\\n",
       "0                            Enriched    52.317880794702 (79/151)   \n",
       "1                            Enriched   51.6556291390728 (78/151)   \n",
       "2                            Enriched   51.6556291390728 (78/151)   \n",
       "3                            Enriched   12.5827814569536 (19/151)   \n",
       "4                            Enriched   13.2450331125828 (20/151)   \n",
       "5                            Enriched   12.5827814569536 (19/151)   \n",
       "6                            Enriched   11.9205298013245 (18/151)   \n",
       "7                            Enriched   11.9205298013245 (18/151)   \n",
       "8                            Enriched    4.63576158940397 (7/151)   \n",
       "9                            Enriched   14.5695364238411 (22/151)   \n",
       "10                           Enriched   15.8940397350993 (24/151)   \n",
       "11                           Enriched    19.205298013245 (29/151)   \n",
       "12                           Enriched   8.60927152317881 (13/151)   \n",
       "13                           Enriched   9.93377483443709 (15/151)   \n",
       "14                           Enriched   13.2450331125828 (20/151)   \n",
       "15                           Enriched   15.8940397350993 (24/151)   \n",
       "16                           Enriched   9.93377483443709 (15/151)   \n",
       "17                           Enriched   41.0596026490066 (62/151)   \n",
       "18                           Enriched    23.841059602649 (36/151)   \n",
       "19                           Enriched   40.3973509933775 (61/151)   \n",
       "20                           Enriched    23.841059602649 (36/151)   \n",
       "21                   Underrepresented  68.8741721854305 (104/151)   \n",
       "22                           Enriched   17.8807947019868 (27/151)   \n",
       "23                             Higher          0.0578795262492446   \n",
       "24                             Higher            31596.0256701031   \n",
       "25                             Higher            8.10106357615894   \n",
       "26                             Higher            4020.90520833333   \n",
       "27                             Higher            18.5992516556291   \n",
       "\n",
       "            Background_Frequency  Corrected_pvalue  \\\n",
       "0     27.456839309429 (827/3012)      1.933990e-07   \n",
       "1    26.8260292164675 (808/3012)      1.933990e-07   \n",
       "2    27.0252324037185 (814/3012)      1.933990e-07   \n",
       "3     2.75564409030544 (83/3012)      2.302700e-05   \n",
       "4     3.15405046480744 (95/3012)      2.851560e-05   \n",
       "5     3.22045152722444 (97/3012)      1.322570e-04   \n",
       "6     2.75564409030544 (83/3012)      5.963010e-05   \n",
       "7    3.51925630810093 (106/3012)      1.100610e-03   \n",
       "8    0.597609561752988 (18/3012)      3.465800e-03   \n",
       "9    3.68525896414343 (111/3012)      2.302700e-05   \n",
       "10   4.44887118193891 (134/3012)      2.851560e-05   \n",
       "11   6.90571049136786 (208/3012)      1.514130e-04   \n",
       "12    1.85922974767596 (56/3012)      8.621080e-04   \n",
       "13    2.52324037184595 (76/3012)      1.100610e-03   \n",
       "14   4.58167330677291 (138/3012)      2.881290e-03   \n",
       "15   6.17529880478088 (186/3012)      2.881290e-03   \n",
       "16    2.78884462151394 (84/3012)      2.988380e-03   \n",
       "17   21.4807436918991 (647/3012)      2.851560e-05   \n",
       "18   9.42895086321381 (284/3012)      5.892290e-05   \n",
       "19   21.2483399734396 (640/3012)      5.892290e-05   \n",
       "20   10.0265604249668 (302/3012)      2.219750e-04   \n",
       "21  84.6945551128818 (2551/3012)      3.117270e-04   \n",
       "22   6.20849933598938 (187/3012)      1.837160e-04   \n",
       "23            0.0487964770423139      1.463610e-03   \n",
       "24              13684.6208433048      1.472360e-03   \n",
       "25              7.31765260398461      2.920790e-03   \n",
       "26              2387.36666641874      2.211010e-04   \n",
       "27              5.82367806520199      4.121010e-04   \n",
       "\n",
       "                Genes_In_Common/Interacting_Gene_Pair  \n",
       "0   SPAC24H6.07;SPAC22F3.09c;SPAC13D6.02c;SPAC1783...  \n",
       "1   SPAC24H6.07;SPAC22F3.09c;SPAC13D6.02c;SPAC1783...  \n",
       "2   SPAC24H6.07;SPAC22F3.09c;SPAC13D6.02c;SPAC1783...  \n",
       "3   SPAC24H6.07;SPAC959.08;SPCC74.05;SPCC364.03;SP...  \n",
       "4   SPAC24H6.07;SPAC959.08;SPCC74.05;SPCC364.03;SP...  \n",
       "5   SPAC24H6.07;SPAC959.08;SPCC74.05;SPCC364.03;SP...  \n",
       "6   SPAC24H6.07;SPAC959.08;SPCC74.05;SPCC364.03;SP...  \n",
       "7   SPAC24H6.07;SPAC959.08;SPCC74.05;SPCC364.03;SP...  \n",
       "8   SPAC24H6.07;SPBC16D10.11c;SPAC22F8.09;SPAC3G9....  \n",
       "9   SPAC24H6.07;SPAC959.08;SPCC74.05;SPAC23H4.10c;...  \n",
       "10  SPAC24H6.07;SPBPB10D8.04c;SPBC11B10.02c;SPAC95...  \n",
       "11  SPAC24H6.07;SPAC13D6.02c;SPBPB10D8.04c;SPAC959...  \n",
       "12  SPAC24H6.07;SPAC959.08;SPCC74.05;SPCC364.03;SP...  \n",
       "13  SPAC24H6.07;SPAC959.08;SPCC74.05;SPCC364.03;SP...  \n",
       "14  SPAC24H6.07;SPAC13D6.02c;SPAC959.08;SPCC74.05;...  \n",
       "15  SPAC24H6.07;SPAC13D6.02c;SPAC959.08;SPCC74.05;...  \n",
       "16  SPAC24H6.07;SPAC959.08;SPCC74.05;SPCC364.03;SP...  \n",
       "17  SPAC22F3.09c;SPBC543.09;SPBC1215.01;SPAC1783.0...  \n",
       "18  SPAC13D6.02c;SPAC1783.07c;SPCC188.07;SPCP1E11....  \n",
       "19  SPAC22F3.09c;SPBC543.09;SPBC1215.01;SPAC1783.0...  \n",
       "20  SPBC543.09;SPBC1215.01;SPAC1783.07c;SPAC3A12.1...  \n",
       "21  SPAC24H6.07;SPAPB18E9.01;SPAC22F3.09c;SPAC13D6...  \n",
       "22  SPAC13D6.02c;SPCC188.07;SPAC23H4.10c;SPCP1E11....  \n",
       "23  SPAC24H6.07;SPAC22A12.04c;SPAC3A12.12;SPAC23H4...  \n",
       "24  SPAC24H6.07;SPAC3A12.12;SPAC23H4.10c;SPAC343.1...  \n",
       "25  SPAC24H6.07;SPAC22A12.04c;SPAC3A12.12;SPAC23H4...  \n",
       "26  SPAC24H6.07;SPAC22A12.04c;SPAC3A12.12;SPAC23H4...  \n",
       "27  SPAC24H6.07;SPAC22A12.04c;SPAC3A12.12;SPAC23H4...  "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "angeli"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/mb/vdj4y7t13zd3tzlk99b8ycd40000gn/T/ipykernel_93734/569773454.py:6: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  angeli['dummy'] = angeli[colnames[0]].astype('category').cat.codes\n",
      "/var/folders/mb/vdj4y7t13zd3tzlk99b8ycd40000gn/T/ipykernel_93734/569773454.py:10: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  angeli['dummy2'] = np.where(indices,  \"darkred\", \"darkgreen\")\n",
      "/var/folders/mb/vdj4y7t13zd3tzlk99b8ycd40000gn/T/ipykernel_93734/569773454.py:14: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  angeli[\"dummy\"] = angeli[\"dummy\"].apply(lambda x: a[x])\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_f757d_row0_col0, #T_f757d_row0_col2, #T_f757d_row0_col4, #T_f757d_row1_col0, #T_f757d_row1_col2, #T_f757d_row1_col4, #T_f757d_row2_col0, #T_f757d_row2_col2, #T_f757d_row2_col4 {\n",
       "  background-color: #440154;\n",
       "  color: darkred;\n",
       "  font-weight: bold;\n",
       "}\n",
       "#T_f757d_row3_col0, #T_f757d_row3_col2, #T_f757d_row3_col4, #T_f757d_row4_col0, #T_f757d_row4_col2, #T_f757d_row4_col4, #T_f757d_row5_col0, #T_f757d_row5_col2, #T_f757d_row5_col4 {\n",
       "  background-color: #46327e;\n",
       "  color: darkred;\n",
       "  font-weight: bold;\n",
       "}\n",
       "#T_f757d_row6_col0, #T_f757d_row6_col2, #T_f757d_row6_col4, #T_f757d_row7_col0, #T_f757d_row7_col2, #T_f757d_row7_col4, #T_f757d_row8_col0, #T_f757d_row8_col2, #T_f757d_row8_col4 {\n",
       "  background-color: #365c8d;\n",
       "  color: darkred;\n",
       "  font-weight: bold;\n",
       "}\n",
       "#T_f757d_row9_col0, #T_f757d_row9_col2, #T_f757d_row9_col4, #T_f757d_row10_col0, #T_f757d_row10_col2, #T_f757d_row10_col4, #T_f757d_row11_col0, #T_f757d_row11_col2, #T_f757d_row11_col4, #T_f757d_row12_col0, #T_f757d_row12_col2, #T_f757d_row12_col4, #T_f757d_row13_col0, #T_f757d_row13_col2, #T_f757d_row13_col4, #T_f757d_row14_col0, #T_f757d_row14_col2, #T_f757d_row14_col4, #T_f757d_row15_col0, #T_f757d_row15_col2, #T_f757d_row15_col4, #T_f757d_row16_col0, #T_f757d_row16_col2, #T_f757d_row16_col4 {\n",
       "  background-color: #277f8e;\n",
       "  color: darkred;\n",
       "  font-weight: bold;\n",
       "}\n",
       "#T_f757d_row17_col0, #T_f757d_row17_col2, #T_f757d_row17_col4, #T_f757d_row18_col0, #T_f757d_row18_col2, #T_f757d_row18_col4, #T_f757d_row19_col0, #T_f757d_row19_col2, #T_f757d_row19_col4, #T_f757d_row20_col0, #T_f757d_row20_col2, #T_f757d_row20_col4 {\n",
       "  background-color: #1fa187;\n",
       "  color: darkred;\n",
       "  font-weight: bold;\n",
       "}\n",
       "#T_f757d_row21_col0, #T_f757d_row21_col2, #T_f757d_row21_col4 {\n",
       "  background-color: #1fa187;\n",
       "  color: darkgreen;\n",
       "  font-weight: bold;\n",
       "}\n",
       "#T_f757d_row22_col0, #T_f757d_row22_col2, #T_f757d_row22_col4 {\n",
       "  background-color: #4ac16d;\n",
       "  color: darkred;\n",
       "  font-weight: bold;\n",
       "}\n",
       "#T_f757d_row23_col0, #T_f757d_row23_col2, #T_f757d_row23_col4, #T_f757d_row24_col0, #T_f757d_row24_col2, #T_f757d_row24_col4, #T_f757d_row25_col0, #T_f757d_row25_col2, #T_f757d_row25_col4 {\n",
       "  background-color: #a0da39;\n",
       "  color: darkred;\n",
       "  font-weight: bold;\n",
       "}\n",
       "#T_f757d_row26_col0, #T_f757d_row26_col2, #T_f757d_row26_col4, #T_f757d_row27_col0, #T_f757d_row27_col2, #T_f757d_row27_col4 {\n",
       "  background-color: #fde725;\n",
       "  color: darkred;\n",
       "  font-weight: bold;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_f757d\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_f757d_level0_col0\" class=\"col_heading level0 col0\" >Category_Name</th>\n",
       "      <th id=\"T_f757d_level0_col2\" class=\"col_heading level0 col2\" >GeneSet_Name</th>\n",
       "      <th id=\"T_f757d_level0_col4\" class=\"col_heading level0 col4\" >Corrected_pvalue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_f757d_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_f757d_row0_col0\" class=\"data row0 col0\" >GO Biological Process</td>\n",
       "      <td id=\"T_f757d_row0_col2\" class=\"data row0 col2\" >biosynthetic process</td>\n",
       "      <td id=\"T_f757d_row0_col4\" class=\"data row0 col4\" >0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f757d_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_f757d_row1_col0\" class=\"data row1 col0\" >GO Biological Process</td>\n",
       "      <td id=\"T_f757d_row1_col2\" class=\"data row1 col2\" >cellular biosynthetic process</td>\n",
       "      <td id=\"T_f757d_row1_col4\" class=\"data row1 col4\" >0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f757d_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_f757d_row2_col0\" class=\"data row2 col0\" >GO Biological Process</td>\n",
       "      <td id=\"T_f757d_row2_col2\" class=\"data row2 col2\" >organic substance biosynthetic process</td>\n",
       "      <td id=\"T_f757d_row2_col4\" class=\"data row2 col4\" >0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f757d_level0_row3\" class=\"row_heading level0 row3\" >3</th>\n",
       "      <td id=\"T_f757d_row3_col0\" class=\"data row3 col0\" >GO Cellular Component</td>\n",
       "      <td id=\"T_f757d_row3_col2\" class=\"data row3 col2\" >cytosolic ribosome</td>\n",
       "      <td id=\"T_f757d_row3_col4\" class=\"data row3 col4\" >0.000023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f757d_level0_row4\" class=\"row_heading level0 row4\" >4</th>\n",
       "      <td id=\"T_f757d_row4_col0\" class=\"data row4 col0\" >GO Cellular Component</td>\n",
       "      <td id=\"T_f757d_row4_col2\" class=\"data row4 col2\" >cytosolic part</td>\n",
       "      <td id=\"T_f757d_row4_col4\" class=\"data row4 col4\" >0.000029</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f757d_level0_row5\" class=\"row_heading level0 row5\" >5</th>\n",
       "      <td id=\"T_f757d_row5_col0\" class=\"data row5 col0\" >GO Cellular Component</td>\n",
       "      <td id=\"T_f757d_row5_col2\" class=\"data row5 col2\" >ribosome</td>\n",
       "      <td id=\"T_f757d_row5_col4\" class=\"data row5 col4\" >0.000132</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f757d_level0_row6\" class=\"row_heading level0 row6\" >6</th>\n",
       "      <td id=\"T_f757d_row6_col0\" class=\"data row6 col0\" >GO Molecular Function</td>\n",
       "      <td id=\"T_f757d_row6_col2\" class=\"data row6 col2\" >structural constituent of ribosome</td>\n",
       "      <td id=\"T_f757d_row6_col4\" class=\"data row6 col4\" >0.000060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f757d_level0_row7\" class=\"row_heading level0 row7\" >7</th>\n",
       "      <td id=\"T_f757d_row7_col0\" class=\"data row7 col0\" >GO Molecular Function</td>\n",
       "      <td id=\"T_f757d_row7_col2\" class=\"data row7 col2\" >structural molecule activity</td>\n",
       "      <td id=\"T_f757d_row7_col4\" class=\"data row7 col4\" >0.001101</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f757d_level0_row8\" class=\"row_heading level0 row8\" >8</th>\n",
       "      <td id=\"T_f757d_row8_col0\" class=\"data row8 col0\" >GO Molecular Function</td>\n",
       "      <td id=\"T_f757d_row8_col2\" class=\"data row8 col2\" >rRNA binding</td>\n",
       "      <td id=\"T_f757d_row8_col4\" class=\"data row8 col4\" >0.003466</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f757d_level0_row9\" class=\"row_heading level0 row9\" >9</th>\n",
       "      <td id=\"T_f757d_row9_col0\" class=\"data row9 col0\" >Gene Expression</td>\n",
       "      <td id=\"T_f757d_row9_col2\" class=\"data row9 col2\" >Ribosomal protein module Homol D E</td>\n",
       "      <td id=\"T_f757d_row9_col4\" class=\"data row9 col4\" >0.000023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f757d_level0_row10\" class=\"row_heading level0 row10\" >10</th>\n",
       "      <td id=\"T_f757d_row10_col0\" class=\"data row10 col0\" >Gene Expression</td>\n",
       "      <td id=\"T_f757d_row10_col2\" class=\"data row10 col2\" >Ribosomal protein module 2</td>\n",
       "      <td id=\"T_f757d_row10_col4\" class=\"data row10 col4\" >0.000029</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f757d_level0_row11\" class=\"row_heading level0 row11\" >11</th>\n",
       "      <td id=\"T_f757d_row11_col0\" class=\"data row11 col0\" >Gene Expression</td>\n",
       "      <td id=\"T_f757d_row11_col2\" class=\"data row11 col2\" >Highly expressed genes</td>\n",
       "      <td id=\"T_f757d_row11_col4\" class=\"data row11 col4\" >0.000151</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f757d_level0_row12\" class=\"row_heading level0 row12\" >12</th>\n",
       "      <td id=\"T_f757d_row12_col0\" class=\"data row12 col0\" >Gene Expression</td>\n",
       "      <td id=\"T_f757d_row12_col2\" class=\"data row12 col2\" >Oxidative Stess Cluster 11</td>\n",
       "      <td id=\"T_f757d_row12_col4\" class=\"data row12 col4\" >0.000862</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f757d_level0_row13\" class=\"row_heading level0 row13\" >13</th>\n",
       "      <td id=\"T_f757d_row13_col0\" class=\"data row13 col0\" >Gene Expression</td>\n",
       "      <td id=\"T_f757d_row13_col2\" class=\"data row13 col2\" >Oxidative Stess Cluster 9</td>\n",
       "      <td id=\"T_f757d_row13_col4\" class=\"data row13 col4\" >0.001101</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f757d_level0_row14\" class=\"row_heading level0 row14\" >14</th>\n",
       "      <td id=\"T_f757d_row14_col0\" class=\"data row14 col0\" >Gene Expression</td>\n",
       "      <td id=\"T_f757d_row14_col2\" class=\"data row14 col2\" >Caffeine and Rapamycin repressed</td>\n",
       "      <td id=\"T_f757d_row14_col4\" class=\"data row14 col4\" >0.002881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f757d_level0_row15\" class=\"row_heading level0 row15\" >15</th>\n",
       "      <td id=\"T_f757d_row15_col0\" class=\"data row15 col0\" >Gene Expression</td>\n",
       "      <td id=\"T_f757d_row15_col2\" class=\"data row15 col2\" >Core Environmental Stress Response repressed</td>\n",
       "      <td id=\"T_f757d_row15_col4\" class=\"data row15 col4\" >0.002881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f757d_level0_row16\" class=\"row_heading level0 row16\" >16</th>\n",
       "      <td id=\"T_f757d_row16_col0\" class=\"data row16 col0\" >Gene Expression</td>\n",
       "      <td id=\"T_f757d_row16_col2\" class=\"data row16 col2\" >Oxidative Stess Cluster 6</td>\n",
       "      <td id=\"T_f757d_row16_col4\" class=\"data row16 col4\" >0.002988</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f757d_level0_row17\" class=\"row_heading level0 row17\" >17</th>\n",
       "      <td id=\"T_f757d_row17_col0\" class=\"data row17 col0\" >Phenotypes (FYPO)</td>\n",
       "      <td id=\"T_f757d_row17_col2\" class=\"data row17 col2\" >decreased cell population growth</td>\n",
       "      <td id=\"T_f757d_row17_col4\" class=\"data row17 col4\" >0.000029</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f757d_level0_row18\" class=\"row_heading level0 row18\" >18</th>\n",
       "      <td id=\"T_f757d_row18_col0\" class=\"data row18 col0\" >Phenotypes (FYPO)</td>\n",
       "      <td id=\"T_f757d_row18_col2\" class=\"data row18 col2\" >slow vegetative cell population growth</td>\n",
       "      <td id=\"T_f757d_row18_col4\" class=\"data row18 col4\" >0.000059</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f757d_level0_row19\" class=\"row_heading level0 row19\" >19</th>\n",
       "      <td id=\"T_f757d_row19_col0\" class=\"data row19 col0\" >Phenotypes (FYPO)</td>\n",
       "      <td id=\"T_f757d_row19_col2\" class=\"data row19 col2\" >decreased vegetative cell population growth</td>\n",
       "      <td id=\"T_f757d_row19_col4\" class=\"data row19 col4\" >0.000059</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f757d_level0_row20\" class=\"row_heading level0 row20\" >20</th>\n",
       "      <td id=\"T_f757d_row20_col0\" class=\"data row20 col0\" >Phenotypes (FYPO)</td>\n",
       "      <td id=\"T_f757d_row20_col2\" class=\"data row20 col2\" >decreased cell population growth on glycerol carbon source</td>\n",
       "      <td id=\"T_f757d_row20_col4\" class=\"data row20 col4\" >0.000222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f757d_level0_row21\" class=\"row_heading level0 row21\" >21</th>\n",
       "      <td id=\"T_f757d_row21_col0\" class=\"data row21 col0\" >Phenotypes (FYPO)</td>\n",
       "      <td id=\"T_f757d_row21_col2\" class=\"data row21 col2\" >viable vegetative cell with normal cell morphology</td>\n",
       "      <td id=\"T_f757d_row21_col4\" class=\"data row21 col4\" >0.000312</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f757d_level0_row22\" class=\"row_heading level0 row22\" >22</th>\n",
       "      <td id=\"T_f757d_row22_col0\" class=\"data row22 col0\" >Phenotypic Profiles</td>\n",
       "      <td id=\"T_f757d_row22_col2\" class=\"data row22 col2\" >Slowly proliferating mutants</td>\n",
       "      <td id=\"T_f757d_row22_col4\" class=\"data row22 col4\" >0.000184</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f757d_level0_row23\" class=\"row_heading level0 row23\" >23</th>\n",
       "      <td id=\"T_f757d_row23_col0\" class=\"data row23 col0\" >Protein Features</td>\n",
       "      <td id=\"T_f757d_row23_col2\" class=\"data row23 col2\" >Arginine</td>\n",
       "      <td id=\"T_f757d_row23_col4\" class=\"data row23 col4\" >0.001464</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f757d_level0_row24\" class=\"row_heading level0 row24\" >24</th>\n",
       "      <td id=\"T_f757d_row24_col0\" class=\"data row24 col0\" >Protein Features</td>\n",
       "      <td id=\"T_f757d_row24_col2\" class=\"data row24 col2\" >Protein copies per proliferating cell</td>\n",
       "      <td id=\"T_f757d_row24_col4\" class=\"data row24 col4\" >0.001472</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f757d_level0_row25\" class=\"row_heading level0 row25\" >25</th>\n",
       "      <td id=\"T_f757d_row25_col0\" class=\"data row25 col0\" >Protein Features</td>\n",
       "      <td id=\"T_f757d_row25_col2\" class=\"data row25 col2\" >Isoelectric point (predicted pH)</td>\n",
       "      <td id=\"T_f757d_row25_col4\" class=\"data row25 col4\" >0.002921</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f757d_level0_row26\" class=\"row_heading level0 row26\" >26</th>\n",
       "      <td id=\"T_f757d_row26_col0\" class=\"data row26 col0\" >Transcript Features</td>\n",
       "      <td id=\"T_f757d_row26_col2\" class=\"data row26 col2\" >mRNA level (WT)</td>\n",
       "      <td id=\"T_f757d_row26_col4\" class=\"data row26 col4\" >0.000221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f757d_level0_row27\" class=\"row_heading level0 row27\" >27</th>\n",
       "      <td id=\"T_f757d_row27_col0\" class=\"data row27 col0\" >Transcript Features</td>\n",
       "      <td id=\"T_f757d_row27_col2\" class=\"data row27 col2\" >mRNA copies per proliferating cell</td>\n",
       "      <td id=\"T_f757d_row27_col4\" class=\"data row27 col4\" >0.000412</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x7fc3d00a7fd0>"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "<Figure size 640x480 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#keep first 4 and 7th column\n",
    "angeli = angeli.iloc[:, [0, 1,2,3, 6]]\n",
    "#create a table where background of row colored by unique entry of first column and text colored by entry of 4th column\n",
    "colnames = angeli.columns.values\n",
    "#create a dummy column with numbers for unique values in column 1\n",
    "angeli['dummy'] = angeli[colnames[0]].astype('category').cat.codes\n",
    "#create dummy for unique values in column 4 is \"Enriched\" or \"Higher\"\n",
    "#find indices where column 4 is \"Enriched\" or \"Higher\"\n",
    "indices = angeli[colnames[3]].isin(['Enriched', 'Higher'])\n",
    "angeli['dummy2'] = np.where(indices,  \"darkred\", \"darkgreen\")\n",
    "a = plt.cm.viridis(np.linspace(0,1,len(np.unique(angeli['dummy']))))\n",
    "a = [matplotlib.colors.rgb2hex(i) for i in a]\n",
    "#subset angeli to values in unique values in a. first unique value in a goes for first unique value in dummy\n",
    "angeli[\"dummy\"] = angeli[\"dummy\"].apply(lambda x: a[x])\n",
    "#final is first third fifth\n",
    "#color entire rows, meaning all columns, based on dummy and text based on dummy2\n",
    "#write function that highlights rows that are enriched\n",
    "def highlight_enriched(row):\n",
    "      #color text based on dummy and color background based on dummy2\n",
    "      return ['background-color: %s; color: %s' % (row['dummy'], row['dummy2'])]*len(row)\n",
    "#make text bold\n",
    "#give black highlight to fonts\n",
    "plt.savefig('/Users/bencekover/Library/CloudStorage/OneDrive-UniversityCollegeLondon/MSci Bahler lab/S.-Pombe-biofilm/Bence folder/Angeli/angeli/angeli.png', dpi = 300)\n",
    "angeli.style.apply(highlight_enriched, axis=1).hide(subset=['dummy', 'dummy2','External_ID','over_represented/under_represented'], axis=1).set_properties(**{'font-weight': 'bold'})\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "#create function that does this all\n",
    "def angeli_function(file,name):\n",
    "      #get wd from file by omitting last element\n",
    "      wd = '/'.join(file.split('/')[:-1])\n",
    "      #import angeli file\n",
    "      angeli = pd.read_csv(file, sep = '\\t', header = 0, index_col = None)\n",
    "      #leave top 10 gene expression\n",
    "      angeli_fypo= angeli[angeli['Category_Name'] == 'Phenotypes (FYPO)'].iloc[0:5]\n",
    "      angeli = angeli[angeli['Category_Name'] != 'Phenotypes (FYPO)']\n",
    "      angeli = pd.concat([angeli_fypo, angeli])\n",
    "      angeli_gene = angeli[angeli['Category_Name'] == 'Gene Expression'].iloc[0:10]\n",
    "      angeli = angeli[angeli['Category_Name'] != 'Gene Expression']\n",
    "      angeli = pd.concat([angeli_gene, angeli])\n",
    "      angeli_mf = angeli[angeli['Category_Name'] == 'GO Molecular Function'].iloc[0:3]\n",
    "      angeli = angeli[angeli['Category_Name'] != 'GO Molecular Function']\n",
    "      angeli = pd.concat([angeli_mf, angeli])\n",
    "      angeli_cc = angeli[angeli['Category_Name'] == 'GO Cellular Component'].iloc[0:3]\n",
    "      angeli = angeli[angeli['Category_Name'] != 'GO Cellular Component']\n",
    "      angeli = pd.concat([angeli_cc, angeli])\n",
    "      angeli_bp = angeli[angeli['Category_Name'] == 'GO Biological Process'].iloc[0:3]\n",
    "      #remove all biological process and add back only the first 3 rows to the top\n",
    "      angeli = angeli[angeli['Category_Name'] != 'GO Biological Process']\n",
    "      #add back to top of angeli\n",
    "      angeli = pd.concat([angeli_bp, angeli])\n",
    "      #reset index\n",
    "      angeli = angeli.reset_index(drop = True)\n",
    "      #keep first 4 and 7th column\n",
    "      angeli = angeli.iloc[:, [0, 1,2,3, 6]]\n",
    "\n",
    "      #rename Category_Name\tto Category\n",
    "      angeli = angeli.rename(columns = {'Category_Name':'Category'})\n",
    "      #rename GeneSet_Name to Gene set\n",
    "      angeli = angeli.rename(columns = {'GeneSet_Name':'Gene set'})\n",
    "      #rename Corrected_pvalue to Corrected P-value\n",
    "      angeli = angeli.rename(columns = {'Corrected_pvalue':'Corrected P-value'})\n",
    "      #in Corrected P-value, add P: to the beginning of each entry\n",
    "      angeli['Corrected P-value'] = 'P= ' + angeli['Corrected P-value'].astype(str)\n",
    "      \n",
    "\n",
    "      #create a table where background of row colored by unique entry of first column and text colored by entry of 4th column\n",
    "      colnames = angeli.columns.values\n",
    "      #create a dummy column with numbers for unique values in column 1\n",
    "      angeli['dummy'] = angeli[colnames[0]].astype('category').cat.codes\n",
    "      #create a dummy column with numbers for unique values in column 1\n",
    "      angeli['dummy'] = angeli[colnames[0]].astype('category').cat.codes\n",
    "      #create dummy for unique values in column 4 is neriched then 1 else 0\n",
    "      indices = angeli[colnames[3]].isin(['Enriched', 'Higher'])\n",
    "      angeli['dummy2'] = np.where(indices,  \"darkred\", \"darkgreen\")\n",
    "      a = plt.cm.viridis(np.linspace(0.4,1,len(np.unique(angeli['dummy']))))\n",
    "      a = [matplotlib.colors.rgb2hex(i) for i in a]\n",
    "      #subset angeli to values in unique values in a. first unique value in a goes for first unique value in dummy\n",
    "      angeli[\"dummy\"] = angeli[\"dummy\"].apply(lambda x: a[x])\n",
    "      #final is first third fifth\n",
    "      #color entire rows, meaning all columns, based on dummy and text based on dummy2\n",
    "      #write function that highlights rows that are enriched\n",
    "      def highlight_enriched(row):\n",
    "            #color text based on dummy and color background based on dummy2\n",
    "            return ['background-color: %s; color: %s' % (row['dummy'], row['dummy2'])]*len(row)\n",
    "      #make text bold\n",
    "      #give black highlight to fonts\n",
    "      plt.savefig('/Users/bencekover/Library/CloudStorage/OneDrive-UniversityCollegeLondon/MSci Bahler lab/S.-Pombe-biofilm/Bence folder/Angeli/angeli/angeli.png', dpi = 300)\n",
    "      #larger font\n",
    "      styled= angeli.style.apply(highlight_enriched, axis=1).hide(subset=['dummy', 'dummy2','External_ID','over_represented/under_represented'], axis=1).set_properties(**{'font-weight': 'bold'})\n",
    "      dfi.export(styled, wd + '/' + name + '.png')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Applications/Google Chrome.app/Contents/MacOS/Google Chrome\n",
      "/Applications/Google Chrome.app/Contents/MacOS/Google Chrome\n",
      "/Applications/Google Chrome.app/Contents/MacOS/Google Chrome\n",
      "/Applications/Google Chrome.app/Contents/MacOS/Google Chrome\n",
      "/Applications/Google Chrome.app/Contents/MacOS/Google Chrome\n",
      "/Applications/Google Chrome.app/Contents/MacOS/Google Chrome\n",
      "/Applications/Google Chrome.app/Contents/MacOS/Google Chrome\n",
      "/Applications/Google Chrome.app/Contents/MacOS/Google Chrome\n",
      "/Applications/Google Chrome.app/Contents/MacOS/Google Chrome\n",
      "/Applications/Google Chrome.app/Contents/MacOS/Google Chrome\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<Figure size 640x480 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "angeli_function(\"/Users/bencekover/Library/CloudStorage/OneDrive-UniversityCollegeLondon/MSci Bahler lab/S.-Pombe-biofilm/Bence folder/Angeli/angeli/bioneer_angeli_results.txt\",\"bioneer\")\n",
    "angeli_function(\"/Users/bencekover/Library/CloudStorage/OneDrive-UniversityCollegeLondon/MSci Bahler lab/S.-Pombe-biofilm/Bence folder/Angeli/angeli/cdk8.txt\",\"cdk8\")\n",
    "angeli_function(\"/Users/bencekover/Library/CloudStorage/OneDrive-UniversityCollegeLondon/MSci Bahler lab/S.-Pombe-biofilm/Bence folder/Angeli/angeli/floc_df.txt\", \"floc\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.9.0 ('base')",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.0"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "2d0398177d7bb10ea33b6a5cd9d022d63e20680fcfd84762562c72ff35bdf44b"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
