{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "import pandas as pd\n",
    "root = \"/Users/k23030440/github_projects/S.-Pombe-MLPs/\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "#read in data\n",
    "\n",
    "#go terms for orthology analysis from: Supplementary material/annotation_orthogroup_genes.xlsx\n",
    "\n",
    "go_terms = pd.read_csv(root + \"internal data/go_terms_formatted.csv\",header=1)\n",
    "\n",
    "#omics data sources\n",
    "omics_sources = pd.read_excel(root + \"internal data/external data sources.xlsx\")\n",
    "\n",
    "#floc_cor_genes /Users/bencekover/Library/CloudStorage/OneDrive-Personal/MSci Bahler lab/S.-Pombe-MLPs - Github/Bence folder/Analysis of Clement-Ziza RNA-seq/filter_corr.csv\n",
    "floc_cor_genes = pd.read_csv(root + \"Bence folder/Analysis of Clement-Ziza RNA-seq/filter_corr.csv\", index_col=0)\n",
    "\n",
    "#structure_sequence_homology /Users/bencekover/Library/CloudStorage/OneDrive-Personal/MSci Bahler lab/S.-Pombe-MLPs - Github/Bence folder/Structure_and_Sequence_comparisons/final_analysis.csv\n",
    "structure_sequence_homology = pd.read_csv(root + \"Bence folder/Structure_and_Sequence_comparisons/final_analysis.csv\")\n",
    "\n",
    "#srb11 DE genes /Users/bencekover/Library/CloudStorage/OneDrive-Personal/MSci Bahler lab/S.-Pombe-MLPs - Github/Bence folder/Analysis of Clement-Ziza RNA-seq/DESeq2/DE_results.csv\n",
    "srb11_DE_genes = pd.read_csv(root + \"Bence folder/Analysis of Clement-Ziza RNA-seq/DESeq2/DE_results.csv\")\n",
    "#rename first col to Systematic name\n",
    "srb11_DE_genes.rename(columns={srb11_DE_genes.columns[0]:\"Systematic name\"}, inplace=True)\n",
    "\n",
    "\n",
    "#segregant genotypes/Users/bencekover/Library/CloudStorage/OneDrive-Personal/MSci Bahler lab/S.-Pombe-MLPs - Github/internal data/updated_genotype_matrix_final_pos.tsv\n",
    "segregant_genotypes = pd.read_csv(root + \"internal data/updated_genotype_matrix_final_pos.tsv\", sep=\"\\t\")\n",
    "\n",
    "\n",
    "wi_adh_across_media = pd.read_excel(root + \"Bence folder/Image processing - Adhesion assay/condition_plot_df.xlsx\",index_col=0)\n",
    "\n",
    "\n",
    "#plasmids '/Users/k23030440/github_projects/S.-Pombe-MLPs/Supplementary material/plasmid_table.xlsx'\n",
    "plasmids=pd.read_excel(root + \"Supplementary material/plasmid_table.xlsx\")\n",
    "\n",
    "#screen_raw from root + \"internal data/Deletion screen raw results.csv\n",
    "screen_raw = pd.read_csv(root + \"internal data/Deletion screen raw results.csv\")\n",
    "\n",
    "#screen enrichment from root + Bence folder/Angeli/angeli/bioneer_angeli_results_recent.txt\n",
    "screen_enrichment = pd.read_csv(root + \"Bence folder/Angeli/angeli/bioneer_angeli_results_recent.txt\", sep=\"\\t\")\n",
    "\n",
    "#screen_final root + \"Bence folder/Bioneer screen/final_results_screen.csv\"\n",
    "screen_final = pd.read_csv(root + \"Bence folder/Bioneer screen/final_results_screen.csv\")\n",
    "\n",
    "\n",
    "#orthology, first col is index\n",
    "orthology = pd.read_csv(root + \"Bence folder/Orthologs/final_orthology_database_new.csv\", index_col=0)\n",
    "\n",
    "#mediator /Users/k23030440/S.-Pombe-MLPs - most recent/Bence folder/Bioneer screen/final_results_mediator.csv\n",
    "mediator = pd.read_csv(root + \"Bence folder/Bioneer screen/final_results_mediator.csv\")\n",
    "\n",
    "#variants S.-Pombe-MLPs/Bence folder/Looking for variants in wild isolates/snp_hunting.csv\n",
    "variants = pd.read_csv(root + \"Bence folder/Looking for variants in wild isolates/snp_hunting.csv\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Import the necessary libraries\n",
    "import pandas as pd\n",
    "\n",
    "# Define the file path for the output xlsx file\n",
    "output_file = root + \"Supplementary material/Supplementary table.xlsx\"\n",
    "\n",
    "# Create a Pandas Excel writer using the output file path\n",
    "writer = pd.ExcelWriter(output_file, engine='xlsxwriter')\n",
    "\n",
    "# Define a dictionary containing the dataframes and their corresponding sheet names\n",
    "dataframes = {\n",
    "    '1_go_fypo_terms': go_terms,\n",
    "    '2_structure_sequence_homology': structure_sequence_homology,\n",
    "    '3_omics_sources': omics_sources,\n",
    "    '4_segregant_genotypes': segregant_genotypes,\n",
    "    '5_orthology_relationships': orthology,\n",
    "    '6_nat_iso_adhesion_across_media' : wi_adh_across_media,\n",
    "    '7_floc_cor_genes': floc_cor_genes,\n",
    "    '8_mediator_results': mediator,\n",
    "    '9_srb11_DE_genes': srb11_DE_genes,\n",
    "    '10_Screen_raw_results': screen_raw,\n",
    "    '11_Screen_enrichment': screen_enrichment,\n",
    "    '12_Screen_final_results': screen_final,\n",
    "    '13_Variants_in_candidate_genes': variants,\n",
    "    '14_Plasmids': plasmids\n",
    "}\n",
    "\n",
    "descriptions = [\n",
    "    \"#Summary of the GO terms, FYPO terms, and phenotype keywords used for the orthology analysis.\",\n",
    "    \"#Summary of results from the structure and sequence homology analysis in a tidy format, ready for plotting.\\n\",\n",
    "    \"#Summary of omics data sources used in the study.\",\n",
    "    \"#Table of segregant genotypes used in the QTL analysis. Y0036 is alternative name for JB759, while 968 is for JB50\",\n",
    "    \"\"\"#Table of orthology relatonships for MLP-related genes. Each combination of orthology relationships is summarized by one of the 7 \"orthopatterns\", representative of the 7 sets in the plotted Venn diagrams.\"\"\",\n",
    "    \"#Measurements of adhesion phenotypes in the natural isolates in a tidy format, assembled from the output of yeastmlp, ready for plotting.\",\n",
    "    \"#Correlation analysis between flocculation (filtering assay) and gene expression (Clement-Ziza RNA-seq) data.\",\n",
    "    \"#Adhesion ratios of mediator deletion strains, assembled from the output of yeastmlp.\",\n",
    "    \"#Differentially expressed genes in the srb11 deletion strain from the output of DESeq2.\",\n",
    "    \"#Raw results from the deletion screen as measured by yeastmlp.\",\n",
    "    \"\"\"#Enrichment analysis based on the top 5 percent of adhesive strains identified in the deletion screen.\"\"\",\n",
    "    \"#Final results from the 31 high-confidence strains, as measured by yeastmlp.\",\n",
    "    \"#Variants identified in candidate genes (from screen and Mediator genes) by short variant analysis in MLP-forming wild isolates.\",\n",
    "    \"#List of plasmids used in this study.\"\n",
    "]\n",
    "\n",
    "# Loop through the dictionary and write each dataframe to a separate sheet in the output file\n",
    "for sheet_name, df, desc in zip(dataframes.keys(), dataframes.values(), descriptions):\n",
    "    # Add a header to the sheet name\n",
    "    sheet_name = sheet_name\n",
    "    title = \"# Title: \" + sheet_name\n",
    "    correspondence = \"# Correspondence: b.heineike@ucl.ac.uk or j.bahler@ucl.ac.uk\"\n",
    "    \n",
    "    # Write the dataframe to the sheet\n",
    "    df.to_excel(writer, sheet_name=sheet_name, index=False, startrow=4)\n",
    "    \n",
    "    # Get the xlsxwriter workbook and worksheet objects\n",
    "    workbook = writer.book\n",
    "    worksheet = writer.sheets[sheet_name]\n",
    "    \n",
    "    # Add the title and correspondance to the sheet\n",
    "    worksheet.write(0, 0, title)\n",
    "    worksheet.write(1, 0, correspondence)\n",
    "    worksheet.write(2, 0, desc)\n",
    "\n",
    "# Save the Excel file\n",
    "writer.close()\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Now assemble .pdf of all images"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Defaulting to user installation because normal site-packages is not writeable\n",
      "Requirement already satisfied: pillow in /Users/k23030440/Library/Python/3.11/lib/python/site-packages (10.1.0)\n",
      "Requirement already satisfied: fpdf2 in /Users/k23030440/Library/Python/3.11/lib/python/site-packages (2.7.6)\n",
      "Requirement already satisfied: defusedxml in /Users/k23030440/Library/Python/3.11/lib/python/site-packages (from fpdf2) (0.7.1)\n",
      "Requirement already satisfied: fonttools>=4.34.0 in /Users/k23030440/Library/Python/3.11/lib/python/site-packages (from fpdf2) (4.43.1)\n",
      "Defaulting to user installation because normal site-packages is not writeable\n",
      "Requirement already satisfied: PyPDF2 in /Users/k23030440/Library/Python/3.11/lib/python/site-packages (3.0.1)\n",
      "^C\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/k23030440/Library/Python/3.11/bin/pip\", line 8, in <module>\n",
      "    sys.exit(main())\n",
      "             ^^^^^^\n",
      "  File \"/Users/k23030440/Library/Python/3.11/lib/python/site-packages/pip/_internal/cli/main.py\", line 79, in main\n",
      "    return command.main(cmd_args)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/Users/k23030440/Library/Python/3.11/lib/python/site-packages/pip/_internal/cli/base_command.py\", line 101, in main\n",
      "    return self._main(args)\n",
      "           ^^^^^^^^^^^^^^^^\n",
      "  File \"/Users/k23030440/Library/Python/3.11/lib/python/site-packages/pip/_internal/cli/base_command.py\", line 236, in _main\n",
      "    self.handle_pip_version_check(options)\n",
      "  File \"/Users/k23030440/Library/Python/3.11/lib/python/site-packages/pip/_internal/cli/req_command.py\", line 177, in handle_pip_version_check\n",
      "    session = self._build_session(\n",
      "              ^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/Users/k23030440/Library/Python/3.11/lib/python/site-packages/pip/_internal/cli/req_command.py\", line 122, in _build_session\n",
      "    session = PipSession(\n",
      "              ^^^^^^^^^^^\n",
      "  File \"/Users/k23030440/Library/Python/3.11/lib/python/site-packages/pip/_internal/network/session.py\", line 342, in __init__\n",
      "    self.headers[\"User-Agent\"] = user_agent()\n",
      "                                 ^^^^^^^^^^^^\n",
      "  File \"/Users/k23030440/Library/Python/3.11/lib/python/site-packages/pip/_internal/network/session.py\", line 182, in user_agent\n",
      "    rustc_output = subprocess.check_output(\n",
      "                   ^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/subprocess.py\", line 466, in check_output\n",
      "    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/subprocess.py\", line 550, in run\n",
      "    stdout, stderr = process.communicate(input, timeout=timeout)\n",
      "                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/subprocess.py\", line 1209, in communicate\n",
      "    stdout, stderr = self._communicate(input, endtime, timeout)\n",
      "                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/subprocess.py\", line 2108, in _communicate\n",
      "    ready = selector.select(timeout)\n",
      "            ^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/selectors.py\", line 415, in select\n",
      "    fd_event_list = self._selector.poll(timeout)\n",
      "                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "KeyboardInterrupt\n"
     ]
    }
   ],
   "source": [
    "!pip install pillow fpdf2\n",
    "!pip install PyPDF2\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "image_paths = [\n",
    "    root + \"Figures/Figure 1/figure1.png\",\n",
    "    root + \"Figures/Figure 2/figure2.png\",\n",
    "    root + \"Figures/Figure 3/figure3.png\",\n",
    "    root + \"Figures/Figure 4/figure4.png\",\n",
    "    root + \"Figures/Figure 5/figure5.png\",\n",
    "    root + \"Figures/Figure 6/figure6.png\",\n",
    "    root + \"Figures/Supplementary figure 1/suppfig1.png\",\n",
    "    root + \"Figures/Supplementary figure 2/suppfig2.png\",\n",
    "    root + \"Figures/Supplementary figure 3/suppfig3.png\",\n",
    "    root + \"Figures/Supplementary figure 4/suppfig4.png\",\n",
    "    root + \"Figures/Supplementary figure 5/suppfig5.png\",\n",
    "    root + \"Figures/Supplementary figure 6/suppfig6.png\",\n",
    "    root + \"Figures/Supplementary figure 7/suppfig7.png\",\n",
    "    root + \"Figures/Supplementary figure 8/suppfig8.png\",\n",
    "    root + \"Figures/Supplementary figure 9/suppfig9.png\",\n",
    "    root + \"Figures/Supplementary figure 10/suppfig10.png\",\n",
    "    root + \"Figures/Supplementary figure 11/suppfig11.png\",\n",
    "    root + \"Figures/Supplementary figure 12/suppfig12.png\"\n",
    "\n",
    "]\n",
    "\n",
    "legends = [\n",
    "    \"\"\"Several regulators of MLP formation are conserved between fission and budding yeast, but cell-adhesion effector proteins are not.\n",
    "(A) Venn diagram of numbers of orthogroups, in which at least one gene in the orthogroup is annotated in GO-terms or phenotypic data related to MLP formation in either S. pombe, S. cerevisiae or C. albicans. Red numbers indicate cell-adhesion proteins. (B) Venn diagram of orthogroups that are conserved across the 3 species (i.e. the middle subset on A) asking whether they are also functionally conserved, i.e. contain at least one gene that is annotated in GO-terms or phenotypic data related to MLP formation in all three species. (C) Functionally conserved genes coloured by their broad functional category as indicated. GO: Gene Ontology, FYPO: Fission Yeast Phenotype Ontology, P: Phenotype annotations.\n",
    "\"\"\",\n",
    "\n",
    "\n",
    "    \"\"\"MLP formation in S. pombe natural isolates varies with nutrient conditions and is associated with decreased growth.\n",
    "(A) Left: Images of the initial observations on the standard laboratory strain JB50 and the natural isolate JB759 showing MLP formation of JB759 in EMM. Cells were grown at 32ºC for 2 days, shaking at 160 rpm. Right: Microscopy images at two magnifications of the JB759 strain grown in EMM for 2 days. (B) Scheme of the high-throughput adhesion assay used to assess MLP formation in S. pombe. (C) Strip plot of adhesion to agar across different conditions in the natural isolate library, with images of representative post-wash agar plates below. Each dot represents the mean adhesion value for a given strain in a specific condition. The red dashed line shows a cut-off for strong phenotypes (intensity after wash >0.2 times than before wash). Each condition was compared to rich media (YES) with the null hypothesis that they do not increase MLP formation. P-values were obtained using a one-sided permutation-based T-test and Bonferroni correction. Comparisons were marked not significant (ns.) where the null hypothesis could not be rejected at significance threshold 0.05. Strains around the edges were not taken into account for any statistical analysis (see Methods). The lab strain JB50 and natural isolate JB759 used in panel (A), as well as the lab strain, JB22, and the highly adhesive strains JB914 and JB953, are highlighted in conditions in which they are adhesive. (D) Histogram showing the number of unique strains forming MLPs in a given number of conditions. (E) Panel of light microscopy images for JB953 and JB914 in EMM and YES, showing representative images of MLP-formation. Scale bars are 100 µm. Rest of the microscopy images from this experiment are displayed in Supplementary Figure 4. Representative images of culture flasks are shown in Supplementary Figure 5.  (F) Scatter plot of mean cell densities (measured in normalised inverse pixel intensity) before and after washing. Each dot represents one strain in one condition, orange dots represent adhesive data points (ratio of before-wash to after-wash intensity >0.2, dashed red line) and purple dots represent non-adhesive data points. The histogram shows the distribution of cell densities before washing as a proxy for growth. The vertical dotted orange and purple lines mark the mean pre-wash densities for the two populations.\n",
    "\"\"\",\n",
    "\n",
    "\n",
    "       \"\"\"MLP formation in JB759 is driven by mbx2 expression and associated with a single-nucleotide deletion on chromosome II.\n",
    "(A) Scheme for the JB759xJB50 segregant library. Red and blue stripes indicate genomic recombination resulting from meiosis. SNPs for the strains inside the grey box were previously identified (49) and genome sequencing data for the strains in the light blue box were generated in this work (Methods). (B) (Left) Example plate scans of segregants grown on EMM or YES after washing, shown in viridis colourmap. (Right) Adhesion to agar of segregant strains on EMM (mean of 10 replicates) compared to YES (mean of 2 replicates), along with significance of the difference obtained using permutation-based T-test. (C) Correlations of mbx2 and flocculin gene expression with flocculation in EMM. Each dot represents a strain from the segregant library. (D) (Left) Box plot of adhesion measurements overlaid comparing adhesion measurements from control, GFP overexpression and mbx2 overexpression strains in both JB22 and JB50 backgrounds, as well as measurements for empty media (Methods). (Right) Representative images of the flocculating mbx2 overexpression strain in YES and EMM. Rest of the flask images are found in Supplementary Figure 7B. (E) (Left) Manhattan plot of QTL analysis results for flocculation in EMM. The red line shows the Bonferroni threshold, while the green dashed line shows the highest possible significance achievable using 20,000 permutations. (Middle) Candidate variant II:2316851 TA>T is associated with increased flocculation in EMM (filtering assay, mean of 3 replicates). (Right) Candidate variant II:2316851 TA>T is associated with increased adhesion on EMM (washing assay). P-values determined using a permutation-based T-test with 1E+6 permutations. (F) Microscopy image of srb11ΔCRISPR in EMM showing flocculation. The inset shows a close-up of the region highlighted in box “a”.\n",
    "\"\"\",\n",
    "\n",
    "\n",
    "\"\"\"Cdk8 kinase module deletions upregulate mbx2 in minimal media, but not in rich media, and lead to MLP formation.\n",
    "(A) Scheme showing how a single nucleotide deletion leads to frameshift and premature stop-codon in srb11. (B) Full Srb11 structure (green) compared with the truncated Srb11 structure (red) as predicted using Colabfold (92). Scheme on the right shows Srb11 in the context of the Cdk8 kinase module of the Mediator complex. The structure was sketched based on structural data (84), and functional roles were summarised based on (84,86). (C) Box plot showing adhesion values from Mediator gene deletion strains from the genome-wide prototrophic deletion library on EMM and YES as indicated. Each dot represents a replicate. P-value represents comparison of each strain against the least adhesive strain in the assay (med27Δ) using a T-test. (D) Differential expression analysis of segregant strains split on the II:2316851 TA>T single-nucleotide deletion. Fold-change values and P-values were obtained from DESeq2 (67). Flocculins and mbx2 are highlighted in red. (E) Overlap of upregulated genes in three CKM mutants based on our srb11 data and data from Linder et al. (68). (F) Simplified model for how CKM deletions lead to MLP formation. (G) Adhesion measurements for the srb11Δ strain obtained from the deletion collection, and its derived strain after mbx2 knock-out with CRISPR. Each dot represents a replicate measurement. Error bars represent the 95% confidence interval for the mean. (H) RT-qPCR showing mbx2 expression in JB22 (WT), srb10Δ::Kan, srb11Δ strains obtained from the genome-wide prototrophic deletion collection in EMM or YES. Expression of each gene is normalised relative to act1 expression. Height of each bar reflects the mean of three biological replicates which are indicated by dots.\n",
    "\"\"\",\n",
    "\n",
    "\n",
    "\"\"\"Deletion library screen identified 31 genes associated with MLP formation on minimal media.\n",
    "(A) Scatter plot of mean cell densities before and after washing. Each dot represents a deletion strain, and colours represent adhesive and non-adhesive strains as indicated. The red line represents the cut-off at the 95th percentile of adhesion values (Supp Fig 10B). The histogram shows the distribution of cell densities before washing as a proxy for growth. The dotted orange and purple lines mark the mean pre-wash intensity values for the two populations. The P-value was determined using a permutation-based T-test with 1E+5 permutations. (B) Bar plot showing fold enrichment for the top 10 most significantly enriched processes from AnGeLi (74). Blue circles indicate terms associated with ribosomes, and orange circles indicate terms associated with decreased growth. Gene sets were sorted based on P-values, with increasing colour intensity representing increasing -log10(P-value). The size of the circle at the end of each bar represents the size of the gene set. GEX: Gene expression; pop.: population. (C) Box plots of adhesion ratios obtained with the washing assay for 4 of the 31 verified hits (rpl15Δ, rpa12Δ, fkh2Δ, SPNCRNA.1234Δ) on EMM (light blue) vs YES (orange). Each dot is an independent observation. Remaining hits are shown in Supp Fig 11. (D) Venn diagram showing functional conservation of the genetically conserved hits from our screen. The hits SPAC607.02c, sre2, meu27, for3 are not shown as they are not genetically conserved (only present in S. pombe). Only fkh2 and res1 are also annotated as being involved in MLP formation in S. cerevisiae and C. albicans. Annotations are assigned using both GO and Phenotype data (GO+P). (E) Scatter plot of adhesion ratios and before-wash colony intensities overlaid by a histogram showing before-wash colony intensities of non-adhesive deletion strains, which were assayed in the middle 60 spots during the original screen. The green-shaded area marks strains that are above the lowest 5th percentile of before-wash colony intensity in the non-adhesive strains, a proxy for growth rate. The srb10/11Δ strains, highlighted in green, are the most adhesive strains among those with growth values above the lowest 5th percentile. (F) Line plot of growth measurements (OD600) from WT (JB22), empty media, srb11Δ, srb10Δ, mus7Δ, rpa12Δ, kgd2Δ and tlg2Δ strains in EMM and YES.\n",
    "\"\"\",\n",
    "\n",
    "\n",
    "\"\"\"Model for EMM-dependent MLP formation of CKM mutants.\n",
    "We propose that the CKM phosphorylates Mbx2 and targets it for degradation, based on similar observations in C. albicans (88) and S. cerevisiae (90). Additionally, mbx2 expression is repressed through an unknown mechanism in YES, as we observed using RT-qPCR. In minimal media, if members of the CKM are deleted, mbx2 becomes upregulated, triggering the expression of flocculin genes, which in turn cause MLP formation. Deletion of ribosomal genes also triggers MLP formation, although it is unclear whether this occurs through upregulation of mbx2 or directly through the flocculin genes. The red arrows indicate our main findings, and the red question marks highlight the main outstanding mechanistic questions. See main text for details.\n",
    "\"\"\",\n",
    "\n",
    "\n",
    "\"\"\"Plate-reader assay and filtering assay for flocculation are correlated.\n",
    "Scatter plot comparison of flocculation for the JB50-JB759 segregant library by measuring each strain with a conventional filtering assay (x-axis, n=3) and our high-throughput plate-reader assay which measures the CV of OD600 within each well (y-axis, n=5) (Methods). Each point represents the mean flocculation values for a strain. P-value was obtained with a one-tailed test on whether the Spearman correlation coefficient is larger than 0.\n",
    "\"\"\",\n",
    "\n",
    "\n",
    "\"\"\"Conservation of MLP-related orthogroups between S. pombe, S. cerevisiae and C. albicans.\n",
    "(A) Venn diagram representing genetic and (B) functional conservation of orthogroups between the three species using data only from GO-terms. (C) Highest alignment scores for either 25 protein sequences annotated as unique to S pombe or 50 random conserved sequences using either BLAST-P or Foldseek to identify similar proteins in the S. cerevisiae or C. albicans proteome (Methods). Each dot represents the highest score using the indicated search algorithm for a given protein. P-values were derived from Mann-Whitney U tests comparing top scores for conserved proteins with either all unique proteins (black) or just unique S. pombe flocculins (green).\n",
    "\"\"\",\n",
    "\n",
    "\n",
    "\"\"\"Further analysis on adhesion phenotypes of the natural isolates across different conditions.\n",
    "(A) Scatter plot comparison of flocculation and adhesion for the natural isolate library by measuring each strain with a high-throughput plate-reader assay (x-axis, n=11) and our high-throughput adhesion assay (y-axis, n=10). Each point represents the mean values for a strain with the standard error of the mean as error bars. Stars mark JB759 (red) and the lab strain JB50 (blue). (B) Elbow plot of the number of strains designated as “MLP-forming” in at least one condition (y-axis) given different cut-off thresholds (x-axis). Empirically we found that 0.2 sits in the elbow of the plot, suggesting that it is a reasonable cut-off. (C) Phosphate starvation induces MLP formation in the lab strain JB50. Images of the JB50 strain growing in 25ml flasks with 10ml media of either EMM (left) or EMM-P (right). Strains were first grown up on YES plates and then transferred to liquid cultures. There are clearly visible flocs forming in EMM-P. (D) Bar plot of log2FC in gene expression during 2 days of phosphate starvation for selected genes, including the transcription factor mbx2 and various flocculins. Data from (81). Red bars represent significant log2FC values.\n",
    "\"\"\",\n",
    "\n",
    "\n",
    "\"\"\"Microscope images of wild isolates exhibiting MLPs.\n",
    "Images of during exponential growth of JB50 (lab strain) and three wild isolates (JB759, JB914, and JB953) in media conditions under which they exhibit MLPs. Note: EMM-P images were taken with a 20x objective with 1x zoom, while EMM and YES images were taken with a 5x objective and 2x optical zoom. Images were scaled to be comparable. The images of JB914 and JB953 in EMM and YES are duplicated from this panel and also included in Figure 2. Scale bars are 100 µm.\n",
    "\"\"\",\n",
    "\n",
    "\"\"\"Culture flask images of wild isolates exhibiting MLPs.\n",
    "Images of during (A) exponential growth and (B) in saturated cultures after two days of JB50 (lab strain) and three wild isolates (JB759, JB914, and JB953) in media conditions under which they exhibit MLPs. \n",
    "\"\"\",\n",
    "\n",
    "\n",
    "\"\"\"MLP formation in S. pombe involves mbx2 and CKM components.\n",
    "(A) Scatter plot comparison of flocculation and adhesion for the JB50-JB759 library by measuring each strain with the filtering assay (x-axis, n=3) and the high-throughput adhesion assay (y-axis, n=10). Each point represents the mean values for a strain with the standard error of the mean as error bars. Stars mark the two parental strains JB759 (red) and JB50 (blue). There is a strong correlation between the two MLPs. (B) Scatter plot of expression levels of four flocculin genes against mbx2 expression in the JB50-JB759 segregant library (data from (49)), demonstrating a strong association. (C) (Left) Images showing flocculating cultures of srb10 and srb11 in 250ml flasks grown in EMM. (Right) Representative microscopy image of srb11∆::Kan (D) Barplot of adhesion values showing that the Kan deletion of srb11 found in the deletion library phenocopies the CRISPR deletion of srb11. Each dot is an independent measurement. Error bars represent the 95% confidence interval for the mean. (E) Box plot showing adhesion values from validation cohort of fresh Mediator gene deletion strains on EMM and YES as indicated. Each dot represents a replicate.\n",
    "\"\"\",\n",
    "\n",
    "\n",
    "\"\"\"Overexpression of mbx2 drives MLP formation. \n",
    "(A) Microscopy and (B) flask images illustrating the effects of mbx2 overexpression on cells grown in EMM or YES for both the h- (JB22) and h90 (JB50) backgrounds and the h90 JB50 background. Flask images were from overnight cultures, and microscope images were from overnight cultures after dilution into fresh media for two hours. The flask images of prRPL2102-mbx2 (JB22) are duplicated and also shown in Figure 3. Scale bars are 75 µm.\n",
    "\"\"\",\n",
    "\n",
    "\n",
    "\"\"\"Mbx2 drives MLP formation in the srb11Δ strain.\n",
    "(A) Venn diagram showing the intersection of three gene sets: (i) Genes upregulated in CKM mutants, meaning upregulated in the med12Δ and srb10Δ microarray data from (68) and in the srb11 truncated segregant RNA-seq data from (49) (Fig 4E), (ii) Genes upregulated after mbx2 overexpression obtained from (27) and (iii) Genes bound by Mbx2 obtained from (27). (B) Images showing non-flocculating cultures of three biological replicates of srb11/mbx2 double deletion strains. (C) Empirical cumulative distribution function of L1 distances between phenotype vectors for the srb11Δ strain and all other gene deletion strains for growth phenotypes across 131 different conditions. For each condition in each strain, a number of -1, 0, or 1 was assigned to denote sensitivity, neutrality and resistance, and the L1 distance is the sum of the absolute value of the differences across all phenotypes as measured in (96). The 10 deletion strains closest in their phenotypes to srb11Δ and med13Δ are highlighted on the plot. The similarity score of med13Δ is in the 21st percentile. (D) Scatter plot of fold changes comparing two different data sources for transcriptomic changes upon fkh2 deletion. Dots mark individual genes in the datasets, while the line of best fit is shown in red. This plot concludes that it is fair to use the two data sources to draw joint conclusions from them. (E) Bar plot showing log2 fold-changes in mbx2 expression levels from experiments in which fkh2 was deleted (red) and in which CKM subunits are deleted (blue). Selected fkh2 deletion experiments included microarray data from (86) and (97), while CKM subunit deletion experiments included med12Δ and srb10Δ microarray data from (68), fold change calculated by grouping RNA-seq data from (48) for segregants with the srb11 truncation vs those without the mutation (Fig 3D), and the RT-qPCR data collected in this work. (F) Bar plot showing log2 fold-changes in gsf1 expression in YES compared to EMM. There is no significant upregulation in either of the two replicates. Data and P-values were taken from (98).\n",
    "\"\"\",\n",
    "\n",
    "\n",
    "\"\"\"The srb11 variant is present in the JB759 and JB914 natural isolates, and segregates with MLP-formation across natural isolates closely related to those strains.\n",
    "(A) S. pombe reference genome visualised for each of the three chromosomes, with vertical lines marking variant loci (called in (45)) compared to the reference genome in at least one of the two strains. Variant loci with the same alternative allele for JB759 and JB914 are marked with purple, while variant loci where alleles do not match are marked with orange. (B) Inset showing the region surrounding the causal srb11 variant. The +/- 20Kb window is highlighted. Vertical lines indicate variant alleles as in A. (C) Dendrogram illustrating genomic similarity of natural isolates based on the +/- 20Kb window around the srb11 frameshift mutation. Red text indicates strains closely related to JB759 considering the whole genome (based on (45)), while blue text indicates strains related to JB914. Heatmap below indicates adhesion ratio in EMM.\n",
    "\"\"\",\n",
    "\n",
    "\n",
    "\"\"\"Deletion library screen for MLP formation on EMM.\n",
    "(A) Histogram showing cell densities, measured by inverse pixel intensity after 4 days of growth, for strains in our deletion library screen before washing. (B) Empirical cumulative distribution function (ECDF) of adhesion ratios from the deletion library screen, with red lines showing the cut-off for the 95th percentile. Strains above the cut-off were used for enrichment analysis.\n",
    "\"\"\",\n",
    "\n",
    "\"\"\"Verified hits from the deletion library screen for MLP formation, assayed both on EMM and YES.\n",
    "Box plots of adhesion ratios obtained with the washing assay for the 31 high-confidence hits on EMM (light blue) vs YES (orange). Each dot is an independent observation. Adhesion was generally specific to EMM, except in the case of sre2Δ. The images of rpl15Δ, rpa12Δ, fkh2Δ, SPNCRNA.1234Δ are duplicated from this panel and also included in Figure 5.\n",
    "\"\"\",\n",
    "\n",
    "\"\"\"MLP-forming hits belong in various phenotypic clusters, but generally exhibit slow growth.\n",
    "(A) Bar plot showing the number of MLP-forming (blue, left axis scale) and total (blue, right axis scale) strains in each broad phenotypic cluster of deletion strains identified in (96). (B) Histogram of mean fitness values, based on growth in solid EMM media, for each strain as defined in (96). MLP-forming hit strains are shown in purple, and all other strains are in yellow. Blue and red dotted lines show the thresholds beyond which strains were defined as slow- or fast-growing, respectively. \n",
    "\"\"\"\n",
    "\n",
    "]\n",
    "output_path = root + \"Supplementary material/figures.pdf\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "from PIL import Image\n",
    "Image.MAX_IMAGE_PIXELS = None \n",
    "from reportlab.lib.pagesizes import A4\n",
    "from reportlab.platypus import SimpleDocTemplate, Image as ReportLabImage, Paragraph, PageBreak\n",
    "from reportlab.lib.styles import ParagraphStyle\n",
    "from reportlab.lib.pagesizes import A4\n",
    "from reportlab.platypus import SimpleDocTemplate, Image as ReportLabImage\n",
    "import os\n",
    "import PyPDF2\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "def compress_pdf(pdf_path):\n",
    "    compressed_path = pdf_path.replace(\".pdf\", \"_compressed.pdf\")\n",
    "    with open(pdf_path, \"rb\") as pdf_file:\n",
    "        pdf_reader = PyPDF2.PdfReader(pdf_file)\n",
    "        pdf_writer = PyPDF2.PdfWriter()\n",
    "\n",
    "        for page in pdf_reader.pages:\n",
    "            pdf_writer.add_page(page)\n",
    "\n",
    "        with open(compressed_path, \"wb\") as compressed_pdf:\n",
    "            pdf_writer.write(compressed_pdf)\n",
    "\n",
    "    os.remove(pdf_path)  # Remove the original uncompressed PDF\n",
    "    os.rename(compressed_path, pdf_path)  # Rename the compressed PDF to original name\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "from reportlab.lib.pagesizes import A4, landscape\n",
    "from reportlab.platypus import SimpleDocTemplate, Image as ReportLabImage, Paragraph, PageBreak\n",
    "from reportlab.lib.styles import ParagraphStyle\n",
    "from reportlab.platypus.doctemplate import NextPageTemplate, PageTemplate\n",
    "from reportlab.platypus.frames import Frame\n",
    "from PIL import Image\n",
    "\n",
    "def create_pdf_from_images(image_paths, output_path, n_main_figs, n_supp_figs, legends, landscape_figures=None):\n",
    "    \"\"\"\n",
    "    Create a PDF from images with support for landscape orientation on specific figures.\n",
    "    \n",
    "    Parameters:\n",
    "    - image_paths: list of image file paths\n",
    "    - output_path: output PDF file path\n",
    "    - n_main_figs: number of main figures\n",
    "    - n_supp_figs: number of supplementary figures\n",
    "    - legends: list of legend text for each figure\n",
    "    - landscape_figures: list of figure indices (0-based) that should be in landscape mode\n",
    "    \"\"\"\n",
    "    if landscape_figures is None:\n",
    "        landscape_figures = []\n",
    "    \n",
    "    # Create document with templates for both orientations\n",
    "    doc = SimpleDocTemplate(output_path, pagesize=A4, rightMargin=10, leftMargin=10, topMargin=10, bottomMargin=10)\n",
    "    \n",
    "    # Define page templates\n",
    "    portrait_frame = Frame(10, 10, A4[0]-20, A4[1]-20, id='portrait')\n",
    "    landscape_frame = Frame(10, 10, landscape(A4)[0]-20, landscape(A4)[1]-20, id='landscape')\n",
    "    \n",
    "    portrait_template = PageTemplate(id='portrait', frames=[portrait_frame], pagesize=A4)\n",
    "    landscape_template = PageTemplate(id='landscape', frames=[landscape_frame], pagesize=landscape(A4))\n",
    "    \n",
    "    doc.addPageTemplates([portrait_template, landscape_template])\n",
    "    \n",
    "    story = []\n",
    "    \n",
    "    for i, (image_path, legend) in enumerate(zip(image_paths, legends)):\n",
    "        # Determine if this figure should be in landscape\n",
    "        is_landscape = i in landscape_figures\n",
    "        \n",
    "        # Switch to appropriate page template\n",
    "        if is_landscape:\n",
    "            story.append(NextPageTemplate('landscape'))\n",
    "            page_width = landscape(A4)[0] - 20  # Account for margins\n",
    "            page_height = landscape(A4)[1] - 20\n",
    "            max_width = page_width - 50  # Leave some margin for text\n",
    "            max_height = page_height - 150  # Leave space for caption\n",
    "        else:\n",
    "            story.append(NextPageTemplate('portrait'))\n",
    "            page_width = A4[0] - 20\n",
    "            page_height = A4[1] - 20\n",
    "            max_width = 550  # Original width for portrait\n",
    "            max_height = 700  # Original max height for portrait\n",
    "        \n",
    "        # Load and resize image\n",
    "        img = Image.open(image_path)\n",
    "        width, height = img.size\n",
    "        aspect_ratio = height / float(width)\n",
    "        \n",
    "        if is_landscape:\n",
    "            # For landscape pages, optimize sizing differently\n",
    "            # Start with width-based sizing\n",
    "            new_width = max_width\n",
    "            new_height = new_width * aspect_ratio\n",
    "            \n",
    "            # If height exceeds limit, scale down based on height\n",
    "            if new_height > max_height:\n",
    "                new_height = max_height\n",
    "                new_width = new_height / aspect_ratio\n",
    "        else:\n",
    "            # Original portrait sizing logic\n",
    "            new_width = max_width\n",
    "            new_height = new_width * aspect_ratio\n",
    "            \n",
    "            if new_height > 700:\n",
    "                new_height = 750\n",
    "                new_width = new_height / aspect_ratio\n",
    "            elif new_height > 600:\n",
    "                new_height = 650\n",
    "                new_width = new_height / aspect_ratio\n",
    "        \n",
    "        # Create ReportLab image\n",
    "        reportlab_img = ReportLabImage(image_path, width=new_width, height=new_height)\n",
    "        \n",
    "        # Create captions\n",
    "        if i < n_main_figs:\n",
    "            caption1 = f\"<strong>Figure {i + 1}</strong>\"\n",
    "            caption2 = f\"<strong>Figure {i + 1}</strong>: {legend}\"\n",
    "        elif i < n_main_figs + n_supp_figs:\n",
    "            caption1 = f\"<strong>Supplementary Figure {i - n_main_figs + 1}</strong>\"\n",
    "            caption2 = f\"<strong>Supplementary Figure {i - n_main_figs + 1}</strong>: {legend}\"\n",
    "        else:\n",
    "            caption1 = \"\"\n",
    "            caption2 = \"\"\n",
    "        \n",
    "        caption_style1 = ParagraphStyle(name='Caption', fontSize=15, leading=20)\n",
    "        caption_style2 = ParagraphStyle(name='Caption', fontSize=9, leading=16)\n",
    "        \n",
    "        # Add content to story\n",
    "        if caption1:\n",
    "            story.append(Paragraph(caption1, caption_style1))\n",
    "            story.append(Paragraph(\"<br/>\", caption_style2))\n",
    "        \n",
    "        story.append(reportlab_img)\n",
    "        \n",
    "        if caption2:\n",
    "            story.append(Paragraph(caption2, caption_style2))\n",
    "        \n",
    "        story.append(PageBreak())\n",
    "    \n",
    "    # Build the document\n",
    "    doc.build(story)\n",
    "    compress_pdf(output_path)\n",
    "\n",
    "# Example usage:\n",
    "# Figures 2 and 5 will be in landscape mode (0-based indexing: 1 and 4)\n",
    "# create_pdf_from_images(image_paths, output_path, 7, 7, legends, landscape_figures=[1, 4])\n",
    "\n",
    "\n",
    "create_pdf_from_images(image_paths, output_path, 6,12, legends, landscape_figures=[])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.4"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
