{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "import pandas as pd\n",
    "root = \"/Users/k23030440/S.-Pombe-MLPs/\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "#read in data\n",
    "\n",
    "#go terms for orthology analysis from: Supplementary material/annotation_orthogroup_genes.xlsx\n",
    "\n",
    "go_terms = pd.read_csv(root + \"internal data/go_terms_formatted.csv\",header=1)\n",
    "\n",
    "#omics data sources\n",
    "omics_sources = pd.read_excel(root + \"internal data/external data sources.xlsx\")\n",
    "\n",
    "#floc_cor_genes /Users/bencekover/Library/CloudStorage/OneDrive-Personal/MSci Bahler lab/S.-Pombe-MLPs - Github/Bence folder/Analysis of Clement-Ziza RNA-seq/filter_corr.csv\n",
    "floc_cor_genes = pd.read_csv(root + \"Bence folder/Analysis of Clement-Ziza RNA-seq/filter_corr.csv\", index_col=0)\n",
    "\n",
    "#structure_sequence_homology /Users/bencekover/Library/CloudStorage/OneDrive-Personal/MSci Bahler lab/S.-Pombe-MLPs - Github/Bence folder/Structure_and_Sequence_comparisons/final_analysis.csv\n",
    "structure_sequence_homology = pd.read_csv(root + \"Bence folder/Structure_and_Sequence_comparisons/final_analysis.csv\")\n",
    "\n",
    "#srb11 DE genes /Users/bencekover/Library/CloudStorage/OneDrive-Personal/MSci Bahler lab/S.-Pombe-MLPs - Github/Bence folder/Analysis of Clement-Ziza RNA-seq/DESeq2/DE_results.csv\n",
    "srb11_DE_genes = pd.read_csv(root + \"Bence folder/Analysis of Clement-Ziza RNA-seq/DESeq2/DE_results.csv\")\n",
    "#rename first col to Systematic name\n",
    "srb11_DE_genes.rename(columns={srb11_DE_genes.columns[0]:\"Systematic name\"}, inplace=True)\n",
    "\n",
    "\n",
    "#segregant genotypes/Users/bencekover/Library/CloudStorage/OneDrive-Personal/MSci Bahler lab/S.-Pombe-MLPs - Github/internal data/updated_genotype_matrix_final_pos.tsv\n",
    "segregant_genotypes = pd.read_csv(root + \"internal data/updated_genotype_matrix_final_pos.tsv\", sep=\"\\t\")\n",
    "\n",
    "\n",
    "wi_adh_across_media = pd.read_excel(root + \"Bence folder/Image processing - Adhesion assay/condition_plot_df.xlsx\",index_col=0)\n",
    "\n",
    "\n",
    "#primers\n",
    "primers=pd.read_excel(root + \"internal data/Primers.xlsx\")\n",
    "\n",
    "#screen_raw from root + \"internal data/Deletion screen raw results.csv\n",
    "screen_raw = pd.read_csv(root + \"internal data/Deletion screen raw results.csv\")\n",
    "\n",
    "#screen enrichment from root + Bence folder/Angeli/angeli/bioneer_angeli_results_recent.txt\n",
    "screen_enrichment = pd.read_csv(root + \"Bence folder/Angeli/angeli/bioneer_angeli_results_recent.txt\", sep=\"\\t\")\n",
    "\n",
    "#screen_final root + \"Bence folder/Bioneer screen/final_results_screen.csv\"\n",
    "screen_final = pd.read_csv(root + \"Bence folder/Bioneer screen/final_results_screen.csv\")\n",
    "\n",
    "\n",
    "#orthology, first col is index\n",
    "orthology = pd.read_csv(root + \"Bence folder/Orthologs/final_orthology_database.csv\", index_col=0)\n",
    "\n",
    "#mediator /Users/k23030440/S.-Pombe-MLPs - most recent/Bence folder/Bioneer screen/final_results_mediator.csv\n",
    "mediator = pd.read_csv(root + \"Bence folder/Bioneer screen/final_results_mediator.csv\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# Import the necessary libraries\n",
    "import pandas as pd\n",
    "\n",
    "# Define the file path for the output xlsx file\n",
    "output_file = root + \"Supplementary material/Supplementary table.xlsx\"\n",
    "\n",
    "# Create a Pandas Excel writer using the output file path\n",
    "writer = pd.ExcelWriter(output_file, engine='xlsxwriter')\n",
    "\n",
    "# Define a dictionary containing the dataframes and their corresponding sheet names\n",
    "dataframes = {\n",
    "    '1_go_fypo_terms': go_terms,\n",
    "    '2_structure_sequence_homology': structure_sequence_homology,\n",
    "    '3_omics_sources': omics_sources,\n",
    "    '4_segregant_genotypes': segregant_genotypes,\n",
    "    '5_orthology_relationships': orthology,\n",
    "    '6_nat_iso_adhesion_across_media' : wi_adh_across_media,\n",
    "    '7_floc_cor_genes': floc_cor_genes,\n",
    "    '8_mediator_results': mediator,\n",
    "    '9_srb11_DE_genes': srb11_DE_genes,\n",
    "    '10_Screen_raw_results': screen_raw,\n",
    "    '11_Screen_enrichment': screen_enrichment,\n",
    "    '12_Screen_final_results': screen_final\n",
    "}\n",
    "descriptions = [\n",
    "    \"#Summary of the GO terms, FYPO terms, and phenotype keywords used for the orthology analysis.\",\n",
    "    \"#Summary of results from the structure and sequence homology analysis in a tidy format, ready for plotting.\\n\",\n",
    "    \"#Summary of omics data sources used in the study.\",\n",
    "    \"#Table of segregant genotypes used in the QTL analysis.\",\n",
    "    \"\"\"#Table of orthology relatonships for MLP-related genes. Each combination of orthology relationships is summarized by one of the 7 \"orthopatterns\", representative of the 7 sets in the plotted Venn diagrams.\"\"\",\n",
    "    \"#Measurements of adhesion phenotypes in the natural isolates in a tidy format, assembled from the output of yeastmlp, ready for plotting.\",\n",
    "    \"#Correlation analysis between flocculation (filtering assay) and gene expression (Clement-Ziza RNA-seq) data.\",\n",
    "    \"#Adhesion ratios of mediator deletion strains, assembled from the output of yeastmlp.\",\n",
    "    \"#Differentially expressed genes in the srb11 deletion strain from the output of DESeq2.\",\n",
    "    \"#Raw results from the deletion screen as measured by yeastmlp.\",\n",
    "    \"\"\"#Enrichment analysis based on the top 5 percent of adhesive strains identified in the deletion screen.\"\"\",\n",
    "    \"#Final results from the 31 high-confidence strains, as measured by yeastmlp.\"]\n",
    "\n",
    "# Loop through the dictionary and write each dataframe to a separate sheet in the output file\n",
    "for sheet_name, df, desc in zip(dataframes.keys(), dataframes.values(), descriptions):\n",
    "    # Add a header to the sheet name\n",
    "    sheet_name = sheet_name\n",
    "    title = \"# Title: \" + sheet_name\n",
    "    correspondence = \"# Correspondence: bence.kover.19@ucl.ac.uk , b.heineike@ucl.ac.uk or j.bahler@ucl.ac.uk\"\n",
    "    \n",
    "    # Write the dataframe to the sheet\n",
    "    df.to_excel(writer, sheet_name=sheet_name, index=False, startrow=4)\n",
    "    \n",
    "    # Get the xlsxwriter workbook and worksheet objects\n",
    "    workbook = writer.book\n",
    "    worksheet = writer.sheets[sheet_name]\n",
    "    \n",
    "    # Add the title and correspondance to the sheet\n",
    "    worksheet.write(0, 0, title)\n",
    "    worksheet.write(1, 0, correspondence)\n",
    "    worksheet.write(2, 0, desc)\n",
    "\n",
    "\n",
    "\n",
    "# Save the Excel file\n",
    "writer.close()\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Now assemble .pdf of all images"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Defaulting to user installation because normal site-packages is not writeable\n",
      "Requirement already satisfied: pillow in /Users/k23030440/Library/Python/3.11/lib/python/site-packages (10.0.1)\n",
      "Requirement already satisfied: fpdf2 in /Users/k23030440/Library/Python/3.11/lib/python/site-packages (2.7.6)\n",
      "Requirement already satisfied: defusedxml in /Users/k23030440/Library/Python/3.11/lib/python/site-packages (from fpdf2) (0.7.1)\n",
      "Requirement already satisfied: fonttools>=4.34.0 in /Users/k23030440/Library/Python/3.11/lib/python/site-packages (from fpdf2) (4.43.1)\n",
      "Defaulting to user installation because normal site-packages is not writeable\n",
      "Requirement already satisfied: PyPDF2 in /Users/k23030440/Library/Python/3.11/lib/python/site-packages (3.0.1)\n"
     ]
    }
   ],
   "source": [
    "!pip install pillow fpdf2\n",
    "!pip install PyPDF2\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "image_paths = [\n",
    "    root + \"Figures/Figure 1/figure1.png\",\n",
    "    root + \"Figures/Figure 2/figure2.png\",\n",
    "    root + \"Figures/Figure 3/figure3.png\",\n",
    "    root + \"Figures/Figure 4/figure4.png\",\n",
    "    root + \"Figures/Figure 5/figure5.png\",\n",
    "    root + \"Figures/Figure 6/figure6.jpg\",\n",
    "    root + \"Figures/Supplementary figure 1./suppfig1.png\",\n",
    "    root + \"Figures/Supplementary figure 2./suppfig2.png\",\n",
    "    root + \"Figures/Supplementary figure 3./suppfig3.png\",\n",
    "    root + \"Figures/Supplementary figure 4./suppfig4.png\",\n",
    "    root + \"Figures/Supplementary figure 5./suppfig5.png\",\n",
    "    root + \"Figures/Supplementary figure 6./suppfig6.png\",\n",
    "    root + \"Figures/Supplementary figure 7./suppfig7.png\",\n",
    "    root + \"Figures/Supplementary figure 8./suppfig8.png\",\n",
    "    root + \"Figures/Supplementary figure 9./suppfig9.png\",\n",
    "    root + \"Figures/Supplementary figure 10./suppfig10.png\"\n",
    "\n",
    "]\n",
    "\n",
    "legends = [\n",
    "    \"\"\"Several regulators of MLP formation are conserved between fission and budding yeast, but cell-adhesion effector proteins are not. \n",
    "A: Venn diagram of orthogroups, in which at least one gene in the orthogroup is annotated in GO-terms or phenotypic data related to MLP formation in either S. pombe, S. cerevisiae or C. albicans. Red numbers indicate cell-adhesion proteins. B: Venn diagram of orthogroups that are conserved across the 3 species (i.e. the middle subset on plot A) asking whether they are also functionally conserved (contain at least one gene that is annotated in GO-terms or phenotypic data related to MLP formation in all three species). C: Table of functionally conserved genes coloured by their broad functional category. GO: Gene Ontology, FYPO: Fission Yeast Phenotype Ontology, P: Phenotype annotations. Venn diagrams were made using matplotlib-venn in Python (120).\"\"\",\n",
    "\n",
    "       \"\"\"MLP formation in S. pombe natural isolates varies with nutrient conditions and is associated with decreased growth. \n",
    "A: Left: Images of our initial observations on the S. pombe lab strain JB50 and the natural isolate JB759 showing MLP formation in the JB759 strain in EMM. Right: Microscopy images at two different magnifications of the JB759 strain grown in EMM for 2 days. B: Scheme of the high-throughput adhesion assay used to assess MLP formation in S. pombe. C: Strip-plot of adhesion to agar across different conditions in the natural isolate library, with images of representative post-wash agar plates below. Each dot represents the mean adhesion value for a given strain in a specific condition. Dashed line shows a cut-off (intensity after wash greater than 0.2 times before wash intensity) for strong phenotypes. P-values were obtained using a one-sided permutation-based T-test and Bonferroni-correction. Strains around the edges were not taken into account for any statistical analysis (see Methods). D: Histogram showing the number of unique strains forming MLPs in a given number of conditions. E: Scatterplot of mean cell densities before and after washing. Each dot represents one strain in one condition, orange dots represent adhesive data points (ratio of before wash to after wash intensity greater than 0.2, solid red line)  and purple dots represent non-adhesive data points. The histogram shows the distribution of cell densities before washing, as a proxy for growth. The dashed orange and purple lines mark the mean pre-wash density for the two populations.\"\"\",\n",
    "\n",
    "       \"\"\"In the JB50-JB759 segregant library, MLP formation on EMM is driven by mbx2 expression and is associated with a single-nucleotide deletion on chromosome II.\n",
    "A: Scheme for the segregant library. Red and blue stripes indicate recombination of the genome resulting from meiosis. SNPs for the strains inside the grey box were previously identified (46) and genome sequencing data for the strains in the light blue box were generated in this work (Methods). B: (Left) Example plates of segregants grown on EMM or YES are shown after washing shown in viridis colormap. (Right) Adhesion to agar of segregant strains on EMM (mean of 10 replicates) compared to YES (mean of 2 replicates). C: Correlation of mbx2 and flocculin genes with flocculation in EMM. Each dot represents a strain from the segregant library. D: Barplot with measurements overlaid comparing adhesion measurements from standard laboratory S. pombe strains JB22 and JB50, and the mbx2 overexpression strain generated in this work (Methods). E: (Left) Manhattan plot of our QTL analysis results for flocculation in EMM. Red line shows the Bonferroni-threshold, while the green dotted line shows the highest possible significance achievable using 20,000 permutations. (Middle, Right) Candidate variant is associated with both increased adhesion to agar (mean of 10 replicates) and increased flocculation in EMM (filtering assay, mean of 3 replicates). P-value determined using a permutation-based T-test with 1E+6 permutations.\"\"\",\n",
    "\"\"\"Cdk8 kinase module deletions upregulate mbx2 in EMM, but not in YES, and lead to MLP formation.\n",
    "A: Scheme showing how a single nucleotide deletion leads to frameshift and premature stop-codon in srb11. B: Full Srb11 structure (green) compared with the truncated Srb11 structure (red) as predicted using Colabfold (121). Scheme on the right shows Srb11 in the context of the Cdk8 kinase module of the Mediator complex. The structure was sketched based on structural data (78). C: Empirical cumulative distribution functions (ECDF) of adhesion values from Mediator gene deletion strains on EMM and YES. Each dot represents a replicate. Shaded regions show the 95% confidence intervals which were determined using 10,000 bootstrap samples. Percentiles where these confidence intervals do not overlap can be considered significantly different between any two strains. D: Differential expression analysis of segregant strains split on the II:2316851 TA>A single-nucleotide deletion. Fold-change values and P-values were obtained from DESeq2 (1). E: Overlap of upregulated genes in CKM mutants based on our data and data from Linder et al. (1). F: Simplified model for how CKM deletions lead to MLP formation. G: Adhesion measurements for the srb11 deletion strain obtained from the deletion collection, and its derived strain after mbx2 knock-out with CRISPR. Each dot represents a replicate. H: RT-qPCR showing mbx2 expression of srb10Δ::Kan, srb11Δ::Kan and WT strains in EMM or YES. Height of each bar reflects the mean of three biological replicates.\"\"\",\n",
    "\n",
    "\"\"\"Our deletion library screen identified 31 genes associated with MLP formation on EMM.\n",
    "A: Histogram showing cell densities of strains in our deletion library screen before washing. B: Barplot showing the fold enrichment of the top-10 most significantly enriched processes, with blue outline for terms associated with ribosomes and orange outline for terms associated with decreased growth. Terms were sorted based on P-values and increasing color intensity represents increasing -log10(P-value). Additionally, the size of the circle at the end of each bar represents the size of the gene set. C: Scatterplot of mean cell densities before and after washing. Each dot represents a deletion strain, and colors represent adhesive and non-adhesive strains. The red line represents the cut-off at the 95th percentile of adhesion values, also shown on Supp Fig 9. The histogram shows the distribution of cell densities before washing as a proxy for growth. The dashed orange and purple lines mark the mean pre-wash intensity value for the two populations. P-value determined using a permutation-based T-test with 1E+5 permutations. D: Empirical cumulative distribution function (ECDF) of adhesion ratios for 4 of the 31 verified hits on EMM (light blue) vs YES (orange). Shaded regions show the 95% confidence intervals which were determined using 10,000 bootstrap samples. Percentiles where these confidence intervals do not overlap can be considered significantly different between the two conditions. E: Venn diagram showing the functional conservation of the genetically conserved hits from our screen. Only fkh2 is annotated as being involved in MLP formation in all three species. Venn diagram was made using matplotlib-venn in Python (120). F: Scatterplot of adhesion ratios and before-wash colony intensities overlaid by a histogram showing before-wash colony intensities of non-adhesive deletion strains which were assayed in the middle 60 spots during the original screen. The green shaded area marks strains that are above the 5th percentile of colony intensities in the non-adhesive strains. The srb10/11Δ strains are highlighted with green, because they are the most adhesive strains from those with growth values above the 5th percentile.\n",
    "\"\"\",\n",
    "\n",
    "\"\"\"Model explaining EMM-dependent MLP formation of CKM mutants. \n",
    "Here we hypothesize that the CKM phosphorylates Mbx2 and targets it for degradation. Additionally, the mbx2 transcript is repressed through an unknown mechanism in YES. In minimal media, when members of the CKM are deleted, mbx2 becomes upregulated, triggering the expression of flocculin genes, which cause MLP formation. Additionally, deletion of ribosomal genes also triggers MLP formation, although it is unclear whether this occurs through upregulation of mbx2 or directly through the flocculin genes. The coloured boxes show our main pathway of interest. The red arrows show our main contributions to the field, while red questions marks show the main outstanding mechanistic questions.\n",
    "\"\"\",\n",
    "\n",
    "\"\"\"Scatterplot comparison of flocculation for the JB50-JB759 segregant library by measuring each strain with a conventional filtering assay (x-axis) and our high-throughput plate-reader assay which measures the CV of OD600 within each well (y-axis) (Methods). Each point represents the mean flocculation values for a strain.\n",
    "\"\"\",\n",
    "\n",
    "\"\"\"Conservation of MLP-related orthogroups between S. pombe, S. cerevisiae and C. albicans.\n",
    "A: Venn diagram representing genetic and functional conservation of orthogroups between the three species using data only from GO-terms. B: Comparison of “unique” and “conserved” genes using BLAST-P and Foldseek, see Methods. Each dot represents a given gene/protein. P-values were derived from Mann-Whitney U tests. Orange P-values were derived using the values of flocculin genes/proteins only, rather than all “unique” MLP-related genes/proteins.\n",
    "\"\"\",\n",
    "\"\"\"Further analysis on adhesion phenotypes of the natural isolates across different conditions.\n",
    "A: Scatterplot comparison of flocculation and adhesion for the natural isolate library by measuring each strain with a high-throughput plate-reader assay (x-axis) and our high-throughput adhesion assay (y-axis). Each point represents the mean values for a strain with the standard error of the mean as error bars. B: Elbow plot of the number of strains designated as “MLP-forming” in at least one condition (y-axis) given different cut-off thresholds (x-axis). Empirically we found that 0.2 sits in the elbow of the plot, suggesting that it is a reasonable cut-off. C: Phosphate starvation induces MLP formation in the lab strain JB50. Images of the JB50 strain growing in 25ml flasks with 10ml media of either EMM (left) or EMM-P (right). Strains were first grown up on YES plates and then transferred to liquid cultures. There are clearly visible flocs forming in EMM-P. E: Barplot of log2FC in mbx2 and flocculin expression at different timepoints during 2 days of phosphate starvation. Red bars represent significant log2FC values. Data taken from (75).\n",
    "\"\"\",\n",
    "\n",
    "\"\"\"Further analysis of the JB50-JB759 segregants and the role of mbx2.\n",
    "A:  Scatterplot comparison of flocculation and adhesion for the JB50-JB759 library by measuring each strain with a filtering assay (x-axis) and our high-throughput adhesion assay (y-axis). Each point represents the mean values for a strain with the standard error of the mean as error bars. There is a strong correlation between the two MLPs. B: Scatterplot of expression levels of four flocculin genes against mbx2 expression in the JB50-JB759 library (data from (46)), demonstrating a strong association. C: Results from the mbx2 overexpression strain on EMM. (Left, Middle) Images of flocculating mbx2OE strain in EMM. (Right) Images of mbx2OE strain after thiamine addition resulting in no flocculation.\n",
    "\"\"\",\n",
    "\n",
    "\"\"\"The srb10Δ and srb11Δ strains show MLP formation.\n",
    "A: Images showing flocculating cultures of srb10 and srb11 in 250ml flasks grown in EMM. B: Validation of srb11Δ::Kan and CRISPR deletion strains used in the paper. Results from gel electrophoresis after PCR. Red rectangles mark the expected bands validating the strains. C: (Left) Barplot of adhesion values showing that the CRISPR deletion of srb11 phenocopies the Kan deletion of srb11 found in the deletion library. Additionally, we note that the CRISPR strain exhibits significantly stronger adhesion, likely because the deletion library strain has accumulated suppression mutations in other genes. Each dot is an independent measurement. (Right) Representative microscopy images of srb11Δ::Kan (top) and srb11Δ (CRISPR) (bottom) strains.\n",
    "\"\"\",\n",
    "\n",
    "\"\"\"Mbx2 drives MLP formation in the srb10Δ strain.\n",
    "A: Venn diagram showing the intersection of three gene sets: (i) Genes upregulated in CKM mutants, meaning upregulated in the med12 and srb10 microarray data from (61) and in the srb11 truncated segregant RNA-seq data from (46), (ii) Genes upregulated after mbx2 overexpression obtained from (25) and (iii) Genes bound by Mbx2 obtained from (25). B: Images showing non-flocculating cultures of three biological repeats (JB1782, JB1783, JB1784) of srb11/mbx2 double delete strains.\n",
    "\"\"\",\n",
    "\n",
    "\"\"\"In terms of the sensitivity/resistance phenotypes measured in (89), srb11 is not any more similar to med13 than to an average deletion strain. Empirical cumulative distribution function of L1 distances (sum of differences across phenotypes; -1,0,1 are sensitive, neutral and resistant). Top 10 deletion strains closest in their phenotypes to srb11 are highlighted on the plot.\n",
    "\"\"\",\n",
    "\n",
    "\"\"\" Further analysis to deepen our understanding of the EMM-dependent upregulation of mbx2 in CKM deletions. Fkh2 is the only well-studied protein known to interact with the CKM in S. pombe, while gsf1 is the only known transcription factor to repress mbx2. A: Scatterplot of fold changes comparing two different data sources for transcriptomic changes upon fkh2 deletion. Each dot is a given gene in the dataset and the red line being a simple line of best fit. This plot concludes that it is fair to use the two data sources to draw joint conclusions from them. B: Barplot showing log2 fold-changes in mbx2 expression levels upon fkh2 deletion from (80) and (90) compared with expected fold changes from med12 and srb10 microarray data from (61) and in the srb11 truncated segregant RNA-seq data from (46) and our RT-qPCR data collected in this work. C: Barplot showing log2 fold-changes in gsf1 expression in YES compared to EMM. There is no significant upregulation in either of the two replicates. Data and P-values were taken from (1).\"\"\",\n",
    "\n",
    "\"\"\"Empirical cumulative distribution function (ECDF) of adhesion ratios from the deletion library screen, with red lines showing the cut-off for the 95th percentile. Strains above the cut-off were used for enrichment analysis.\n",
    "\"\"\",\n",
    "\n",
    "\"\"\"Empirical cumulative distribution function (ECDF) of adhesion ratios for the 31 verified hits on EMM (light blue) vs YES (orange). Shaded regions show the 95% confidence intervals which were determined using 10,000 bootstrap samples. Percentiles where these confidence intervals do not overlap can be considered significantly different between the two conditions. Note that there is only an overlap in the sre2Δ strain.\n",
    "\"\"\"]\n",
    "\n",
    "\n",
    "\n",
    "output_path = root + \"Supplementary material/figures.pdf\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "def compress_pdf(pdf_path):\n",
    "    compressed_path = pdf_path.replace(\".pdf\", \"_compressed.pdf\")\n",
    "    with open(pdf_path, \"rb\") as pdf_file:\n",
    "        pdf_reader = PyPDF2.PdfReader(pdf_file)\n",
    "        pdf_writer = PyPDF2.PdfWriter()\n",
    "\n",
    "        for page in pdf_reader.pages:\n",
    "            pdf_writer.add_page(page)\n",
    "\n",
    "        with open(compressed_path, \"wb\") as compressed_pdf:\n",
    "            pdf_writer.write(compressed_pdf)\n",
    "\n",
    "    os.remove(pdf_path)  # Remove the original uncompressed PDF\n",
    "    os.rename(compressed_path, pdf_path)  # Rename the compressed PDF to original name\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "from PIL import Image\n",
    "Image.MAX_IMAGE_PIXELS = None \n",
    "from reportlab.lib.pagesizes import A4\n",
    "from reportlab.platypus import SimpleDocTemplate, Image as ReportLabImage, Paragraph, PageBreak\n",
    "from reportlab.lib.styles import ParagraphStyle\n",
    "from reportlab.lib.pagesizes import A4\n",
    "from reportlab.platypus import SimpleDocTemplate, Image as ReportLabImage\n",
    "import os\n",
    "import PyPDF2\n",
    "\n",
    "def create_pdf_from_images(image_paths, output_path, n_main_figs, n_supp_figs, legends):\n",
    "    #thin margins\n",
    "    doc = SimpleDocTemplate(output_path, pagesize=A4, rightMargin=10, leftMargin=10, topMargin=10, bottomMargin=10)\n",
    "    story = []\n",
    "\n",
    "    for i, (image_path, legend) in enumerate(zip(image_paths, legends)):\n",
    "        img = Image.open(image_path)\n",
    "        width, height = img.size\n",
    "        aspect_ratio = height / float(width)\n",
    "        new_width = 550  # You can adjust this width as needed for A4 layout\n",
    "        new_height = new_width * aspect_ratio\n",
    "        if new_height > 700:\n",
    "            new_height = 809\n",
    "            new_width = new_height / aspect_ratio\n",
    "        elif new_height > 600:\n",
    "            new_height = 650\n",
    "            new_width = new_height / aspect_ratio\n",
    "        \n",
    "        img.thumbnail((new_width, new_height), Image.LANCZOS)\n",
    "\n",
    "        reportlab_img = ReportLabImage(image_path, width=new_width, height=new_height)\n",
    "\n",
    "        # Add a caption to the PDF document\n",
    "        if i < n_main_figs:\n",
    "            caption1 = f\"<strong>Figure {i + 1}</strong>\"\n",
    "            caption2 = f\"<strong>Figure {i + 1}</strong>: {legend}\"\n",
    "        elif i < n_main_figs + n_supp_figs:\n",
    "            caption1 = f\"<strong>Supplementary Figure {i - n_main_figs + 1}</strong>\"\n",
    "            caption2 = f\"<strong>Supplementary Figure {i - n_main_figs + 1}</strong>: {legend}\"\n",
    "        else:\n",
    "            caption = \"\"\n",
    "        caption_style1 = ParagraphStyle(name='Caption', fontSize=15, leading=20)\n",
    "        caption_style2 = ParagraphStyle(name='Caption', fontSize=9, leading=16)\n",
    "        story.append(Paragraph(caption1, caption_style1))\n",
    "        # paragraph\n",
    "        story.append(Paragraph(\"<br/>\", caption_style2))\n",
    "        story.append(reportlab_img)\n",
    "        story.append(Paragraph(caption2, caption_style2))\n",
    "        \n",
    "\n",
    "        story.append(PageBreak())\n",
    "\n",
    "    doc.build(story)\n",
    "    compress_pdf(output_path)\n",
    "    \n",
    "\n",
    "\n",
    "\n",
    "create_pdf_from_images(image_paths, output_path, 6, 10, legends)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.9.0 ('base')",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "2d0398177d7bb10ea33b6a5cd9d022d63e20680fcfd84762562c72ff35bdf44b"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
