{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#find snps\n",
    "#import data science libraries\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import allel\n",
    "import bokeh\n",
    "import iqplot\n",
    "import numpy as np\n",
    "import scipy.stats as st\n",
    "\n",
    "\n",
    "root = \"/Users/bencekover/Library/CloudStorage/OneDrive-Personal/MSci Bahler lab/S.-Pombe-MLPs/\"\n",
    "\n",
    "\n",
    "\n",
    "snps = pd.read_csv(root + 'Bence folder/Looking for variants in wild isolates/snp_hunting.csv')\n",
    "#washing phenotype /Users/bencekover/Library/CloudStorage/OneDrive-UniversityCollegeLondon/MSci Bahler lab/S.-Pombe-biofilm/Bence folder/QTL mapping/washing_phenotypes.csv\n",
    "wash_corr = pd.read_csv(root + 'Bence folder/Analysis of Clement-Ziza RNA-seq/wash_corr.csv')\n",
    "#full snp dataset /Users/bencekover/Library/CloudStorage/OneDrive-UniversityCollegeLondon/MSci Bahler lab/S.-Pombe-biofilm/Bence folder/QTL mapping/SupplementaryDataset_S7_genotype.tsv\n",
    "orthologs = pd.read_csv(root + 'Bence folder/Orthologs/final_orthology_database.csv')\n",
    "#import /Users/bencekover/Library/CloudStorage/OneDrive-UniversityCollegeLondon/MSci Bahler lab/S.-Pombe-biofilm/Bence folder/Phenotypes/phenotype_gwas.csv\n",
    "wash_data = pd.read_csv(root + 'Bence folder/Phenotypes/phenotype_yes_gwas.csv')\n",
    "\n",
    "snpsdata = '/Users/bencekover/Library/CloudStorage/OneDrive-UniversityCollegeLondon/MSci Bahler lab/Jeffares copy/Jeffares SNPs/ERZ289255/Spombe.2013-01-02.filt3c.nr57-final.snps.anno-snpeff3.cleaned4.vcf'\n",
    "indelsdata = '/Users/bencekover/Library/CloudStorage/OneDrive-UniversityCollegeLondon/MSci Bahler lab/Jeffares copy/Jeffares SNPs/ERZ289254/Spombe.indels.unique.nr57.final.annotated.2014-01-23.cleaned4.vcf.gz'\n",
    "\n",
    "\n",
    "snpsdata = allel.read_vcf(snpsdata,fields=['samples','variants/CHROM','variants/POS','calldata/GT'])\n",
    "indelsdata = allel.read_vcf(indelsdata,fields=['samples','variants/CHROM','variants/POS','calldata/GT'])\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#pandas read in \"/Users/bencekover/Library/CloudStorage/OneDrive-UniversityCollegeLondon/MSci Bahler lab/S.-Pombe-biofilm/external data/Pombe genome annotation.gff3\"\n",
    "import pandas as pd\n",
    "df = pd.read_csv(root + 'external data/Pombase files/Schizosaccharomyces_pombe_all_chromosomes.gff3', sep='\\t', header=1)\n",
    "#break last column into 2 at ;\n",
    "df[['a','b']] = df[\"ID=SPAC1002.01;Name=mrx11\"].str.split(';',expand=True)\n",
    "#remove column ID=SPAC1002.01;Name=mrx11\n",
    "df.drop(columns =[\"ID=SPAC1002.01;Name=mrx11\"], inplace = True)\n",
    "#remove ID= from column a and rename it to ID\n",
    "df['ID'] = df['a'].str.replace('ID=', '')\n",
    "#remove column a\n",
    "df.drop(columns =['a','PomBase',\".\",\"..1\"], inplace = True)\n",
    "#rename b to Parent\n",
    "df.rename(columns={'b':'Parent','gene':'object','I':'chromosome','1798347':\"start\",'1798835':\"end\",\"+\":\"strand\"}, inplace=True)\n",
    "#In parent remove anything before =\n",
    "df['Parent'] = df['Parent'].str.replace('Parent=', '')\n",
    "#or Name=\n",
    "df['Parent'] = df['Parent'].str.replace('Name=', '')\n",
    "#create an aempty column called detail\n",
    "df['detail'] = \"\"\n",
    "#split ID column at :, and if there is something after put it in detail column\n",
    "df['detail'] = df['ID'].str.split(':', expand=True)[1]\n",
    "#split ID column at :, and if there is something before put it in ID column\n",
    "df['ID'] = df['ID'].str.split(':', expand=True)[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#find a subset of df, where snps[position] is +- 1000 of df[start] or df[end]. Create a result df, and say which snps were there\n",
    "#suppress warnings\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "\n",
    "result = pd.DataFrame()\n",
    "for i in range(len(snps)):\n",
    "    for j in range(len(df)):\n",
    "        if snps['position'][i] >= df['start'][j] - 30 and snps['position'][i] <= df['end'][j] + 30 :\n",
    "            #and if chromsome is chromosome_2 for df chromsome II etc\n",
    "            if snps['chromosome'][i] == df['chromosome'][j]:\n",
    "                result = result.append({'position': snps['position'][i], 'ID': df['ID'][j], 'Parent': df['Parent'][j], 'object': df['object'][j], 'chromosome': df['chromosome'][j], 'start': df['start'][j], 'end': df['end'][j], 'strand': df['strand'][j], 'detail': df['detail'][j], \"p_value\": snps[\"p_value\"][i], \"type\":snps[\"type\"][i]}, ignore_index=True)\n",
    "result\n",
    "\n",
    "#if the last two characters are .1, remove them\n",
    "for i in range(len(result)):\n",
    "    if result['ID'][i][-2:] == \".1\":\n",
    "        result['ID'][i] = result['ID'][i][:-2]\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#match ID with systematic name in wash_corr and add columns gene\tcorrelation\tp-value\n",
    "#create np.arrays with NAs\n",
    "corrs = np.array([np.nan]*len(result))\n",
    "pvals = np.array([np.nan]*len(result))\n",
    "genes = np.array([np.nan]*len(result), dtype=object)\n",
    "for i in range(len(result)):\n",
    "    for j in range(len(wash_corr)):\n",
    "        if result['ID'][i] == wash_corr['systematic_names'][j]:\n",
    "            corrs[i] = wash_corr['correlation'][j]\n",
    "            pvals[i] = wash_corr['p-value'][j]\n",
    "            genes[i] = wash_corr['gene'][j]\n",
    "\n",
    "result['correlation'] = corrs\n",
    "result['p_value_corr'] = pvals\n",
    "result['gene'] = genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "result_final__ = result.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#iterate through result_final__. If ID is found in orthologs, then add the albicans_name(s)cerevisiae_name(s) to the df\n",
    "from numpy import dtype\n",
    "\n",
    "albicans_names = np.array([np.nan]*len(result_final__),dtype=object)\n",
    "cerevisiae_names = np.array([np.nan]*len(result_final__),dtype=object)\n",
    "for i in range(len(result_final__)):\n",
    "    for j in range(len(orthologs)):\n",
    "        if result_final__['ID'][i] in orthologs['pombe_name(s)'][j]:\n",
    "            albicans_names[i] = orthologs['albicans_name(s)'][j]\n",
    "            cerevisiae_names[i] = orthologs['cerevisiae_name(s)'][j]\n",
    "#add to df\n",
    "result_final__['albicans_name(s)'] = albicans_names\n",
    "result_final__['cerevisiae_name(s)'] = cerevisiae_names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#iterate through positions in result_final__  and find the corresponding alleles in snps. Then add the two values in X968_allele\tY0036_allele to the df in two new columns at the start \n",
    "#add two columns to result_final__\n",
    "result_final__['allele_1'] = np.nan\n",
    "result_final__['allele_2'] = np.nan\n",
    "for i in range(len(result_final__)):\n",
    "    for j in range(len(snps)):\n",
    "        if result_final__['position'][i] == snps['position'][j]:\n",
    "            result_final__['allele_1'][i] = snps['allele1'][j]\n",
    "            result_final__['allele_2'][i] = snps['allele2'][j]\n",
    "result_final__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys, re\n",
    "\n",
    "f=open(root + 'external data/Pombase files/Schizosaccharomyces_pombe_all_chromosomes.fa','r')\n",
    "lines=f.readlines()\n",
    "\n",
    "hre=re.compile('>(\\S+)')\n",
    "lre=re.compile('^(\\S+)$')\n",
    "\n",
    "regions={}\n",
    "\n",
    "for line in lines:\n",
    "        outh = hre.search(line)\n",
    "        if outh:\n",
    "                id=outh.group(1)\n",
    "        else:\n",
    "                outl=lre.search(line)\n",
    "                if(id in regions.keys()):\n",
    "                        regions[id] += outl.group(1)\n",
    "                else:\n",
    "                        regions[id]  =outl.group(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#result_final___ only keep rows where snp is between start and end\n",
    "result_final___ = result_final__.copy()\n",
    "result_final___ = result_final___[(result_final___['position'] >= result_final___['start']) & (result_final___['position'] <= result_final___['end'])]\n",
    "\n",
    "gene_exons = df.copy()\n",
    "#only keep rows where object is CDS\n",
    "gene_exons = gene_exons[gene_exons['object'] == 'CDS']\n",
    "#only keep rows where ID is in result_final__['ID']\n",
    "gene_exons_final = pd.DataFrame()\n",
    "for i in np.unique(result_final___['ID']):\n",
    "      #keep ones where ID is i or i.1\n",
    "      df_subset = gene_exons[(gene_exons['ID'] == i) | (gene_exons['ID'] == i + \".1\")]\n",
    "      gene_exons_final = gene_exons_final.append(df_subset)\n",
    "#reset index\n",
    "gene_exons_final.reset_index(inplace=True)\n",
    "#make a column exon_number\n",
    "gene_exons_final['exon_number'] = np.nan\n",
    "#for those where ID is multiple times, number the exons by increasing start if strand is +, or decreasing start if strand is -\n",
    "for i in np.unique(gene_exons_final['ID']):\n",
    "    df_subset = gene_exons_final[gene_exons_final['ID'] == i]\n",
    "    if df_subset['strand'].values[0]== '+':\n",
    "        df_subset = df_subset.sort_values(by=['start'], ascending=True)\n",
    "    if df_subset['strand'].values[0] == '-':\n",
    "        df_subset = df_subset.sort_values(by=['end'], ascending=False)\n",
    "    df_subset['exon_number'] = np.arange(1, len(df_subset) + 1)\n",
    "    gene_exons_final[gene_exons_final['ID'] == i] = df_subset\n",
    "\n",
    "\n",
    "#REMOVE entries wehre ID contains SPNCRNA\n",
    "gene_exons_final = gene_exons_final[~gene_exons_final['ID'].str.contains('SPNCRNA')]\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "#make a column called real_start\n",
    "gene_exons_final['old_start'] = gene_exons_final['start']\n",
    "gene_exons_final['old_end'] = gene_exons_final['end']\n",
    "#for rows where strand is + look at the load the bases +- 5 around the start from regions\n",
    "for i in np.unique(gene_exons_final['ID']):\n",
    "    df_subset = gene_exons_final[gene_exons_final['ID'] == i]\n",
    "    df_subset = df_subset[df_subset['exon_number'] == 1]\n",
    "    if df_subset['strand'].values[0]== '+':\n",
    "        seq = regions[df_subset['chromosome'].values[0]][int(df_subset['start'].values[0]-5):int(df_subset['start'].values[0]+5)]\n",
    "        #where does ATG occur in seq\n",
    "        real_start = seq.find('ATG')-4\n",
    "        real_start = df_subset['start'] + real_start\n",
    "        #for the row in gene_exons_final where ID is i and exon_number is 1, change the start to real_start\n",
    "        gene_exons_final.loc[(gene_exons_final['ID'] == i) & (gene_exons_final['exon_number'] == 1), 'start'] = real_start\n",
    "\n",
    "\n",
    "        \n",
    "\n",
    "\n",
    "\n",
    "    else:\n",
    "        seq_rev = regions[df_subset['chromosome'].values[0]][int(df_subset['end'].values[0]-5):int(df_subset['end'].values[0]+5)]\n",
    "        real_end = seq_rev.find('CAT')+2-4\n",
    "        real_end = df_subset['end'] + real_end\n",
    "        #for the row in gene_exons_final where ID is i and exon_number is 1, change the end to real_end\n",
    "        gene_exons_final.loc[(gene_exons_final['ID'] == i) & (gene_exons_final['exon_number'] == 1), 'end'] = real_end\n",
    "\n",
    "\n",
    "    df_subset = gene_exons_final[gene_exons_final['ID'] == i]\n",
    "    highest_exon = df_subset['exon_number'].max()\n",
    "    df_subset = df_subset[df_subset['exon_number'] == highest_exon]\n",
    "    if df_subset['strand'].values[0]== '+':\n",
    "\n",
    "        seq2 = regions[df_subset['chromosome'].values[0]][int(df_subset['end'].values[0]-5):int(df_subset['end'].values[0]+5)]\n",
    "        real_end = seq2.find('TAA')+2-4\n",
    "        real_end = df_subset['end'] + real_end\n",
    "        #for the row in gene_exons_final where ID is i and exon_number is highest_exon, change the end to real_end\n",
    "        gene_exons_final.loc[(gene_exons_final['ID'] == i) & (gene_exons_final['exon_number'] == highest_exon), 'end'] = real_end\n",
    "    else:\n",
    "        seq2_rev = regions[df_subset['chromosome'].values[0]][int(df_subset['start'].values[0]-5):int(df_subset['start'].values[0]+5)]\n",
    "        real_start = seq2_rev.find('TCA')+2-4\n",
    "        real_start = df_subset['start'] + real_start\n",
    "        #for the row in gene_exons_final where ID is i and exon_number is highest_exon, change the start to real_start\n",
    "        gene_exons_final.loc[(gene_exons_final['ID'] == i) & (gene_exons_final['exon_number'] == highest_exon), 'start'] = real_start\n",
    "\n",
    "\n",
    "    \n",
    "    \n",
    "\n",
    "\n",
    "\n",
    "#for each row calculate the length of the exon\n",
    "gene_exons_final['exon_length'] = abs(gene_exons_final['end'] - gene_exons_final['start'])+1\n",
    "\n",
    "#remove rows where ID is NaN\n",
    "gene_exons_final = gene_exons_final[~gene_exons_final['ID'].isna()]\n",
    "#for each row calculate the length of the CDS until that exon starts\n",
    "gene_exons_final['CDS_length_so_far'] = np.nan\n",
    "for i in np.unique(gene_exons_final['ID']):\n",
    "    df_subset = gene_exons_final[gene_exons_final['ID'] == i]\n",
    "    #order by exon_number\n",
    "    df_subset = df_subset.sort_values(by=['exon_number'])\n",
    "    \n",
    "    df_subset['CDS_length_so_far'] = np.append(np.array(0),df_subset['exon_length'].cumsum())[:-1]\n",
    "    gene_exons_final[gene_exons_final['ID'] == i] = df_subset\n",
    "#add total cds length\n",
    "gene_exons_final['total_CDS_length'] = np.nan\n",
    "for i in np.unique(gene_exons_final['ID']):\n",
    "    df_subset = gene_exons_final[gene_exons_final['ID'] == i]\n",
    "    df_subset['total_CDS_length'] = sum(df_subset['exon_length'])\n",
    "    gene_exons_final[gene_exons_final['ID'] == i] = df_subset\n",
    "\n",
    "gene_exons_final\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#iterate through positions in result_final__ and look up the chromosome and position in regions, and find the corresponding letter +/- 5 bp in regions. Add it to a new column\n",
    "result_final__['reference_seq']=np.nan\n",
    "for i in range(len(result_final__)):\n",
    "    for j in range(len(regions)):\n",
    "        if result_final__['chromosome'][i] == list(regions.keys())[j]:\n",
    "            #add | | characters left and right of tletter at position-1\n",
    "            \n",
    "            result_final__['reference_seq'][i] = regions[list(regions.keys())[j]][result_final__['position'][i]-4:result_final__['position'][i]-1] + ' |' + regions[list(regions.keys())[j]][result_final__['position'][i]-1] + '| ' + regions[list(regions.keys())[j]][result_final__['position'][i]:result_final__['position'][i]+3] \n",
    "\n",
    "#if strand is -, reverse reference_seq\n",
    "for i in range(len(result_final__)):\n",
    "    if result_final__['strand'][i] == '-':\n",
    "        result_final__['reference_seq'][i] = result_final__['reference_seq'][i][::-1]\n",
    "        #also flip the alleles\n",
    "        result_final__['allele_1'][i] = result_final__['allele_1'][i][::-1]\n",
    "        result_final__['allele_2'][i] = result_final__['allele_2'][i][::-1]\n",
    "#also create reference_seq_ without the | | characters\n",
    "result_final__['reference_seq_'] = result_final__['reference_seq'].str.replace('|', '')\n",
    "result_final__['reference_seq_'] = result_final__['reference_seq_'].str.replace(' ', '')\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "result_final__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#create column, whole gene as codons\n",
    "result_final__['wt_gene'] = np.nan\n",
    "#iterate through result_final__ positions. If object is CDS, then look up the sequence from start to end in regions.\n",
    "#add the sequence to the df\n",
    "for i in range(len(result_final__)):\n",
    "    \n",
    "    if result_final__['object'][i] == 'CDS':\n",
    "    \n",
    "        start = result_final__['start'][i]\n",
    "        end = result_final__['end'][i]\n",
    "        chrom = result_final__['chromosome'][i]\n",
    "        pos = result_final__['position'][i]\n",
    "        gene = regions[chrom][start-1:end]\n",
    "        #gene_exons_final where start is the same as start in result_final__\n",
    "        exon = gene_exons_final[(gene_exons_final['start'] < start+20) & (gene_exons_final['end'] > end-20)] \n",
    "        #if exon is empty, then print gene name\n",
    "        if exon.empty:\n",
    "            print(result_final__['ID'][i])\n",
    "            continue\n",
    "        exon_len_so_far=0\n",
    "        exon_len_so_far = exon['CDS_length_so_far'].values[0]\n",
    "        exon_len_so_far=exon_len_so_far%3\n",
    "        if pos > start and pos < end:\n",
    "        \n",
    "            if result_final__['strand'][i] == '-':\n",
    "                \n",
    "                gene = gene[::-1]\n",
    "            \n",
    "                gene = ''.join(gene)\n",
    "            gene = 'A'*int(exon_len_so_far) + gene\n",
    "            #write the gene as triplets of letters like AAA, ATC, etc\n",
    "            #if divisible by 3\n",
    "            if len(gene)%3 != 0:\n",
    "                gene = gene[:-(len(gene)%3)]\n",
    "            gene = [gene[i:i+3] for i in range(0, len(gene), 3)]\n",
    "                #only triplets near the position\n",
    "            if result_final__['strand'][i] == '+':\n",
    "                gene = gene[(pos-(start-int(exon_len_so_far)))//3-2:(pos-end)//3+25]\n",
    "            else: \n",
    "                gene = gene[(end-pos)//3-2:(start-int(exon_len_so_far)-pos)//3+25]\n",
    "            \n",
    "                \n",
    "                \n",
    "\n",
    "\n",
    "            result_final__['wt_gene'][i] = gene\n",
    "\n",
    "result_final__['mutant allele'] = np.nan\n",
    "for i in range(len(result_final__)):\n",
    "    if result_final__['object'][i] == 'CDS':\n",
    "    \n",
    "        start = result_final__['start'][i]\n",
    "        end = result_final__['end'][i]\n",
    "        chrom = result_final__['chromosome'][i]\n",
    "        pos = result_final__['position'][i]\n",
    "        gene = regions[chrom][start-1:end]\n",
    "         #gene_exons_final where start is the same as start in result_final__\n",
    "        exon = gene_exons_final[(gene_exons_final['start'] < start+20) & (gene_exons_final['end'] > end-20)] \n",
    "        #if exon is empty, then print gene name\n",
    "        if exon.empty:\n",
    "            print(result_final__['ID'][i])\n",
    "            continue\n",
    "        exon_len_so_far=0\n",
    "        exon_len_so_far = exon['CDS_length_so_far'].values[0]\n",
    "        exon_len_so_far=exon_len_so_far%3\n",
    "        if pos > start and pos < end:\n",
    "        \n",
    "            if result_final__['strand'][i] == '+':\n",
    "                #decide which allele to use\n",
    "                #check \n",
    "                #check how long the y0036 allele is \n",
    "                x968_len = len(result_final__['allele_1'][i])\n",
    "                y0036_len = len(result_final__['allele_2'][i])\n",
    "                \n",
    "                mutation = result_final__['allele_2'][i]\n",
    "               \n",
    "\n",
    "                gene = gene[:pos-start] + mutation + gene[pos-start+x968_len:]\n",
    "            elif result_final__['strand'][i] == '-':\n",
    "                x968_len = len(result_final__['allele_1'][i])\n",
    "                y0036_len = len(result_final__['allele_2'][i])\n",
    "                gene = gene[::-1]\n",
    "                gene = ''.join(gene)\n",
    "                mutation = result_final__['allele_2'][i]\n",
    "                \n",
    "                gene = gene[:end-pos] + mutation + gene[end-pos+x968_len:]\n",
    "                \n",
    "            gene = 'A'*int(exon_len_so_far) + gene\n",
    "            if len(gene)%3 != 0:\n",
    "                gene = gene[:-(len(gene)%3)]\n",
    "            gene = [gene[k:k+3] for k in range(0, len(gene), 3)]\n",
    "                #only triplets near the position\n",
    "            if result_final__['strand'][i] == '+':\n",
    "                gene = gene[(pos-(start-int(exon_len_so_far)))//3-2:(pos-end)//3+25]\n",
    "            else:\n",
    "                gene = gene[(end-pos)//3-2:(start-int(exon_len_so_far)-pos)//3+25]\n",
    "            result_final__[\"mutant allele\"][i] = gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#show all rows of results_final__\n",
    "pd.set_option('display.max_rows', None)\n",
    "#order and by p_value_corr\n",
    "result_final__ = result_final__.sort_values(by=['p_value_corr'],ascending=True)\n",
    "result_final__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#CreATE A DF WHERE WHOLE GENE IS NOT NAN\n",
    "stop_codon_df = result_final__[result_final__['wt_gene'].notna()]\n",
    "stop_codon_df[\"stop_codon_predicted\"] = np.nan\n",
    "#iterate through mutant allele, and see whether it contains\n",
    "#reset index\n",
    "stop_codon_df = stop_codon_df.reset_index(drop=True)\n",
    "\n",
    "for i in range(len(stop_codon_df)):\n",
    "    for j in range(len(stop_codon_df['mutant allele'][i])):\n",
    "        if stop_codon_df[\"strand\"][i] == '+':\n",
    "            if stop_codon_df['mutant allele'][i][j] == 'TAA' or stop_codon_df['mutant allele'][i][j] == 'TAG' or stop_codon_df['mutant allele'][i][j] == 'TGA':\n",
    "                stop_codon_df['stop_codon_predicted'][i] = 'yes'\n",
    "        else:\n",
    "            if stop_codon_df['mutant allele'][i][j] == 'ATT' or stop_codon_df['mutant allele'][i][j] == 'ATC' or stop_codon_df['mutant allele'][i][j] == 'ACT':\n",
    "                stop_codon_df['stop_codon_predicted'][i] = 'yes'\n",
    "#show which entries have a stop codon predicted\n",
    "stop_codon_df[stop_codon_df['stop_codon_predicted'] == 'yes']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#in stop_codon_df create a column \"possible frameshift\" and fill it with nan. Then loop through X96_allele and Y0036_allele and check if the length is different. If yes and the difference is not divisible by 3, fill the column with \"yes\". Then show the entries where the column is \"yes\"\n",
    "frameshift_df = stop_codon_df.copy()\n",
    "frameshift_df['possible frameshift'] = np.nan\n",
    "for i in range(len(frameshift_df)):\n",
    "    if len(frameshift_df['allele_1'][i]) != len(frameshift_df['allele_2'][i]):\n",
    "        if (len(frameshift_df['allele_1'][i]) - len(frameshift_df['allele_2'][i]))%3 != 0:\n",
    "            frameshift_df['possible frameshift'][i] = 'yes'\n",
    "frameshift_df[frameshift_df['possible frameshift'] == 'yes']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#import a codon table as a dictionary\n",
    "codon_table = {\n",
    "    'TTT': 'F', 'TTC': 'F', 'TTA': 'L', 'TTG': 'L',\n",
    "    'TCT': 'S', 'TCC': 'S', 'TCA': 'S', 'TCG': 'S',\n",
    "    'TAT': 'Y', 'TAC': 'Y', 'TAA': '*', 'TAG': '*',\n",
    "    'TGT': 'C', 'TGC': 'C', 'TGA': '*', 'TGG': 'W',\n",
    "    'CTT': 'L', 'CTC': 'L', 'CTA': 'L', 'CTG': 'L',\n",
    "    'CCT': 'P', 'CCC': 'P', 'CCA': 'P', 'CCG': 'P',\n",
    "    'CAT': 'H', 'CAC': 'H', 'CAA': 'Q', 'CAG': 'Q', \n",
    "    'CGT': 'R', 'CGC': 'R', 'CGA': 'R', 'CGG': 'R',\n",
    "    'ATT': 'I', 'ATC': 'I', 'ATA': 'I', 'ATG': 'M',\n",
    "    'ACT': 'T', 'ACC': 'T', 'ACA': 'T', 'ACG': 'T',\n",
    "    'AAT': 'N', 'AAC': 'N', 'AAA': 'K', 'AAG': 'K',\n",
    "    'AGT': 'S', 'AGC': 'S', 'AGA': 'R', 'AGG': 'R',\n",
    "    'GTT': 'V', 'GTC': 'V', 'GTA': 'V', 'GTG': 'V',\n",
    "    'GCT': 'A', 'GCC': 'A', 'GCA': 'A', 'GCG': 'A',\n",
    "    'GAT': 'D', 'GAC': 'D', 'GAA': 'E', 'GAG': 'E',\n",
    "    'GGT': 'G', 'GGC': 'G', 'GGA': 'G', 'GGG': 'G',\n",
    "}\n",
    "#iterate through frameshift_df and trnslate the lists of triplets found in wt_gene and mutant allele\n",
    "frameshift_df['wt_gene_translated'] = np.nan\n",
    "frameshift_df['mutant allele_translated'] = np.nan\n",
    "for i in range(len(frameshift_df)):\n",
    "    #if strand is -, then change a to t and so on\n",
    "    if frameshift_df['strand'][i] == '-':\n",
    "        temp_wt_allele = frameshift_df['wt_gene'][i]\n",
    "        temp_mut_allele = frameshift_df['mutant allele'][i]\n",
    "        for j in range(len(temp_wt_allele)):\n",
    "            temp_wt_allele[j] = temp_wt_allele[j].replace('A','t')\n",
    "            temp_wt_allele[j] = temp_wt_allele[j].replace('T','a')\n",
    "            temp_wt_allele[j] = temp_wt_allele[j].replace('C','g')\n",
    "            temp_wt_allele[j] = temp_wt_allele[j].replace('G','c')\n",
    "            temp_wt_allele[j] = temp_wt_allele[j].upper()\n",
    "        for j in range(len(temp_mut_allele)):\n",
    "            temp_mut_allele[j] = temp_mut_allele[j].replace('A','t')\n",
    "            temp_mut_allele[j] = temp_mut_allele[j].replace('T','a')\n",
    "            temp_mut_allele[j] = temp_mut_allele[j].replace('C','g')\n",
    "            temp_mut_allele[j] = temp_mut_allele[j].replace('G','c')\n",
    "            temp_mut_allele[j] = temp_mut_allele[j].upper()\n",
    "        frameshift_df['wt_gene_translated'][i] = ''.join([codon_table[codon] for codon in temp_wt_allele])\n",
    "        frameshift_df['mutant allele_translated'][i] = ''.join([codon_table[codon] for codon in temp_mut_allele])\n",
    "    else:\n",
    "        frameshift_df['wt_gene_translated'][i] = ''.join([codon_table[codon] for codon in frameshift_df['wt_gene'][i]])\n",
    "        frameshift_df['mutant allele_translated'][i] = ''.join([codon_table[codon] for codon in frameshift_df['mutant allele'][i]])\n",
    "\n",
    "        \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "frameshift_df['non-synonymous mutation'] = \"no\"\n",
    "for i in range(len(frameshift_df)):\n",
    "      if frameshift_df[\"wt_gene_translated\"][i] != frameshift_df[\"mutant allele_translated\"][i]:\n",
    "            frameshift_df['non-synonymous mutation'][i] = 'yes'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#show for entries where ID doesnt start wiht SPNCRNA\n",
    "#show all columns\n",
    "pd.set_option('display.max_columns', None)\n",
    "#show me rows where the difference is not a D TO N mutation\n",
    "#add 2 columns: aa wt to aa mut. iterate through frameshift_df and add the third value from each wt_gene_translated and mutant allele_translated to aa_wt and aa_mut\n",
    "#reset index\n",
    "#make column aa_wt and aa_mut\n",
    "frameshift_df['aa_wt'] = np.nan\n",
    "frameshift_df['aa_mut'] = np.nan\n",
    "frameshift_df.reset_index(inplace = True,drop=True)\n",
    "for i in range(len(frameshift_df)):\n",
    "      if frameshift_df['wt_gene_translated'][i] == \"\":\n",
    "            continue\n",
    "      frameshift_df['aa_wt'][i] = frameshift_df['wt_gene_translated'][i][3]\n",
    "      frameshift_df['aa_mut'][i] = frameshift_df['mutant allele_translated'][i][3]\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>position</th>\n",
       "      <th>ID</th>\n",
       "      <th>Parent</th>\n",
       "      <th>object</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>detail</th>\n",
       "      <th>p_value</th>\n",
       "      <th>type</th>\n",
       "      <th>correlation</th>\n",
       "      <th>p_value_corr</th>\n",
       "      <th>gene</th>\n",
       "      <th>albicans_name(s)</th>\n",
       "      <th>cerevisiae_name(s)</th>\n",
       "      <th>allele_1</th>\n",
       "      <th>allele_2</th>\n",
       "      <th>reference_seq</th>\n",
       "      <th>reference_seq_</th>\n",
       "      <th>wt_gene</th>\n",
       "      <th>mutant allele</th>\n",
       "      <th>stop_codon_predicted</th>\n",
       "      <th>possible frameshift</th>\n",
       "      <th>wt_gene_translated</th>\n",
       "      <th>mutant allele_translated</th>\n",
       "      <th>non-synonymous mutation</th>\n",
       "      <th>aa_wt</th>\n",
       "      <th>aa_mut</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>4020241</td>\n",
       "      <td>SPAC27E2.07</td>\n",
       "      <td>SPAC27E2.07.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>4020082</td>\n",
       "      <td>4021251</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.05</td>\n",
       "      <td>indel</td>\n",
       "      <td>-0.535368</td>\n",
       "      <td>0.000126</td>\n",
       "      <td>pvg2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>GAAGAATTGGGCGAAA</td>\n",
       "      <td>GTT |G| AAG</td>\n",
       "      <td>GTTGAAG</td>\n",
       "      <td>[ACC, GTT, GAA, GAA, TTG, GGC, GAA, AAA, GAA, ...</td>\n",
       "      <td>[ACC, GTT, GAA, GAA, TTG, GGC, GAA, AAA, GAA, ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TVEELGEKELELSKTVTIDIYRKAMCM</td>\n",
       "      <td>TVEELGEKELGEKELELSKTVTIDIYRKAMCM</td>\n",
       "      <td>yes</td>\n",
       "      <td>E</td>\n",
       "      <td>E</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133</th>\n",
       "      <td>2955003</td>\n",
       "      <td>SPAPB24D3.05c</td>\n",
       "      <td>SPAPB24D3.05c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>2954983</td>\n",
       "      <td>2955219</td>\n",
       "      <td>-</td>\n",
       "      <td>pseudogenic_exon</td>\n",
       "      <td>0.05</td>\n",
       "      <td>indel</td>\n",
       "      <td>-0.157143</td>\n",
       "      <td>0.296965</td>\n",
       "      <td>SPAPB24D3.05c</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>AC</td>\n",
       "      <td>TAA |C| TTC</td>\n",
       "      <td>TAACTTC</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169</th>\n",
       "      <td>5025947</td>\n",
       "      <td>SPAC22E12.04</td>\n",
       "      <td>SPAC22E12.04.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>5025279</td>\n",
       "      <td>5026036</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.05</td>\n",
       "      <td>indel</td>\n",
       "      <td>-0.098119</td>\n",
       "      <td>0.516513</td>\n",
       "      <td>ccs1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ACCTCCTGCTGCTCTCAAGAAAAGAAAAGCTGTTGCACATCAGAAAAAC</td>\n",
       "      <td>A</td>\n",
       "      <td>AAA |A| CCT</td>\n",
       "      <td>AAAACCT</td>\n",
       "      <td>[GAA, AAA, ACC, TCC, TGC, TGC, TCT, CAA, GAA, ...</td>\n",
       "      <td>[GAA, AAA, ACC, TCG, TGC, TGT, TCT, AAC, GGA, ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EKTSCCSQEKKSCCTSEKPSCCSNGKS</td>\n",
       "      <td>EKTSCCSNGKS</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170</th>\n",
       "      <td>5025947</td>\n",
       "      <td>SPAC22E12.04</td>\n",
       "      <td>SPAC22E12.04.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>5025279</td>\n",
       "      <td>5026036</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.05</td>\n",
       "      <td>indel</td>\n",
       "      <td>-0.098119</td>\n",
       "      <td>0.516513</td>\n",
       "      <td>ccs1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ACCTCCTGCTGCTCTCAAGAAAAGAAAAGCTGTTGCACATCAGAAAAAC</td>\n",
       "      <td>A</td>\n",
       "      <td>AAA |A| CCT</td>\n",
       "      <td>AAAACCT</td>\n",
       "      <td>[GAA, AAA, ACC, TCC, TGC, TGC, TCT, CAA, GAA, ...</td>\n",
       "      <td>[GAA, AAA, ACC, TCG, TGC, TGT, TCT, AAC, GGA, ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EKTSCCSQEKKSCCTSEKPSCCSNGKS</td>\n",
       "      <td>EKTSCCSNGKS</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202</th>\n",
       "      <td>2316851</td>\n",
       "      <td>SPBC12D12.06</td>\n",
       "      <td>SPBC12D12.06.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2316770</td>\n",
       "      <td>2317136</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.05</td>\n",
       "      <td>indel</td>\n",
       "      <td>-0.063583</td>\n",
       "      <td>0.674632</td>\n",
       "      <td>srb11</td>\n",
       "      <td>['C3_05740C_A']</td>\n",
       "      <td>['YNL025C']</td>\n",
       "      <td>TA</td>\n",
       "      <td>T</td>\n",
       "      <td>TGC |T| AAA</td>\n",
       "      <td>TGCTAAA</td>\n",
       "      <td>[TAC, ATG, CTA, AAA, AAA, AAT, GAG, GAA, AAG, ...</td>\n",
       "      <td>[TAC, ATG, CTA, AAA, AAA, ATG, AGG, AAA, AGG, ...</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>YMLKKNEEKGFSLEALVATCIYLSCKV</td>\n",
       "      <td>YMLKKMRKRVFLWRR*WLHAFIYHAKW</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>223</th>\n",
       "      <td>758540</td>\n",
       "      <td>SPAC4G8.03c</td>\n",
       "      <td>SPAC4G8.03c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>758208</td>\n",
       "      <td>760550</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.05</td>\n",
       "      <td>indel</td>\n",
       "      <td>0.030281</td>\n",
       "      <td>0.841663</td>\n",
       "      <td>puf5</td>\n",
       "      <td>['C1_13570W_A' 'C3_00730W_A']</td>\n",
       "      <td>['YGL014W' 'YGL178W']</td>\n",
       "      <td>TG</td>\n",
       "      <td>G</td>\n",
       "      <td>GTT |G| GAA</td>\n",
       "      <td>GTTGGAA</td>\n",
       "      <td>[ATT, CAA, CCT, TAC, ACA, GAG, CGA, ATC, ATT, ...</td>\n",
       "      <td>[ATT, CAA, CTT, ACA, CAG, AGC, GAA, TCA, TTG, ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>IQPYTERIIEKFFGNICKLSLQKFSSN</td>\n",
       "      <td>IQLTQSESLRNFLVTFASFLFKNSAQ</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     position             ID           Parent object chromosome    start  \\\n",
       "5     4020241    SPAC27E2.07    SPAC27E2.07.1    CDS          I  4020082   \n",
       "133   2955003  SPAPB24D3.05c  SPAPB24D3.05c.1    CDS          I  2954983   \n",
       "169   5025947   SPAC22E12.04   SPAC22E12.04.1    CDS          I  5025279   \n",
       "170   5025947   SPAC22E12.04   SPAC22E12.04.1    CDS          I  5025279   \n",
       "202   2316851   SPBC12D12.06   SPBC12D12.06.1    CDS         II  2316770   \n",
       "223    758540    SPAC4G8.03c    SPAC4G8.03c.1    CDS          I   758208   \n",
       "\n",
       "         end strand            detail  p_value   type  correlation  \\\n",
       "5    4021251      +              exon     0.05  indel    -0.535368   \n",
       "133  2955219      -  pseudogenic_exon     0.05  indel    -0.157143   \n",
       "169  5026036      +              exon     0.05  indel    -0.098119   \n",
       "170  5026036      +              exon     0.05  indel    -0.098119   \n",
       "202  2317136      +              exon     0.05  indel    -0.063583   \n",
       "223   760550      -              exon     0.05  indel     0.030281   \n",
       "\n",
       "     p_value_corr           gene               albicans_name(s)  \\\n",
       "5        0.000126           pvg2                            NaN   \n",
       "133      0.296965  SPAPB24D3.05c                            NaN   \n",
       "169      0.516513           ccs1                            NaN   \n",
       "170      0.516513           ccs1                            NaN   \n",
       "202      0.674632          srb11                ['C3_05740C_A']   \n",
       "223      0.841663           puf5  ['C1_13570W_A' 'C3_00730W_A']   \n",
       "\n",
       "        cerevisiae_name(s)                                           allele_1  \\\n",
       "5                      NaN                                                  G   \n",
       "133                    NaN                                                  C   \n",
       "169                    NaN  ACCTCCTGCTGCTCTCAAGAAAAGAAAAGCTGTTGCACATCAGAAAAAC   \n",
       "170                    NaN  ACCTCCTGCTGCTCTCAAGAAAAGAAAAGCTGTTGCACATCAGAAAAAC   \n",
       "202            ['YNL025C']                                                 TA   \n",
       "223  ['YGL014W' 'YGL178W']                                                 TG   \n",
       "\n",
       "             allele_2 reference_seq reference_seq_  \\\n",
       "5    GAAGAATTGGGCGAAA   GTT |G| AAG        GTTGAAG   \n",
       "133                AC   TAA |C| TTC        TAACTTC   \n",
       "169                 A   AAA |A| CCT        AAAACCT   \n",
       "170                 A   AAA |A| CCT        AAAACCT   \n",
       "202                 T   TGC |T| AAA        TGCTAAA   \n",
       "223                 G   GTT |G| GAA        GTTGGAA   \n",
       "\n",
       "                                               wt_gene  \\\n",
       "5    [ACC, GTT, GAA, GAA, TTG, GGC, GAA, AAA, GAA, ...   \n",
       "133                                                 []   \n",
       "169  [GAA, AAA, ACC, TCC, TGC, TGC, TCT, CAA, GAA, ...   \n",
       "170  [GAA, AAA, ACC, TCC, TGC, TGC, TCT, CAA, GAA, ...   \n",
       "202  [TAC, ATG, CTA, AAA, AAA, AAT, GAG, GAA, AAG, ...   \n",
       "223  [ATT, CAA, CCT, TAC, ACA, GAG, CGA, ATC, ATT, ...   \n",
       "\n",
       "                                         mutant allele stop_codon_predicted  \\\n",
       "5    [ACC, GTT, GAA, GAA, TTG, GGC, GAA, AAA, GAA, ...                  NaN   \n",
       "133                                                 []                  NaN   \n",
       "169  [GAA, AAA, ACC, TCG, TGC, TGT, TCT, AAC, GGA, ...                  NaN   \n",
       "170  [GAA, AAA, ACC, TCG, TGC, TGT, TCT, AAC, GGA, ...                  NaN   \n",
       "202  [TAC, ATG, CTA, AAA, AAA, ATG, AGG, AAA, AGG, ...                  yes   \n",
       "223  [ATT, CAA, CTT, ACA, CAG, AGC, GAA, TCA, TTG, ...                  NaN   \n",
       "\n",
       "    possible frameshift           wt_gene_translated  \\\n",
       "5                   NaN  TVEELGEKELELSKTVTIDIYRKAMCM   \n",
       "133                 yes                                \n",
       "169                 NaN  EKTSCCSQEKKSCCTSEKPSCCSNGKS   \n",
       "170                 NaN  EKTSCCSQEKKSCCTSEKPSCCSNGKS   \n",
       "202                 yes  YMLKKNEEKGFSLEALVATCIYLSCKV   \n",
       "223                 yes  IQPYTERIIEKFFGNICKLSLQKFSSN   \n",
       "\n",
       "             mutant allele_translated non-synonymous mutation aa_wt aa_mut  \n",
       "5    TVEELGEKELGEKELELSKTVTIDIYRKAMCM                     yes     E      E  \n",
       "133                                                        no   NaN    NaN  \n",
       "169                       EKTSCCSNGKS                     yes   NaN    NaN  \n",
       "170                       EKTSCCSNGKS                     yes   NaN    NaN  \n",
       "202       YMLKKMRKRVFLWRR*WLHAFIYHAKW                     yes   NaN    NaN  \n",
       "223        IQLTQSESLRNFLVTFASFLFKNSAQ                     yes   NaN    NaN  "
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#frameshift_df stop_codon_predicted ==yes\n",
    "frameshift_df[frameshift_df['type'] == 'indel']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#iterate through snps unnamed: 0\n",
    "plots=[]\n",
    "for i in range(len(snps_final)):\n",
    "    #if type is snp\n",
    "    if snps_final[\"type\"][i] == \"snp\":\n",
    "        loc=snps_final[\"position\"][i]\n",
    "        strains = snpsdata[\"samples\"][snpsdata['calldata/GT'][np.where(snpsdata['variants/POS'] == float(loc))[0][0],:,0] == 1]\n",
    "    else:\n",
    "        loc=snps[\"position\"][i]\n",
    "        strains =  indelsdata[\"samples\"][indelsdata['calldata/GT'][np.where( indelsdata['variants/POS']== float(loc))[0][0],:,0] == 1]\n",
    "\n",
    "    #find strains ratio value of strains in wash_data\n",
    "    mutated_vals = wash_data[wash_data[\"strain\"].isin(strains)][\"ratio\"]\n",
    "    wt_vals = wash_data[~wash_data[\"strain\"].isin(strains)][\"ratio\"]\n",
    "    #find p value\n",
    "    p = st.ttest_ind(mutated_vals,wt_vals)[1]\n",
    "    #make plot if p_value < 0.05\n",
    "    if p < 0.01:\n",
    "        #create df with values and mut or wt\n",
    "        df = pd.DataFrame({\"Ratio\":np.concatenate((wt_vals,mutated_vals)),\"Mutated\":np.concatenate((np.repeat(\"Ref. allele\",len(wt_vals)),np.repeat(\"Mutant\",len(mutated_vals))))})\n",
    "        genes = np.unique(result_final__.loc[result_final__[\"position\"] == int(loc)][\"gene\"].dropna().values)\n",
    "        #put genes in title but make it look not like a list, but keep commas\n",
    "        plots.append(iqplot.stripbox(df,\"Ratio\",\"Mutated\",title=str(snps[\"position\"][i]) + \"  genes: \" + str(genes).replace(\"[\",\"\").replace(\"]\",\"\").replace(\"'\",\"\").replace(\" \",\", \"),width=600,height=300, q_axis=\"y\"))\n",
    "#allow bokeh output in notebook\n",
    "bokeh.io.output_notebook()\n",
    "bokeh.io.show(bokeh.layouts.gridplot(plots,ncols=5))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.9.12 ('base')",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.0"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "2d0398177d7bb10ea33b6a5cd9d022d63e20680fcfd84762562c72ff35bdf44b"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
