{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "#find snps\n",
    "#import data science libraries\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import allel\n",
    "import bokeh\n",
    "import iqplot\n",
    "import numpy as np\n",
    "import scipy.stats as st\n",
    "#suppress warnings\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "root = \"/Users/k23030440/\"\n",
    "snps_file = root + 'Jeffares copy/Jeffares SNPs/ERZ289255/Spombe.2013-01-02.filt3c.nr57-final.snps.anno-snpeff3.cleaned4.vcf'\n",
    "indels_file = root + 'Jeffares copy/Jeffares SNPs/ERZ289254/Spombe.indels.unique.nr57.final.annotated.2014-01-23.cleaned4.vcf.gz'\n",
    "final_results_screen = pd.read_csv(root + \"S.-Pombe-MLPs - most recent/Bence folder/Bioneer screen/final_results_screen.csv\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "#genome coordinates for genes\n",
    "df = pd.read_csv(root + 'S.-Pombe-MLPs - most recent/external data/Pombase files/Schizosaccharomyces_pombe_all_chromosomes.gff3', sep='\\t', header=1)\n",
    "#break last column into 2 at ;\n",
    "df[['a','b']] = df[\"ID=SPAC1002.01;Name=mrx11\"].str.split(';',expand=True)\n",
    "#remove column ID=SPAC1002.01;Name=mrx11\n",
    "df.drop(columns =[\"ID=SPAC1002.01;Name=mrx11\"], inplace = True)\n",
    "#remove ID= from column a and rename it to ID\n",
    "df['ID'] = df['a'].str.replace('ID=', '')\n",
    "#remove column a\n",
    "df.drop(columns =['a','PomBase',\".\",\"..1\"], inplace = True)\n",
    "#rename b to Parent\n",
    "df.rename(columns={'b':'Parent','gene':'object','I':'chromosome','1798347':\"start\",'1798835':\"end\",\"+\":\"strand\"}, inplace=True)\n",
    "#In parent remove anything before =\n",
    "df['Parent'] = df['Parent'].str.replace('Parent=', '')\n",
    "#or Name=\n",
    "df['Parent'] = df['Parent'].str.replace('Name=', '')\n",
    "#create an aempty column called detail\n",
    "df['detail'] = \"\"\n",
    "#split ID column at :, and if there is something after put it in detail column\n",
    "df['detail'] = df['ID'].str.split(':', expand=True)[1]\n",
    "#split ID column at :, and if there is something before put it in ID column\n",
    "df['ID'] = df['ID'].str.split(':', expand=True)[0]#load in vcf file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(36, 8)"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = df.loc[df[\"object\"] == \"gene\"]\n",
    "mediator = [\"SPAC23H4.17c\",\"SPBC12D12.06\",\"SPAC688.08\",\"SPAC589.02c\",\"SPAC5D6.05\",\"SPCC1450.05c\"]\n",
    "\n",
    "df = df.loc[df[\"ID\"].isin(mediator) | df[\"ID\"].isin(final_results_screen[\"strain\"].values) | df[\"Parent\"].isin(final_results_screen[\"strain\"].values )]\n",
    "df.reset_index(drop=True, inplace=True)\n",
    "df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "snps_data = allel.read_vcf(snps_file,fields=['samples','variants/CHROM','variants/POS','calldata/GT','variants/ALT','variants/REF'])\n",
    "samples = snps_data['samples']\n",
    "interesting_strains = samples\n",
    "interesting_strains = [\"JB914\",\"JB871\",\"JB1117\", \"JB872\", \"JB934\", \"JB759\", \"JB848\",\"JB953\",\"JB869\"]\n",
    "\n",
    "#finding the index of the interesting strains\n",
    "interesting_strains_index = []\n",
    "for i in range(len(samples)):\n",
    "      if samples[i] in interesting_strains:\n",
    "            interesting_strains_index.append(i)\n",
    "\n",
    "calls= snps_data['calldata/GT'][:,:,0]\n",
    "chromosomes = snps_data['variants/CHROM']\n",
    "positions = snps_data['variants/POS']\n",
    "alt = snps_data['variants/ALT']\n",
    "ref= snps_data['variants/REF']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "#iterate through positions\n",
    "#results is a df with columns: alleles\tn\tsum_x\ty_transpose_x\tbeta\tstandard_error\tt_stat\tp_value\ttype\tallele1\tallele2\tchromosome\tposition\n",
    "results = pd.DataFrame(columns=[\"strain\",\"ID\",\"chromosome\",\"position\",\"gene\",\"type\",\"allele1\",\"allele2\",\"p_value\"])\n",
    "#rough gene coordinates\n",
    "#take mean of start and end\n",
    "rough_gene_coordinates = (df[\"start\"]+df[\"end\"])/2\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "for i in range(len(positions)):\n",
    "      #if positions[i] is not near to any of the rough_gene_coordinates  +-10000\n",
    "      \n",
    "      if np.isclose(positions[i],rough_gene_coordinates,atol=20000).any() == False:\n",
    "            \n",
    "            continue\n",
    "      for j in range(len(df)):\n",
    "            #for + strand\n",
    "            if positions[i]<(df[\"end\"][j]+200) and positions[i]>(df[\"start\"][j]-200) and chromosomes[i]==df[\"chromosome\"][j]:\n",
    "                  #iterate through interesting strains\n",
    "                  for k in range(len(interesting_strains_index)):\n",
    "                        #take the value of the genotype of the interesting strain at the position\n",
    "                        call_value = calls[i][interesting_strains_index[k]]\n",
    "                        #if this value is 0 or 1, and there is only max 2 strains with this value, then add the strain name to the detail column\n",
    "                        if call_value == 0 or call_value == 1 or call_value == 2:\n",
    "                              if sum(calls[i]==call_value) <= 5:\n",
    "                                    \n",
    "                                    #results = results.append({\"strain\":interesting_strains[k],\"ID\":df[\"ID\"][j],\"chromosome\":chromosomes[i],\"position\":positions[i], \"gene\":df[\"Parent\"][j],\"type\":\"snp\",\"allele1\":ref[i],\"allele2\":alt[i][call_value-1],\"p_value\":0.05},ignore_index=True)\n",
    "                                    results = pd.concat([results,pd.DataFrame({\"strain\":interesting_strains[k],\"ID\":df[\"ID\"][j],\"chromosome\":chromosomes[i],\"position\":positions[i], \"gene\":df[\"Parent\"][j],\"type\":\"snp\",\"allele1\":ref[i],\"allele2\":alt[i][call_value-1],\"p_value\":0.05},index=[0])],ignore_index=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "indels_data = allel.read_vcf(indels_file,fields=['samples','variants/CHROM','variants/POS','calldata/GT','variants/ALT','variants/REF'])\n",
    "\n",
    "calls= indels_data['calldata/GT'][:,:,0]\n",
    "chromosomes = indels_data['variants/CHROM']\n",
    "positions = indels_data['variants/POS']\n",
    "alt = indels_data['variants/ALT']\n",
    "ref= indels_data['variants/REF']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "for i in range(len(positions)):\n",
    "   \n",
    "      if np.isclose(positions[i],rough_gene_coordinates,atol=20000).any() == False:\n",
    "            continue\n",
    "      for j in range(len(df)):\n",
    "            #for + strand\n",
    "            if positions[i]<(df[\"end\"][j]+200) and positions[i]>(df[\"start\"][j]-200) and chromosomes[i]==df[\"chromosome\"][j]:\n",
    "                  #iterate through interesting strains\n",
    "                  for k in range(len(interesting_strains_index)):\n",
    "                        #take the value of the genotype of the interesting strain at the position\n",
    "                        call_value = calls[i][interesting_strains_index[k]]\n",
    "                        #if this value is 0 or 1, and there is only max 3 strains with this value, then add the strain name to the detail column\n",
    "                        if call_value == 0 or call_value == 1 or call_value == 2 or call_value == 3:\n",
    "                              if sum(calls[i]==call_value) <= 3:\n",
    "                                    \n",
    "                                    results = pd.concat([results,pd.DataFrame({\"strain\":interesting_strains[k],\"ID\":df[\"ID\"][j],\"chromosome\":chromosomes[i],\"position\":positions[i], \"gene\":df[\"Parent\"][j],\"type\":\"indel\",\"allele1\":ref[i],\"allele2\":alt[i][call_value-1],\"p_value\":0.05},index=[0])],ignore_index=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>strain</th>\n",
       "      <th>ID</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>gene</th>\n",
       "      <th>type</th>\n",
       "      <th>allele1</th>\n",
       "      <th>allele2</th>\n",
       "      <th>p_value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>JB872</td>\n",
       "      <td>SPAC1D4.06c</td>\n",
       "      <td>I</td>\n",
       "      <td>648673</td>\n",
       "      <td>csk1</td>\n",
       "      <td>snp</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>JB953</td>\n",
       "      <td>SPAC1D4.06c</td>\n",
       "      <td>I</td>\n",
       "      <td>648875</td>\n",
       "      <td>csk1</td>\n",
       "      <td>snp</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>JB872</td>\n",
       "      <td>SPAC1D4.06c</td>\n",
       "      <td>I</td>\n",
       "      <td>648962</td>\n",
       "      <td>csk1</td>\n",
       "      <td>snp</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>JB872</td>\n",
       "      <td>SPAC1D4.06c</td>\n",
       "      <td>I</td>\n",
       "      <td>649208</td>\n",
       "      <td>csk1</td>\n",
       "      <td>snp</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>JB872</td>\n",
       "      <td>SPAC1D4.06c</td>\n",
       "      <td>I</td>\n",
       "      <td>649624</td>\n",
       "      <td>csk1</td>\n",
       "      <td>snp</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>253</th>\n",
       "      <td>JB871</td>\n",
       "      <td>SPBC23G7.12c</td>\n",
       "      <td>II</td>\n",
       "      <td>2122791</td>\n",
       "      <td>rpt6</td>\n",
       "      <td>indel</td>\n",
       "      <td>T</td>\n",
       "      <td>TA</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>254</th>\n",
       "      <td>JB1117</td>\n",
       "      <td>SPBC12D12.06</td>\n",
       "      <td>II</td>\n",
       "      <td>2316851</td>\n",
       "      <td>srb11</td>\n",
       "      <td>indel</td>\n",
       "      <td>TA</td>\n",
       "      <td>T</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>255</th>\n",
       "      <td>JB759</td>\n",
       "      <td>SPBC12D12.06</td>\n",
       "      <td>II</td>\n",
       "      <td>2316851</td>\n",
       "      <td>srb11</td>\n",
       "      <td>indel</td>\n",
       "      <td>TA</td>\n",
       "      <td>T</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>256</th>\n",
       "      <td>JB872</td>\n",
       "      <td>SPBC405.07</td>\n",
       "      <td>II</td>\n",
       "      <td>3153756</td>\n",
       "      <td>rpl3602</td>\n",
       "      <td>indel</td>\n",
       "      <td>T</td>\n",
       "      <td>TTA</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>257</th>\n",
       "      <td>JB953</td>\n",
       "      <td>SPBC16G5.15c</td>\n",
       "      <td>II</td>\n",
       "      <td>4239834</td>\n",
       "      <td>fkh2</td>\n",
       "      <td>indel</td>\n",
       "      <td>T</td>\n",
       "      <td>TA</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>258 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     strain            ID chromosome position     gene   type allele1 allele2  \\\n",
       "0     JB872   SPAC1D4.06c          I   648673     csk1    snp       A       G   \n",
       "1     JB953   SPAC1D4.06c          I   648875     csk1    snp       C       T   \n",
       "2     JB872   SPAC1D4.06c          I   648962     csk1    snp       A       G   \n",
       "3     JB872   SPAC1D4.06c          I   649208     csk1    snp       T       C   \n",
       "4     JB872   SPAC1D4.06c          I   649624     csk1    snp       T       C   \n",
       "..      ...           ...        ...      ...      ...    ...     ...     ...   \n",
       "253   JB871  SPBC23G7.12c         II  2122791     rpt6  indel       T      TA   \n",
       "254  JB1117  SPBC12D12.06         II  2316851    srb11  indel      TA       T   \n",
       "255   JB759  SPBC12D12.06         II  2316851    srb11  indel      TA       T   \n",
       "256   JB872    SPBC405.07         II  3153756  rpl3602  indel       T     TTA   \n",
       "257   JB953  SPBC16G5.15c         II  4239834     fkh2  indel       T      TA   \n",
       "\n",
       "     p_value  \n",
       "0       0.05  \n",
       "1       0.05  \n",
       "2       0.05  \n",
       "3       0.05  \n",
       "4       0.05  \n",
       "..       ...  \n",
       "253     0.05  \n",
       "254     0.05  \n",
       "255     0.05  \n",
       "256     0.05  \n",
       "257     0.05  \n",
       "\n",
       "[258 rows x 9 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#make csv file\n",
    "#results.to_csv(root + \"/MSci Bahler lab/S.-Pombe-MLPs - Github/Bence folder/Looking for variants in wild isolates/snp_hunting.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#analysis of nearby 100 SNPs to see whether JB914 and JB759 are different\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[10, 29, 44]"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "interesting_strains_index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [],
   "source": [
    "snps_data = allel.read_vcf(snps_file,fields=['samples','variants/CHROM','variants/POS','calldata/GT','variants/ALT','variants/REF'])\n",
    "samples = snps_data['samples']\n",
    "interesting_strains = samples\n",
    "interesting_strains = [\"JB914\",\"JB759\",\"JB22\",\"JB869\"]\n",
    "#in df keep only row where Parent is srb11\n",
    "df = df.loc[df[\"Parent\"] == \"srb11\"]\n",
    "#reset index\n",
    "df.reset_index(drop=True, inplace=True)\n",
    "\n",
    "#finding the index of the interesting strains\n",
    "#create np array with same shape as interesting strains\n",
    "interesting_strains_index =[]\n",
    "#find index of each interesting strain in samples. If it is not in samples, put 0 in the array\n",
    "for i in range(len(interesting_strains)):\n",
    "      if interesting_strains[i] in samples:\n",
    "            interesting_strains_index.append(np.where(samples==interesting_strains[i])[0][0])\n",
    "      else:\n",
    "            interesting_strains_index.append(0)\n",
    "\n",
    "\n",
    "#if interesting_strains_index zero remove from interesting strains and interesting strains index\n",
    "for i in range(len(interesting_strains_index)):\n",
    "      if interesting_strains_index[i] == 0:\n",
    "            interesting_strains.remove(interesting_strains[i])\n",
    "            interesting_strains_index.remove(interesting_strains_index[i])\n",
    "            break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[44, 10, 29]"
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "interesting_strains_index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['JB914', 'JB22', 'JB869']"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "interesting_strains "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "calls= snps_data['calldata/GT'][:,:,0]\n",
    "chromosomes = snps_data['variants/CHROM']\n",
    "positions = snps_data['variants/POS']\n",
    "alt = snps_data['variants/ALT']\n",
    "ref= snps_data['variants/REF']\n",
    "\n",
    "#iterate through positions\n",
    "#results is a df with columns: alleles\tn\tsum_x\ty_transpose_x\tbeta\tstandard_error\tt_stat\tp_value\ttype\tallele1\tallele2\tchromosome\tposition\n",
    "results = pd.DataFrame(columns=[\"strain\",\"ID\",\"chromosome\",\"position\",\"gene\",\"type\",\"allele1\",\"allele2\",\"p_value\"])\n",
    "#rough gene coordinates\n",
    "#take mean of start and end\n",
    "rough_gene_coordinates = (df[\"start\"]+df[\"end\"])/2\n",
    "\n",
    "\n",
    "for i in range(len(positions)):\n",
    "      #if positions[i] is not near to any of the rough_gene_coordinates  +-10000\n",
    "      \n",
    "      if np.isclose(positions[i],rough_gene_coordinates,atol=100000).any() == False:\n",
    "            \n",
    "            continue\n",
    "      for j in range(len(df)):\n",
    "            #for + strand\n",
    "            if positions[i]<(df[\"end\"][j]+50000) and positions[i]>(df[\"start\"][j]-100000) and chromosomes[i]==df[\"chromosome\"][j]:\n",
    "                  #iterate through interesting strains\n",
    "                  for k in range(len(interesting_strains_index)):\n",
    "                        #take the value of the genotype of the interesting strain at the position\n",
    "                        call_value = calls[i][interesting_strains_index[k]]\n",
    "                        #if this value is 0 or 1, and there is only max 2 strains with this value, then add the strain name to the detail column\n",
    "                        results = pd.concat([results,pd.DataFrame({\"strain\":interesting_strains[k],\"ID\":df[\"ID\"][j],\"chromosome\":chromosomes[i],\"position\":positions[i], \"gene\":df[\"Parent\"][j],\"type\":\"snp\",\"allele1\":ref[i],\"allele2\":alt[i][call_value-1],\"p_value\":0.05},index=[0])],ignore_index=True)\n",
    "\n",
    "\n",
    "\n",
    "indels_data = allel.read_vcf(indels_file,fields=['samples','variants/CHROM','variants/POS','calldata/GT','variants/ALT','variants/REF'])\n",
    "\n",
    "calls= indels_data['calldata/GT'][:,:,0]\n",
    "chromosomes = indels_data['variants/CHROM']\n",
    "positions = indels_data['variants/POS']\n",
    "alt = indels_data['variants/ALT']\n",
    "ref= indels_data['variants/REF']\n",
    "\n",
    "\n",
    "for i in range(len(positions)):\n",
    "   \n",
    "      if np.isclose(positions[i],rough_gene_coordinates,atol=100000).any() == False:\n",
    "            continue\n",
    "      for j in range(len(df)):\n",
    "            #for + strand\n",
    "            if positions[i]<(df[\"end\"][j]+50000) and positions[i]>(df[\"start\"][j]-100000) and chromosomes[i]==df[\"chromosome\"][j]:\n",
    "                  #iterate through interesting strains\n",
    "                  for k in range(len(interesting_strains_index)):\n",
    "                        #take the value of the genotype of the interesting strain at the position\n",
    "                        call_value = calls[i][interesting_strains_index[k]]\n",
    "                        #if this value is 0 or 1, and there is only max 3 strains with this value, then add the strain name to the detail column\n",
    "                        results = pd.concat([results,pd.DataFrame({\"strain\":interesting_strains[k],\"ID\":df[\"ID\"][j],\"chromosome\":chromosomes[i],\"position\":positions[i], \"gene\":df[\"Parent\"][j],\"type\":\"indel\",\"allele1\":ref[i],\"allele2\":alt[i][call_value-1],\"p_value\":0.05},index=[0])],ignore_index=True)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "metadata": {},
   "outputs": [],
   "source": [
    "results_final=results.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>strain</th>\n",
       "      <th>ID</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>gene</th>\n",
       "      <th>type</th>\n",
       "      <th>allele1</th>\n",
       "      <th>allele2</th>\n",
       "      <th>p_value</th>\n",
       "      <th>allele</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>JB914</td>\n",
       "      <td>SPBC12D12.06</td>\n",
       "      <td>II</td>\n",
       "      <td>2216902</td>\n",
       "      <td>srb11</td>\n",
       "      <td>snp</td>\n",
       "      <td>C</td>\n",
       "      <td></td>\n",
       "      <td>0.05</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>JB22</td>\n",
       "      <td>SPBC12D12.06</td>\n",
       "      <td>II</td>\n",
       "      <td>2216902</td>\n",
       "      <td>srb11</td>\n",
       "      <td>snp</td>\n",
       "      <td>C</td>\n",
       "      <td></td>\n",
       "      <td>0.05</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>JB869</td>\n",
       "      <td>SPBC12D12.06</td>\n",
       "      <td>II</td>\n",
       "      <td>2216902</td>\n",
       "      <td>srb11</td>\n",
       "      <td>snp</td>\n",
       "      <td>C</td>\n",
       "      <td></td>\n",
       "      <td>0.05</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>JB914</td>\n",
       "      <td>SPBC12D12.06</td>\n",
       "      <td>II</td>\n",
       "      <td>2216992</td>\n",
       "      <td>srb11</td>\n",
       "      <td>snp</td>\n",
       "      <td>G</td>\n",
       "      <td></td>\n",
       "      <td>0.05</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>JB22</td>\n",
       "      <td>SPBC12D12.06</td>\n",
       "      <td>II</td>\n",
       "      <td>2216992</td>\n",
       "      <td>srb11</td>\n",
       "      <td>snp</td>\n",
       "      <td>G</td>\n",
       "      <td></td>\n",
       "      <td>0.05</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8785</th>\n",
       "      <td>JB22</td>\n",
       "      <td>SPBC12D12.06</td>\n",
       "      <td>II</td>\n",
       "      <td>2366722</td>\n",
       "      <td>srb11</td>\n",
       "      <td>indel</td>\n",
       "      <td>TA</td>\n",
       "      <td></td>\n",
       "      <td>0.05</td>\n",
       "      <td>TA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8786</th>\n",
       "      <td>JB869</td>\n",
       "      <td>SPBC12D12.06</td>\n",
       "      <td>II</td>\n",
       "      <td>2366722</td>\n",
       "      <td>srb11</td>\n",
       "      <td>indel</td>\n",
       "      <td>TA</td>\n",
       "      <td></td>\n",
       "      <td>0.05</td>\n",
       "      <td>TA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8787</th>\n",
       "      <td>JB914</td>\n",
       "      <td>SPBC12D12.06</td>\n",
       "      <td>II</td>\n",
       "      <td>2366778</td>\n",
       "      <td>srb11</td>\n",
       "      <td>indel</td>\n",
       "      <td>AT</td>\n",
       "      <td></td>\n",
       "      <td>0.05</td>\n",
       "      <td>AT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8788</th>\n",
       "      <td>JB22</td>\n",
       "      <td>SPBC12D12.06</td>\n",
       "      <td>II</td>\n",
       "      <td>2366778</td>\n",
       "      <td>srb11</td>\n",
       "      <td>indel</td>\n",
       "      <td>AT</td>\n",
       "      <td></td>\n",
       "      <td>0.05</td>\n",
       "      <td>AT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8789</th>\n",
       "      <td>JB869</td>\n",
       "      <td>SPBC12D12.06</td>\n",
       "      <td>II</td>\n",
       "      <td>2366778</td>\n",
       "      <td>srb11</td>\n",
       "      <td>indel</td>\n",
       "      <td>AT</td>\n",
       "      <td></td>\n",
       "      <td>0.05</td>\n",
       "      <td>AT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8790 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     strain            ID chromosome position   gene   type allele1 allele2  \\\n",
       "0     JB914  SPBC12D12.06         II  2216902  srb11    snp       C           \n",
       "1      JB22  SPBC12D12.06         II  2216902  srb11    snp       C           \n",
       "2     JB869  SPBC12D12.06         II  2216902  srb11    snp       C           \n",
       "3     JB914  SPBC12D12.06         II  2216992  srb11    snp       G           \n",
       "4      JB22  SPBC12D12.06         II  2216992  srb11    snp       G           \n",
       "...     ...           ...        ...      ...    ...    ...     ...     ...   \n",
       "8785   JB22  SPBC12D12.06         II  2366722  srb11  indel      TA           \n",
       "8786  JB869  SPBC12D12.06         II  2366722  srb11  indel      TA           \n",
       "8787  JB914  SPBC12D12.06         II  2366778  srb11  indel      AT           \n",
       "8788   JB22  SPBC12D12.06         II  2366778  srb11  indel      AT           \n",
       "8789  JB869  SPBC12D12.06         II  2366778  srb11  indel      AT           \n",
       "\n",
       "      p_value allele  \n",
       "0        0.05      C  \n",
       "1        0.05      C  \n",
       "2        0.05      C  \n",
       "3        0.05      G  \n",
       "4        0.05      G  \n",
       "...       ...    ...  \n",
       "8785     0.05     TA  \n",
       "8786     0.05     TA  \n",
       "8787     0.05     AT  \n",
       "8788     0.05     AT  \n",
       "8789     0.05     AT  \n",
       "\n",
       "[8790 rows x 10 columns]"
      ]
     },
     "execution_count": 153,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#add column allele, which is allele2\n",
    "results_final[\"allele\"] = results_final[\"allele2\"]\n",
    "#where allele is empty, make it allele1\n",
    "results_final.loc[results_final[\"allele\"]==\"\",\"allele\"] = results_final[\"allele1\"]\n",
    "results_final\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>strain</th>\n",
       "      <th>position</th>\n",
       "      <th>allele</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>JB914</td>\n",
       "      <td>2216902</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>JB22</td>\n",
       "      <td>2216902</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>JB869</td>\n",
       "      <td>2216902</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>JB914</td>\n",
       "      <td>2216992</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>JB22</td>\n",
       "      <td>2216992</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8785</th>\n",
       "      <td>JB22</td>\n",
       "      <td>2366722</td>\n",
       "      <td>TA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8786</th>\n",
       "      <td>JB869</td>\n",
       "      <td>2366722</td>\n",
       "      <td>TA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8787</th>\n",
       "      <td>JB914</td>\n",
       "      <td>2366778</td>\n",
       "      <td>AT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8788</th>\n",
       "      <td>JB22</td>\n",
       "      <td>2366778</td>\n",
       "      <td>AT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8789</th>\n",
       "      <td>JB869</td>\n",
       "      <td>2366778</td>\n",
       "      <td>AT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8790 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     strain position allele\n",
       "0     JB914  2216902      C\n",
       "1      JB22  2216902      C\n",
       "2     JB869  2216902      C\n",
       "3     JB914  2216992      G\n",
       "4      JB22  2216992      G\n",
       "...     ...      ...    ...\n",
       "8785   JB22  2366722     TA\n",
       "8786  JB869  2366722     TA\n",
       "8787  JB914  2366778     AT\n",
       "8788   JB22  2366778     AT\n",
       "8789  JB869  2366778     AT\n",
       "\n",
       "[8790 rows x 3 columns]"
      ]
     },
     "execution_count": 154,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#only keep strain position allele\n",
    "results_final= results_final[[\"strain\",\"position\",\"allele\"]]\n",
    "results_final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "metadata": {},
   "outputs": [],
   "source": [
    "#remove duplicates for strain and position\n",
    "results_final.drop_duplicates(subset=[\"strain\",\"position\"],inplace=True)\n",
    "#reset index\n",
    "results_final.reset_index(drop=True, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>strain</th>\n",
       "      <th>position</th>\n",
       "      <th>allele</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>JB914</td>\n",
       "      <td>2216902</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>JB22</td>\n",
       "      <td>2216902</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>JB869</td>\n",
       "      <td>2216902</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>JB914</td>\n",
       "      <td>2216992</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>JB22</td>\n",
       "      <td>2216992</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8698</th>\n",
       "      <td>JB22</td>\n",
       "      <td>2366722</td>\n",
       "      <td>TA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8699</th>\n",
       "      <td>JB869</td>\n",
       "      <td>2366722</td>\n",
       "      <td>TA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8700</th>\n",
       "      <td>JB914</td>\n",
       "      <td>2366778</td>\n",
       "      <td>AT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8701</th>\n",
       "      <td>JB22</td>\n",
       "      <td>2366778</td>\n",
       "      <td>AT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8702</th>\n",
       "      <td>JB869</td>\n",
       "      <td>2366778</td>\n",
       "      <td>AT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8703 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     strain position allele\n",
       "0     JB914  2216902      C\n",
       "1      JB22  2216902      C\n",
       "2     JB869  2216902      C\n",
       "3     JB914  2216992      G\n",
       "4      JB22  2216992      G\n",
       "...     ...      ...    ...\n",
       "8698   JB22  2366722     TA\n",
       "8699  JB869  2366722     TA\n",
       "8700  JB914  2366778     AT\n",
       "8701   JB22  2366778     AT\n",
       "8702  JB869  2366778     AT\n",
       "\n",
       "[8703 rows x 3 columns]"
      ]
     },
     "execution_count": 156,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "results_final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>strain</th>\n",
       "      <th>JB22</th>\n",
       "      <th>JB869</th>\n",
       "      <th>JB914</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>position</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2216902</th>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2216992</th>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2217037</th>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2217041</th>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2217061</th>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2367068</th>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2367343</th>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2367371</th>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2367450</th>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2367500</th>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2901 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "strain   JB22 JB869 JB914\n",
       "position                 \n",
       "2216902     C     C     C\n",
       "2216992     G     G     G\n",
       "2217037     G     G     G\n",
       "2217041     T     T     T\n",
       "2217061     G     G     G\n",
       "...       ...   ...   ...\n",
       "2367068     G     A     G\n",
       "2367343     T     T     T\n",
       "2367371     T     T     C\n",
       "2367450     G     G     A\n",
       "2367500     G     G     G\n",
       "\n",
       "[2901 rows x 3 columns]"
      ]
     },
     "execution_count": 157,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#rearrange to make three columns for each strain\n",
    "results_final = results_final.pivot(index='position', columns='strain', values='allele')\n",
    "results_final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>strain</th>\n",
       "      <th>JB22</th>\n",
       "      <th>JB869</th>\n",
       "      <th>JB914</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>position</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2219195</th>\n",
       "      <td>AT</td>\n",
       "      <td>AT</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2219764</th>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2221916</th>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2222112</th>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2222671</th>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2225261</th>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2228418</th>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2229030</th>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2231797</th>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2232111</th>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2234377</th>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2234644</th>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2238429</th>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2242167</th>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2242168</th>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2252724</th>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2254255</th>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2254586</th>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>CTCAAATATAGAAA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2254649</th>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2258722</th>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2261354</th>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2263164</th>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2264537</th>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2265296</th>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2266539</th>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2266580</th>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2281958</th>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2302895</th>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2304929</th>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2305695</th>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2310884</th>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2311107</th>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2312083</th>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2316554</th>\n",
       "      <td>TA</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2316851</th>\n",
       "      <td>TA</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2317549</th>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2323914</th>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2326615</th>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2327939</th>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2328246</th>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2328339</th>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2328518</th>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2329425</th>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2329747</th>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2330408</th>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2334988</th>\n",
       "      <td>CT</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2338603</th>\n",
       "      <td>CA</td>\n",
       "      <td>CA</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2339289</th>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>AC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2339585</th>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2339590</th>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "strain   JB22 JB869           JB914\n",
       "position                           \n",
       "2219195    AT    AT               A\n",
       "2219764     T     T               C\n",
       "2221916     G     G               A\n",
       "2222112     G     G              GA\n",
       "2222671     T     T               C\n",
       "2225261     A     A               G\n",
       "2228418     T     C               C\n",
       "2229030     A     G               A\n",
       "2231797     C     C               A\n",
       "2232111     G     G               A\n",
       "2234377     G     G               A\n",
       "2234644     T     T               G\n",
       "2238429     C     C               G\n",
       "2242167     A     A               G\n",
       "2242168     T     T               C\n",
       "2252724     C     C               T\n",
       "2254255     A     A               G\n",
       "2254586     C     C  CTCAAATATAGAAA\n",
       "2254649     G     G               T\n",
       "2258722     C     C               T\n",
       "2261354     A     A               G\n",
       "2263164     A     A               G\n",
       "2264537     A     A               G\n",
       "2265296     T     T               A\n",
       "2266539     C     C               T\n",
       "2266580     G     G               T\n",
       "2281958     C     C               T\n",
       "2302895     A     T               T\n",
       "2304929     A     G               G\n",
       "2305695     C     T               T\n",
       "2310884     C     T               T\n",
       "2311107     G     C               C\n",
       "2312083     C     T               T\n",
       "2316554    TA     T               T\n",
       "2316851    TA     T               T\n",
       "2317549     C     A               A\n",
       "2323914     A     G               G\n",
       "2326615     C     A               A\n",
       "2327939     G     A               A\n",
       "2328246     C     T               T\n",
       "2328339     C     T               T\n",
       "2328518     C     C               T\n",
       "2329425     G     C               G\n",
       "2329747     A     G               G\n",
       "2330408     C     A               A\n",
       "2334988    CT     C               C\n",
       "2338603    CA    CA               C\n",
       "2339289     A     A              AC\n",
       "2339585     A     A               T\n",
       "2339590     A     A               T"
      ]
     },
     "execution_count": 158,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#how many of the rows show a difference between JB914 and JB759\n",
    "diff= results_final.loc[(results_final[\"JB914\"]!=results_final[\"JB22\"])|(results_final[\"JB869\"]!=results_final[\"JB22\"])]\n",
    "diff.head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The watermark extension is already loaded. To reload it, use:\n",
      "  %reload_ext watermark\n",
      "Last updated: 2023-11-03T18:58:41.209602+00:00\n",
      "\n",
      "Python implementation: CPython\n",
      "Python version       : 3.9.0\n",
      "IPython version      : 7.31.1\n",
      "\n",
      "Compiler    : Clang 11.0.0 \n",
      "OS          : Darwin\n",
      "Release     : 21.5.0\n",
      "Machine     : x86_64\n",
      "Processor   : i386\n",
      "CPU cores   : 8\n",
      "Architecture: 64bit\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%load_ext watermark\n",
    "%watermark\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "numpy     : 1.23.4\n",
      "matplotlib: 3.8.0\n",
      "allel     : 1.3.5\n",
      "pandas    : 1.4.4\n",
      "bokeh     : 3.2.2\n",
      "scipy     : 1.11.3\n",
      "iqplot    : 0.3.3\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%watermark --iversions\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "2d0398177d7bb10ea33b6a5cd9d022d63e20680fcfd84762562c72ff35bdf44b"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
