{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "#find snps\n",
    "#import data science libraries\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import allel\n",
    "import numpy as np\n",
    "import scipy.stats as st\n",
    "#suppress warnings\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "root = \"/Users/k23030440/github_projects/S.-Pombe-MLPs/\"\n",
    "snps_file = '/Users/k23030440/Jeffares copy/Jeffares SNPs/ERZ289255/Spombe.2013-01-02.filt3c.nr57-final.snps.anno-snpeff3.cleaned4.vcf'\n",
    "indels_file = '/Users/k23030440/Jeffares copy/Jeffares SNPs/ERZ289254/Spombe.indels.unique.nr57.final.annotated.2014-01-23.cleaned4.vcf.gz'\n",
    "final_results_screen = pd.read_csv(root + \"/Bence folder/Bioneer screen/final_results_screen.csv\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "#genome coordinates for genes\n",
    "df = pd.read_csv(root + '/external data/Pombase files/Schizosaccharomyces_pombe_all_chromosomes.gff3', sep='\\t', header=1)\n",
    "#break last column into 2 at ;\n",
    "df[['a','b']] = df[\"ID=SPAC1002.01;Name=mrx11\"].str.split(';',expand=True)\n",
    "#remove column ID=SPAC1002.01;Name=mrx11\n",
    "df.drop(columns =[\"ID=SPAC1002.01;Name=mrx11\"], inplace = True)\n",
    "#remove ID= from column a and rename it to ID\n",
    "df['ID'] = df['a'].str.replace('ID=', '')\n",
    "#remove column a\n",
    "df.drop(columns =['a','PomBase',\".\",\"..1\"], inplace = True)\n",
    "#rename b to Parent\n",
    "df.rename(columns={'b':'Parent','gene':'object','I':'chromosome','1798347':\"start\",'1798835':\"end\",\"+\":\"strand\"}, inplace=True)\n",
    "#In parent remove anything before =\n",
    "df['Parent'] = df['Parent'].str.replace('Parent=', '')\n",
    "#or Name=\n",
    "df['Parent'] = df['Parent'].str.replace('Name=', '')\n",
    "#create an aempty column called detail\n",
    "df['detail'] = \"\"\n",
    "#split ID column at :, and if there is something after put it in detail column\n",
    "df['detail'] = df['ID'].str.split(':', expand=True)[1]\n",
    "#split ID column at :, and if there is something before put it in ID column\n",
    "df['ID'] = df['ID'].str.split(':', expand=True)[0]#load in vcf file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>object</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>Parent</th>\n",
       "      <th>ID</th>\n",
       "      <th>detail</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>I</td>\n",
       "      <td>gene</td>\n",
       "      <td>2192037</td>\n",
       "      <td>2193615</td>\n",
       "      <td>-</td>\n",
       "      <td>hst4</td>\n",
       "      <td>SPAC1783.04c</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>I</td>\n",
       "      <td>gene</td>\n",
       "      <td>648301</td>\n",
       "      <td>649565</td>\n",
       "      <td>-</td>\n",
       "      <td>csk1</td>\n",
       "      <td>SPAC1D4.06c</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>I</td>\n",
       "      <td>gene</td>\n",
       "      <td>686535</td>\n",
       "      <td>689179</td>\n",
       "      <td>+</td>\n",
       "      <td>res2</td>\n",
       "      <td>SPAC22F3.09c</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>I</td>\n",
       "      <td>gene</td>\n",
       "      <td>1576707</td>\n",
       "      <td>1578028</td>\n",
       "      <td>+</td>\n",
       "      <td>srb10</td>\n",
       "      <td>SPAC23H4.17c</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>I</td>\n",
       "      <td>gene</td>\n",
       "      <td>3069322</td>\n",
       "      <td>3071099</td>\n",
       "      <td>-</td>\n",
       "      <td>shm1</td>\n",
       "      <td>SPAC24C9.12c</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>I</td>\n",
       "      <td>gene</td>\n",
       "      <td>5131870</td>\n",
       "      <td>5133058</td>\n",
       "      <td>-</td>\n",
       "      <td>rrp17</td>\n",
       "      <td>SPAC29A4.09</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>I</td>\n",
       "      <td>gene</td>\n",
       "      <td>2832707</td>\n",
       "      <td>2834985</td>\n",
       "      <td>+</td>\n",
       "      <td>his6</td>\n",
       "      <td>SPAC3F10.09</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>I</td>\n",
       "      <td>gene</td>\n",
       "      <td>3095075</td>\n",
       "      <td>3098667</td>\n",
       "      <td>-</td>\n",
       "      <td>med13</td>\n",
       "      <td>SPAC589.02c</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>I</td>\n",
       "      <td>gene</td>\n",
       "      <td>1503761</td>\n",
       "      <td>1504867</td>\n",
       "      <td>-</td>\n",
       "      <td>med18</td>\n",
       "      <td>SPAC5D6.05</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>I</td>\n",
       "      <td>gene</td>\n",
       "      <td>2039370</td>\n",
       "      <td>2040417</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>SPAC607.02c</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>I</td>\n",
       "      <td>gene</td>\n",
       "      <td>3123163</td>\n",
       "      <td>3126653</td>\n",
       "      <td>+</td>\n",
       "      <td>srb8</td>\n",
       "      <td>SPAC688.08</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>I</td>\n",
       "      <td>gene</td>\n",
       "      <td>2408136</td>\n",
       "      <td>2414047</td>\n",
       "      <td>-</td>\n",
       "      <td>mus7</td>\n",
       "      <td>SPAC6B12.02c</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>I</td>\n",
       "      <td>gene</td>\n",
       "      <td>2433380</td>\n",
       "      <td>2435623</td>\n",
       "      <td>+</td>\n",
       "      <td>tom70</td>\n",
       "      <td>SPAC6B12.12</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>I</td>\n",
       "      <td>gene</td>\n",
       "      <td>3273944</td>\n",
       "      <td>3278265</td>\n",
       "      <td>+</td>\n",
       "      <td>puf4</td>\n",
       "      <td>SPAC6G9.14</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>I</td>\n",
       "      <td>gene</td>\n",
       "      <td>2588668</td>\n",
       "      <td>2590090</td>\n",
       "      <td>-</td>\n",
       "      <td>tlg2</td>\n",
       "      <td>SPAC823.05c</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>I</td>\n",
       "      <td>gene</td>\n",
       "      <td>3400528</td>\n",
       "      <td>3401152</td>\n",
       "      <td>+</td>\n",
       "      <td>rpl2102</td>\n",
       "      <td>SPAC959.08</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>II</td>\n",
       "      <td>gene</td>\n",
       "      <td>2316055</td>\n",
       "      <td>2317654</td>\n",
       "      <td>+</td>\n",
       "      <td>srb11</td>\n",
       "      <td>SPBC12D12.06</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>II</td>\n",
       "      <td>gene</td>\n",
       "      <td>4239333</td>\n",
       "      <td>4242718</td>\n",
       "      <td>-</td>\n",
       "      <td>fkh2</td>\n",
       "      <td>SPBC16G5.15c</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>II</td>\n",
       "      <td>gene</td>\n",
       "      <td>2121444</td>\n",
       "      <td>2122816</td>\n",
       "      <td>-</td>\n",
       "      <td>rpt6</td>\n",
       "      <td>SPBC23G7.12c</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>II</td>\n",
       "      <td>gene</td>\n",
       "      <td>2978736</td>\n",
       "      <td>2979730</td>\n",
       "      <td>-</td>\n",
       "      <td>mmp1</td>\n",
       "      <td>SPBC2D10.07c</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>II</td>\n",
       "      <td>gene</td>\n",
       "      <td>3456962</td>\n",
       "      <td>3460319</td>\n",
       "      <td>-</td>\n",
       "      <td>dcd1</td>\n",
       "      <td>SPBC2G2.13c</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>II</td>\n",
       "      <td>gene</td>\n",
       "      <td>551130</td>\n",
       "      <td>555295</td>\n",
       "      <td>-</td>\n",
       "      <td>sre2</td>\n",
       "      <td>SPBC354.05c</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>II</td>\n",
       "      <td>gene</td>\n",
       "      <td>3153488</td>\n",
       "      <td>3154041</td>\n",
       "      <td>+</td>\n",
       "      <td>rpl3602</td>\n",
       "      <td>SPBC405.07</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>II</td>\n",
       "      <td>gene</td>\n",
       "      <td>1171307</td>\n",
       "      <td>1174232</td>\n",
       "      <td>-</td>\n",
       "      <td>mtx1</td>\n",
       "      <td>SPBC409.19c</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>II</td>\n",
       "      <td>gene</td>\n",
       "      <td>2707956</td>\n",
       "      <td>2711825</td>\n",
       "      <td>+</td>\n",
       "      <td>vps901</td>\n",
       "      <td>SPBC4F6.10</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>II</td>\n",
       "      <td>gene</td>\n",
       "      <td>3206576</td>\n",
       "      <td>3208733</td>\n",
       "      <td>-</td>\n",
       "      <td>kgd2</td>\n",
       "      <td>SPBC776.15c</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>III</td>\n",
       "      <td>gene</td>\n",
       "      <td>1040100</td>\n",
       "      <td>1041016</td>\n",
       "      <td>+</td>\n",
       "      <td>rpa12</td>\n",
       "      <td>SPCC1259.03</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>III</td>\n",
       "      <td>gene</td>\n",
       "      <td>1061503</td>\n",
       "      <td>1064195</td>\n",
       "      <td>-</td>\n",
       "      <td>meu27</td>\n",
       "      <td>SPCC1259.14c</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>III</td>\n",
       "      <td>gene</td>\n",
       "      <td>1734260</td>\n",
       "      <td>1735312</td>\n",
       "      <td>-</td>\n",
       "      <td>rox3</td>\n",
       "      <td>SPCC1450.05c</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>III</td>\n",
       "      <td>gene</td>\n",
       "      <td>2099846</td>\n",
       "      <td>2100555</td>\n",
       "      <td>+</td>\n",
       "      <td>rpl15</td>\n",
       "      <td>SPCC576.11</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>III</td>\n",
       "      <td>gene</td>\n",
       "      <td>1939237</td>\n",
       "      <td>1939907</td>\n",
       "      <td>+</td>\n",
       "      <td>rpl2702</td>\n",
       "      <td>SPCC74.05</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>III</td>\n",
       "      <td>gene</td>\n",
       "      <td>1002750</td>\n",
       "      <td>1007515</td>\n",
       "      <td>+</td>\n",
       "      <td>for3</td>\n",
       "      <td>SPCC895.05</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>III</td>\n",
       "      <td>gene</td>\n",
       "      <td>553174</td>\n",
       "      <td>553840</td>\n",
       "      <td>+</td>\n",
       "      <td>rps1201</td>\n",
       "      <td>SPCC962.04</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>III</td>\n",
       "      <td>gene</td>\n",
       "      <td>1837499</td>\n",
       "      <td>1840092</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>SPNCRNA.1234</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>I</td>\n",
       "      <td>gene</td>\n",
       "      <td>1954820</td>\n",
       "      <td>1956332</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>SPNCRNA.781</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>I</td>\n",
       "      <td>gene</td>\n",
       "      <td>3300812</td>\n",
       "      <td>3302082</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>SPNCRNA.900</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   chromosome object    start      end strand   Parent            ID detail\n",
       "0           I   gene  2192037  2193615      -     hst4  SPAC1783.04c   None\n",
       "1           I   gene   648301   649565      -     csk1   SPAC1D4.06c   None\n",
       "2           I   gene   686535   689179      +     res2  SPAC22F3.09c   None\n",
       "3           I   gene  1576707  1578028      +    srb10  SPAC23H4.17c   None\n",
       "4           I   gene  3069322  3071099      -     shm1  SPAC24C9.12c   None\n",
       "5           I   gene  5131870  5133058      -    rrp17   SPAC29A4.09   None\n",
       "6           I   gene  2832707  2834985      +     his6   SPAC3F10.09   None\n",
       "7           I   gene  3095075  3098667      -    med13   SPAC589.02c   None\n",
       "8           I   gene  1503761  1504867      -    med18    SPAC5D6.05   None\n",
       "9           I   gene  2039370  2040417      -     None   SPAC607.02c   None\n",
       "10          I   gene  3123163  3126653      +     srb8    SPAC688.08   None\n",
       "11          I   gene  2408136  2414047      -     mus7  SPAC6B12.02c   None\n",
       "12          I   gene  2433380  2435623      +    tom70   SPAC6B12.12   None\n",
       "13          I   gene  3273944  3278265      +     puf4    SPAC6G9.14   None\n",
       "14          I   gene  2588668  2590090      -     tlg2   SPAC823.05c   None\n",
       "15          I   gene  3400528  3401152      +  rpl2102    SPAC959.08   None\n",
       "16         II   gene  2316055  2317654      +    srb11  SPBC12D12.06   None\n",
       "17         II   gene  4239333  4242718      -     fkh2  SPBC16G5.15c   None\n",
       "18         II   gene  2121444  2122816      -     rpt6  SPBC23G7.12c   None\n",
       "19         II   gene  2978736  2979730      -     mmp1  SPBC2D10.07c   None\n",
       "20         II   gene  3456962  3460319      -     dcd1   SPBC2G2.13c   None\n",
       "21         II   gene   551130   555295      -     sre2   SPBC354.05c   None\n",
       "22         II   gene  3153488  3154041      +  rpl3602    SPBC405.07   None\n",
       "23         II   gene  1171307  1174232      -     mtx1   SPBC409.19c   None\n",
       "24         II   gene  2707956  2711825      +   vps901    SPBC4F6.10   None\n",
       "25         II   gene  3206576  3208733      -     kgd2   SPBC776.15c   None\n",
       "26        III   gene  1040100  1041016      +    rpa12   SPCC1259.03   None\n",
       "27        III   gene  1061503  1064195      -    meu27  SPCC1259.14c   None\n",
       "28        III   gene  1734260  1735312      -     rox3  SPCC1450.05c   None\n",
       "29        III   gene  2099846  2100555      +    rpl15    SPCC576.11   None\n",
       "30        III   gene  1939237  1939907      +  rpl2702     SPCC74.05   None\n",
       "31        III   gene  1002750  1007515      +     for3    SPCC895.05   None\n",
       "32        III   gene   553174   553840      +  rps1201    SPCC962.04   None\n",
       "33        III   gene  1837499  1840092      -     None  SPNCRNA.1234   None\n",
       "34          I   gene  1954820  1956332      +     None   SPNCRNA.781   None\n",
       "35          I   gene  3300812  3302082      -     None   SPNCRNA.900   None"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = df.loc[df[\"object\"] == \"gene\"]\n",
    "mediator = [\"SPAC23H4.17c\",\"SPBC12D12.06\",\"SPAC688.08\",\"SPAC589.02c\",\"SPAC5D6.05\",\"SPCC1450.05c\"]\n",
    "\n",
    "df = df.loc[df[\"ID\"].isin(mediator) | df[\"ID\"].isin(final_results_screen[\"strain\"].values) | df[\"Parent\"].isin(final_results_screen[\"strain\"].values )]\n",
    "df.reset_index(drop=True, inplace=True)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['JB1110', 'JB1117', 'JB1154', 'JB1171', 'JB1174', 'JB1180',\n",
       "       'JB1197', 'JB1205', 'JB1206', 'JB1207', 'JB22', 'JB4', 'JB758',\n",
       "       'JB760', 'JB762', 'JB837', 'JB838', 'JB840', 'JB841', 'JB842',\n",
       "       'JB845', 'JB846', 'JB848', 'JB852', 'JB853', 'JB854', 'JB858',\n",
       "       'JB862', 'JB864', 'JB869', 'JB870', 'JB871', 'JB872', 'JB873',\n",
       "       'JB874', 'JB875', 'JB878', 'JB879', 'JB884', 'JB899', 'JB900',\n",
       "       'JB902', 'JB910', 'JB913', 'JB914', 'JB916', 'JB917', 'JB918',\n",
       "       'JB929', 'JB930', 'JB931', 'JB934', 'JB938', 'JB939', 'JB942',\n",
       "       'JB943', 'JB953'], dtype=object)"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['JB914', 'JB871', 'JB1117', 'JB872', 'JB934', 'JB848', 'JB953', 'JB869']"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "interesting_strains"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Strain JB759 not found in samples, removing from list.\n"
     ]
    }
   ],
   "source": [
    "snps_data = allel.read_vcf(snps_file,fields=['samples','variants/CHROM','variants/POS','calldata/GT','variants/ALT','variants/REF'])\n",
    "samples = snps_data['samples']\n",
    "interesting_strains = samples\n",
    "interesting_strains = [\"JB914\",\"JB871\",\"JB1117\", \"JB872\", \"JB934\", \"JB759\", \"JB848\",\"JB953\",\"JB869\"]\n",
    "\n",
    "#finding the index of the interesting strains\n",
    "interesting_strains_index = []\n",
    "interesting_strains_final = []\n",
    "for strain in interesting_strains:\n",
    "    try:\n",
    "        index = np.where(samples == strain)[0][0]\n",
    "    except:\n",
    "        #remove strain from interesting_strains if not found\n",
    "        \n",
    "        print(f'Strain {strain} not found in samples, removing from list.')\n",
    "        continue\n",
    "    interesting_strains_final.append(strain)\n",
    "    interesting_strains_index.append(index)\n",
    "\n",
    "interesting_strains = interesting_strains_final.copy()\n",
    "calls= snps_data['calldata/GT'][:,:,0]\n",
    "chromosomes = snps_data['variants/CHROM']\n",
    "positions = snps_data['variants/POS']\n",
    "alt = snps_data['variants/ALT']\n",
    "ref= snps_data['variants/REF']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [],
   "source": [
    "#iterate through positions\n",
    "#results is a df with columns: alleles\tn\tsum_x\ty_transpose_x\tbeta\tstandard_error\tt_stat\tp_value\ttype\tallele1\tallele2\tchromosome\tposition\n",
    "results = pd.DataFrame(columns=[\"strain\",\"ID\",\"chromosome\",\"position\",\"gene\",\"type\",\"allele1\",\"allele2\",\"p_value\"])\n",
    "#rough gene coordinates\n",
    "#take mean of start and end\n",
    "rough_gene_coordinates = (df[\"start\"]+df[\"end\"])/2\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "for i in range(len(positions)):\n",
    "      #if positions[i] is not near to any of the rough_gene_coordinates  +-10000\n",
    "      \n",
    "      if np.isclose(positions[i],rough_gene_coordinates,atol=20000).any() == False:\n",
    "            \n",
    "            continue\n",
    "      for j in range(len(df)):\n",
    "            #for + strand\n",
    "            if positions[i]<(df[\"end\"][j]+200) and positions[i]>(df[\"start\"][j]-200) and chromosomes[i]==df[\"chromosome\"][j]:\n",
    "                  #iterate through interesting strains\n",
    "                  for k in range(len(interesting_strains_index)):\n",
    "                        #take the value of the genotype of the interesting strain at the position\n",
    "                        call_value = calls[i][interesting_strains_index[k]]\n",
    "                        #if this value is 0 or 1, and there is only max 2 strains with this value, then add the strain name to the detail column\n",
    "                        if call_value == 0 or call_value == 1 or call_value == 2:\n",
    "                              if sum(calls[i]==call_value) <= 5:\n",
    "                                    \n",
    "                                    #results = results.append({\"strain\":interesting_strains[k],\"ID\":df[\"ID\"][j],\"chromosome\":chromosomes[i],\"position\":positions[i], \"gene\":df[\"Parent\"][j],\"type\":\"snp\",\"allele1\":ref[i],\"allele2\":alt[i][call_value-1],\"p_value\":0.05},ignore_index=True)\n",
    "                                    results = pd.concat([results,pd.DataFrame({\"strain\":interesting_strains[k],\"ID\":df[\"ID\"][j],\"chromosome\":chromosomes[i],\"position\":positions[i], \"gene\":df[\"Parent\"][j],\"type\":\"snp\",\"allele1\":ref[i],\"allele2\":alt[i][call_value-1],\"p_value\":0.05},index=[0])],ignore_index=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [],
   "source": [
    "indels_data = allel.read_vcf(indels_file,fields=['samples','variants/CHROM','variants/POS','calldata/GT','variants/ALT','variants/REF'])\n",
    "\n",
    "calls= indels_data['calldata/GT'][:,:,0]\n",
    "chromosomes = indels_data['variants/CHROM']\n",
    "positions = indels_data['variants/POS']\n",
    "alt = indels_data['variants/ALT']\n",
    "ref= indels_data['variants/REF']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "for i in range(len(positions)):\n",
    "   \n",
    "      if np.isclose(positions[i],rough_gene_coordinates,atol=20000).any() == False:\n",
    "            continue\n",
    "      for j in range(len(df)):\n",
    "            #for + strand\n",
    "            if positions[i]<(df[\"end\"][j]+200) and positions[i]>(df[\"start\"][j]-200) and chromosomes[i]==df[\"chromosome\"][j]:\n",
    "                  #iterate through interesting strains\n",
    "                  for k in range(len(interesting_strains_index)):\n",
    "                        #take the value of the genotype of the interesting strain at the position\n",
    "                        call_value = calls[i][interesting_strains_index[k]]\n",
    "                        #if this value is 0 or 1, and there is only max 3 strains with this value, then add the strain name to the detail column\n",
    "                        if call_value == 0 or call_value == 1 or call_value == 2 or call_value == 3:\n",
    "                              if sum(calls[i]==call_value) <= 3:\n",
    "                                    \n",
    "                                    results = pd.concat([results,pd.DataFrame({\"strain\":interesting_strains[k],\"ID\":df[\"ID\"][j],\"chromosome\":chromosomes[i],\"position\":positions[i], \"gene\":df[\"Parent\"][j],\"type\":\"indel\",\"allele1\":ref[i],\"allele2\":alt[i][call_value-1],\"p_value\":0.05},index=[0])],ignore_index=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>strain</th>\n",
       "      <th>ID</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>gene</th>\n",
       "      <th>type</th>\n",
       "      <th>allele1</th>\n",
       "      <th>allele2</th>\n",
       "      <th>p_value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>208</th>\n",
       "      <td>JB953</td>\n",
       "      <td>SPCC895.05</td>\n",
       "      <td>III</td>\n",
       "      <td>1004996</td>\n",
       "      <td>for3</td>\n",
       "      <td>snp</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>209</th>\n",
       "      <td>JB953</td>\n",
       "      <td>SPCC895.05</td>\n",
       "      <td>III</td>\n",
       "      <td>1005353</td>\n",
       "      <td>for3</td>\n",
       "      <td>snp</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210</th>\n",
       "      <td>JB953</td>\n",
       "      <td>SPCC895.05</td>\n",
       "      <td>III</td>\n",
       "      <td>1005518</td>\n",
       "      <td>for3</td>\n",
       "      <td>snp</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>211</th>\n",
       "      <td>JB914</td>\n",
       "      <td>SPCC895.05</td>\n",
       "      <td>III</td>\n",
       "      <td>1006171</td>\n",
       "      <td>for3</td>\n",
       "      <td>snp</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>212</th>\n",
       "      <td>JB953</td>\n",
       "      <td>SPCC895.05</td>\n",
       "      <td>III</td>\n",
       "      <td>1006799</td>\n",
       "      <td>for3</td>\n",
       "      <td>snp</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>213</th>\n",
       "      <td>JB871</td>\n",
       "      <td>SPCC1259.03</td>\n",
       "      <td>III</td>\n",
       "      <td>1040021</td>\n",
       "      <td>rpa12</td>\n",
       "      <td>snp</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>214</th>\n",
       "      <td>JB953</td>\n",
       "      <td>SPCC1259.03</td>\n",
       "      <td>III</td>\n",
       "      <td>1040021</td>\n",
       "      <td>rpa12</td>\n",
       "      <td>snp</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>215</th>\n",
       "      <td>JB953</td>\n",
       "      <td>SPCC1259.03</td>\n",
       "      <td>III</td>\n",
       "      <td>1040280</td>\n",
       "      <td>rpa12</td>\n",
       "      <td>snp</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>216</th>\n",
       "      <td>JB953</td>\n",
       "      <td>SPCC1259.03</td>\n",
       "      <td>III</td>\n",
       "      <td>1040452</td>\n",
       "      <td>rpa12</td>\n",
       "      <td>snp</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>217</th>\n",
       "      <td>JB953</td>\n",
       "      <td>SPCC1259.03</td>\n",
       "      <td>III</td>\n",
       "      <td>1040467</td>\n",
       "      <td>rpa12</td>\n",
       "      <td>snp</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>218</th>\n",
       "      <td>JB953</td>\n",
       "      <td>SPCC1259.03</td>\n",
       "      <td>III</td>\n",
       "      <td>1040509</td>\n",
       "      <td>rpa12</td>\n",
       "      <td>snp</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>219</th>\n",
       "      <td>JB871</td>\n",
       "      <td>SPCC1259.03</td>\n",
       "      <td>III</td>\n",
       "      <td>1040971</td>\n",
       "      <td>rpa12</td>\n",
       "      <td>snp</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>220</th>\n",
       "      <td>JB953</td>\n",
       "      <td>SPCC1259.03</td>\n",
       "      <td>III</td>\n",
       "      <td>1041198</td>\n",
       "      <td>rpa12</td>\n",
       "      <td>snp</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>221</th>\n",
       "      <td>JB871</td>\n",
       "      <td>SPCC1259.03</td>\n",
       "      <td>III</td>\n",
       "      <td>1041201</td>\n",
       "      <td>rpa12</td>\n",
       "      <td>snp</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>222</th>\n",
       "      <td>JB871</td>\n",
       "      <td>SPCC1259.14c</td>\n",
       "      <td>III</td>\n",
       "      <td>1061455</td>\n",
       "      <td>meu27</td>\n",
       "      <td>snp</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>223</th>\n",
       "      <td>JB871</td>\n",
       "      <td>SPCC1259.14c</td>\n",
       "      <td>III</td>\n",
       "      <td>1062455</td>\n",
       "      <td>meu27</td>\n",
       "      <td>snp</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>224</th>\n",
       "      <td>JB871</td>\n",
       "      <td>SPCC1259.14c</td>\n",
       "      <td>III</td>\n",
       "      <td>1063099</td>\n",
       "      <td>meu27</td>\n",
       "      <td>snp</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>225</th>\n",
       "      <td>JB871</td>\n",
       "      <td>SPCC1450.05c</td>\n",
       "      <td>III</td>\n",
       "      <td>1734146</td>\n",
       "      <td>rox3</td>\n",
       "      <td>snp</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>226</th>\n",
       "      <td>JB871</td>\n",
       "      <td>SPCC1450.05c</td>\n",
       "      <td>III</td>\n",
       "      <td>1734959</td>\n",
       "      <td>rox3</td>\n",
       "      <td>snp</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>227</th>\n",
       "      <td>JB871</td>\n",
       "      <td>SPNCRNA.1234</td>\n",
       "      <td>III</td>\n",
       "      <td>1837529</td>\n",
       "      <td>None</td>\n",
       "      <td>snp</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>228</th>\n",
       "      <td>JB953</td>\n",
       "      <td>SPNCRNA.1234</td>\n",
       "      <td>III</td>\n",
       "      <td>1837628</td>\n",
       "      <td>None</td>\n",
       "      <td>snp</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>229</th>\n",
       "      <td>JB871</td>\n",
       "      <td>SPNCRNA.1234</td>\n",
       "      <td>III</td>\n",
       "      <td>1837786</td>\n",
       "      <td>None</td>\n",
       "      <td>snp</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>230</th>\n",
       "      <td>JB953</td>\n",
       "      <td>SPNCRNA.1234</td>\n",
       "      <td>III</td>\n",
       "      <td>1838396</td>\n",
       "      <td>None</td>\n",
       "      <td>snp</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>231</th>\n",
       "      <td>JB871</td>\n",
       "      <td>SPCC74.05</td>\n",
       "      <td>III</td>\n",
       "      <td>1939313</td>\n",
       "      <td>rpl2702</td>\n",
       "      <td>snp</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>232</th>\n",
       "      <td>JB953</td>\n",
       "      <td>SPCC74.05</td>\n",
       "      <td>III</td>\n",
       "      <td>1939313</td>\n",
       "      <td>rpl2702</td>\n",
       "      <td>snp</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233</th>\n",
       "      <td>JB914</td>\n",
       "      <td>SPCC74.05</td>\n",
       "      <td>III</td>\n",
       "      <td>1939314</td>\n",
       "      <td>rpl2702</td>\n",
       "      <td>snp</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>234</th>\n",
       "      <td>JB953</td>\n",
       "      <td>SPCC74.05</td>\n",
       "      <td>III</td>\n",
       "      <td>1939508</td>\n",
       "      <td>rpl2702</td>\n",
       "      <td>snp</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>235</th>\n",
       "      <td>JB848</td>\n",
       "      <td>SPCC576.11</td>\n",
       "      <td>III</td>\n",
       "      <td>2100079</td>\n",
       "      <td>rpl15</td>\n",
       "      <td>snp</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236</th>\n",
       "      <td>JB871</td>\n",
       "      <td>SPCC576.11</td>\n",
       "      <td>III</td>\n",
       "      <td>2100593</td>\n",
       "      <td>rpl15</td>\n",
       "      <td>snp</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>237</th>\n",
       "      <td>JB871</td>\n",
       "      <td>SPAC5D6.05</td>\n",
       "      <td>I</td>\n",
       "      <td>1504578</td>\n",
       "      <td>med18</td>\n",
       "      <td>indel</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>238</th>\n",
       "      <td>JB871</td>\n",
       "      <td>SPAC5D6.05</td>\n",
       "      <td>I</td>\n",
       "      <td>1504592</td>\n",
       "      <td>med18</td>\n",
       "      <td>indel</td>\n",
       "      <td>A</td>\n",
       "      <td>ATG</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>239</th>\n",
       "      <td>JB871</td>\n",
       "      <td>SPNCRNA.781</td>\n",
       "      <td>I</td>\n",
       "      <td>1954990</td>\n",
       "      <td>None</td>\n",
       "      <td>indel</td>\n",
       "      <td>T</td>\n",
       "      <td>TA</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>240</th>\n",
       "      <td>JB871</td>\n",
       "      <td>SPNCRNA.781</td>\n",
       "      <td>I</td>\n",
       "      <td>1955306</td>\n",
       "      <td>None</td>\n",
       "      <td>indel</td>\n",
       "      <td>TA</td>\n",
       "      <td>T</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>241</th>\n",
       "      <td>JB953</td>\n",
       "      <td>SPNCRNA.781</td>\n",
       "      <td>I</td>\n",
       "      <td>1955306</td>\n",
       "      <td>None</td>\n",
       "      <td>indel</td>\n",
       "      <td>TA</td>\n",
       "      <td>T</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>242</th>\n",
       "      <td>JB871</td>\n",
       "      <td>SPNCRNA.781</td>\n",
       "      <td>I</td>\n",
       "      <td>1956409</td>\n",
       "      <td>None</td>\n",
       "      <td>indel</td>\n",
       "      <td>A</td>\n",
       "      <td>AAATTAATTAATT</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>243</th>\n",
       "      <td>JB1117</td>\n",
       "      <td>SPNCRNA.781</td>\n",
       "      <td>I</td>\n",
       "      <td>1956475</td>\n",
       "      <td>None</td>\n",
       "      <td>indel</td>\n",
       "      <td>T</td>\n",
       "      <td>TGGGAGCGAGA</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>244</th>\n",
       "      <td>JB1117</td>\n",
       "      <td>SPNCRNA.781</td>\n",
       "      <td>I</td>\n",
       "      <td>1956475</td>\n",
       "      <td>None</td>\n",
       "      <td>indel</td>\n",
       "      <td>T</td>\n",
       "      <td>TTTACAAACGAAAAATGGGAGCGAGA</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>245</th>\n",
       "      <td>JB871</td>\n",
       "      <td>SPAC607.02c</td>\n",
       "      <td>I</td>\n",
       "      <td>2040419</td>\n",
       "      <td>None</td>\n",
       "      <td>indel</td>\n",
       "      <td>G</td>\n",
       "      <td>GTAA</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>246</th>\n",
       "      <td>JB953</td>\n",
       "      <td>SPAC1783.04c</td>\n",
       "      <td>I</td>\n",
       "      <td>2193230</td>\n",
       "      <td>hst4</td>\n",
       "      <td>indel</td>\n",
       "      <td>C</td>\n",
       "      <td>CTT</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>247</th>\n",
       "      <td>JB871</td>\n",
       "      <td>SPAC6B12.02c</td>\n",
       "      <td>I</td>\n",
       "      <td>2408225</td>\n",
       "      <td>mus7</td>\n",
       "      <td>indel</td>\n",
       "      <td>AAAAATAAAT</td>\n",
       "      <td>A</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248</th>\n",
       "      <td>JB848</td>\n",
       "      <td>SPAC6B12.02c</td>\n",
       "      <td>I</td>\n",
       "      <td>2408225</td>\n",
       "      <td>mus7</td>\n",
       "      <td>indel</td>\n",
       "      <td>AAAAATAAAT</td>\n",
       "      <td>A</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249</th>\n",
       "      <td>JB871</td>\n",
       "      <td>SPAC823.05c</td>\n",
       "      <td>I</td>\n",
       "      <td>2589315</td>\n",
       "      <td>tlg2</td>\n",
       "      <td>indel</td>\n",
       "      <td>CG</td>\n",
       "      <td>C</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>250</th>\n",
       "      <td>JB848</td>\n",
       "      <td>SPAC823.05c</td>\n",
       "      <td>I</td>\n",
       "      <td>2589992</td>\n",
       "      <td>tlg2</td>\n",
       "      <td>indel</td>\n",
       "      <td>CT</td>\n",
       "      <td>C</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>251</th>\n",
       "      <td>JB872</td>\n",
       "      <td>SPNCRNA.900</td>\n",
       "      <td>I</td>\n",
       "      <td>3300652</td>\n",
       "      <td>None</td>\n",
       "      <td>indel</td>\n",
       "      <td>A</td>\n",
       "      <td>AAAATAAAATAAAATAAAATAAAATAAAAT</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>252</th>\n",
       "      <td>JB848</td>\n",
       "      <td>SPNCRNA.900</td>\n",
       "      <td>I</td>\n",
       "      <td>3301581</td>\n",
       "      <td>None</td>\n",
       "      <td>indel</td>\n",
       "      <td>C</td>\n",
       "      <td>CCTCTTTTCTCTCCTCTCTCCTCTCTCCT</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>253</th>\n",
       "      <td>JB848</td>\n",
       "      <td>SPBC23G7.12c</td>\n",
       "      <td>II</td>\n",
       "      <td>2122791</td>\n",
       "      <td>rpt6</td>\n",
       "      <td>indel</td>\n",
       "      <td>T</td>\n",
       "      <td>TA</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>254</th>\n",
       "      <td>JB914</td>\n",
       "      <td>SPBC12D12.06</td>\n",
       "      <td>II</td>\n",
       "      <td>2316851</td>\n",
       "      <td>srb11</td>\n",
       "      <td>indel</td>\n",
       "      <td>TA</td>\n",
       "      <td>T</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>255</th>\n",
       "      <td>JB869</td>\n",
       "      <td>SPBC12D12.06</td>\n",
       "      <td>II</td>\n",
       "      <td>2316851</td>\n",
       "      <td>srb11</td>\n",
       "      <td>indel</td>\n",
       "      <td>TA</td>\n",
       "      <td>T</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>256</th>\n",
       "      <td>JB871</td>\n",
       "      <td>SPBC405.07</td>\n",
       "      <td>II</td>\n",
       "      <td>3153756</td>\n",
       "      <td>rpl3602</td>\n",
       "      <td>indel</td>\n",
       "      <td>T</td>\n",
       "      <td>TTA</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>257</th>\n",
       "      <td>JB953</td>\n",
       "      <td>SPBC16G5.15c</td>\n",
       "      <td>II</td>\n",
       "      <td>4239834</td>\n",
       "      <td>fkh2</td>\n",
       "      <td>indel</td>\n",
       "      <td>T</td>\n",
       "      <td>TA</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     strain            ID chromosome position     gene   type     allele1  \\\n",
       "208   JB953    SPCC895.05        III  1004996     for3    snp           T   \n",
       "209   JB953    SPCC895.05        III  1005353     for3    snp           C   \n",
       "210   JB953    SPCC895.05        III  1005518     for3    snp           C   \n",
       "211   JB914    SPCC895.05        III  1006171     for3    snp           C   \n",
       "212   JB953    SPCC895.05        III  1006799     for3    snp           C   \n",
       "213   JB871   SPCC1259.03        III  1040021    rpa12    snp           C   \n",
       "214   JB953   SPCC1259.03        III  1040021    rpa12    snp           C   \n",
       "215   JB953   SPCC1259.03        III  1040280    rpa12    snp           G   \n",
       "216   JB953   SPCC1259.03        III  1040452    rpa12    snp           C   \n",
       "217   JB953   SPCC1259.03        III  1040467    rpa12    snp           G   \n",
       "218   JB953   SPCC1259.03        III  1040509    rpa12    snp           T   \n",
       "219   JB871   SPCC1259.03        III  1040971    rpa12    snp           T   \n",
       "220   JB953   SPCC1259.03        III  1041198    rpa12    snp           C   \n",
       "221   JB871   SPCC1259.03        III  1041201    rpa12    snp           T   \n",
       "222   JB871  SPCC1259.14c        III  1061455    meu27    snp           A   \n",
       "223   JB871  SPCC1259.14c        III  1062455    meu27    snp           C   \n",
       "224   JB871  SPCC1259.14c        III  1063099    meu27    snp           T   \n",
       "225   JB871  SPCC1450.05c        III  1734146     rox3    snp           G   \n",
       "226   JB871  SPCC1450.05c        III  1734959     rox3    snp           T   \n",
       "227   JB871  SPNCRNA.1234        III  1837529     None    snp           C   \n",
       "228   JB953  SPNCRNA.1234        III  1837628     None    snp           T   \n",
       "229   JB871  SPNCRNA.1234        III  1837786     None    snp           A   \n",
       "230   JB953  SPNCRNA.1234        III  1838396     None    snp           G   \n",
       "231   JB871     SPCC74.05        III  1939313  rpl2702    snp           C   \n",
       "232   JB953     SPCC74.05        III  1939313  rpl2702    snp           C   \n",
       "233   JB914     SPCC74.05        III  1939314  rpl2702    snp           C   \n",
       "234   JB953     SPCC74.05        III  1939508  rpl2702    snp           G   \n",
       "235   JB848    SPCC576.11        III  2100079    rpl15    snp           C   \n",
       "236   JB871    SPCC576.11        III  2100593    rpl15    snp           T   \n",
       "237   JB871    SPAC5D6.05          I  1504578    med18  indel           G   \n",
       "238   JB871    SPAC5D6.05          I  1504592    med18  indel           A   \n",
       "239   JB871   SPNCRNA.781          I  1954990     None  indel           T   \n",
       "240   JB871   SPNCRNA.781          I  1955306     None  indel          TA   \n",
       "241   JB953   SPNCRNA.781          I  1955306     None  indel          TA   \n",
       "242   JB871   SPNCRNA.781          I  1956409     None  indel           A   \n",
       "243  JB1117   SPNCRNA.781          I  1956475     None  indel           T   \n",
       "244  JB1117   SPNCRNA.781          I  1956475     None  indel           T   \n",
       "245   JB871   SPAC607.02c          I  2040419     None  indel           G   \n",
       "246   JB953  SPAC1783.04c          I  2193230     hst4  indel           C   \n",
       "247   JB871  SPAC6B12.02c          I  2408225     mus7  indel  AAAAATAAAT   \n",
       "248   JB848  SPAC6B12.02c          I  2408225     mus7  indel  AAAAATAAAT   \n",
       "249   JB871   SPAC823.05c          I  2589315     tlg2  indel          CG   \n",
       "250   JB848   SPAC823.05c          I  2589992     tlg2  indel          CT   \n",
       "251   JB872   SPNCRNA.900          I  3300652     None  indel           A   \n",
       "252   JB848   SPNCRNA.900          I  3301581     None  indel           C   \n",
       "253   JB848  SPBC23G7.12c         II  2122791     rpt6  indel           T   \n",
       "254   JB914  SPBC12D12.06         II  2316851    srb11  indel          TA   \n",
       "255   JB869  SPBC12D12.06         II  2316851    srb11  indel          TA   \n",
       "256   JB871    SPBC405.07         II  3153756  rpl3602  indel           T   \n",
       "257   JB953  SPBC16G5.15c         II  4239834     fkh2  indel           T   \n",
       "\n",
       "                            allele2  p_value  \n",
       "208                               C     0.05  \n",
       "209                               T     0.05  \n",
       "210                               T     0.05  \n",
       "211                               T     0.05  \n",
       "212                               T     0.05  \n",
       "213                               T     0.05  \n",
       "214                               T     0.05  \n",
       "215                               T     0.05  \n",
       "216                               A     0.05  \n",
       "217                               A     0.05  \n",
       "218                               C     0.05  \n",
       "219                               A     0.05  \n",
       "220                               A     0.05  \n",
       "221                               C     0.05  \n",
       "222                               G     0.05  \n",
       "223                               T     0.05  \n",
       "224                               C     0.05  \n",
       "225                               A     0.05  \n",
       "226                               C     0.05  \n",
       "227                               T     0.05  \n",
       "228                               C     0.05  \n",
       "229                               G     0.05  \n",
       "230                               A     0.05  \n",
       "231                               T     0.05  \n",
       "232                               T     0.05  \n",
       "233                               T     0.05  \n",
       "234                               A     0.05  \n",
       "235                               T     0.05  \n",
       "236                               G     0.05  \n",
       "237                              GA     0.05  \n",
       "238                             ATG     0.05  \n",
       "239                              TA     0.05  \n",
       "240                               T     0.05  \n",
       "241                               T     0.05  \n",
       "242                   AAATTAATTAATT     0.05  \n",
       "243                     TGGGAGCGAGA     0.05  \n",
       "244      TTTACAAACGAAAAATGGGAGCGAGA     0.05  \n",
       "245                            GTAA     0.05  \n",
       "246                             CTT     0.05  \n",
       "247                               A     0.05  \n",
       "248                               A     0.05  \n",
       "249                               C     0.05  \n",
       "250                               C     0.05  \n",
       "251  AAAATAAAATAAAATAAAATAAAATAAAAT     0.05  \n",
       "252   CCTCTTTTCTCTCCTCTCTCCTCTCTCCT     0.05  \n",
       "253                              TA     0.05  \n",
       "254                               T     0.05  \n",
       "255                               T     0.05  \n",
       "256                             TTA     0.05  \n",
       "257                              TA     0.05  "
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "results.tail(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [],
   "source": [
    "#make csv file\n",
    "results.to_csv(\"/Users/k23030440/github_projects/S.-Pombe-MLPs/Bence folder/Looking for variants in wild isolates/snp_hunting.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>strain</th>\n",
       "      <th>ID</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>gene</th>\n",
       "      <th>type</th>\n",
       "      <th>allele1</th>\n",
       "      <th>allele2</th>\n",
       "      <th>p_value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>253</th>\n",
       "      <td>JB848</td>\n",
       "      <td>SPBC23G7.12c</td>\n",
       "      <td>II</td>\n",
       "      <td>2122791</td>\n",
       "      <td>rpt6</td>\n",
       "      <td>indel</td>\n",
       "      <td>T</td>\n",
       "      <td>TA</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>254</th>\n",
       "      <td>JB914</td>\n",
       "      <td>SPBC12D12.06</td>\n",
       "      <td>II</td>\n",
       "      <td>2316851</td>\n",
       "      <td>srb11</td>\n",
       "      <td>indel</td>\n",
       "      <td>TA</td>\n",
       "      <td>T</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>255</th>\n",
       "      <td>JB869</td>\n",
       "      <td>SPBC12D12.06</td>\n",
       "      <td>II</td>\n",
       "      <td>2316851</td>\n",
       "      <td>srb11</td>\n",
       "      <td>indel</td>\n",
       "      <td>TA</td>\n",
       "      <td>T</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>256</th>\n",
       "      <td>JB871</td>\n",
       "      <td>SPBC405.07</td>\n",
       "      <td>II</td>\n",
       "      <td>3153756</td>\n",
       "      <td>rpl3602</td>\n",
       "      <td>indel</td>\n",
       "      <td>T</td>\n",
       "      <td>TTA</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>257</th>\n",
       "      <td>JB953</td>\n",
       "      <td>SPBC16G5.15c</td>\n",
       "      <td>II</td>\n",
       "      <td>4239834</td>\n",
       "      <td>fkh2</td>\n",
       "      <td>indel</td>\n",
       "      <td>T</td>\n",
       "      <td>TA</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    strain            ID chromosome position     gene   type allele1 allele2  \\\n",
       "253  JB848  SPBC23G7.12c         II  2122791     rpt6  indel       T      TA   \n",
       "254  JB914  SPBC12D12.06         II  2316851    srb11  indel      TA       T   \n",
       "255  JB869  SPBC12D12.06         II  2316851    srb11  indel      TA       T   \n",
       "256  JB871    SPBC405.07         II  3153756  rpl3602  indel       T     TTA   \n",
       "257  JB953  SPBC16G5.15c         II  4239834     fkh2  indel       T      TA   \n",
       "\n",
       "     p_value  \n",
       "253     0.05  \n",
       "254     0.05  \n",
       "255     0.05  \n",
       "256     0.05  \n",
       "257     0.05  "
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "results.tail(5)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The watermark extension is already loaded. To reload it, use:\n",
      "  %reload_ext watermark\n",
      "Last updated: 2023-11-03T18:58:41.209602+00:00\n",
      "\n",
      "Python implementation: CPython\n",
      "Python version       : 3.9.0\n",
      "IPython version      : 7.31.1\n",
      "\n",
      "Compiler    : Clang 11.0.0 \n",
      "OS          : Darwin\n",
      "Release     : 21.5.0\n",
      "Machine     : x86_64\n",
      "Processor   : i386\n",
      "CPU cores   : 8\n",
      "Architecture: 64bit\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%load_ext watermark\n",
    "%watermark\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "numpy     : 1.23.4\n",
      "matplotlib: 3.8.0\n",
      "allel     : 1.3.5\n",
      "pandas    : 1.4.4\n",
      "bokeh     : 3.2.2\n",
      "scipy     : 1.11.3\n",
      "iqplot    : 0.3.3\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%watermark --iversions\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.4"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "2d0398177d7bb10ea33b6a5cd9d022d63e20680fcfd84762562c72ff35bdf44b"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
