{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "#find snps\n",
    "#import data science libraries\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "#read in data\n",
    "root = \"/Users/k23030440/S.-Pombe-MLPs - most recent/\"\n",
    "snps = pd.read_csv(root + 'Bence folder/QTL mapping/permutations/results_new_filtering.csv')\n",
    "#washing phenotype /Users/bencekover/Library/CloudStorage/OneDrive-UniversityCollegeLondon/MSci Bahler lab/S.-Pombe-biofilm/Bence folder/QTL mapping/washing_phenotypes.csv\n",
    "wash_pheno = pd.read_csv(root + 'Bence folder/QTL mapping/washing_phenotypes.csv')\n",
    "#import rna seq corrs /Users/bencekover/Library/CloudStorage/OneDrive-UniversityCollegeLondon/MSci Bahler lab/S.-Pombe-biofilm/Bence folder/Analysis of Clement-Ziza RNA-seq/wash_corr.csv\n",
    "wash_corr = pd.read_csv(root + 'Bence folder/Analysis of Clement-Ziza RNA-seq/wash_corr.csv')\n",
    "#full snp dataset /Users/bencekover/Library/CloudStorage/OneDrive-UniversityCollegeLondon/MSci Bahler lab/S.-Pombe-biofilm/Bence folder/QTL mapping/SupplementaryDataset_S7_genotype.tsv\n",
    "full_snps = pd.read_csv(root + 'internal data/updated_genotype_matrix_final_pos.tsv', sep='\\t', header=0)\n",
    "#load orthologs database /Users/bencekover/Library/CloudStorage/OneDrive-UniversityCollegeLondon/MSci Bahler lab/S.-Pombe-biofilm/Bence folder/Orthologs/final_orthology_database.csv\n",
    "orthologs = pd.read_csv(root + 'Bence folder/Orthologs/final_orthology_database.csv')\n",
    "full_snps_og = full_snps.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/g6/4flmdffs7l35hmmt1wkgrfcm0000gr/T/ipykernel_25014/3198399901.py:11: FutureWarning: The behavior of DataFrame concatenation with empty or all-NA entries is deprecated. In a future version, this will no longer exclude empty or all-NA columns when determining the result dtypes. To retain the old behavior, exclude the relevant entries before the concat operation.\n",
      "  pca_pheno = pd.concat([pca_pheno, row])\n"
     ]
    }
   ],
   "source": [
    "#remove first 4 columns\n",
    "full_snps = full_snps.iloc[:,4:]\n",
    "#rename column 968 to X968\n",
    "full_snps.rename(columns={'968':'X968'}, inplace=True)\n",
    "#re order wash_pheno such that order of strains matches columns in full_snps column names\n",
    "pca_pheno = pd.DataFrame(columns = ['strain','ratio'])\n",
    "for i in full_snps.columns:\n",
    "    #see what is the column name, and find the row in wash_pheno where strain has the same name\n",
    "    \n",
    "    row = wash_pheno.loc[wash_pheno['strain'] == i]\n",
    "    pca_pheno = pd.concat([pca_pheno, row])\n",
    "#pcapheno remove column Unnamed: 0\n",
    "pca_pheno.drop(columns=['Unnamed: 0'], inplace=True)\n",
    "#reset index\n",
    "pca_pheno.reset_index(inplace=True)\n",
    "#remove chromosome_ from chromosome column\n",
    "snps['chromosome'] = snps['chromosome'].str.replace('chromosome_', '')\n",
    "#change chromosome number to roman numbers\n",
    "snps['chromosome'] = snps['chromosome'].str.replace('1','I')\n",
    "snps['chromosome'] = snps['chromosome'].str.replace('2','II')\n",
    "snps['chromosome'] = snps['chromosome'].str.replace('3','III')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>968_allele</th>\n",
       "      <th>Y0036_allele</th>\n",
       "      <th>968</th>\n",
       "      <th>Y0036</th>\n",
       "      <th>R1_1</th>\n",
       "      <th>R1_2</th>\n",
       "      <th>R1_3</th>\n",
       "      <th>R1_5</th>\n",
       "      <th>...</th>\n",
       "      <th>R1_9</th>\n",
       "      <th>R1_34</th>\n",
       "      <th>R1_44</th>\n",
       "      <th>R1_49</th>\n",
       "      <th>R1_50</th>\n",
       "      <th>R1_51</th>\n",
       "      <th>R1_52</th>\n",
       "      <th>R1_53</th>\n",
       "      <th>R1_54</th>\n",
       "      <th>R1_4</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>chromosome_1</td>\n",
       "      <td>162963</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>chromosome_1</td>\n",
       "      <td>163061</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>chromosome_1</td>\n",
       "      <td>164921</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>chromosome_1</td>\n",
       "      <td>165673</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>chromosome_1</td>\n",
       "      <td>168534</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>chromosome_1</td>\n",
       "      <td>170928</td>\n",
       "      <td>C</td>\n",
       "      <td>CT</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>chromosome_1</td>\n",
       "      <td>173079</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>chromosome_1</td>\n",
       "      <td>173651</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8 rows × 58 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      chromosome  position 968_allele Y0036_allele  968  Y0036  R1_1  R1_2  \\\n",
       "52  chromosome_1    162963          G            A    0    1.0   1.0   0.0   \n",
       "53  chromosome_1    163061          C            T    0    1.0   1.0   0.0   \n",
       "54  chromosome_1    164921          G            A    0    1.0   1.0   0.0   \n",
       "55  chromosome_1    165673          A            G    0    1.0   1.0   0.0   \n",
       "56  chromosome_1    168534          T            G    0    1.0   1.0   0.0   \n",
       "57  chromosome_1    170928          C           CT    0    1.0   1.0   0.0   \n",
       "58  chromosome_1    173079          A            C    0    1.0   1.0   0.0   \n",
       "59  chromosome_1    173651          T            G    0    1.0   1.0   0.0   \n",
       "\n",
       "    R1_3  R1_5  ...  R1_9  R1_34  R1_44  R1_49  R1_50  R1_51  R1_52  R1_53  \\\n",
       "52   0.0   1.0  ...   0.0    1.0    0.0    NaN    0.0    0.0    0.0    1.0   \n",
       "53   0.0   1.0  ...   0.0    1.0    0.0    NaN    0.0    0.0    0.0    1.0   \n",
       "54   0.0   1.0  ...   0.0    1.0    0.0    NaN    0.0    0.0    0.0    1.0   \n",
       "55   0.0   1.0  ...   0.0    1.0    0.0    NaN    0.0    0.0    0.0    1.0   \n",
       "56   0.0   1.0  ...   0.0    1.0    0.0    NaN    0.0    0.0    0.0    1.0   \n",
       "57   0.0   1.0  ...   0.0    1.0    0.0    NaN    0.0    0.0    0.0    1.0   \n",
       "58   0.0   1.0  ...   0.0    1.0    0.0    NaN    0.0    0.0    0.0    1.0   \n",
       "59   0.0   1.0  ...   0.0    1.0    0.0    NaN    0.0    0.0    0.0    1.0   \n",
       "\n",
       "    R1_54  R1_4  \n",
       "52    1.0   0.0  \n",
       "53    1.0   0.0  \n",
       "54    1.0   0.0  \n",
       "55    1.0   0.0  \n",
       "56    1.0   0.0  \n",
       "57    1.0   0.0  \n",
       "58    1.0   0.0  \n",
       "59    1.0   0.0  \n",
       "\n",
       "[8 rows x 58 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#show snps at chromosome_1 in position +- 10000 around 100011 in full_snps_og\n",
    "full_snps_og.loc[(full_snps_og['chromosome'] == \"chromosome_1\") & (full_snps_og['position'] > 159850) & (full_snps_og['position'] < 179850)]\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "snps_t = snps.transpose()\n",
    "#remove first 5 rows\n",
    "snps_t = snps_t.iloc[5:]\n",
    "#by matching index of snps_t to column strain in wash_pheno, we can add the phenotype to the snps_t dataframe\n",
    "snps_t['phenotype'] = wash_pheno.set_index('strain')['ratio']\n",
    "#rename all but last column to snps df position \n",
    "#snps_t all but last columns rename as snps[\"position\"]\n",
    "ncols = len(snps_t.columns)\n",
    "snps_t.columns = np.append(snps[\"position\"].values, snps_t.columns[ncols-1]).flatten()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "#pandas read in \"/Users/bencekover/Library/CloudStorage/OneDrive-UniversityCollegeLondon/MSci Bahler lab/S.-Pombe-biofilm/external data/Pombe genome annotation.gff3\"\n",
    "import pandas as pd\n",
    "df = pd.read_csv(root + 'external data/Pombase files/Schizosaccharomyces_pombe_all_chromosomes.gff3', sep='\\t', header=1)\n",
    "\n",
    "#break last column into 2 at ;\n",
    "df[['a','b']] = df[\"ID=SPAC1002.01;Name=mrx11\"].str.split(';',expand=True)\n",
    "#remove column ID=SPAC1002.01;Name=mrx11\n",
    "df.drop(columns =[\"ID=SPAC1002.01;Name=mrx11\"], inplace = True)\n",
    "#remove ID= from column a and rename it to ID\n",
    "df['ID'] = df['a'].str.replace('ID=', '')\n",
    "#remove column a\n",
    "df.drop(columns =['a','PomBase',\".\",\"..1\"], inplace = True)\n",
    "#rename b to Parent\n",
    "df.rename(columns={'b':'Parent','gene':'object','I':'chromosome','1798347':\"start\",'1798835':\"end\",\"+\":\"strand\"}, inplace=True)\n",
    "#In parent remove anything before =\n",
    "df['Parent'] = df['Parent'].str.replace('Parent=', '')\n",
    "#or Name=\n",
    "df['Parent'] = df['Parent'].str.replace('Name=', '')\n",
    "#create an aempty column called detail\n",
    "df['detail'] = \"\"\n",
    "#split ID column at :, and if there is something after put it in detail column\n",
    "df['detail'] = df['ID'].str.split(':', expand=True)[1]\n",
    "#split ID column at :, and if there is something before put it in ID column\n",
    "df['ID'] = df['ID'].str.split(':', expand=True)[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "#find a subset of df, where snps[position] is +- 1000 of df[start] or df[end]. Create a result df, and say which snps were there\n",
    "#suppress warnings\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "\n",
    "result = pd.DataFrame()\n",
    "for i in range(len(snps)):\n",
    "    for j in range(len(df)):\n",
    "        if snps['position'][i] >= df['start'][j] - 500 and snps['position'][i] <= df['end'][j] + 500:\n",
    "            #and if chromsome is chromosome_2 for df chromsome II etc\n",
    "            if snps['chromosome'][i] == df['chromosome'][j]:\n",
    "                new_row = pd.DataFrame({'position': [snps['position'][i]], \n",
    "                        'ID': [df['ID'][j]], \n",
    "                        'Parent': [df['Parent'][j]], \n",
    "                        'object': [df['object'][j]], \n",
    "                        'chromosome': [df['chromosome'][j]], \n",
    "                        'start': [df['start'][j]], \n",
    "                        'end': [df['end'][j]], \n",
    "                        'strand': [df['strand'][j]], \n",
    "                        'detail': [df['detail'][j]]})\n",
    "\n",
    "                result = pd.concat([result, new_row], ignore_index=True)\n",
    "    \n",
    "\n",
    "#if the last two characters are .1, remove them\n",
    "for i in range(len(result)):\n",
    "    if result['ID'][i][-2:] == \".1\":\n",
    "        result['ID'][i] = result['ID'][i][:-2]\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "#find the p_value of the snp position matches snps_t column_name\n",
    "#add a column to result called p_value\n",
    "pvals = np.array([])\n",
    "for i in range(len(result)):\n",
    "    for j in range(len(snps_t.columns)):\n",
    "        #skip if colum nname is phenotype\n",
    "        if snps_t.columns[j] == 'phenotype':\n",
    "            continue\n",
    "        if result['position'][i] == int(snps_t.columns[j]):\n",
    "            pvals = np.append(pvals, snps_t.iloc[46,j])\n",
    "result['p_value'] = pvals\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "#match ID with systematic name in wash_corr and add columns gene\tcorrelation\tp-value\n",
    "#create np.arrays with NAs\n",
    "corrs = np.array([np.nan]*len(result))\n",
    "pvals = np.array([np.nan]*len(result))\n",
    "genes = np.array([np.nan]*len(result), dtype=object)\n",
    "for i in range(len(result)):\n",
    "    for j in range(len(wash_corr)):\n",
    "        if result['ID'][i] == wash_corr['systematic_names'][j]:\n",
    "            corrs[i] = wash_corr['correlation'][j]\n",
    "            pvals[i] = wash_corr['p-value'][j]\n",
    "            genes[i] = wash_corr['gene'][j]\n",
    "\n",
    "            \n",
    "\n",
    "\n",
    "result['correlation'] = corrs\n",
    "result['p_value_corr'] = pvals\n",
    "result['gene'] = genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "import sys, re\n",
    "\n",
    "f=open(root + 'external data/Pombase files/Schizosaccharomyces_pombe_all_chromosomes.fa','r')\n",
    "lines=f.readlines()\n",
    "\n",
    "hre=re.compile('>(\\S+)')\n",
    "lre=re.compile('^(\\S+)$')\n",
    "\n",
    "regions={}\n",
    "\n",
    "for line in lines:\n",
    "        outh = hre.search(line)\n",
    "        if outh:\n",
    "                id=outh.group(1)\n",
    "        else:\n",
    "                outl=lre.search(line)\n",
    "                if(id in regions.keys()):\n",
    "                        regions[id] += outl.group(1)\n",
    "                else:\n",
    "                        regions[id]  =outl.group(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>object</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>Parent</th>\n",
       "      <th>ID</th>\n",
       "      <th>detail</th>\n",
       "      <th>exon_number</th>\n",
       "      <th>old_start</th>\n",
       "      <th>old_end</th>\n",
       "      <th>exon_length</th>\n",
       "      <th>CDS_length_so_far</th>\n",
       "      <th>total_CDS_length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>II</td>\n",
       "      <td>CDS</td>\n",
       "      <td>2461357</td>\n",
       "      <td>2461495</td>\n",
       "      <td>-</td>\n",
       "      <td>SPBC12C2.08.1</td>\n",
       "      <td>SPBC12C2.08.1</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2461357</td>\n",
       "      <td>2461495</td>\n",
       "      <td>139</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2347.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>II</td>\n",
       "      <td>CDS</td>\n",
       "      <td>2459018</td>\n",
       "      <td>2461225</td>\n",
       "      <td>-</td>\n",
       "      <td>SPBC12C2.08.1</td>\n",
       "      <td>SPBC12C2.08.1</td>\n",
       "      <td>exon</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2459019</td>\n",
       "      <td>2461225</td>\n",
       "      <td>2208</td>\n",
       "      <td>139.0</td>\n",
       "      <td>2347.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>II</td>\n",
       "      <td>CDS</td>\n",
       "      <td>2452052</td>\n",
       "      <td>2452181</td>\n",
       "      <td>+</td>\n",
       "      <td>SPBC12C2.10c.1</td>\n",
       "      <td>SPBC12C2.10c.1</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2452052</td>\n",
       "      <td>2452181</td>\n",
       "      <td>130</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4566.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>II</td>\n",
       "      <td>CDS</td>\n",
       "      <td>2452319</td>\n",
       "      <td>2456754</td>\n",
       "      <td>+</td>\n",
       "      <td>SPBC12C2.10c.1</td>\n",
       "      <td>SPBC12C2.10c.1</td>\n",
       "      <td>exon</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2452319</td>\n",
       "      <td>2456757</td>\n",
       "      <td>4436</td>\n",
       "      <td>130.0</td>\n",
       "      <td>4566.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>II</td>\n",
       "      <td>CDS</td>\n",
       "      <td>2450850</td>\n",
       "      <td>2450854</td>\n",
       "      <td>-</td>\n",
       "      <td>SPBC12C2.11.1</td>\n",
       "      <td>SPBC12C2.11.1</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2450850</td>\n",
       "      <td>2450854</td>\n",
       "      <td>5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2094.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>5</td>\n",
       "      <td>II</td>\n",
       "      <td>CDS</td>\n",
       "      <td>2450711</td>\n",
       "      <td>2450781</td>\n",
       "      <td>-</td>\n",
       "      <td>SPBC12C2.11.1</td>\n",
       "      <td>SPBC12C2.11.1</td>\n",
       "      <td>exon</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2450711</td>\n",
       "      <td>2450781</td>\n",
       "      <td>71</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2094.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>6</td>\n",
       "      <td>II</td>\n",
       "      <td>CDS</td>\n",
       "      <td>2448488</td>\n",
       "      <td>2450505</td>\n",
       "      <td>-</td>\n",
       "      <td>SPBC12C2.11.1</td>\n",
       "      <td>SPBC12C2.11.1</td>\n",
       "      <td>exon</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2448491</td>\n",
       "      <td>2450505</td>\n",
       "      <td>2018</td>\n",
       "      <td>76.0</td>\n",
       "      <td>2094.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>7</td>\n",
       "      <td>II</td>\n",
       "      <td>CDS</td>\n",
       "      <td>2446698</td>\n",
       "      <td>2447606</td>\n",
       "      <td>+</td>\n",
       "      <td>SPBC12C2.12c.1</td>\n",
       "      <td>SPBC12C2.12c.1</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2446698</td>\n",
       "      <td>2447606</td>\n",
       "      <td>909</td>\n",
       "      <td>0.0</td>\n",
       "      <td>909.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>8</td>\n",
       "      <td>II</td>\n",
       "      <td>CDS</td>\n",
       "      <td>2444329</td>\n",
       "      <td>2445924</td>\n",
       "      <td>+</td>\n",
       "      <td>SPBC12C2.13c.1</td>\n",
       "      <td>SPBC12C2.13c.1</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2444329</td>\n",
       "      <td>2445924</td>\n",
       "      <td>1596</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1596.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>9</td>\n",
       "      <td>II</td>\n",
       "      <td>CDS</td>\n",
       "      <td>2304182</td>\n",
       "      <td>2305726</td>\n",
       "      <td>+</td>\n",
       "      <td>SPBC12D12.01.1</td>\n",
       "      <td>SPBC12D12.01.1</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2304182</td>\n",
       "      <td>2305726</td>\n",
       "      <td>1545</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1545.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   index chromosome object    start      end strand          Parent  \\\n",
       "0      0         II    CDS  2461357  2461495      -   SPBC12C2.08.1   \n",
       "1      1         II    CDS  2459018  2461225      -   SPBC12C2.08.1   \n",
       "2      2         II    CDS  2452052  2452181      +  SPBC12C2.10c.1   \n",
       "3      3         II    CDS  2452319  2456754      +  SPBC12C2.10c.1   \n",
       "4      4         II    CDS  2450850  2450854      -   SPBC12C2.11.1   \n",
       "5      5         II    CDS  2450711  2450781      -   SPBC12C2.11.1   \n",
       "6      6         II    CDS  2448488  2450505      -   SPBC12C2.11.1   \n",
       "7      7         II    CDS  2446698  2447606      +  SPBC12C2.12c.1   \n",
       "8      8         II    CDS  2444329  2445924      +  SPBC12C2.13c.1   \n",
       "9      9         II    CDS  2304182  2305726      +  SPBC12D12.01.1   \n",
       "\n",
       "               ID detail  exon_number  old_start  old_end  exon_length  \\\n",
       "0   SPBC12C2.08.1   exon          1.0    2461357  2461495          139   \n",
       "1   SPBC12C2.08.1   exon          2.0    2459019  2461225         2208   \n",
       "2  SPBC12C2.10c.1   exon          1.0    2452052  2452181          130   \n",
       "3  SPBC12C2.10c.1   exon          2.0    2452319  2456757         4436   \n",
       "4   SPBC12C2.11.1   exon          1.0    2450850  2450854            5   \n",
       "5   SPBC12C2.11.1   exon          2.0    2450711  2450781           71   \n",
       "6   SPBC12C2.11.1   exon          3.0    2448491  2450505         2018   \n",
       "7  SPBC12C2.12c.1   exon          1.0    2446698  2447606          909   \n",
       "8  SPBC12C2.13c.1   exon          1.0    2444329  2445924         1596   \n",
       "9  SPBC12D12.01.1   exon          1.0    2304182  2305726         1545   \n",
       "\n",
       "   CDS_length_so_far  total_CDS_length  \n",
       "0                0.0            2347.0  \n",
       "1              139.0            2347.0  \n",
       "2                0.0            4566.0  \n",
       "3              130.0            4566.0  \n",
       "4                0.0            2094.0  \n",
       "5                5.0            2094.0  \n",
       "6               76.0            2094.0  \n",
       "7                0.0             909.0  \n",
       "8                0.0            1596.0  \n",
       "9                0.0            1545.0  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#result_final___ only keep rows where snp is between start and end\n",
    "result_final__ = result.copy()\n",
    "result_final___ = result_final__.copy()\n",
    "result_final___ = result_final___[(result_final___['position'] >= result_final___['start']-10) & (result_final___['position'] <= result_final___['end']+100)]\n",
    "\n",
    "gene_exons = df.copy()\n",
    "#only keep rows where object is CDS\n",
    "gene_exons = gene_exons[gene_exons['object'] == 'CDS']\n",
    "#only keep rows where ID is in result_final__['ID']\n",
    "gene_exons_final = pd.DataFrame()\n",
    "for i in np.unique(result_final___['ID']):\n",
    "    df_subset = gene_exons[gene_exons['ID'] == str(i + \".1\")]\n",
    "    gene_exons_final = pd.concat([gene_exons_final, df_subset], ignore_index=True)\n",
    "#reset index\n",
    "gene_exons_final.reset_index(inplace=True)\n",
    "#make a column exon_number\n",
    "gene_exons_final['exon_number'] = np.nan\n",
    "#for those where ID is multiple times, number the exons by increasing start if strand is +, or decreasing start if strand is -\n",
    "for i in np.unique(gene_exons_final['ID']):\n",
    "    df_subset = gene_exons_final[gene_exons_final['ID'] == i]\n",
    "    if df_subset['strand'].values[0]== '+':\n",
    "        df_subset = df_subset.sort_values(by=['start'], ascending=True)\n",
    "    if df_subset['strand'].values[0] == '-':\n",
    "        df_subset = df_subset.sort_values(by=['end'], ascending=False)\n",
    "    df_subset['exon_number'] = np.arange(1, len(df_subset) + 1)\n",
    "    gene_exons_final[gene_exons_final['ID'] == i] = df_subset\n",
    "\n",
    "\n",
    "#REMOVE entries wehre ID contains SPNCRNA\n",
    "gene_exons_final = gene_exons_final[~gene_exons_final['ID'].str.contains('SPNCRNA')]\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "#make a column called real_start\n",
    "gene_exons_final['old_start'] = gene_exons_final['start']\n",
    "gene_exons_final['old_end'] = gene_exons_final['end']\n",
    "#for rows where strand is + look at the load the bases +- 5 around the start from regions\n",
    "for i in np.unique(gene_exons_final['ID']):\n",
    "    df_subset = gene_exons_final[gene_exons_final['ID'] == i]\n",
    "    df_subset = df_subset[df_subset['exon_number'] == 1]\n",
    "    if df_subset['strand'].values[0]== '+':\n",
    "        seq = regions[df_subset['chromosome'].values[0]][int(df_subset['start'].values[0]-5):int(df_subset['start'].values[0]+5)]\n",
    "        #where does ATG occur in seq\n",
    "        real_start = seq.find('ATG')-4\n",
    "        real_start = df_subset['start'] + real_start\n",
    "        #for the row in gene_exons_final where ID is i and exon_number is 1, change the start to real_start\n",
    "        gene_exons_final.loc[(gene_exons_final['ID'] == i) & (gene_exons_final['exon_number'] == 1), 'start'] = real_start\n",
    "\n",
    "\n",
    "\n",
    "    else:\n",
    "        seq_rev = regions[df_subset['chromosome'].values[0]][int(df_subset['end'].values[0]-5):int(df_subset['end'].values[0]+5)]\n",
    "        real_end = seq_rev.find('CAT')+2-4\n",
    "        real_end = df_subset['end'] + real_end\n",
    "        #for the row in gene_exons_final where ID is i and exon_number is 1, change the end to real_end\n",
    "        gene_exons_final.loc[(gene_exons_final['ID'] == i) & (gene_exons_final['exon_number'] == 1), 'end'] = real_end\n",
    "\n",
    "\n",
    "    df_subset = gene_exons_final[gene_exons_final['ID'] == i]\n",
    "    highest_exon = df_subset['exon_number'].max()\n",
    "    df_subset = df_subset[df_subset['exon_number'] == highest_exon]\n",
    "    if df_subset['strand'].values[0]== '+':\n",
    "\n",
    "        seq2 = regions[df_subset['chromosome'].values[0]][int(df_subset['end'].values[0]-5):int(df_subset['end'].values[0]+5)]\n",
    "        real_end = seq2.find('TAA')+2-4\n",
    "        real_end = df_subset['end'] + real_end\n",
    "        #for the row in gene_exons_final where ID is i and exon_number is highest_exon, change the end to real_end\n",
    "        gene_exons_final.loc[(gene_exons_final['ID'] == i) & (gene_exons_final['exon_number'] == highest_exon), 'end'] = real_end\n",
    "    else:\n",
    "        seq2_rev = regions[df_subset['chromosome'].values[0]][int(df_subset['start'].values[0]-5):int(df_subset['start'].values[0]+5)]\n",
    "        real_start = seq2_rev.find('TCA')+2-4\n",
    "        real_start = df_subset['start'] + real_start\n",
    "        #for the row in gene_exons_final where ID is i and exon_number is highest_exon, change the start to real_start\n",
    "        gene_exons_final.loc[(gene_exons_final['ID'] == i) & (gene_exons_final['exon_number'] == highest_exon), 'start'] = real_start\n",
    "\n",
    "\n",
    "    \n",
    "    \n",
    "\n",
    "\n",
    "\n",
    "#for each row calculate the length of the exon\n",
    "gene_exons_final['exon_length'] = abs(gene_exons_final['end'] - gene_exons_final['start'])+1\n",
    "\n",
    "#remove rows where ID is NaN\n",
    "gene_exons_final = gene_exons_final[~gene_exons_final['ID'].isna()]\n",
    "#for each row calculate the length of the CDS until that exon starts\n",
    "gene_exons_final['CDS_length_so_far'] = np.nan\n",
    "for i in np.unique(gene_exons_final['ID']):\n",
    "    df_subset = gene_exons_final[gene_exons_final['ID'] == i]\n",
    "    #order by exon_number\n",
    "    df_subset = df_subset.sort_values(by=['exon_number'])\n",
    "    \n",
    "    df_subset['CDS_length_so_far'] = np.append(np.array(0),df_subset['exon_length'].cumsum())[:-1]\n",
    "    gene_exons_final[gene_exons_final['ID'] == i] = df_subset\n",
    "#add total cds length\n",
    "gene_exons_final['total_CDS_length'] = np.nan\n",
    "for i in np.unique(gene_exons_final['ID']):\n",
    "    df_subset = gene_exons_final[gene_exons_final['ID'] == i]\n",
    "    df_subset['total_CDS_length'] = sum(df_subset['exon_length'])\n",
    "    gene_exons_final[gene_exons_final['ID'] == i] = df_subset\n",
    "\n",
    "gene_exons_final.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "#add to each row the unique number of positions that ID had\n",
    "#add a column called unique\n",
    "#remove duplicates of ID and position\n",
    "#keep only entries with \"object\" is CDS\n",
    "result = result[result['object'] == 'CDS']\n",
    "result_final = result.drop_duplicates(subset=['ID', 'position'])\n",
    "#reset index\n",
    "result_final = result_final.reset_index(drop=True)\n",
    "unique = np.array([])\n",
    "for i in range(len(result_final)):\n",
    "    #get the ID\n",
    "    ID = result_final['ID'][i]\n",
    "    #get the number of times ID appears in result\n",
    "    count = len(result_final[result_final['ID'] == ID])\n",
    "    #add count to unique\n",
    "    unique = np.append(unique, count)\n",
    "result_final['unique'] = unique\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "#show SPBC290.07 in result\n",
    "#gather positions for each ID, and put in a single entry as a list. Add that to the df. THen remove duplicated IDs\n",
    "result_final_ = result_final.groupby('ID')['position'].apply(list).reset_index(name='positions')\n",
    "result_final__ = pd.merge(result_final, result_final_, on='ID')\n",
    "\n",
    "#remove duplicates of ID\n",
    "result_final__ = result_final__.drop_duplicates(subset=['ID'])\n",
    "\n",
    "#reset index\n",
    "result_final__ = result_final__.reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "#iterate through result_final__. If ID is found in orthologs, then add the albicans_name(s)cerevisiae_name(s) to the df\n",
    "from numpy import dtype\n",
    "\n",
    "albicans_names = np.array([np.nan]*len(result_final__),dtype=object)\n",
    "cerevisiae_names = np.array([np.nan]*len(result_final__),dtype=object)\n",
    "for i in range(len(result_final__)):\n",
    "    for j in range(len(orthologs)):\n",
    "        if result_final__['ID'][i] in orthologs['pombe_name(s)'][j]:\n",
    "            albicans_names[i] = orthologs['albicans_name(s)'][j]\n",
    "            cerevisiae_names[i] = orthologs['cerevisiae_name(s)'][j]\n",
    "#add to df\n",
    "result_final__['albicans_name(s)'] = albicans_names\n",
    "result_final__['cerevisiae_name(s)'] = cerevisiae_names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>position</th>\n",
       "      <th>ID</th>\n",
       "      <th>Parent</th>\n",
       "      <th>object</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>detail</th>\n",
       "      <th>p_value</th>\n",
       "      <th>correlation</th>\n",
       "      <th>p_value_corr</th>\n",
       "      <th>gene</th>\n",
       "      <th>unique</th>\n",
       "      <th>positions</th>\n",
       "      <th>albicans_name(s)</th>\n",
       "      <th>cerevisiae_name(s)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>2367068</td>\n",
       "      <td>SPBC19G7.09</td>\n",
       "      <td>SPBC19G7.09.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2367117</td>\n",
       "      <td>2368823</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.373650</td>\n",
       "      <td>0.011462</td>\n",
       "      <td>ulp1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[2367068]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>2435580</td>\n",
       "      <td>SPBC21D10.06c</td>\n",
       "      <td>SPBC21D10.06c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2435636</td>\n",
       "      <td>2438482</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.367721</td>\n",
       "      <td>0.012952</td>\n",
       "      <td>map4</td>\n",
       "      <td>5.0</td>\n",
       "      <td>[2435580, 2436559, 2436911, 2437248, 2438335]</td>\n",
       "      <td>[nan]</td>\n",
       "      <td>[nan]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>41504</td>\n",
       "      <td>SPNCRNA.1097</td>\n",
       "      <td>SPNCRNA.1097.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>III</td>\n",
       "      <td>39627</td>\n",
       "      <td>41424</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.364295</td>\n",
       "      <td>0.013886</td>\n",
       "      <td>SPNCRNA.1097</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[41504]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>47493</td>\n",
       "      <td>SPCC757.03c</td>\n",
       "      <td>SPCC757.03c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>III</td>\n",
       "      <td>47741</td>\n",
       "      <td>48475</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.357839</td>\n",
       "      <td>0.015803</td>\n",
       "      <td>hsp3101</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[47493]</td>\n",
       "      <td>['C3_02610C_A']</td>\n",
       "      <td>['YDR533C' 'YMR322C' 'YOR391C' 'YPL280W']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>2427851</td>\n",
       "      <td>SPBC21D10.09c</td>\n",
       "      <td>SPBC21D10.09c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2426785</td>\n",
       "      <td>2431588</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.343215</td>\n",
       "      <td>0.020990</td>\n",
       "      <td>rkr1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[2427851]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>2380243</td>\n",
       "      <td>SPNCRNA.1508</td>\n",
       "      <td>SPNCRNA.1508.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2380550</td>\n",
       "      <td>2382306</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.298155</td>\n",
       "      <td>0.046667</td>\n",
       "      <td>iws1-antisense-1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[2380243]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>2418555</td>\n",
       "      <td>SPBC1921.06c</td>\n",
       "      <td>SPBC1921.06c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2417166</td>\n",
       "      <td>2418302</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.286957</td>\n",
       "      <td>0.055978</td>\n",
       "      <td>pvg3</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[2418555]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>2446853</td>\n",
       "      <td>SPBC12C2.12c</td>\n",
       "      <td>SPBC12C2.12c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2446698</td>\n",
       "      <td>2447606</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.283794</td>\n",
       "      <td>0.058861</td>\n",
       "      <td>glo1</td>\n",
       "      <td>3.0</td>\n",
       "      <td>[2446853, 2447183, 2447678]</td>\n",
       "      <td>['C1_00500C_A']</td>\n",
       "      <td>['YML004C']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>2334988</td>\n",
       "      <td>SPNCRNA.1505</td>\n",
       "      <td>SPNCRNA.1505.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2335158</td>\n",
       "      <td>2337882</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.267852</td>\n",
       "      <td>0.075255</td>\n",
       "      <td>SPBC24C6.09c-antisense-1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[2334988]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>2327939</td>\n",
       "      <td>SPBC24C6.05</td>\n",
       "      <td>SPBC24C6.05.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2327506</td>\n",
       "      <td>2327536</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.256126</td>\n",
       "      <td>0.089459</td>\n",
       "      <td>sec28</td>\n",
       "      <td>3.0</td>\n",
       "      <td>[2327939, 2328246, 2328339]</td>\n",
       "      <td>['C5_00080C_A']</td>\n",
       "      <td>['YIL076W']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>2328246</td>\n",
       "      <td>SPNCRNA.1504</td>\n",
       "      <td>SPNCRNA.1504.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2328687</td>\n",
       "      <td>2329696</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-0.252042</td>\n",
       "      <td>0.094868</td>\n",
       "      <td>SPNCRNA.1504</td>\n",
       "      <td>5.0</td>\n",
       "      <td>[2328246, 2328339, 2329401, 2329425, 2329747]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>2327939</td>\n",
       "      <td>SPNCRNA.1503</td>\n",
       "      <td>SPNCRNA.1503.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2327472</td>\n",
       "      <td>2330741</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.250725</td>\n",
       "      <td>0.096666</td>\n",
       "      <td>SPNCRNA.1503</td>\n",
       "      <td>7.0</td>\n",
       "      <td>[2327939, 2328246, 2328339, 2329401, 2329425, ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2228418</td>\n",
       "      <td>SPBC14C8.14c</td>\n",
       "      <td>SPBC14C8.14c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2228791</td>\n",
       "      <td>2228988</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-0.229908</td>\n",
       "      <td>0.128686</td>\n",
       "      <td>pol5</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[2228418]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>2326615</td>\n",
       "      <td>SPBC24C6.04</td>\n",
       "      <td>SPBC24C6.04.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2325256</td>\n",
       "      <td>2326902</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-0.222266</td>\n",
       "      <td>0.142246</td>\n",
       "      <td>put2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[2326615]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>53819</td>\n",
       "      <td>SPCC757.05c</td>\n",
       "      <td>SPCC757.05c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>III</td>\n",
       "      <td>52910</td>\n",
       "      <td>54112</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.216864</td>\n",
       "      <td>0.152452</td>\n",
       "      <td>SPCC757.05c</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[53819]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>2452181</td>\n",
       "      <td>SPBC12C2.10c</td>\n",
       "      <td>SPBC12C2.10c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2452052</td>\n",
       "      <td>2452181</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.211331</td>\n",
       "      <td>0.163453</td>\n",
       "      <td>pst1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>[2452181, 2457217]</td>\n",
       "      <td>['C1_00930C_A']</td>\n",
       "      <td>['YOL004W']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>2387677</td>\n",
       "      <td>SPBC36B7.03</td>\n",
       "      <td>SPBC36B7.03.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2387093</td>\n",
       "      <td>2387207</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.189196</td>\n",
       "      <td>0.213231</td>\n",
       "      <td>sec63</td>\n",
       "      <td>2.0</td>\n",
       "      <td>[2387677, 2388281]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>2385042</td>\n",
       "      <td>SPBC36B7.02</td>\n",
       "      <td>SPBC36B7.02.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2385439</td>\n",
       "      <td>2386500</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.185375</td>\n",
       "      <td>0.222790</td>\n",
       "      <td>svf2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[2385042]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>2438335</td>\n",
       "      <td>SPNCRNA.1513</td>\n",
       "      <td>SPNCRNA.1513.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2438665</td>\n",
       "      <td>2439695</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.178393</td>\n",
       "      <td>0.241015</td>\n",
       "      <td>ucp3-antisense-1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>[2438335, 2439430]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>2450844</td>\n",
       "      <td>SPNCRNA.1515</td>\n",
       "      <td>SPNCRNA.1515.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2449774</td>\n",
       "      <td>2450647</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.177997</td>\n",
       "      <td>0.242077</td>\n",
       "      <td>SPBC12C2.11-antisense-1</td>\n",
       "      <td>3.0</td>\n",
       "      <td>[2450844, 2450917, 2451069]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>2359346</td>\n",
       "      <td>SPBC19G7.06</td>\n",
       "      <td>SPBC19G7.06.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2359429</td>\n",
       "      <td>2359479</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.173254</td>\n",
       "      <td>0.255057</td>\n",
       "      <td>mbx1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[2359346]</td>\n",
       "      <td>['C7_00890C_A']</td>\n",
       "      <td>['YMR042W' 'YMR043W']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>2461512</td>\n",
       "      <td>SPNCRNA.1516</td>\n",
       "      <td>SPNCRNA.1516.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2461903</td>\n",
       "      <td>2463113</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.153755</td>\n",
       "      <td>0.313259</td>\n",
       "      <td>SPNCRNA.1516</td>\n",
       "      <td>6.0</td>\n",
       "      <td>[2461512, 2461523, 2461663, 2461833, 2461868, ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2310884</td>\n",
       "      <td>SPBC12D12.04c</td>\n",
       "      <td>SPBC12D12.04c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2309787</td>\n",
       "      <td>2312837</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.153360</td>\n",
       "      <td>0.314520</td>\n",
       "      <td>pck2</td>\n",
       "      <td>3.0</td>\n",
       "      <td>[2310884, 2311107, 2312083]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>2444081</td>\n",
       "      <td>SPNCRNA.1514</td>\n",
       "      <td>SPNCRNA.1514.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2444131</td>\n",
       "      <td>2444830</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.143612</td>\n",
       "      <td>0.346624</td>\n",
       "      <td>fnx1-antisense-1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>[2444081, 2444308]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2305695</td>\n",
       "      <td>SPBC12D12.02c</td>\n",
       "      <td>SPBC12D12.02c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2305758</td>\n",
       "      <td>2306240</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.142029</td>\n",
       "      <td>0.352020</td>\n",
       "      <td>cdm1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[2305695]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>2330408</td>\n",
       "      <td>SPBC24C6.06</td>\n",
       "      <td>SPBC24C6.06.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2330338</td>\n",
       "      <td>2331561</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.123979</td>\n",
       "      <td>0.417130</td>\n",
       "      <td>gpa1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[2330408]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>2334988</td>\n",
       "      <td>SPBC24C6.09c</td>\n",
       "      <td>SPBC24C6.09c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2335262</td>\n",
       "      <td>2337739</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-0.113966</td>\n",
       "      <td>0.456014</td>\n",
       "      <td>SPBC24C6.09c</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[2334988]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>41504</td>\n",
       "      <td>SPNCRNA.07</td>\n",
       "      <td>SPNCRNA.07.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>III</td>\n",
       "      <td>41985</td>\n",
       "      <td>42728</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.111045</td>\n",
       "      <td>0.467714</td>\n",
       "      <td>meu3</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[41504]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>2380243</td>\n",
       "      <td>SPBC19G7.16</td>\n",
       "      <td>SPBC19G7.16.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2380663</td>\n",
       "      <td>2381949</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.110408</td>\n",
       "      <td>0.470287</td>\n",
       "      <td>iws1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[2380243]</td>\n",
       "      <td>['C1_06590C_A']</td>\n",
       "      <td>['YPR133C']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>2323914</td>\n",
       "      <td>SPBC24C6.03</td>\n",
       "      <td>SPBC24C6.03.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2323024</td>\n",
       "      <td>2324301</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.108300</td>\n",
       "      <td>0.478856</td>\n",
       "      <td>SPBC24C6.03</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[2323914]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81</th>\n",
       "      <td>47493</td>\n",
       "      <td>SPNCRNA.1100</td>\n",
       "      <td>SPNCRNA.1100.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>III</td>\n",
       "      <td>47513</td>\n",
       "      <td>48882</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.106192</td>\n",
       "      <td>0.487506</td>\n",
       "      <td>hsp3101-antisense-1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[47493]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>47493</td>\n",
       "      <td>SPNCRNA.1099</td>\n",
       "      <td>SPNCRNA.1099.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>III</td>\n",
       "      <td>46987</td>\n",
       "      <td>47566</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.104348</td>\n",
       "      <td>0.495141</td>\n",
       "      <td>SPNCRNA.1099</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[47493]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>2364604</td>\n",
       "      <td>SPBC19G7.08c</td>\n",
       "      <td>SPBC19G7.08c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2364279</td>\n",
       "      <td>2365730</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.100395</td>\n",
       "      <td>0.511708</td>\n",
       "      <td>art1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[2364604]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>2367068</td>\n",
       "      <td>SPNCRNA.1507</td>\n",
       "      <td>SPNCRNA.1507.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2366962</td>\n",
       "      <td>2367928</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.098055</td>\n",
       "      <td>0.521645</td>\n",
       "      <td>ulp1-antisense-1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[2367068]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>2378455</td>\n",
       "      <td>SPBC19G7.15</td>\n",
       "      <td>SPBC19G7.15.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2378035</td>\n",
       "      <td>2378050</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.091304</td>\n",
       "      <td>0.550841</td>\n",
       "      <td>nup44</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[2378455]</td>\n",
       "      <td>['CR_02610C_A']</td>\n",
       "      <td>['YGR119C']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>2439430</td>\n",
       "      <td>SPBC21D10.05c</td>\n",
       "      <td>SPBC21D10.05c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2439122</td>\n",
       "      <td>2439236</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.081950</td>\n",
       "      <td>0.592533</td>\n",
       "      <td>ucp3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>[2439430, 2440301]</td>\n",
       "      <td>['CR_08290W_A']</td>\n",
       "      <td>['YGL181W']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>2380243</td>\n",
       "      <td>SPBC19G7.19</td>\n",
       "      <td>SPBC19G7.19.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2379937</td>\n",
       "      <td>2380002</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.063768</td>\n",
       "      <td>0.677296</td>\n",
       "      <td>tom5</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[2380243]</td>\n",
       "      <td>['C1_06640C_A']</td>\n",
       "      <td>['YPR133W-A']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>2449202</td>\n",
       "      <td>SPBC12C2.11</td>\n",
       "      <td>SPBC12C2.11.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2448491</td>\n",
       "      <td>2450505</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.061792</td>\n",
       "      <td>0.686776</td>\n",
       "      <td>gfa1</td>\n",
       "      <td>4.0</td>\n",
       "      <td>[2449202, 2450844, 2450917, 2451069]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2305695</td>\n",
       "      <td>SPBC12D12.01</td>\n",
       "      <td>SPBC12D12.01.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2304182</td>\n",
       "      <td>2305726</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-0.051647</td>\n",
       "      <td>0.736167</td>\n",
       "      <td>sad1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[2305695]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>2457217</td>\n",
       "      <td>SPBC12C2.09c</td>\n",
       "      <td>SPBC12C2.09c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2457715</td>\n",
       "      <td>2458689</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.050725</td>\n",
       "      <td>0.740713</td>\n",
       "      <td>izh2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[2457217]</td>\n",
       "      <td>['C1_05010C_A']</td>\n",
       "      <td>['YOL002C']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>2444081</td>\n",
       "      <td>SPBC12C2.13c</td>\n",
       "      <td>SPBC12C2.13c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2444329</td>\n",
       "      <td>2445924</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.048221</td>\n",
       "      <td>0.753097</td>\n",
       "      <td>fnx1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>[2444081, 2444308]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>2316554</td>\n",
       "      <td>SPBC12D12.06</td>\n",
       "      <td>SPBC12D12.06.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2316168</td>\n",
       "      <td>2316199</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-0.044401</td>\n",
       "      <td>0.772115</td>\n",
       "      <td>srb11</td>\n",
       "      <td>3.0</td>\n",
       "      <td>[2316554, 2316851, 2317549]</td>\n",
       "      <td>['C3_05740C_A']</td>\n",
       "      <td>['YNL025C']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>2420149</td>\n",
       "      <td>SPBC1921.07c</td>\n",
       "      <td>SPBC1921.07c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2419533</td>\n",
       "      <td>2419981</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.036891</td>\n",
       "      <td>0.809874</td>\n",
       "      <td>sgf29</td>\n",
       "      <td>2.0</td>\n",
       "      <td>[2420149, 2420208]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>2363654</td>\n",
       "      <td>SPBC19G7.07c</td>\n",
       "      <td>SPBC19G7.07c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2363086</td>\n",
       "      <td>2363340</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.035968</td>\n",
       "      <td>0.814543</td>\n",
       "      <td>ppr3</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[2363654]</td>\n",
       "      <td>['C6_03440W_A']</td>\n",
       "      <td>['YGR150C']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>2317549</td>\n",
       "      <td>SPBC12D12.07c</td>\n",
       "      <td>SPBC12D12.07c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2317984</td>\n",
       "      <td>2318251</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.035310</td>\n",
       "      <td>0.817882</td>\n",
       "      <td>trx2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[2317549]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2202682</td>\n",
       "      <td>SPBC1815.01</td>\n",
       "      <td>SPBC1815.01.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2201237</td>\n",
       "      <td>2202556</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.027931</td>\n",
       "      <td>0.855478</td>\n",
       "      <td>eno101</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[2202682]</td>\n",
       "      <td>['C1_08500C_A']</td>\n",
       "      <td>['YGR254W' 'YHR174W' 'YMR323W' 'YOR393W' 'YPL2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>2418555</td>\n",
       "      <td>SPNCRNA.1512</td>\n",
       "      <td>SPNCRNA.1512.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2417501</td>\n",
       "      <td>2418682</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.026557</td>\n",
       "      <td>0.862521</td>\n",
       "      <td>pvg3-antisense-1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[2418555]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>2425776</td>\n",
       "      <td>SPBC21D10.10</td>\n",
       "      <td>SPBC21D10.10.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2424574</td>\n",
       "      <td>2425473</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.012780</td>\n",
       "      <td>0.933596</td>\n",
       "      <td>bdc1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[2425776]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>2461512</td>\n",
       "      <td>SPBC12C2.08</td>\n",
       "      <td>SPBC12C2.08.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2461357</td>\n",
       "      <td>2461495</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.012516</td>\n",
       "      <td>0.934962</td>\n",
       "      <td>dnm1</td>\n",
       "      <td>6.0</td>\n",
       "      <td>[2461512, 2461523, 2461663, 2461833, 2461868, ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2202682</td>\n",
       "      <td>SPNCRNA.5536</td>\n",
       "      <td>SPNCRNA.5536.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2202759</td>\n",
       "      <td>2204257</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[2202682]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    position             ID           Parent object chromosome    start  \\\n",
       "27   2367068    SPBC19G7.09    SPBC19G7.09.1    CDS         II  2367117   \n",
       "49   2435580  SPBC21D10.06c  SPBC21D10.06c.1    CDS         II  2435636   \n",
       "77     41504   SPNCRNA.1097   SPNCRNA.1097.1    CDS        III    39627   \n",
       "79     47493    SPCC757.03c    SPCC757.03c.1    CDS        III    47741   \n",
       "48   2427851  SPBC21D10.09c  SPBC21D10.09c.1    CDS         II  2426785   \n",
       "34   2380243   SPNCRNA.1508   SPNCRNA.1508.1    CDS         II  2380550   \n",
       "40   2418555   SPBC1921.06c   SPBC1921.06c.1    CDS         II  2417166   \n",
       "61   2446853   SPBC12C2.12c   SPBC12C2.12c.1    CDS         II  2446698   \n",
       "21   2334988   SPNCRNA.1505   SPNCRNA.1505.1    CDS         II  2335158   \n",
       "16   2327939    SPBC24C6.05    SPBC24C6.05.1    CDS         II  2327506   \n",
       "18   2328246   SPNCRNA.1504   SPNCRNA.1504.1    CDS         II  2328687   \n",
       "17   2327939   SPNCRNA.1503   SPNCRNA.1503.1    CDS         II  2327472   \n",
       "2    2228418   SPBC14C8.14c   SPBC14C8.14c.1    CDS         II  2228791   \n",
       "14   2326615    SPBC24C6.04    SPBC24C6.04.1    CDS         II  2325256   \n",
       "82     53819    SPCC757.05c    SPCC757.05c.1    CDS        III    52910   \n",
       "68   2452181   SPBC12C2.10c   SPBC12C2.10c.1    CDS         II  2452052   \n",
       "37   2387677    SPBC36B7.03    SPBC36B7.03.1    CDS         II  2387093   \n",
       "35   2385042    SPBC36B7.02    SPBC36B7.02.1    CDS         II  2385439   \n",
       "52   2438335   SPNCRNA.1513   SPNCRNA.1513.1    CDS         II  2438665   \n",
       "65   2450844   SPNCRNA.1515   SPNCRNA.1515.1    CDS         II  2449774   \n",
       "23   2359346    SPBC19G7.06    SPBC19G7.06.1    CDS         II  2359429   \n",
       "75   2461512   SPNCRNA.1516   SPNCRNA.1516.1    CDS         II  2461903   \n",
       "6    2310884  SPBC12D12.04c  SPBC12D12.04c.1    CDS         II  2309787   \n",
       "60   2444081   SPNCRNA.1514   SPNCRNA.1514.1    CDS         II  2444131   \n",
       "5    2305695  SPBC12D12.02c  SPBC12D12.02c.1    CDS         II  2305758   \n",
       "19   2330408    SPBC24C6.06    SPBC24C6.06.1    CDS         II  2330338   \n",
       "20   2334988   SPBC24C6.09c   SPBC24C6.09c.1    CDS         II  2335262   \n",
       "76     41504     SPNCRNA.07     SPNCRNA.07.1    CDS        III    41985   \n",
       "32   2380243    SPBC19G7.16    SPBC19G7.16.1    CDS         II  2380663   \n",
       "12   2323914    SPBC24C6.03    SPBC24C6.03.1    CDS         II  2323024   \n",
       "81     47493   SPNCRNA.1100   SPNCRNA.1100.1    CDS        III    47513   \n",
       "80     47493   SPNCRNA.1099   SPNCRNA.1099.1    CDS        III    46987   \n",
       "26   2364604   SPBC19G7.08c   SPBC19G7.08c.1    CDS         II  2364279   \n",
       "28   2367068   SPNCRNA.1507   SPNCRNA.1507.1    CDS         II  2366962   \n",
       "31   2378455    SPBC19G7.15    SPBC19G7.15.1    CDS         II  2378035   \n",
       "53   2439430  SPBC21D10.05c  SPBC21D10.05c.1    CDS         II  2439122   \n",
       "33   2380243    SPBC19G7.19    SPBC19G7.19.1    CDS         II  2379937   \n",
       "64   2449202    SPBC12C2.11    SPBC12C2.11.1    CDS         II  2448491   \n",
       "4    2305695   SPBC12D12.01   SPBC12D12.01.1    CDS         II  2304182   \n",
       "70   2457217   SPBC12C2.09c   SPBC12C2.09c.1    CDS         II  2457715   \n",
       "59   2444081   SPBC12C2.13c   SPBC12C2.13c.1    CDS         II  2444329   \n",
       "9    2316554   SPBC12D12.06   SPBC12D12.06.1    CDS         II  2316168   \n",
       "43   2420149   SPBC1921.07c   SPBC1921.07c.1    CDS         II  2419533   \n",
       "24   2363654   SPBC19G7.07c   SPBC19G7.07c.1    CDS         II  2363086   \n",
       "10   2317549  SPBC12D12.07c  SPBC12D12.07c.1    CDS         II  2317984   \n",
       "0    2202682    SPBC1815.01    SPBC1815.01.1    CDS         II  2201237   \n",
       "41   2418555   SPNCRNA.1512   SPNCRNA.1512.1    CDS         II  2417501   \n",
       "44   2425776   SPBC21D10.10   SPBC21D10.10.1    CDS         II  2424574   \n",
       "74   2461512    SPBC12C2.08    SPBC12C2.08.1    CDS         II  2461357   \n",
       "1    2202682   SPNCRNA.5536   SPNCRNA.5536.1    CDS         II  2202759   \n",
       "\n",
       "        end strand detail  p_value  correlation  p_value_corr  \\\n",
       "27  2368823      +   exon      0.0    -0.373650      0.011462   \n",
       "49  2438482      +   exon      0.0     0.367721      0.012952   \n",
       "77    41424      -   exon      0.0     0.364295      0.013886   \n",
       "79    48475      -   exon      0.0     0.357839      0.015803   \n",
       "48  2431588      +   exon      0.0    -0.343215      0.020990   \n",
       "34  2382306      -   exon      0.0     0.298155      0.046667   \n",
       "40  2418302      -   exon      0.0    -0.286957      0.055978   \n",
       "61  2447606      +   exon      0.0    -0.283794      0.058861   \n",
       "21  2337882      +   exon      1.0     0.267852      0.075255   \n",
       "16  2327536      +   exon      1.0     0.256126      0.089459   \n",
       "18  2329696      +   exon      1.0    -0.252042      0.094868   \n",
       "17  2330741      -   exon      1.0     0.250725      0.096666   \n",
       "2   2228988      -   exon      1.0    -0.229908      0.128686   \n",
       "14  2326902      +   exon      1.0    -0.222266      0.142246   \n",
       "82    54112      -   exon      0.0     0.216864      0.152452   \n",
       "68  2452181      +   exon      0.0    -0.211331      0.163453   \n",
       "37  2387207      +   exon      0.0    -0.189196      0.213231   \n",
       "35  2386500      +   exon      0.0    -0.185375      0.222790   \n",
       "52  2439695      -   exon      0.0     0.178393      0.241015   \n",
       "65  2450647      +   exon      0.0     0.177997      0.242077   \n",
       "23  2359479      +   exon      0.0    -0.173254      0.255057   \n",
       "75  2463113      +   exon      0.0     0.153755      0.313259   \n",
       "6   2312837      -   exon      1.0     0.153360      0.314520   \n",
       "60  2444830      -   exon      0.0     0.143612      0.346624   \n",
       "5   2306240      -   exon      1.0     0.142029      0.352020   \n",
       "19  2331561      +   exon      1.0     0.123979      0.417130   \n",
       "20  2337739      -   exon      1.0    -0.113966      0.456014   \n",
       "76    42728      -   exon      0.0     0.111045      0.467714   \n",
       "32  2381949      +   exon      0.0     0.110408      0.470287   \n",
       "12  2324301      +   exon      1.0     0.108300      0.478856   \n",
       "81    48882      +   exon      0.0     0.106192      0.487506   \n",
       "80    47566      +   exon      0.0    -0.104348      0.495141   \n",
       "26  2365730      -   exon      0.0     0.100395      0.511708   \n",
       "28  2367928      -   exon      0.0    -0.098055      0.521645   \n",
       "31  2378050      +   exon      0.0    -0.091304      0.550841   \n",
       "53  2439236      +   exon      0.0     0.081950      0.592533   \n",
       "33  2380002      -   exon      0.0    -0.063768      0.677296   \n",
       "64  2450505      -   exon      0.0     0.061792      0.686776   \n",
       "4   2305726      +   exon      1.0    -0.051647      0.736167   \n",
       "70  2458689      +   exon      0.0     0.050725      0.740713   \n",
       "59  2445924      +   exon      0.0     0.048221      0.753097   \n",
       "9   2316199      +   exon      1.0    -0.044401      0.772115   \n",
       "43  2419981      -   exon      0.0    -0.036891      0.809874   \n",
       "24  2363340      -   exon      0.0     0.035968      0.814543   \n",
       "10  2318251      -   exon      1.0     0.035310      0.817882   \n",
       "0   2202556      +   exon      1.0     0.027931      0.855478   \n",
       "41  2418682      +   exon      0.0    -0.026557      0.862521   \n",
       "44  2425473      -   exon      0.0     0.012780      0.933596   \n",
       "74  2461495      -   exon      0.0    -0.012516      0.934962   \n",
       "1   2204257      +   exon      1.0          NaN           NaN   \n",
       "\n",
       "                        gene  unique  \\\n",
       "27                      ulp1     1.0   \n",
       "49                      map4     5.0   \n",
       "77              SPNCRNA.1097     1.0   \n",
       "79                   hsp3101     1.0   \n",
       "48                      rkr1     1.0   \n",
       "34          iws1-antisense-1     1.0   \n",
       "40                      pvg3     1.0   \n",
       "61                      glo1     3.0   \n",
       "21  SPBC24C6.09c-antisense-1     1.0   \n",
       "16                     sec28     3.0   \n",
       "18              SPNCRNA.1504     5.0   \n",
       "17              SPNCRNA.1503     7.0   \n",
       "2                       pol5     1.0   \n",
       "14                      put2     1.0   \n",
       "82               SPCC757.05c     1.0   \n",
       "68                      pst1     2.0   \n",
       "37                     sec63     2.0   \n",
       "35                      svf2     1.0   \n",
       "52          ucp3-antisense-1     2.0   \n",
       "65   SPBC12C2.11-antisense-1     3.0   \n",
       "23                      mbx1     1.0   \n",
       "75              SPNCRNA.1516     6.0   \n",
       "6                       pck2     3.0   \n",
       "60          fnx1-antisense-1     2.0   \n",
       "5                       cdm1     1.0   \n",
       "19                      gpa1     1.0   \n",
       "20              SPBC24C6.09c     1.0   \n",
       "76                      meu3     1.0   \n",
       "32                      iws1     1.0   \n",
       "12               SPBC24C6.03     1.0   \n",
       "81       hsp3101-antisense-1     1.0   \n",
       "80              SPNCRNA.1099     1.0   \n",
       "26                      art1     1.0   \n",
       "28          ulp1-antisense-1     1.0   \n",
       "31                     nup44     1.0   \n",
       "53                      ucp3     2.0   \n",
       "33                      tom5     1.0   \n",
       "64                      gfa1     4.0   \n",
       "4                       sad1     1.0   \n",
       "70                      izh2     1.0   \n",
       "59                      fnx1     2.0   \n",
       "9                      srb11     3.0   \n",
       "43                     sgf29     2.0   \n",
       "24                      ppr3     1.0   \n",
       "10                      trx2     1.0   \n",
       "0                     eno101     1.0   \n",
       "41          pvg3-antisense-1     1.0   \n",
       "44                      bdc1     1.0   \n",
       "74                      dnm1     6.0   \n",
       "1                        NaN     1.0   \n",
       "\n",
       "                                            positions albicans_name(s)  \\\n",
       "27                                          [2367068]              NaN   \n",
       "49      [2435580, 2436559, 2436911, 2437248, 2438335]            [nan]   \n",
       "77                                            [41504]              NaN   \n",
       "79                                            [47493]  ['C3_02610C_A']   \n",
       "48                                          [2427851]              NaN   \n",
       "34                                          [2380243]              NaN   \n",
       "40                                          [2418555]              NaN   \n",
       "61                        [2446853, 2447183, 2447678]  ['C1_00500C_A']   \n",
       "21                                          [2334988]              NaN   \n",
       "16                        [2327939, 2328246, 2328339]  ['C5_00080C_A']   \n",
       "18      [2328246, 2328339, 2329401, 2329425, 2329747]              NaN   \n",
       "17  [2327939, 2328246, 2328339, 2329401, 2329425, ...              NaN   \n",
       "2                                           [2228418]              NaN   \n",
       "14                                          [2326615]              NaN   \n",
       "82                                            [53819]              NaN   \n",
       "68                                 [2452181, 2457217]  ['C1_00930C_A']   \n",
       "37                                 [2387677, 2388281]              NaN   \n",
       "35                                          [2385042]              NaN   \n",
       "52                                 [2438335, 2439430]              NaN   \n",
       "65                        [2450844, 2450917, 2451069]              NaN   \n",
       "23                                          [2359346]  ['C7_00890C_A']   \n",
       "75  [2461512, 2461523, 2461663, 2461833, 2461868, ...              NaN   \n",
       "6                         [2310884, 2311107, 2312083]              NaN   \n",
       "60                                 [2444081, 2444308]              NaN   \n",
       "5                                           [2305695]              NaN   \n",
       "19                                          [2330408]              NaN   \n",
       "20                                          [2334988]              NaN   \n",
       "76                                            [41504]              NaN   \n",
       "32                                          [2380243]  ['C1_06590C_A']   \n",
       "12                                          [2323914]              NaN   \n",
       "81                                            [47493]              NaN   \n",
       "80                                            [47493]              NaN   \n",
       "26                                          [2364604]              NaN   \n",
       "28                                          [2367068]              NaN   \n",
       "31                                          [2378455]  ['CR_02610C_A']   \n",
       "53                                 [2439430, 2440301]  ['CR_08290W_A']   \n",
       "33                                          [2380243]  ['C1_06640C_A']   \n",
       "64               [2449202, 2450844, 2450917, 2451069]              NaN   \n",
       "4                                           [2305695]              NaN   \n",
       "70                                          [2457217]  ['C1_05010C_A']   \n",
       "59                                 [2444081, 2444308]              NaN   \n",
       "9                         [2316554, 2316851, 2317549]  ['C3_05740C_A']   \n",
       "43                                 [2420149, 2420208]              NaN   \n",
       "24                                          [2363654]  ['C6_03440W_A']   \n",
       "10                                          [2317549]              NaN   \n",
       "0                                           [2202682]  ['C1_08500C_A']   \n",
       "41                                          [2418555]              NaN   \n",
       "44                                          [2425776]              NaN   \n",
       "74  [2461512, 2461523, 2461663, 2461833, 2461868, ...              NaN   \n",
       "1                                           [2202682]              NaN   \n",
       "\n",
       "                                   cerevisiae_name(s)  \n",
       "27                                                NaN  \n",
       "49                                              [nan]  \n",
       "77                                                NaN  \n",
       "79          ['YDR533C' 'YMR322C' 'YOR391C' 'YPL280W']  \n",
       "48                                                NaN  \n",
       "34                                                NaN  \n",
       "40                                                NaN  \n",
       "61                                        ['YML004C']  \n",
       "21                                                NaN  \n",
       "16                                        ['YIL076W']  \n",
       "18                                                NaN  \n",
       "17                                                NaN  \n",
       "2                                                 NaN  \n",
       "14                                                NaN  \n",
       "82                                                NaN  \n",
       "68                                        ['YOL004W']  \n",
       "37                                                NaN  \n",
       "35                                                NaN  \n",
       "52                                                NaN  \n",
       "65                                                NaN  \n",
       "23                              ['YMR042W' 'YMR043W']  \n",
       "75                                                NaN  \n",
       "6                                                 NaN  \n",
       "60                                                NaN  \n",
       "5                                                 NaN  \n",
       "19                                                NaN  \n",
       "20                                                NaN  \n",
       "76                                                NaN  \n",
       "32                                        ['YPR133C']  \n",
       "12                                                NaN  \n",
       "81                                                NaN  \n",
       "80                                                NaN  \n",
       "26                                                NaN  \n",
       "28                                                NaN  \n",
       "31                                        ['YGR119C']  \n",
       "53                                        ['YGL181W']  \n",
       "33                                      ['YPR133W-A']  \n",
       "64                                                NaN  \n",
       "4                                                 NaN  \n",
       "70                                        ['YOL002C']  \n",
       "59                                                NaN  \n",
       "9                                         ['YNL025C']  \n",
       "43                                                NaN  \n",
       "24                                        ['YGR150C']  \n",
       "10                                                NaN  \n",
       "0   ['YGR254W' 'YHR174W' 'YMR323W' 'YOR393W' 'YPL2...  \n",
       "41                                                NaN  \n",
       "44                                                NaN  \n",
       "74                                                NaN  \n",
       "1                                                 NaN  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#sort by p_value\n",
    "result_final__ = result_final__.sort_values(by=['p_value_corr'])\n",
    "result_final__.head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "result_final__ = result.copy().reset_index(drop=True, inplace=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "#iterate through positions in result_final__  and find the corresponding alleles in snps. Then add the two values in X968_allele\tY0036_allele to the df in two new columns at the start \n",
    "#add two columns to result_final__\n",
    "result_final__['X968_allele'] = np.nan\n",
    "result_final__['Y0036_allele'] = np.nan\n",
    "for i in range(len(result_final__)):\n",
    "    for j in range(len(snps)):\n",
    "        if result_final__['position'][i] == snps['position'][j]:\n",
    "            result_final__['X968_allele'][i] = snps['X968_allele'][j]\n",
    "            result_final__['Y0036_allele'][i] = snps['Y0036_allele'][j]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "#iterate through positions in result_final__ and look up the chromosome and position in regions, and find the corresponding letter +/- 5 bp in regions. Add it to a new column\n",
    "result_final__['reference_seq']=np.nan\n",
    "for i in range(len(result_final__)):\n",
    "    for j in range(len(regions)):\n",
    "        if result_final__['chromosome'][i] == list(regions.keys())[j]:\n",
    "            #add | | characters left and right of tletter at position-1\n",
    "            \n",
    "            result_final__['reference_seq'][i] = regions[list(regions.keys())[j]][result_final__['position'][i]-4:result_final__['position'][i]-1] + ' |' + regions[list(regions.keys())[j]][result_final__['position'][i]-1] + '| ' + regions[list(regions.keys())[j]][result_final__['position'][i]:result_final__['position'][i]+3] \n",
    "\n",
    "#if strand is -, reverse reference_seq\n",
    "for i in range(len(result_final__)):\n",
    "    if result_final__['strand'][i] == '-':\n",
    "        result_final__['reference_seq'][i] = result_final__['reference_seq'][i][::-1]\n",
    "        #also flip the alleles\n",
    "        result_final__['X968_allele'][i] = result_final__['X968_allele'][i][::-1]\n",
    "        result_final__['Y0036_allele'][i] = result_final__['Y0036_allele'][i][::-1]\n",
    "#also create reference_seq_ without the | | characters\n",
    "result_final__['reference_seq_'] = result_final__['reference_seq'].str.replace('|', '')\n",
    "result_final__['reference_seq_'] = result_final__['reference_seq_'].str.replace(' ', '')\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SPNCRNA.5536\n",
      "SPNCRNA.5547\n",
      "SPNCRNA.5597\n",
      "SPNCRNA.5600\n",
      "SPNCRNA.1503\n",
      "SPNCRNA.1503\n",
      "SPNCRNA.1504\n",
      "SPNCRNA.1503\n",
      "SPNCRNA.1504\n",
      "SPNCRNA.1503\n",
      "SPNCRNA.1504\n",
      "SPNCRNA.1503\n",
      "SPNCRNA.1504\n",
      "SPNCRNA.1503\n",
      "SPNCRNA.1504\n",
      "SPNCRNA.1503\n",
      "SPBC24C6.09c\n",
      "SPNCRNA.1505\n",
      "SPNCRNA.5612\n",
      "SPNCRNA.5615\n",
      "SPNCRNA.5615\n",
      "SPBC19G7.09\n",
      "SPNCRNA.1507\n",
      "SPNCRNA.5616\n",
      "SPNCRNA.5617\n",
      "SPBC19G7.16\n",
      "SPBC19G7.19\n",
      "SPBC19G7.19\n",
      "SPBC19G7.19\n",
      "SPNCRNA.1508\n",
      "SPNCRNA.5627\n",
      "SPNCRNA.5627\n",
      "SPNCRNA.5629\n",
      "SPNCRNA.1512\n",
      "SPNCRNA.5647\n",
      "SPBC1921.07c\n",
      "SPNCRNA.5647\n",
      "SPBC1921.07c\n",
      "SPNCRNA.5647\n",
      "SPBC21D10.10\n",
      "SPNCRNA.5652\n",
      "SPNCRNA.5653\n",
      "SPNCRNA.5654\n",
      "SPNCRNA.5657\n",
      "SPNCRNA.1513\n",
      "SPNCRNA.1513\n",
      "SPNCRNA.5659\n",
      "SPNCRNA.5660\n",
      "SPNCRNA.5661\n",
      "SPNCRNA.5659\n",
      "SPNCRNA.5660\n",
      "SPNCRNA.5661\n",
      "SPNCRNA.5659\n",
      "SPNCRNA.5660\n",
      "SPNCRNA.5661\n",
      "SPNCRNA.5662\n",
      "SPNCRNA.5660\n",
      "SPNCRNA.5661\n",
      "SPNCRNA.5662\n",
      "SPNCRNA.1514\n",
      "SPNCRNA.5662\n",
      "SPNCRNA.1514\n",
      "SPNCRNA.5663\n",
      "SPNCRNA.5663\n",
      "SPNCRNA.5663\n",
      "SPSNORNA.38\n",
      "SPNCRNA.1515\n",
      "SPNCRNA.5664\n",
      "SPNCRNA.1515\n",
      "SPNCRNA.5664\n",
      "SPNCRNA.1515\n",
      "SPNCRNA.5664\n",
      "SPNCRNA.5665\n",
      "SPNCRNA.5665\n",
      "SPBC12C2.09c\n",
      "SPNCRNA.5669\n",
      "SPNCRNA.5670\n",
      "SPNCRNA.5671\n",
      "SPNCRNA.1516\n",
      "SPNCRNA.1516\n",
      "SPNCRNA.1516\n",
      "SPNCRNA.1516\n",
      "SPNCRNA.1516\n",
      "SPNCRNA.1516\n",
      "SPNCRNA.07\n",
      "SPNCRNA.1097\n",
      "SPNCRNA.6682\n",
      "SPNCRNA.1099\n",
      "SPNCRNA.1100\n",
      "SPNCRNA.6688\n",
      "SPNCRNA.5536\n",
      "SPNCRNA.5547\n",
      "SPNCRNA.5597\n",
      "SPNCRNA.5600\n",
      "SPNCRNA.1503\n",
      "SPNCRNA.1503\n",
      "SPNCRNA.1504\n",
      "SPNCRNA.1503\n",
      "SPNCRNA.1504\n",
      "SPNCRNA.1503\n",
      "SPNCRNA.1504\n",
      "SPNCRNA.1503\n",
      "SPNCRNA.1504\n",
      "SPNCRNA.1503\n",
      "SPNCRNA.1504\n",
      "SPNCRNA.1503\n",
      "SPBC24C6.09c\n",
      "SPNCRNA.1505\n",
      "SPNCRNA.5612\n",
      "SPNCRNA.5615\n",
      "SPNCRNA.5615\n",
      "SPBC19G7.09\n",
      "SPNCRNA.1507\n",
      "SPNCRNA.5616\n",
      "SPNCRNA.5617\n",
      "SPBC19G7.16\n",
      "SPBC19G7.19\n",
      "SPBC19G7.19\n",
      "SPBC19G7.19\n",
      "SPNCRNA.1508\n",
      "SPNCRNA.5627\n",
      "SPNCRNA.5627\n",
      "SPNCRNA.5629\n",
      "SPNCRNA.1512\n",
      "SPNCRNA.5647\n",
      "SPBC1921.07c\n",
      "SPNCRNA.5647\n",
      "SPBC1921.07c\n",
      "SPNCRNA.5647\n",
      "SPBC21D10.10\n",
      "SPNCRNA.5652\n",
      "SPNCRNA.5653\n",
      "SPNCRNA.5654\n",
      "SPNCRNA.5657\n",
      "SPNCRNA.1513\n",
      "SPNCRNA.1513\n",
      "SPNCRNA.5659\n",
      "SPNCRNA.5660\n",
      "SPNCRNA.5661\n",
      "SPNCRNA.5659\n",
      "SPNCRNA.5660\n",
      "SPNCRNA.5661\n",
      "SPNCRNA.5659\n",
      "SPNCRNA.5660\n",
      "SPNCRNA.5661\n",
      "SPNCRNA.5662\n",
      "SPNCRNA.5660\n",
      "SPNCRNA.5661\n",
      "SPNCRNA.5662\n",
      "SPNCRNA.1514\n",
      "SPNCRNA.5662\n",
      "SPNCRNA.1514\n",
      "SPNCRNA.5663\n",
      "SPNCRNA.5663\n",
      "SPNCRNA.5663\n",
      "SPSNORNA.38\n",
      "SPNCRNA.1515\n",
      "SPNCRNA.5664\n",
      "SPNCRNA.1515\n",
      "SPNCRNA.5664\n",
      "SPNCRNA.1515\n",
      "SPNCRNA.5664\n",
      "SPNCRNA.5665\n",
      "SPNCRNA.5665\n",
      "SPBC12C2.09c\n",
      "SPNCRNA.5669\n",
      "SPNCRNA.5670\n",
      "SPNCRNA.5671\n",
      "SPNCRNA.1516\n",
      "SPNCRNA.1516\n",
      "SPNCRNA.1516\n",
      "SPNCRNA.1516\n",
      "SPNCRNA.1516\n",
      "SPNCRNA.1516\n",
      "SPNCRNA.07\n",
      "SPNCRNA.1097\n",
      "SPNCRNA.6682\n",
      "SPNCRNA.1099\n",
      "SPNCRNA.1100\n",
      "SPNCRNA.6688\n"
     ]
    }
   ],
   "source": [
    "#create column, whole gene as codons\n",
    "result_final__['wt_gene'] = np.nan\n",
    "#iterate through result_final__ positions. If object is CDS, then look up the sequence from start to end in regions.\n",
    "#add the sequence to the df\n",
    "for i in range(len(result_final__)):\n",
    "    \n",
    "    if result_final__['object'][i] == 'CDS':\n",
    "    \n",
    "        start = result_final__['start'][i]\n",
    "        end = result_final__['end'][i]\n",
    "        chrom = result_final__['chromosome'][i]\n",
    "        pos = result_final__['position'][i]\n",
    "        gene = regions[chrom][start-1:end]\n",
    "        #gene_exons_final where start is the same as start in result_final__\n",
    "        exon = gene_exons_final[(gene_exons_final['start'] < start+20) & (gene_exons_final['end'] > end-20)] \n",
    "        #if exon is empty, then print gene name\n",
    "        if exon.empty:\n",
    "            print(result_final__['ID'][i])\n",
    "            continue\n",
    "        exon_len_so_far=0\n",
    "        exon_len_so_far = exon['CDS_length_so_far'].values[0]\n",
    "        exon_len_so_far=exon_len_so_far%3\n",
    "        if pos > start and pos < end:\n",
    "        \n",
    "            if result_final__['strand'][i] == '-':\n",
    "                \n",
    "                gene = gene[::-1]\n",
    "            \n",
    "                gene = ''.join(gene)\n",
    "            gene = 'A'*int(exon_len_so_far) + gene\n",
    "            #write the gene as triplets of letters like AAA, ATC, etc\n",
    "            #if divisible by 3\n",
    "            if len(gene)%3 != 0:\n",
    "                gene = gene[:-(len(gene)%3)]\n",
    "            gene = [gene[i:i+3] for i in range(0, len(gene), 3)]\n",
    "                #only triplets near the position\n",
    "            if result_final__['strand'][i] == '+':\n",
    "                gene = gene[(pos-(start-int(exon_len_so_far)))//3-2:(pos-end)//3+25]\n",
    "            else: \n",
    "                gene = gene[(end-pos)//3-2:(start-int(exon_len_so_far)-pos)//3+25]\n",
    "            \n",
    "                \n",
    "                \n",
    "\n",
    "\n",
    "            result_final__['wt_gene'][i] = gene\n",
    "\n",
    "result_final__['mutant allele'] = np.nan\n",
    "for i in range(len(result_final__)):\n",
    "    if result_final__['object'][i] == 'CDS':\n",
    "    \n",
    "        start = result_final__['start'][i]\n",
    "        end = result_final__['end'][i]\n",
    "        chrom = result_final__['chromosome'][i]\n",
    "        pos = result_final__['position'][i]\n",
    "        gene = regions[chrom][start-1:end]\n",
    "         #gene_exons_final where start is the same as start in result_final__\n",
    "        exon = gene_exons_final[(gene_exons_final['start'] < start+20) & (gene_exons_final['end'] > end-20)] \n",
    "        #if exon is empty, then print gene name\n",
    "        if exon.empty:\n",
    "            print(result_final__['ID'][i])\n",
    "            continue\n",
    "        exon_len_so_far=0\n",
    "        exon_len_so_far = exon['CDS_length_so_far'].values[0]\n",
    "        exon_len_so_far=exon_len_so_far%3\n",
    "        if pos > start and pos < end:\n",
    "        \n",
    "            if result_final__['strand'][i] == '+':\n",
    "                #decide which allele to use\n",
    "                #check \n",
    "                #check how long the y0036 allele is \n",
    "                x968_len = len(result_final__['X968_allele'][i])\n",
    "                y0036_len = len(result_final__['Y0036_allele'][i])\n",
    "                \n",
    "                mutation = result_final__['Y0036_allele'][i]\n",
    "               \n",
    "\n",
    "                gene = gene[:pos-start] + mutation + gene[pos-start+x968_len:]\n",
    "            elif result_final__['strand'][i] == '-':\n",
    "                x968_len = len(result_final__['X968_allele'][i])\n",
    "                y0036_len = len(result_final__['Y0036_allele'][i])\n",
    "                gene = gene[::-1]\n",
    "                gene = ''.join(gene)\n",
    "                mutation = result_final__['Y0036_allele'][i]\n",
    "                \n",
    "                gene = gene[:end-pos] + mutation + gene[end-pos+x968_len:]\n",
    "                \n",
    "            gene = 'A'*int(exon_len_so_far) + gene\n",
    "            if len(gene)%3 != 0:\n",
    "                gene = gene[:-(len(gene)%3)]\n",
    "            gene = [gene[k:k+3] for k in range(0, len(gene), 3)]\n",
    "                #only triplets near the position\n",
    "            if result_final__['strand'][i] == '+':\n",
    "                gene = gene[(pos-(start-int(exon_len_so_far)))//3-2:(pos-end)//3+25]\n",
    "            else:\n",
    "                gene = gene[(end-pos)//3-2:(start-int(exon_len_so_far)-pos)//3+25]\n",
    "            result_final__[\"mutant allele\"][i] = gene\n",
    "                \n",
    "           \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>position</th>\n",
       "      <th>ID</th>\n",
       "      <th>Parent</th>\n",
       "      <th>object</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>detail</th>\n",
       "      <th>p_value</th>\n",
       "      <th>correlation</th>\n",
       "      <th>p_value_corr</th>\n",
       "      <th>gene</th>\n",
       "      <th>X968_allele</th>\n",
       "      <th>Y0036_allele</th>\n",
       "      <th>reference_seq</th>\n",
       "      <th>reference_seq_</th>\n",
       "      <th>wt_gene</th>\n",
       "      <th>mutant allele</th>\n",
       "      <th>stop_codon_predicted</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2310884</td>\n",
       "      <td>SPNCRNA.5593</td>\n",
       "      <td>SPNCRNA.5593.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2310853</td>\n",
       "      <td>2311405</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>CAG |C| ATC</td>\n",
       "      <td>CAGCATC</td>\n",
       "      <td>[TTC, ACA, GCA, TCT, AGG, GAA, GGA, GGT, TTC, ...</td>\n",
       "      <td>[TTC, ACA, GTA, TCT, AGG, GAA, GGA, GGT, TTC, ...</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2311107</td>\n",
       "      <td>SPNCRNA.5593</td>\n",
       "      <td>SPNCRNA.5593.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2310853</td>\n",
       "      <td>2311405</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>AAC |G| GAT</td>\n",
       "      <td>AACGGAT</td>\n",
       "      <td>[ATA, GAA, ACG, GAT, GCT, TTG, TCA, TTT, TCC, ...</td>\n",
       "      <td>[ATA, GAA, ACC, GAT, GCT, TTG, TCA, TTT, TCC, ...</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>2316851</td>\n",
       "      <td>SPBC12D12.06</td>\n",
       "      <td>SPBC12D12.06.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2316770</td>\n",
       "      <td>2317136</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-0.044401</td>\n",
       "      <td>0.772115</td>\n",
       "      <td>srb11</td>\n",
       "      <td>TA</td>\n",
       "      <td>T</td>\n",
       "      <td>TGC |T| AAA</td>\n",
       "      <td>TGCTAAA</td>\n",
       "      <td>[TAC, ATG, CTA, AAA, AAA, AAT, GAG, GAA, AAG, ...</td>\n",
       "      <td>[TAC, ATG, CTA, AAA, AAA, ATG, AGG, AAA, AGG, ...</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>2326615</td>\n",
       "      <td>SPNCRNA.5602</td>\n",
       "      <td>SPNCRNA.5602.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2325881</td>\n",
       "      <td>2326882</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>ACT |C| TAG</td>\n",
       "      <td>ACTCTAG</td>\n",
       "      <td>[GCT, ACT, CTA, GCG, TTT, GTC, GCA, GAT, TCA, ...</td>\n",
       "      <td>[GCT, ACT, ATA, GCG, TTT, GTC, GCA, GAT, TCA, ...</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>2387677</td>\n",
       "      <td>SPNCRNA.5628</td>\n",
       "      <td>SPNCRNA.5628.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2387488</td>\n",
       "      <td>2388232</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>AAA |A| TAG</td>\n",
       "      <td>AAAATAG</td>\n",
       "      <td>[CGC, TCA, AAA, TAG, TAG, TCA, ATT, TCG, AGA, ...</td>\n",
       "      <td>[CGC, TCA, AAG, TAG, TAG, TCA, ATT, TCG, AGA, ...</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>2436559</td>\n",
       "      <td>SPNCRNA.5658</td>\n",
       "      <td>SPNCRNA.5658.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2436317</td>\n",
       "      <td>2436671</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>CCA |T| GAC</td>\n",
       "      <td>CCATGAC</td>\n",
       "      <td>[CGT, TCC, ATG, ACC, ATC, GCT, AAC, AAC, TTC, ...</td>\n",
       "      <td>[CGT, TCC, ACG, ACC, ATC, GCT, AAC, AAC, TTC, ...</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>2442808</td>\n",
       "      <td>SPRRNA.40</td>\n",
       "      <td>SPRRNA.40.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2442700</td>\n",
       "      <td>2442852</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>GGT |G| TGG</td>\n",
       "      <td>GGTGTGG</td>\n",
       "      <td>[ATG, CCG, GTG, TGG, ATC, CGC, TTT, TGT, GGT, ...</td>\n",
       "      <td>[ATG, CCG, GTA, TGG, ATC, CGC, TTT, TGT, GGT, ...</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    position            ID          Parent object chromosome    start  \\\n",
       "3    2310884  SPNCRNA.5593  SPNCRNA.5593.1    CDS         II  2310853   \n",
       "5    2311107  SPNCRNA.5593  SPNCRNA.5593.1    CDS         II  2310853   \n",
       "7    2316851  SPBC12D12.06  SPBC12D12.06.1    CDS         II  2316770   \n",
       "10   2326615  SPNCRNA.5602  SPNCRNA.5602.1    CDS         II  2325881   \n",
       "18   2387677  SPNCRNA.5628  SPNCRNA.5628.1    CDS         II  2387488   \n",
       "22   2436559  SPNCRNA.5658  SPNCRNA.5658.1    CDS         II  2436317   \n",
       "27   2442808     SPRRNA.40     SPRRNA.40.1    CDS         II  2442700   \n",
       "\n",
       "        end strand detail  p_value  correlation  p_value_corr   gene  \\\n",
       "3   2311405      +   exon      1.0          NaN           NaN    NaN   \n",
       "5   2311405      +   exon      1.0          NaN           NaN    NaN   \n",
       "7   2317136      +   exon      1.0    -0.044401      0.772115  srb11   \n",
       "10  2326882      -   exon      1.0          NaN           NaN    NaN   \n",
       "18  2388232      -   exon      0.0          NaN           NaN    NaN   \n",
       "22  2436671      -   exon      0.0          NaN           NaN    NaN   \n",
       "27  2442852      -   exon      0.0          NaN           NaN    NaN   \n",
       "\n",
       "   X968_allele Y0036_allele reference_seq reference_seq_  \\\n",
       "3            C            T   CAG |C| ATC        CAGCATC   \n",
       "5            G            C   AAC |G| GAT        AACGGAT   \n",
       "7           TA            T   TGC |T| AAA        TGCTAAA   \n",
       "10           C            A   ACT |C| TAG        ACTCTAG   \n",
       "18           A            G   AAA |A| TAG        AAAATAG   \n",
       "22           T            C   CCA |T| GAC        CCATGAC   \n",
       "27           G            A   GGT |G| TGG        GGTGTGG   \n",
       "\n",
       "                                              wt_gene  \\\n",
       "3   [TTC, ACA, GCA, TCT, AGG, GAA, GGA, GGT, TTC, ...   \n",
       "5   [ATA, GAA, ACG, GAT, GCT, TTG, TCA, TTT, TCC, ...   \n",
       "7   [TAC, ATG, CTA, AAA, AAA, AAT, GAG, GAA, AAG, ...   \n",
       "10  [GCT, ACT, CTA, GCG, TTT, GTC, GCA, GAT, TCA, ...   \n",
       "18  [CGC, TCA, AAA, TAG, TAG, TCA, ATT, TCG, AGA, ...   \n",
       "22  [CGT, TCC, ATG, ACC, ATC, GCT, AAC, AAC, TTC, ...   \n",
       "27  [ATG, CCG, GTG, TGG, ATC, CGC, TTT, TGT, GGT, ...   \n",
       "\n",
       "                                        mutant allele stop_codon_predicted  \n",
       "3   [TTC, ACA, GTA, TCT, AGG, GAA, GGA, GGT, TTC, ...                  yes  \n",
       "5   [ATA, GAA, ACC, GAT, GCT, TTG, TCA, TTT, TCC, ...                  yes  \n",
       "7   [TAC, ATG, CTA, AAA, AAA, ATG, AGG, AAA, AGG, ...                  yes  \n",
       "10  [GCT, ACT, ATA, GCG, TTT, GTC, GCA, GAT, TCA, ...                  yes  \n",
       "18  [CGC, TCA, AAG, TAG, TAG, TCA, ATT, TCG, AGA, ...                  yes  \n",
       "22  [CGT, TCC, ACG, ACC, ATC, GCT, AAC, AAC, TTC, ...                  yes  \n",
       "27  [ATG, CCG, GTA, TGG, ATC, CGC, TTT, TGT, GGT, ...                  yes  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#CreATE A DF WHERE WHOLE GENE IS NOT NAN\n",
    "stop_codon_df = result_final__[result_final__['wt_gene'].notna()]\n",
    "stop_codon_df[\"stop_codon_predicted\"] = np.nan\n",
    "#iterate through mutant allele, and see whether it contains\n",
    "#reset index\n",
    "stop_codon_df = stop_codon_df.reset_index(drop=True)\n",
    "\n",
    "for i in range(len(stop_codon_df)):\n",
    "    for j in range(len(stop_codon_df['mutant allele'][i])):\n",
    "        if stop_codon_df[\"strand\"][i] == '+':\n",
    "            if stop_codon_df['mutant allele'][i][j] == 'TAA' or stop_codon_df['mutant allele'][i][j] == 'TAG' or stop_codon_df['mutant allele'][i][j] == 'TGA':\n",
    "                stop_codon_df['stop_codon_predicted'][i] = 'yes'\n",
    "        else:\n",
    "            if stop_codon_df['mutant allele'][i][j] == 'ATT' or stop_codon_df['mutant allele'][i][j] == 'ATC' or stop_codon_df['mutant allele'][i][j] == 'ACT':\n",
    "                stop_codon_df['stop_codon_predicted'][i] = 'yes'\n",
    "#show which entries have a stop codon predicted\n",
    "stop_codon_df[stop_codon_df['stop_codon_predicted'] == 'yes']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "#in stop_codon_df create a column \"possible frameshift\" and fill it with nan. Then loop through X96_allele and Y0036_allele and check if the length is different. If yes and the difference is not divisible by 3, fill the column with \"yes\". Then show the entries where the column is \"yes\"\n",
    "frameshift_df = stop_codon_df.copy()\n",
    "frameshift_df['possible frameshift'] = np.nan\n",
    "\n",
    "for i in range(len(frameshift_df)):\n",
    "    if len(frameshift_df['X968_allele'][i]) != len(frameshift_df['Y0036_allele'][i]):\n",
    "        if (len(frameshift_df['X968_allele'][i]) - len(frameshift_df['Y0036_allele'][i]))%3 != 0:\n",
    "            frameshift_df['possible frameshift'][i] = 'yes'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "#import a codon table as a dictionary\n",
    "codon_table = {\n",
    "    'TTT': 'F', 'TTC': 'F', 'TTA': 'L', 'TTG': 'L',\n",
    "    'TCT': 'S', 'TCC': 'S', 'TCA': 'S', 'TCG': 'S',\n",
    "    'TAT': 'Y', 'TAC': 'Y', 'TAA': '*', 'TAG': '*',\n",
    "    'TGT': 'C', 'TGC': 'C', 'TGA': '*', 'TGG': 'W',\n",
    "    'CTT': 'L', 'CTC': 'L', 'CTA': 'L', 'CTG': 'L',\n",
    "    'CCT': 'P', 'CCC': 'P', 'CCA': 'P', 'CCG': 'P',\n",
    "    'CAT': 'H', 'CAC': 'H', 'CAA': 'Q', 'CAG': 'Q', \n",
    "    'CGT': 'R', 'CGC': 'R', 'CGA': 'R', 'CGG': 'R',\n",
    "    'ATT': 'I', 'ATC': 'I', 'ATA': 'I', 'ATG': 'M',\n",
    "    'ACT': 'T', 'ACC': 'T', 'ACA': 'T', 'ACG': 'T',\n",
    "    'AAT': 'N', 'AAC': 'N', 'AAA': 'K', 'AAG': 'K',\n",
    "    'AGT': 'S', 'AGC': 'S', 'AGA': 'R', 'AGG': 'R',\n",
    "    'GTT': 'V', 'GTC': 'V', 'GTA': 'V', 'GTG': 'V',\n",
    "    'GCT': 'A', 'GCC': 'A', 'GCA': 'A', 'GCG': 'A',\n",
    "    'GAT': 'D', 'GAC': 'D', 'GAA': 'E', 'GAG': 'E',\n",
    "    'GGT': 'G', 'GGC': 'G', 'GGA': 'G', 'GGG': 'G',\n",
    "}\n",
    "#iterate through frameshift_df and trnslate the lists of triplets found in wt_gene and mutant allele\n",
    "frameshift_df['wt_gene_translated'] = np.nan\n",
    "frameshift_df['mutant allele_translated'] = np.nan\n",
    "for i in range(len(frameshift_df)):\n",
    "    #if strand is -, then change a to t and so on\n",
    "    if frameshift_df['strand'][i] == '-':\n",
    "        temp_wt_allele = frameshift_df['wt_gene'][i]\n",
    "        temp_mut_allele = frameshift_df['mutant allele'][i]\n",
    "        for j in range(len(temp_wt_allele)):\n",
    "            temp_wt_allele[j] = temp_wt_allele[j].replace('A','t')\n",
    "            temp_wt_allele[j] = temp_wt_allele[j].replace('T','a')\n",
    "            temp_wt_allele[j] = temp_wt_allele[j].replace('C','g')\n",
    "            temp_wt_allele[j] = temp_wt_allele[j].replace('G','c')\n",
    "            temp_wt_allele[j] = temp_wt_allele[j].upper()\n",
    "        for j in range(len(temp_mut_allele)):\n",
    "            temp_mut_allele[j] = temp_mut_allele[j].replace('A','t')\n",
    "            temp_mut_allele[j] = temp_mut_allele[j].replace('T','a')\n",
    "            temp_mut_allele[j] = temp_mut_allele[j].replace('C','g')\n",
    "            temp_mut_allele[j] = temp_mut_allele[j].replace('G','c')\n",
    "            temp_mut_allele[j] = temp_mut_allele[j].upper()\n",
    "        frameshift_df['wt_gene_translated'][i] = ''.join([codon_table[codon] for codon in temp_wt_allele])\n",
    "        frameshift_df['mutant allele_translated'][i] = ''.join([codon_table[codon] for codon in temp_mut_allele])\n",
    "    else:\n",
    "        frameshift_df['wt_gene_translated'][i] = ''.join([codon_table[codon] for codon in frameshift_df['wt_gene'][i]])\n",
    "        frameshift_df['mutant allele_translated'][i] = ''.join([codon_table[codon] for codon in frameshift_df['mutant allele'][i]])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "13"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#count how many times wt_gene_translated is not equal to mutant allele_translated when ID doesnt contain SPNC\n",
    "frameshift_df[(frameshift_df['ID'].str.contains('RNA') == False) & (frameshift_df['wt_gene_translated'] != frameshift_df['mutant allele_translated'])].shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "26"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#display all cols\n",
    "pd.set_option('display.max_columns', None)\n",
    "#number of unique position, from rows where ID doesnt contain RNA\n",
    "frameshift_df[frameshift_df['ID'].str.contains('RNA') == False]['position'].nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>position</th>\n",
       "      <th>ID</th>\n",
       "      <th>Parent</th>\n",
       "      <th>object</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>detail</th>\n",
       "      <th>p_value</th>\n",
       "      <th>correlation</th>\n",
       "      <th>p_value_corr</th>\n",
       "      <th>gene</th>\n",
       "      <th>X968_allele</th>\n",
       "      <th>Y0036_allele</th>\n",
       "      <th>reference_seq</th>\n",
       "      <th>reference_seq_</th>\n",
       "      <th>wt_gene</th>\n",
       "      <th>mutant allele</th>\n",
       "      <th>stop_codon_predicted</th>\n",
       "      <th>possible frameshift</th>\n",
       "      <th>wt_gene_translated</th>\n",
       "      <th>mutant allele_translated</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>2316851</td>\n",
       "      <td>SPBC12D12.06</td>\n",
       "      <td>SPBC12D12.06.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2316770</td>\n",
       "      <td>2317136</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-0.044401</td>\n",
       "      <td>0.772115</td>\n",
       "      <td>srb11</td>\n",
       "      <td>TA</td>\n",
       "      <td>T</td>\n",
       "      <td>TGC |T| AAA</td>\n",
       "      <td>TGCTAAA</td>\n",
       "      <td>[TAC, ATG, CTA, AAA, AAA, AAT, GAG, GAA, AAG, ...</td>\n",
       "      <td>[TAC, ATG, CTA, AAA, AAA, ATG, AGG, AAA, AGG, ...</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>YMLKKNEEKGFSLEALVATCIYLSCKV</td>\n",
       "      <td>YMLKKMRKRVFLWRR*WLHAFIYHAKW</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>2442808</td>\n",
       "      <td>SPRRNA.40</td>\n",
       "      <td>SPRRNA.40.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2442700</td>\n",
       "      <td>2442852</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>GGT |G| TGG</td>\n",
       "      <td>GGTGTGG</td>\n",
       "      <td>[TAC, GGC, CAC, ACC, TAG, GCG, AAA, ACA, CCA, ...</td>\n",
       "      <td>[TAC, GGC, CAT, ACC, TAG, GCG, AAA, ACA, CCA, ...</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>YGHT*AKTPVPVRSLQLSV*GPR*YYGW</td>\n",
       "      <td>YGHT*AKTPVPVRSLQLSV*GPR*YYGW</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    position            ID          Parent object chromosome    start  \\\n",
       "7    2316851  SPBC12D12.06  SPBC12D12.06.1    CDS         II  2316770   \n",
       "27   2442808     SPRRNA.40     SPRRNA.40.1    CDS         II  2442700   \n",
       "\n",
       "        end strand detail  p_value  correlation  p_value_corr   gene  \\\n",
       "7   2317136      +   exon      1.0    -0.044401      0.772115  srb11   \n",
       "27  2442852      -   exon      0.0          NaN           NaN    NaN   \n",
       "\n",
       "   X968_allele Y0036_allele reference_seq reference_seq_  \\\n",
       "7           TA            T   TGC |T| AAA        TGCTAAA   \n",
       "27           G            A   GGT |G| TGG        GGTGTGG   \n",
       "\n",
       "                                              wt_gene  \\\n",
       "7   [TAC, ATG, CTA, AAA, AAA, AAT, GAG, GAA, AAG, ...   \n",
       "27  [TAC, GGC, CAC, ACC, TAG, GCG, AAA, ACA, CCA, ...   \n",
       "\n",
       "                                        mutant allele stop_codon_predicted  \\\n",
       "7   [TAC, ATG, CTA, AAA, AAA, ATG, AGG, AAA, AGG, ...                  yes   \n",
       "27  [TAC, GGC, CAT, ACC, TAG, GCG, AAA, ACA, CCA, ...                  yes   \n",
       "\n",
       "   possible frameshift            wt_gene_translated  \\\n",
       "7                  yes   YMLKKNEEKGFSLEALVATCIYLSCKV   \n",
       "27                 NaN  YGHT*AKTPVPVRSLQLSV*GPR*YYGW   \n",
       "\n",
       "        mutant allele_translated  \n",
       "7    YMLKKMRKRVFLWRR*WLHAFIYHAKW  \n",
       "27  YGHT*AKTPVPVRSLQLSV*GPR*YYGW  "
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#show where gene doesnt start with SPNC and possible frameshift\n",
    "#show all rows and stop_codon_predicted\n",
    "pd.set_option('display.max_rows', None)\n",
    "frameshift_df[(frameshift_df['ID'].str.contains('SPNC') == False) & (frameshift_df['stop_codon_predicted'] == 'yes')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Last updated: 2023-11-05T10:32:25.211706+00:00\n",
      "\n",
      "Python implementation: CPython\n",
      "Python version       : 3.9.0\n",
      "IPython version      : 7.31.1\n",
      "\n",
      "Compiler    : Clang 11.0.0 \n",
      "OS          : Darwin\n",
      "Release     : 21.5.0\n",
      "Machine     : x86_64\n",
      "Processor   : i386\n",
      "CPU cores   : 8\n",
      "Architecture: 64bit\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%load_ext watermark\n",
    "%watermark\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "matplotlib: 3.8.0\n",
      "seaborn   : 0.13.0\n",
      "pandas    : 1.4.4\n",
      "sys       : 3.9.0 | packaged by conda-forge | (default, Nov 26 2020, 07:54:06) \n",
      "[Clang 11.0.0 ]\n",
      "re        : 2.2.1\n",
      "scipy     : 1.11.3\n",
      "numpy     : 1.23.4\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%watermark --iversions\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.9.12 ('base')",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "2d0398177d7bb10ea33b6a5cd9d022d63e20680fcfd84762562c72ff35bdf44b"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
