{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "#find snps\n",
    "#import data science libraries\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "#import /Users/bencekover/Library/CloudStorage/OneDrive-UniversityCollegeLondon/MSci Bahler lab/S.-Pombe-biofilm/Bence folder/QTL mapping/permutations/results.csv\n",
    "#read in data\n",
    "root = \"/Users/bencekover/Library/CloudStorage/OneDrive-Personal/MSci Bahler lab/S.-Pombe-MLPs - Github/\"\n",
    "snps = pd.read_csv(root + 'Bence folder/QTL mapping/permutations_emm_p/results_emm_p.csv')\n",
    "#washing phenotype /Users/bencekover/Library/CloudStorage/OneDrive-UniversityCollegeLondon/MSci Bahler lab/S.-Pombe-biofilm/Bence folder/QTL mapping/washing_phenotypes.csv\n",
    "wash_pheno = pd.read_csv(root + 'Bence folder/QTL mapping/washing_phenotypes.csv')\n",
    "#import rna seq corrs /Users/bencekover/Library/CloudStorage/OneDrive-UniversityCollegeLondon/MSci Bahler lab/S.-Pombe-biofilm/Bence folder/Analysis of Clement-Ziza RNA-seq/wash_corr.csv\n",
    "wash_corr = pd.read_csv(root + 'Bence folder/Analysis of Clement-Ziza RNA-seq/wash_corr.csv')\n",
    "#full snp dataset /Users/bencekover/Library/CloudStorage/OneDrive-UniversityCollegeLondon/MSci Bahler lab/S.-Pombe-biofilm/Bence folder/QTL mapping/SupplementaryDataset_S7_genotype.tsv\n",
    "full_snps = pd.read_csv(root + 'internal data/updated_genotype_matrix_final_pos.tsv', sep='\\t', header=0)\n",
    "#load orthologs database /Users/bencekover/Library/CloudStorage/OneDrive-UniversityCollegeLondon/MSci Bahler lab/S.-Pombe-biofilm/Bence folder/Orthologs/final_orthology_database.csv\n",
    "orthologs = pd.read_csv(root + 'Bence folder/Orthologs/final_orthology_database.csv')\n",
    "full_snps_og = full_snps.copy()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "full_snps.rename(columns={'968':'X968'}, inplace=True)\n",
    "full_snps.rename(columns={'968_allele':'X968_allele'}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "#iterate through snps and take a given row and the next one. If the Unnamed: 0 is different by less then 500 fill in the gap from full_snps\n",
    "\n",
    "snps = snps.rename(columns={'Unnamed: 0': 'number'})\n",
    "snps['number'] = snps['number'] - 1\n",
    "\n",
    "#create snpsfinal dataframe\n",
    "snpsfinal = pd.DataFrame()\n",
    "for i,j in enumerate(snps.number):\n",
    "    if i == len(snps.number) - 1:\n",
    "        break\n",
    "    if (j != snps.number[i+1] - 1) & (snps.number[i+1] - j < 100):\n",
    "        indices_between = list(range(j+1, snps.number[i+1]))\n",
    "        #append rows from full_snps to snps to the right index, not end of dataframe\n",
    "        if snpsfinal.empty:\n",
    "            snpsfinal = pd.concat([snps.iloc[:i+1], full_snps.iloc[indices_between], snps.iloc[i+1:]], ignore_index=True)\n",
    "        else:\n",
    "            snpsfinal = pd.concat([snpsfinal.iloc[:i+1], full_snps.iloc[indices_between], snpsfinal.iloc[i+1:]], ignore_index=True)\n",
    "\n",
    "snps = snpsfinal.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "#remove first 4 columns\n",
    "full_snps = full_snps.iloc[:,4:]\n",
    "\n",
    "#re order wash_pheno such that order of strains matches columns in full_snps column names\n",
    "pca_pheno = pd.DataFrame(columns = ['strain','ratio'])\n",
    "for i in full_snps.columns:\n",
    "    #see what is the column name, and find the row in wash_pheno where strain has the same name\n",
    "    \n",
    "    row = wash_pheno.loc[wash_pheno['strain'] == i]\n",
    "    pca_pheno = pca_pheno.append(row)\n",
    "#pcapheno remove column Unnamed: 0\n",
    "pca_pheno.drop(columns=['Unnamed: 0'], inplace=True)\n",
    "#reset index\n",
    "pca_pheno.reset_index(inplace=True)\n",
    "#remove chromosome_ from chromosome column\n",
    "snps['chromosome'] = snps['chromosome'].str.replace('chromosome_', '')\n",
    "#change chromosome number to roman numbers\n",
    "snps['chromosome'] = snps['chromosome'].str.replace('1','I')\n",
    "snps['chromosome'] = snps['chromosome'].str.replace('2','II')\n",
    "snps['chromosome'] = snps['chromosome'].str.replace('3','III')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "snps_t = snps.transpose()\n",
    "#remove first 5 rows\n",
    "snps_t = snps_t.iloc[5:]\n",
    "#by matching index of snps_t to column strain in wash_pheno, we can add the phenotype to the snps_t dataframe\n",
    "snps_t['phenotype'] = wash_pheno.set_index('strain')['ratio']\n",
    "#rename all but last column to snps df position \n",
    "#snps_t all but last columns rename as snps[\"position\"]\n",
    "ncols = len(snps_t.columns)\n",
    "snps_t.columns = np.append(snps[\"position\"].values, snps_t.columns[ncols-1]).flatten()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>2604820</th>\n",
       "      <th>2605777</th>\n",
       "      <th>2611769</th>\n",
       "      <th>2612379</th>\n",
       "      <th>2613145</th>\n",
       "      <th>2616395</th>\n",
       "      <th>2617567</th>\n",
       "      <th>2617646</th>\n",
       "      <th>2622695</th>\n",
       "      <th>2624384</th>\n",
       "      <th>...</th>\n",
       "      <th>4582389</th>\n",
       "      <th>4583262</th>\n",
       "      <th>4584481</th>\n",
       "      <th>4585323</th>\n",
       "      <th>4585663</th>\n",
       "      <th>4587327</th>\n",
       "      <th>4588747</th>\n",
       "      <th>4595875</th>\n",
       "      <th>4602196</th>\n",
       "      <th>phenotype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>X968</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.04793</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Y0036</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.172119</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.054247</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_2</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.061369</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_3</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.033055</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_5</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.061934</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_6</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.036447</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_7</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.020331</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_8</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.04712</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_10</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.028066</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_11</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.040438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_12</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.421832</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_13</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.039811</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_16</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.023504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_15</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.029537</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_17</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.025863</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_18</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.151611</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_19</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.046443</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_20</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.035152</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_21</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.043023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_22</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.01935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_23</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.034654</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_24</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.078209</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_14</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.042707</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_25</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.038677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_26</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.030804</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_27</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.038478</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_28</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.015525</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_29</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.019658</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_30</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.020797</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_31</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.041101</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_32</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.028581</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_33</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.016066</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_35</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.066556</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_36</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.516687</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_37</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.03192</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_38</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.025814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_39</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.364</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_40</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.03628</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_41</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.014202</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_42</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.073556</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_43</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.058999</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_46</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.017455</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_47</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.099641</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_9</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.034268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_34</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.480026</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_44</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.03895</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_49</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.133768</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_50</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.40269</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_51</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.868284</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_52</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.534445</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_53</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.018397</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_54</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.02434</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R1_4</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.019548</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>p_vals</th>\n",
       "      <td>0.002038</td>\n",
       "      <td>0.002038</td>\n",
       "      <td>0.001651</td>\n",
       "      <td>0.001651</td>\n",
       "      <td>0.001651</td>\n",
       "      <td>0.001651</td>\n",
       "      <td>0.001651</td>\n",
       "      <td>0.001651</td>\n",
       "      <td>0.001651</td>\n",
       "      <td>0.001651</td>\n",
       "      <td>...</td>\n",
       "      <td>0.039653</td>\n",
       "      <td>0.039653</td>\n",
       "      <td>0.039653</td>\n",
       "      <td>0.039653</td>\n",
       "      <td>0.039653</td>\n",
       "      <td>0.039653</td>\n",
       "      <td>0.039653</td>\n",
       "      <td>0.039653</td>\n",
       "      <td>0.039653</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>55 rows Ã— 410 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         2604820   2605777   2611769   2612379   2613145   2616395   2617567  \\\n",
       "X968           0         0         0         0         0         0         0   \n",
       "Y0036        1.0       1.0       1.0       1.0       1.0       1.0       1.0   \n",
       "R1_1         0.0       0.0       0.0       0.0       0.0       0.0       0.0   \n",
       "R1_2         1.0       1.0       1.0       1.0       1.0       1.0       1.0   \n",
       "R1_3         1.0       1.0       1.0       1.0       1.0       1.0       1.0   \n",
       "R1_5         0.0       0.0       0.0       0.0       0.0       0.0       0.0   \n",
       "R1_6         1.0       1.0       1.0       1.0       1.0       1.0       1.0   \n",
       "R1_7         0.0       0.0       0.0       0.0       0.0       0.0       0.0   \n",
       "R1_8         0.0       0.0       0.0       0.0       0.0       0.0       0.0   \n",
       "R1_10        0.0       0.0       0.0       0.0       0.0       0.0       0.0   \n",
       "R1_11        0.0       0.0       0.0       0.0       0.0       0.0       0.0   \n",
       "R1_12        1.0       1.0       1.0       1.0       1.0       1.0       1.0   \n",
       "R1_13        1.0       1.0       1.0       1.0       1.0       1.0       1.0   \n",
       "R1_16        1.0       1.0       1.0       1.0       1.0       1.0       1.0   \n",
       "R1_15        1.0       1.0       1.0       1.0       1.0       1.0       1.0   \n",
       "R1_17        1.0       1.0       1.0       1.0       1.0       1.0       1.0   \n",
       "R1_18        1.0       1.0       1.0       1.0       1.0       1.0       1.0   \n",
       "R1_19        1.0       1.0       1.0       1.0       1.0       1.0       1.0   \n",
       "R1_20        1.0       1.0       1.0       1.0       1.0       1.0       1.0   \n",
       "R1_21        0.0       0.0       0.0       0.0       0.0       0.0       0.0   \n",
       "R1_22        1.0       1.0       1.0       1.0       1.0       1.0       1.0   \n",
       "R1_23        1.0       1.0       1.0       1.0       1.0       1.0       1.0   \n",
       "R1_24        1.0       1.0       1.0       1.0       1.0       1.0       1.0   \n",
       "R1_14        1.0       1.0       1.0       1.0       1.0       1.0       1.0   \n",
       "R1_25        1.0       1.0       1.0       1.0       1.0       1.0       1.0   \n",
       "R1_26        1.0       1.0       1.0       1.0       1.0       1.0       1.0   \n",
       "R1_27        0.0       0.0       0.0       0.0       0.0       0.0       0.0   \n",
       "R1_28        1.0       1.0       1.0       1.0       1.0       1.0       1.0   \n",
       "R1_29        0.0       0.0       0.0       0.0       0.0       0.0       0.0   \n",
       "R1_30        1.0       1.0       1.0       1.0       1.0       1.0       1.0   \n",
       "R1_31        1.0       1.0       1.0       1.0       1.0       1.0       1.0   \n",
       "R1_32        1.0       1.0       1.0       1.0       1.0       1.0       1.0   \n",
       "R1_33        1.0       1.0       1.0       1.0       1.0       1.0       1.0   \n",
       "R1_35        1.0       1.0       1.0       1.0       1.0       1.0       1.0   \n",
       "R1_36        0.0       0.0       0.0       0.0       0.0       0.0       0.0   \n",
       "R1_37        1.0       1.0       1.0       1.0       1.0       1.0       1.0   \n",
       "R1_38        1.0       1.0       1.0       1.0       1.0       1.0       1.0   \n",
       "R1_39        0.0       0.0       0.0       0.0       0.0       0.0       0.0   \n",
       "R1_40        1.0       1.0       1.0       1.0       1.0       1.0       1.0   \n",
       "R1_41        1.0       1.0       1.0       1.0       1.0       1.0       1.0   \n",
       "R1_42        0.0       0.0       0.0       0.0       0.0       0.0       0.0   \n",
       "R1_43        1.0       1.0       1.0       1.0       1.0       1.0       1.0   \n",
       "R1_46        NaN       NaN       1.0       1.0       1.0       1.0       1.0   \n",
       "R1_47        1.0       1.0       1.0       1.0       1.0       1.0       1.0   \n",
       "R1_9         1.0       1.0       1.0       1.0       1.0       1.0       1.0   \n",
       "R1_34        0.0       0.0       0.0       0.0       0.0       0.0       0.0   \n",
       "R1_44        1.0       1.0       1.0       1.0       1.0       1.0       1.0   \n",
       "R1_49        NaN       NaN       NaN       NaN       NaN       NaN       NaN   \n",
       "R1_50        NaN       NaN       NaN       NaN       NaN       NaN       NaN   \n",
       "R1_51        NaN       NaN       NaN       NaN       NaN       NaN       NaN   \n",
       "R1_52        1.0       1.0       1.0       1.0       1.0       1.0       1.0   \n",
       "R1_53        1.0       1.0       1.0       1.0       1.0       1.0       1.0   \n",
       "R1_54        0.0       0.0       0.0       0.0       0.0       0.0       0.0   \n",
       "R1_4         0.0       0.0       0.0       0.0       0.0       0.0       0.0   \n",
       "p_vals  0.002038  0.002038  0.001651  0.001651  0.001651  0.001651  0.001651   \n",
       "\n",
       "         2617646   2622695   2624384  ...   4582389   4583262   4584481  \\\n",
       "X968           0         0         0  ...         0         0         0   \n",
       "Y0036        1.0       1.0       1.0  ...       1.0       1.0       1.0   \n",
       "R1_1         0.0       0.0       0.0  ...       0.0       0.0       0.0   \n",
       "R1_2         1.0       1.0       1.0  ...       1.0       1.0       1.0   \n",
       "R1_3         1.0       1.0       1.0  ...       1.0       1.0       1.0   \n",
       "R1_5         0.0       0.0       0.0  ...       1.0       1.0       1.0   \n",
       "R1_6         1.0       1.0       1.0  ...       1.0       1.0       1.0   \n",
       "R1_7         0.0       0.0       0.0  ...       0.0       0.0       0.0   \n",
       "R1_8         0.0       0.0       0.0  ...       0.0       0.0       0.0   \n",
       "R1_10        0.0       0.0       0.0  ...       0.0       0.0       0.0   \n",
       "R1_11        0.0       0.0       0.0  ...       0.0       0.0       0.0   \n",
       "R1_12        1.0       1.0       1.0  ...       1.0       1.0       1.0   \n",
       "R1_13        1.0       1.0       1.0  ...       1.0       1.0       1.0   \n",
       "R1_16        1.0       1.0       1.0  ...       1.0       1.0       1.0   \n",
       "R1_15        1.0       1.0       1.0  ...       1.0       1.0       1.0   \n",
       "R1_17        1.0       1.0       1.0  ...       1.0       1.0       1.0   \n",
       "R1_18        1.0       1.0       1.0  ...       1.0       1.0       1.0   \n",
       "R1_19        1.0       1.0       1.0  ...       1.0       1.0       1.0   \n",
       "R1_20        1.0       1.0       1.0  ...       1.0       1.0       1.0   \n",
       "R1_21        0.0       0.0       0.0  ...       0.0       0.0       0.0   \n",
       "R1_22        1.0       1.0       1.0  ...       1.0       1.0       1.0   \n",
       "R1_23        1.0       1.0       1.0  ...       0.0       0.0       0.0   \n",
       "R1_24        1.0       1.0       1.0  ...       1.0       1.0       1.0   \n",
       "R1_14        1.0       1.0       1.0  ...       0.0       0.0       0.0   \n",
       "R1_25        1.0       1.0       1.0  ...       1.0       1.0       1.0   \n",
       "R1_26        1.0       1.0       1.0  ...       0.0       0.0       0.0   \n",
       "R1_27        0.0       0.0       0.0  ...       0.0       0.0       0.0   \n",
       "R1_28        1.0       1.0       1.0  ...       1.0       1.0       1.0   \n",
       "R1_29        0.0       0.0       0.0  ...       0.0       0.0       0.0   \n",
       "R1_30        1.0       1.0       1.0  ...       1.0       1.0       1.0   \n",
       "R1_31        1.0       1.0       1.0  ...       1.0       1.0       1.0   \n",
       "R1_32        1.0       1.0       1.0  ...       1.0       1.0       1.0   \n",
       "R1_33        1.0       1.0       1.0  ...       1.0       1.0       1.0   \n",
       "R1_35        1.0       1.0       1.0  ...       0.0       0.0       0.0   \n",
       "R1_36        0.0       0.0       0.0  ...       1.0       1.0       1.0   \n",
       "R1_37        1.0       1.0       1.0  ...       1.0       1.0       1.0   \n",
       "R1_38        1.0       1.0       1.0  ...       1.0       1.0       1.0   \n",
       "R1_39        0.0       0.0       0.0  ...       0.0       0.0       0.0   \n",
       "R1_40        1.0       1.0       1.0  ...       1.0       1.0       1.0   \n",
       "R1_41        1.0       1.0       1.0  ...       1.0       1.0       1.0   \n",
       "R1_42        0.0       0.0       0.0  ...       0.0       0.0       0.0   \n",
       "R1_43        1.0       1.0       1.0  ...       0.0       0.0       0.0   \n",
       "R1_46        1.0       1.0       1.0  ...       1.0       1.0       1.0   \n",
       "R1_47        1.0       1.0       1.0  ...       1.0       1.0       1.0   \n",
       "R1_9         1.0       1.0       1.0  ...       1.0       1.0       1.0   \n",
       "R1_34        0.0       0.0       0.0  ...       0.0       0.0       0.0   \n",
       "R1_44        1.0       1.0       1.0  ...       1.0       1.0       1.0   \n",
       "R1_49        NaN       NaN       NaN  ...       NaN       NaN       NaN   \n",
       "R1_50        NaN       NaN       NaN  ...       NaN       NaN       NaN   \n",
       "R1_51        NaN       NaN       NaN  ...       NaN       NaN       NaN   \n",
       "R1_52        1.0       1.0       1.0  ...       1.0       1.0       1.0   \n",
       "R1_53        1.0       1.0       1.0  ...       1.0       1.0       1.0   \n",
       "R1_54        0.0       0.0       0.0  ...       0.0       0.0       0.0   \n",
       "R1_4         0.0       0.0       0.0  ...       0.0       0.0       0.0   \n",
       "p_vals  0.001651  0.001651  0.001651  ...  0.039653  0.039653  0.039653   \n",
       "\n",
       "         4585323   4585663   4587327   4588747   4595875   4602196 phenotype  \n",
       "X968           0         0         0         0         0         0   0.04793  \n",
       "Y0036        1.0       1.0       1.0       1.0       1.0       1.0  0.172119  \n",
       "R1_1         0.0       0.0       0.0       0.0       0.0       0.0  0.054247  \n",
       "R1_2         1.0       1.0       1.0       1.0       1.0       1.0  0.061369  \n",
       "R1_3         1.0       1.0       1.0       1.0       1.0       1.0  0.033055  \n",
       "R1_5         1.0       1.0       1.0       1.0       1.0       1.0  0.061934  \n",
       "R1_6         1.0       1.0       1.0       1.0       1.0       1.0  0.036447  \n",
       "R1_7         0.0       0.0       0.0       0.0       0.0       0.0  1.020331  \n",
       "R1_8         0.0       0.0       0.0       0.0       0.0       0.0   0.04712  \n",
       "R1_10        0.0       0.0       0.0       0.0       0.0       0.0  0.028066  \n",
       "R1_11        0.0       0.0       0.0       0.0       0.0       0.0  0.040438  \n",
       "R1_12        1.0       1.0       1.0       1.0       1.0       1.0  0.421832  \n",
       "R1_13        1.0       1.0       1.0       1.0       1.0       1.0  0.039811  \n",
       "R1_16        1.0       1.0       1.0       1.0       1.0       1.0  0.023504  \n",
       "R1_15        1.0       1.0       1.0       1.0       1.0       1.0  0.029537  \n",
       "R1_17        1.0       1.0       1.0       1.0       1.0       1.0  0.025863  \n",
       "R1_18        1.0       1.0       1.0       1.0       1.0       1.0  0.151611  \n",
       "R1_19        1.0       1.0       1.0       1.0       1.0       1.0  0.046443  \n",
       "R1_20        1.0       1.0       1.0       1.0       1.0       1.0  0.035152  \n",
       "R1_21        0.0       0.0       0.0       0.0       0.0       0.0  0.043023  \n",
       "R1_22        1.0       1.0       1.0       1.0       1.0       1.0   0.01935  \n",
       "R1_23        0.0       0.0       0.0       0.0       0.0       0.0  0.034654  \n",
       "R1_24        1.0       1.0       1.0       1.0       1.0       1.0  0.078209  \n",
       "R1_14        0.0       0.0       0.0       0.0       0.0       0.0  0.042707  \n",
       "R1_25        1.0       1.0       1.0       1.0       1.0       1.0  0.038677  \n",
       "R1_26        0.0       0.0       0.0       0.0       0.0       0.0  0.030804  \n",
       "R1_27        0.0       0.0       0.0       0.0       0.0       0.0  0.038478  \n",
       "R1_28        1.0       1.0       1.0       1.0       1.0       1.0  0.015525  \n",
       "R1_29        0.0       0.0       0.0       0.0       0.0       0.0  0.019658  \n",
       "R1_30        1.0       1.0       1.0       1.0       1.0       1.0  0.020797  \n",
       "R1_31        1.0       1.0       1.0       1.0       1.0       1.0  0.041101  \n",
       "R1_32        1.0       1.0       1.0       1.0       1.0       1.0  0.028581  \n",
       "R1_33        1.0       1.0       1.0       1.0       1.0       1.0  0.016066  \n",
       "R1_35        0.0       0.0       0.0       0.0       0.0       0.0  0.066556  \n",
       "R1_36        1.0       1.0       1.0       1.0       1.0       1.0  0.516687  \n",
       "R1_37        1.0       1.0       1.0       1.0       1.0       1.0   0.03192  \n",
       "R1_38        1.0       1.0       1.0       1.0       1.0       1.0  0.025814  \n",
       "R1_39        0.0       0.0       0.0       0.0       0.0       0.0     0.364  \n",
       "R1_40        1.0       1.0       1.0       1.0       1.0       1.0   0.03628  \n",
       "R1_41        1.0       1.0       1.0       1.0       1.0       1.0  0.014202  \n",
       "R1_42        0.0       0.0       0.0       0.0       0.0       0.0  0.073556  \n",
       "R1_43        0.0       0.0       0.0       0.0       0.0       0.0  0.058999  \n",
       "R1_46        1.0       1.0       1.0       1.0       1.0       1.0  0.017455  \n",
       "R1_47        1.0       1.0       1.0       1.0       1.0       1.0  0.099641  \n",
       "R1_9         1.0       1.0       1.0       1.0       1.0       1.0  0.034268  \n",
       "R1_34        0.0       0.0       0.0       0.0       0.0       0.0  0.480026  \n",
       "R1_44        1.0       1.0       1.0       1.0       1.0       1.0   0.03895  \n",
       "R1_49        NaN       NaN       NaN       NaN       NaN       NaN  0.133768  \n",
       "R1_50        NaN       NaN       NaN       NaN       NaN       NaN   0.40269  \n",
       "R1_51        NaN       NaN       NaN       NaN       NaN       NaN  0.868284  \n",
       "R1_52        1.0       1.0       1.0       1.0       1.0       1.0  0.534445  \n",
       "R1_53        1.0       1.0       1.0       1.0       1.0       1.0  0.018397  \n",
       "R1_54        0.0       0.0       0.0       0.0       0.0       0.0   1.02434  \n",
       "R1_4         0.0       0.0       0.0       0.0       0.0       0.0  1.019548  \n",
       "p_vals  0.039653  0.039653  0.039653  0.039653  0.039653  0.039653        NA  \n",
       "\n",
       "[55 rows x 410 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#for each column perform a t-test between phenotypes of rows with 0 and 1\n",
    "#store p-values in a list\n",
    "import scipy.stats as st\n",
    "\n",
    "p_values = []\n",
    "#iterate through columns\n",
    "for i in range(0, len(snps_t.columns)-1):\n",
    "    #get column\n",
    "    col = snps_t.iloc[:,i]\n",
    "    #get phenotype column\n",
    "    pheno = snps_t.iloc[:,-1]\n",
    "    #get rows with 0 and 1\n",
    "    col_0 = col[col == 0]\n",
    "    col_1 = col[col == 1]\n",
    "    #get phenotype rows with 0 and 1\n",
    "    pheno_0 = pheno[col == 0]\n",
    "    pheno_1 = pheno[col == 1]\n",
    "    #perform t-test\n",
    "    t, p = st.ttest_ind(pheno_0, pheno_1)\n",
    "    p_values.append(p)\n",
    "#add final entry NA\n",
    "p_values.append(\"NA\")\n",
    "#add row with index p_vals\n",
    "snps_t.loc['p_vals'] = p_values\n",
    "snps_t\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "could not convert string to float: 'chromosome_1'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m/var/folders/mb/vdj4y7t13zd3tzlk99b8ycd40000gn/T/ipykernel_64309/964638402.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     11\u001b[0m     \u001b[0mfull_snps_pca\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0miloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfull_snps_pca\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0miloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfillna\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrandom\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mchoice\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     12\u001b[0m \u001b[0;31m#standardise data\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 13\u001b[0;31m \u001b[0mX_std\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mStandardScaler\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfit_transform\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfull_snps_pca\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     14\u001b[0m \u001b[0;31m#perform PCA\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     15\u001b[0m \u001b[0mpca\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mPCA\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mn_components\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m2\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/sklearn/base.py\u001b[0m in \u001b[0;36mfit_transform\u001b[0;34m(self, X, y, **fit_params)\u001b[0m\n\u001b[1;32m    697\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0my\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    698\u001b[0m             \u001b[0;31m# fit method of arity 1 (unsupervised transformation)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 699\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mX\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mfit_params\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtransform\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mX\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    700\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    701\u001b[0m             \u001b[0;31m# fit method of arity 2 (supervised transformation)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/sklearn/preprocessing/_data.py\u001b[0m in \u001b[0;36mfit\u001b[0;34m(self, X, y, sample_weight)\u001b[0m\n\u001b[1;32m    728\u001b[0m         \u001b[0;31m# Reset internal state before fitting\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    729\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_reset\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 730\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpartial_fit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mX\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msample_weight\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    731\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    732\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mpartial_fit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mX\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msample_weight\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/sklearn/preprocessing/_data.py\u001b[0m in \u001b[0;36mpartial_fit\u001b[0;34m(self, X, y, sample_weight)\u001b[0m\n\u001b[1;32m    764\u001b[0m         \"\"\"\n\u001b[1;32m    765\u001b[0m         \u001b[0mfirst_call\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mhasattr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"n_samples_seen_\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 766\u001b[0;31m         X = self._validate_data(X, accept_sparse=('csr', 'csc'),\n\u001b[0m\u001b[1;32m    767\u001b[0m                                 \u001b[0mestimator\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mFLOAT_DTYPES\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    768\u001b[0m                                 force_all_finite='allow-nan', reset=first_call)\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/sklearn/base.py\u001b[0m in \u001b[0;36m_validate_data\u001b[0;34m(self, X, y, reset, validate_separately, **check_params)\u001b[0m\n\u001b[1;32m    419\u001b[0m             \u001b[0mout\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mX\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    420\u001b[0m         \u001b[0;32melif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0my\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstr\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0my\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'no_validation'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 421\u001b[0;31m             \u001b[0mX\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcheck_array\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mX\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mcheck_params\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    422\u001b[0m             \u001b[0mout\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mX\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    423\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/sklearn/utils/validation.py\u001b[0m in \u001b[0;36minner_f\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m     61\u001b[0m             \u001b[0mextra_args\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m-\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mall_args\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     62\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mextra_args\u001b[0m \u001b[0;34m<=\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 63\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     64\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     65\u001b[0m             \u001b[0;31m# extra_args > 0\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/sklearn/utils/validation.py\u001b[0m in \u001b[0;36mcheck_array\u001b[0;34m(array, accept_sparse, accept_large_sparse, dtype, order, copy, force_all_finite, ensure_2d, allow_nd, ensure_min_samples, ensure_min_features, estimator)\u001b[0m\n\u001b[1;32m    671\u001b[0m                     \u001b[0marray\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0marray\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mastype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcasting\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"unsafe\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcopy\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    672\u001b[0m                 \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 673\u001b[0;31m                     \u001b[0marray\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0masarray\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marray\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0morder\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0morder\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    674\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mComplexWarning\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mcomplex_warning\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    675\u001b[0m                 raise ValueError(\"Complex data not supported\\n\"\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36m__array__\u001b[0;34m(self, dtype)\u001b[0m\n\u001b[1;32m   2062\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2063\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__array__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mnpt\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mDTypeLike\u001b[0m \u001b[0;34m|\u001b[0m \u001b[0;32mNone\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m->\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mndarray\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2064\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0masarray\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_values\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2065\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2066\u001b[0m     def __array_wrap__(\n",
      "\u001b[0;31mValueError\u001b[0m: could not convert string to float: 'chromosome_1'"
     ]
    }
   ],
   "source": [
    "#create PCA plot\n",
    "#take entries except last column and last row\n",
    "full_snps_pca = full_snps.T\n",
    "#perform PCA and label with rownames on plot, and colour by respecitve phenotypes (stored in snps_t [\"phenotype\"])\n",
    "from sklearn.decomposition import PCA\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "#for NaN values randomly replace them with 0 or 1\n",
    "for i in range(0, len(full_snps_pca.columns)):\n",
    "    full_snps_pca.iloc[:,i] = full_snps_pca.iloc[:,i].fillna(np.random.choice([0,1]))\n",
    "#standardise data\n",
    "X_std = StandardScaler().fit_transform(full_snps_pca)  \n",
    "#perform PCA\n",
    "pca = PCA(n_components=2)\n",
    "principalComponents = pca.fit_transform(X_std)\n",
    "#add principal components to dataframe\n",
    "principalDf = pd.DataFrame(data = principalComponents, columns = ['principal component 1', 'principal component 2'])\n",
    "#add phenotype column\n",
    "principalDf['phenotype'] = snps_t.iloc[:-1,-1].values\n",
    "#change phenotype column to float\n",
    "principalDf['phenotype'] = pca_pheno[\"ratio\"].astype(float)\n",
    "\n",
    "#plot\n",
    "fig = plt.figure(figsize = (8,8))\n",
    "ax = fig.add_subplot(1,1,1)\n",
    "ax.set_xlabel('Principal Component 1', fontsize = 15)\n",
    "ax.set_ylabel('Principal Component 2', fontsize = 15)\n",
    "ax.set_title('2 component PCA', fontsize = 20)\n",
    "#plot each point and color by viridis based on phenotype value\n",
    "ax.scatter(principalDf['principal component 1'], principalDf['principal component 2'], c=principalDf['phenotype'], cmap='viridis')\n",
    "#add rownames for points with phenotype above 0.2 or with names X968 and Y0036. Use snps_t_pca.index\n",
    "for i in range(0, len(principalDf.index)):\n",
    "    if principalDf.iloc[i,2] > 0.2 or principalDf.index[i] == 'X968' or principalDf.index[i] == 'Y0036':\n",
    "        ax.annotate(pca_pheno[\"strain\"][i], (principalDf.iloc[i,0], principalDf.iloc[i,1]))\n",
    "#add variance explained by each component\n",
    "print(pca.explained_variance_ratio_)\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'2316554'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mget_loc\u001b[0;34m(self, key, method, tolerance)\u001b[0m\n\u001b[1;32m   3628\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3629\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcasted_key\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3630\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/pandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/pandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: '2316554'",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m/var/folders/mb/vdj4y7t13zd3tzlk99b8ycd40000gn/T/ipykernel_64309/3320528611.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0mdf_plot\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'phenotype'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msnps_t\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0miloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m  \u001b[0;31m#which column is called 2528629\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 5\u001b[0;31m \u001b[0mcol_index\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msnps_t\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"2316554\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      6\u001b[0m \u001b[0mdf_plot\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'allele'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msnps_t\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0miloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mcol_index\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      7\u001b[0m \u001b[0mdf_plot\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'allele'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdf_plot\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'allele'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreplace\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'Ref. allele'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mget_loc\u001b[0;34m(self, key, method, tolerance)\u001b[0m\n\u001b[1;32m   3629\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcasted_key\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3630\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3631\u001b[0;31m                 \u001b[0;32mraise\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3632\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mTypeError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3633\u001b[0m                 \u001b[0;31m# If we have a listlike key, _check_indexing_error will raise\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: '2316554'"
     ]
    }
   ],
   "source": [
    "#create a new df with 2 columns: phenotype and colum 2528629 if 0 say Ref. allele, if 1 say Mutant\n",
    "df_plot = pd.DataFrame(columns = ['phenotype','allele'])\n",
    "df_plot['phenotype'] = snps_t.iloc[:,-1]\n",
    " #which column is called 2528629\n",
    "col_index = snps_t.columns.get_loc(\"2316554\")\n",
    "df_plot['allele'] = snps_t.iloc[:,col_index]\n",
    "df_plot['allele'] = df_plot['allele'].replace(0, 'Ref. allele')\n",
    "df_plot['allele'] = df_plot['allele'].replace(1, 'Mutant')\n",
    "#keep only rows with Ref. allele or Mutant\n",
    "df_plot = df_plot[(df_plot['allele'] == 'Ref. allele') | (df_plot['allele'] == 'Mutant')]\n",
    "\n",
    "#PLOT data now with matplotlib\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "#keep only rows with Ref. allele or Mutant\n",
    "df_plot = df_plot[(df_plot['allele'] == 'Ref. allele') | (df_plot['allele'] == 'Mutant')]\n",
    "#create boxplot with outliers as circles and overlay data with scatterplot\n",
    "sns.boxplot(x=\"allele\", y=\"phenotype\", data=df_plot, whis=np.inf)\n",
    "#create a jitter of the data and plot that on a scatter plot by adding jitter the Ref. allele and Mutant columns\n",
    "df_plot['x_coords'] = df_plot['allele'].apply(lambda x: np.random.normal(0, 0.05) if x == 'Ref. allele' else np.random.normal(1, 0.05))\n",
    "sns.scatterplot(x=\"x_coords\", y=\"phenotype\", data=df_plot, color=\"black\", alpha=0.5)\n",
    "#plot datapoints with red if their index is in R1_34, R1_44, R1_45, R1_48, R1_49, R1_50, R1_51, R1_52, R1_53, R1_54\n",
    "for i in range(0, len(df_plot.index)):\n",
    "      if df_plot.index[i] in ['R1_34', 'R1_44', 'R1_45', 'R1_48', 'R1_49', 'R1_50', 'R1_51', 'R1_52', 'R1_53', 'R1_54']:\n",
    "            sns.scatterplot(x=\"x_coords\", y=\"phenotype\", data=df_plot.iloc[i:i+1], color=\"red\", alpha=0.5, s=100)\n",
    "      \n",
    "\n",
    "\n",
    "plt.title(\"Washing phenotype: II:2316554 TA vs. A\")\n",
    "plt.xlabel(\"Allele\")\n",
    "plt.ylabel(\"Washing phenotype\")\n",
    "#also label each point with strain name\n",
    "\n",
    "plt.show(p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'2527314'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mget_loc\u001b[0;34m(self, key, method, tolerance)\u001b[0m\n\u001b[1;32m   3628\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3629\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcasted_key\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3630\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/pandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/pandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: '2527314'",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m/var/folders/mb/vdj4y7t13zd3tzlk99b8ycd40000gn/T/ipykernel_64309/2009972868.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0mdf_plot\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'phenotype'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msnps_t\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0miloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m  \u001b[0;31m#which column is called 2528629\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 5\u001b[0;31m \u001b[0mcol_index\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msnps_t\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"2527314\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      6\u001b[0m \u001b[0mdf_plot\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'allele'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msnps_t\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0miloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mcol_index\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      7\u001b[0m \u001b[0mdf_plot\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'allele'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdf_plot\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'allele'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreplace\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'Ref. allele'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mget_loc\u001b[0;34m(self, key, method, tolerance)\u001b[0m\n\u001b[1;32m   3629\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcasted_key\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3630\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3631\u001b[0;31m                 \u001b[0;32mraise\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3632\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mTypeError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3633\u001b[0m                 \u001b[0;31m# If we have a listlike key, _check_indexing_error will raise\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: '2527314'"
     ]
    }
   ],
   "source": [
    "#create a new df with 2 columns: phenotype and colum 2528629 if 0 say Ref. allele, if 1 say Mutant\n",
    "df_plot = pd.DataFrame(columns = ['phenotype','allele'])\n",
    "df_plot['phenotype'] = snps_t.iloc[:,-1]\n",
    " #which column is called 2528629\n",
    "col_index = snps_t.columns.get_loc(\"2527314\")\n",
    "df_plot['allele'] = snps_t.iloc[:,col_index]\n",
    "df_plot['allele'] = df_plot['allele'].replace(0, 'Ref. allele')\n",
    "df_plot['allele'] = df_plot['allele'].replace(1, 'Mutant')\n",
    "#keep only rows with Ref. allele or Mutant\n",
    "df_plot = df_plot[(df_plot['allele'] == 'Ref. allele') | (df_plot['allele'] == 'Mutant')]\n",
    "\n",
    "#PLOT data now with matplotlib\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "#keep only rows with Ref. allele or Mutant\n",
    "df_plot = df_plot[(df_plot['allele'] == 'Ref. allele') | (df_plot['allele'] == 'Mutant')]\n",
    "#create boxplot with outliers as circles and overlay data with scatterplot\n",
    "sns.boxplot(x=\"allele\", y=\"phenotype\", data=df_plot, whis=np.inf)\n",
    "#create a jitter of the data and plot that on a scatter plot by adding jitter the Ref. allele and Mutant columns\n",
    "df_plot['x_coords'] = df_plot['allele'].apply(lambda x: np.random.normal(0, 0.05) if x == 'Ref. allele' else np.random.normal(1, 0.05))\n",
    "sns.scatterplot(x=\"x_coords\", y=\"phenotype\", data=df_plot, color=\"black\", alpha=0.5)\n",
    "#plot datapoints with red if their index is in R1_34, R1_44, R1_45, R1_48, R1_49, R1_50, R1_51, R1_52, R1_53, R1_54\n",
    "for i in range(0, len(df_plot.index)):\n",
    "      if df_plot.index[i] in ['R1_34', 'R1_44', 'R1_45', 'R1_48', 'R1_49', 'R1_50', 'R1_51', 'R1_52', 'R1_53', 'R1_54']:\n",
    "            sns.scatterplot(x=\"x_coords\", y=\"phenotype\", data=df_plot.iloc[i:i+1], color=\"red\", alpha=0.5, s=100)\n",
    "      \n",
    "\n",
    "\n",
    "plt.title(\"Washing phenotype: II:2527314 T vs. A\")\n",
    "plt.xlabel(\"Allele\")\n",
    "plt.ylabel(\"Washing phenotype\")\n",
    "#also label each point with strain name\n",
    "\n",
    "plt.show(p)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "#pandas read in \"/Users/bencekover/Library/CloudStorage/OneDrive-UniversityCollegeLondon/MSci Bahler lab/S.-Pombe-biofilm/external data/Pombe genome annotation.gff3\"\n",
    "import pandas as pd\n",
    "df = pd.read_csv(root + 'external data/Pombase files/Schizosaccharomyces_pombe_all_chromosomes.gff3', sep='\\t', header=1)\n",
    "\n",
    "#break last column into 2 at ;\n",
    "df[['a','b']] = df[\"ID=SPAC1002.01;Name=mrx11\"].str.split(';',expand=True)\n",
    "#remove column ID=SPAC1002.01;Name=mrx11\n",
    "df.drop(columns =[\"ID=SPAC1002.01;Name=mrx11\"], inplace = True)\n",
    "#remove ID= from column a and rename it to ID\n",
    "df['ID'] = df['a'].str.replace('ID=', '')\n",
    "#remove column a\n",
    "df.drop(columns =['a','PomBase',\".\",\"..1\"], inplace = True)\n",
    "#rename b to Parent\n",
    "df.rename(columns={'b':'Parent','gene':'object','I':'chromosome','1798347':\"start\",'1798835':\"end\",\"+\":\"strand\"}, inplace=True)\n",
    "#In parent remove anything before =\n",
    "df['Parent'] = df['Parent'].str.replace('Parent=', '')\n",
    "#or Name=\n",
    "df['Parent'] = df['Parent'].str.replace('Name=', '')\n",
    "#create an aempty column called detail\n",
    "df['detail'] = \"\"\n",
    "#split ID column at :, and if there is something after put it in detail column\n",
    "df['detail'] = df['ID'].str.split(':', expand=True)[1]\n",
    "#split ID column at :, and if there is something before put it in ID column\n",
    "df['ID'] = df['ID'].str.split(':', expand=True)[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "#find a subset of df, where snps[position] is +- 1000 of df[start] or df[end]. Create a result df, and say which snps were there\n",
    "#suppress warnings\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "\n",
    "result = pd.DataFrame()\n",
    "for i in range(len(snps)):\n",
    "    for j in range(len(df)):\n",
    "        if snps['position'][i] >= df['start'][j] - 500 and snps['position'][i] <= df['end'][j] + 500:\n",
    "            #and if chromsome is chromosome_2 for df chromsome II etc\n",
    "            if snps['chromosome'][i] == df['chromosome'][j]:\n",
    "                result = result.append({'position': snps['position'][i], 'ID': df['ID'][j], 'Parent': df['Parent'][j], 'object': df['object'][j], 'chromosome': df['chromosome'][j], 'start': df['start'][j], 'end': df['end'][j], 'strand': df['strand'][j], 'detail': df['detail'][j]}, ignore_index=True)\n",
    "\n",
    "\n",
    "#if the last two characters are .1, remove them\n",
    "for i in range(len(result)):\n",
    "    if result['ID'][i][-2:] == \".1\":\n",
    "        result['ID'][i] = result['ID'][i][:-2]\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "#find the p_value of the snp position matches snps_t column_name\n",
    "#add a column to result called p_value\n",
    "pvals = np.array([])\n",
    "for i in range(len(result)):\n",
    "    for j in range(len(snps_t.columns)):\n",
    "        #skip if colum nname is phenotype\n",
    "        if snps_t.columns[j] == 'phenotype':\n",
    "            continue\n",
    "        if result['position'][i] == int(snps_t.columns[j]):\n",
    "            pvals = np.append(pvals, snps_t.iloc[46,j])\n",
    "result['p_value'] = pvals\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "#match ID with systematic name in wash_corr and add columns gene\tcorrelation\tp-value\n",
    "#create np.arrays with NAs\n",
    "corrs = np.array([np.nan]*len(result))\n",
    "pvals = np.array([np.nan]*len(result))\n",
    "genes = np.array([np.nan]*len(result), dtype=object)\n",
    "for i in range(len(result)):\n",
    "    for j in range(len(wash_corr)):\n",
    "        if result['ID'][i] == wash_corr['systematic_names'][j]:\n",
    "            corrs[i] = wash_corr['correlation'][j]\n",
    "            pvals[i] = wash_corr['p-value'][j]\n",
    "            genes[i] = wash_corr['gene'][j]\n",
    "\n",
    "            \n",
    "\n",
    "\n",
    "result['correlation'] = corrs\n",
    "result['p_value_corr'] = pvals\n",
    "result['gene'] = genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "import sys, re\n",
    "\n",
    "f=open(root + 'external data/Pombase files/Schizosaccharomyces_pombe_all_chromosomes.fa','r')\n",
    "lines=f.readlines()\n",
    "\n",
    "hre=re.compile('>(\\S+)')\n",
    "lre=re.compile('^(\\S+)$')\n",
    "\n",
    "regions={}\n",
    "\n",
    "for line in lines:\n",
    "        outh = hre.search(line)\n",
    "        if outh:\n",
    "                id=outh.group(1)\n",
    "        else:\n",
    "                outl=lre.search(line)\n",
    "                if(id in regions.keys()):\n",
    "                        regions[id] += outl.group(1)\n",
    "                else:\n",
    "                        regions[id]  =outl.group(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>object</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>Parent</th>\n",
       "      <th>ID</th>\n",
       "      <th>detail</th>\n",
       "      <th>exon_number</th>\n",
       "      <th>old_start</th>\n",
       "      <th>old_end</th>\n",
       "      <th>exon_length</th>\n",
       "      <th>CDS_length_so_far</th>\n",
       "      <th>total_CDS_length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>955</td>\n",
       "      <td>I</td>\n",
       "      <td>CDS</td>\n",
       "      <td>4570495</td>\n",
       "      <td>4575441</td>\n",
       "      <td>-</td>\n",
       "      <td>SPAC12B10.01c.1</td>\n",
       "      <td>SPAC12B10.01c.1</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4570498</td>\n",
       "      <td>4575441</td>\n",
       "      <td>4947</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4947.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>965</td>\n",
       "      <td>I</td>\n",
       "      <td>CDS</td>\n",
       "      <td>4578627</td>\n",
       "      <td>4580258</td>\n",
       "      <td>+</td>\n",
       "      <td>SPAC12B10.03.1</td>\n",
       "      <td>SPAC12B10.03.1</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4578627</td>\n",
       "      <td>4580258</td>\n",
       "      <td>1632</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1632.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>979</td>\n",
       "      <td>I</td>\n",
       "      <td>CDS</td>\n",
       "      <td>4582378</td>\n",
       "      <td>4582549</td>\n",
       "      <td>+</td>\n",
       "      <td>SPAC12B10.05.1</td>\n",
       "      <td>SPAC12B10.05.1</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4582378</td>\n",
       "      <td>4582549</td>\n",
       "      <td>172</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1458.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>981</td>\n",
       "      <td>I</td>\n",
       "      <td>CDS</td>\n",
       "      <td>4582595</td>\n",
       "      <td>4582697</td>\n",
       "      <td>+</td>\n",
       "      <td>SPAC12B10.05.1</td>\n",
       "      <td>SPAC12B10.05.1</td>\n",
       "      <td>exon</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4582595</td>\n",
       "      <td>4582697</td>\n",
       "      <td>103</td>\n",
       "      <td>172.0</td>\n",
       "      <td>1458.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>983</td>\n",
       "      <td>I</td>\n",
       "      <td>CDS</td>\n",
       "      <td>4582747</td>\n",
       "      <td>4582767</td>\n",
       "      <td>+</td>\n",
       "      <td>SPAC12B10.05.1</td>\n",
       "      <td>SPAC12B10.05.1</td>\n",
       "      <td>exon</td>\n",
       "      <td>3.0</td>\n",
       "      <td>4582747</td>\n",
       "      <td>4582767</td>\n",
       "      <td>21</td>\n",
       "      <td>275.0</td>\n",
       "      <td>1458.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>352</th>\n",
       "      <td>15957</td>\n",
       "      <td>I</td>\n",
       "      <td>CDS</td>\n",
       "      <td>4532852</td>\n",
       "      <td>4532924</td>\n",
       "      <td>+</td>\n",
       "      <td>SPATRNAASP.01.1</td>\n",
       "      <td>SPATRNAASP.01.1</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4532857</td>\n",
       "      <td>4532927</td>\n",
       "      <td>73</td>\n",
       "      <td>0.0</td>\n",
       "      <td>73.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>353</th>\n",
       "      <td>16038</td>\n",
       "      <td>I</td>\n",
       "      <td>CDS</td>\n",
       "      <td>3086305</td>\n",
       "      <td>3086348</td>\n",
       "      <td>+</td>\n",
       "      <td>SPATRNALYS.05.1</td>\n",
       "      <td>SPATRNALYS.05.1</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3086310</td>\n",
       "      <td>3086348</td>\n",
       "      <td>44</td>\n",
       "      <td>0.0</td>\n",
       "      <td>84.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>354</th>\n",
       "      <td>16040</td>\n",
       "      <td>I</td>\n",
       "      <td>CDS</td>\n",
       "      <td>3086357</td>\n",
       "      <td>3086396</td>\n",
       "      <td>+</td>\n",
       "      <td>SPATRNALYS.05.1</td>\n",
       "      <td>SPATRNALYS.05.1</td>\n",
       "      <td>exon</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3086357</td>\n",
       "      <td>3086392</td>\n",
       "      <td>40</td>\n",
       "      <td>44.0</td>\n",
       "      <td>84.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>355</th>\n",
       "      <td>16096</td>\n",
       "      <td>I</td>\n",
       "      <td>CDS</td>\n",
       "      <td>3164744</td>\n",
       "      <td>3164817</td>\n",
       "      <td>+</td>\n",
       "      <td>SPATRNATHR.04.1</td>\n",
       "      <td>SPATRNATHR.04.1</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3164749</td>\n",
       "      <td>3164820</td>\n",
       "      <td>74</td>\n",
       "      <td>0.0</td>\n",
       "      <td>74.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>510</th>\n",
       "      <td>57736</td>\n",
       "      <td>I</td>\n",
       "      <td>CDS</td>\n",
       "      <td>2976785</td>\n",
       "      <td>2976983</td>\n",
       "      <td>+</td>\n",
       "      <td>SPRRNA.13.1</td>\n",
       "      <td>SPRRNA.13.1</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2976790</td>\n",
       "      <td>2976979</td>\n",
       "      <td>199</td>\n",
       "      <td>0.0</td>\n",
       "      <td>199.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>357 rows Ã— 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     index chromosome object    start      end strand           Parent  \\\n",
       "0      955          I    CDS  4570495  4575441      -  SPAC12B10.01c.1   \n",
       "1      965          I    CDS  4578627  4580258      +   SPAC12B10.03.1   \n",
       "2      979          I    CDS  4582378  4582549      +   SPAC12B10.05.1   \n",
       "3      981          I    CDS  4582595  4582697      +   SPAC12B10.05.1   \n",
       "4      983          I    CDS  4582747  4582767      +   SPAC12B10.05.1   \n",
       "..     ...        ...    ...      ...      ...    ...              ...   \n",
       "352  15957          I    CDS  4532852  4532924      +  SPATRNAASP.01.1   \n",
       "353  16038          I    CDS  3086305  3086348      +  SPATRNALYS.05.1   \n",
       "354  16040          I    CDS  3086357  3086396      +  SPATRNALYS.05.1   \n",
       "355  16096          I    CDS  3164744  3164817      +  SPATRNATHR.04.1   \n",
       "510  57736          I    CDS  2976785  2976983      +      SPRRNA.13.1   \n",
       "\n",
       "                  ID detail  exon_number  old_start  old_end  exon_length  \\\n",
       "0    SPAC12B10.01c.1   exon          1.0    4570498  4575441         4947   \n",
       "1     SPAC12B10.03.1   exon          1.0    4578627  4580258         1632   \n",
       "2     SPAC12B10.05.1   exon          1.0    4582378  4582549          172   \n",
       "3     SPAC12B10.05.1   exon          2.0    4582595  4582697          103   \n",
       "4     SPAC12B10.05.1   exon          3.0    4582747  4582767           21   \n",
       "..               ...    ...          ...        ...      ...          ...   \n",
       "352  SPATRNAASP.01.1   exon          1.0    4532857  4532927           73   \n",
       "353  SPATRNALYS.05.1   exon          1.0    3086310  3086348           44   \n",
       "354  SPATRNALYS.05.1   exon          2.0    3086357  3086392           40   \n",
       "355  SPATRNATHR.04.1   exon          1.0    3164749  3164820           74   \n",
       "510      SPRRNA.13.1   exon          1.0    2976790  2976979          199   \n",
       "\n",
       "     CDS_length_so_far  total_CDS_length  \n",
       "0                  0.0            4947.0  \n",
       "1                  0.0            1632.0  \n",
       "2                  0.0            1458.0  \n",
       "3                172.0            1458.0  \n",
       "4                275.0            1458.0  \n",
       "..                 ...               ...  \n",
       "352                0.0              73.0  \n",
       "353                0.0              84.0  \n",
       "354               44.0              84.0  \n",
       "355                0.0              74.0  \n",
       "510                0.0             199.0  \n",
       "\n",
       "[357 rows x 15 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#result_final___ only keep rows where snp is between start and end\n",
    "result_final__ = result.copy()\n",
    "result_final___ = result_final__.copy()\n",
    "result_final___ = result_final___[(result_final___['position'] >= result_final___['start']-10) & (result_final___['position'] <= result_final___['end']+100)]\n",
    "\n",
    "gene_exons = df.copy()\n",
    "#only keep rows where object is CDS\n",
    "gene_exons = gene_exons[gene_exons['object'] == 'CDS']\n",
    "#only keep rows where ID is in result_final__['ID']\n",
    "gene_exons_final = pd.DataFrame()\n",
    "for i in np.unique(result_final___['ID']):\n",
    "      df_subset = gene_exons[gene_exons['ID'] == str(i + \".1\")]\n",
    "      gene_exons_final = gene_exons_final.append(df_subset)\n",
    "#reset index\n",
    "gene_exons_final.reset_index(inplace=True)\n",
    "#make a column exon_number\n",
    "gene_exons_final['exon_number'] = np.nan\n",
    "#for those where ID is multiple times, number the exons by increasing start if strand is +, or decreasing start if strand is -\n",
    "for i in np.unique(gene_exons_final['ID']):\n",
    "    df_subset = gene_exons_final[gene_exons_final['ID'] == i]\n",
    "    if df_subset['strand'].values[0]== '+':\n",
    "        df_subset = df_subset.sort_values(by=['start'], ascending=True)\n",
    "    if df_subset['strand'].values[0] == '-':\n",
    "        df_subset = df_subset.sort_values(by=['end'], ascending=False)\n",
    "    df_subset['exon_number'] = np.arange(1, len(df_subset) + 1)\n",
    "    gene_exons_final[gene_exons_final['ID'] == i] = df_subset\n",
    "\n",
    "\n",
    "#REMOVE entries wehre ID contains SPNCRNA\n",
    "gene_exons_final = gene_exons_final[~gene_exons_final['ID'].str.contains('SPNCRNA')]\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "#make a column called real_start\n",
    "gene_exons_final['old_start'] = gene_exons_final['start']\n",
    "gene_exons_final['old_end'] = gene_exons_final['end']\n",
    "#for rows where strand is + look at the load the bases +- 5 around the start from regions\n",
    "for i in np.unique(gene_exons_final['ID']):\n",
    "    df_subset = gene_exons_final[gene_exons_final['ID'] == i]\n",
    "    df_subset = df_subset[df_subset['exon_number'] == 1]\n",
    "    if df_subset['strand'].values[0]== '+':\n",
    "        seq = regions[df_subset['chromosome'].values[0]][int(df_subset['start'].values[0]-5):int(df_subset['start'].values[0]+5)]\n",
    "        #where does ATG occur in seq\n",
    "        real_start = seq.find('ATG')-4\n",
    "        real_start = df_subset['start'] + real_start\n",
    "        #for the row in gene_exons_final where ID is i and exon_number is 1, change the start to real_start\n",
    "        gene_exons_final.loc[(gene_exons_final['ID'] == i) & (gene_exons_final['exon_number'] == 1), 'start'] = real_start\n",
    "\n",
    "\n",
    "\n",
    "    else:\n",
    "        seq_rev = regions[df_subset['chromosome'].values[0]][int(df_subset['end'].values[0]-5):int(df_subset['end'].values[0]+5)]\n",
    "        real_end = seq_rev.find('CAT')+2-4\n",
    "        real_end = df_subset['end'] + real_end\n",
    "        #for the row in gene_exons_final where ID is i and exon_number is 1, change the end to real_end\n",
    "        gene_exons_final.loc[(gene_exons_final['ID'] == i) & (gene_exons_final['exon_number'] == 1), 'end'] = real_end\n",
    "\n",
    "\n",
    "    df_subset = gene_exons_final[gene_exons_final['ID'] == i]\n",
    "    highest_exon = df_subset['exon_number'].max()\n",
    "    df_subset = df_subset[df_subset['exon_number'] == highest_exon]\n",
    "    if df_subset['strand'].values[0]== '+':\n",
    "\n",
    "        seq2 = regions[df_subset['chromosome'].values[0]][int(df_subset['end'].values[0]-5):int(df_subset['end'].values[0]+5)]\n",
    "        real_end = seq2.find('TAA')+2-4\n",
    "        real_end = df_subset['end'] + real_end\n",
    "        #for the row in gene_exons_final where ID is i and exon_number is highest_exon, change the end to real_end\n",
    "        gene_exons_final.loc[(gene_exons_final['ID'] == i) & (gene_exons_final['exon_number'] == highest_exon), 'end'] = real_end\n",
    "    else:\n",
    "        seq2_rev = regions[df_subset['chromosome'].values[0]][int(df_subset['start'].values[0]-5):int(df_subset['start'].values[0]+5)]\n",
    "        real_start = seq2_rev.find('TCA')+2-4\n",
    "        real_start = df_subset['start'] + real_start\n",
    "        #for the row in gene_exons_final where ID is i and exon_number is highest_exon, change the start to real_start\n",
    "        gene_exons_final.loc[(gene_exons_final['ID'] == i) & (gene_exons_final['exon_number'] == highest_exon), 'start'] = real_start\n",
    "\n",
    "\n",
    "    \n",
    "    \n",
    "\n",
    "\n",
    "\n",
    "#for each row calculate the length of the exon\n",
    "gene_exons_final['exon_length'] = abs(gene_exons_final['end'] - gene_exons_final['start'])+1\n",
    "\n",
    "#remove rows where ID is NaN\n",
    "gene_exons_final = gene_exons_final[~gene_exons_final['ID'].isna()]\n",
    "#for each row calculate the length of the CDS until that exon starts\n",
    "gene_exons_final['CDS_length_so_far'] = np.nan\n",
    "for i in np.unique(gene_exons_final['ID']):\n",
    "    df_subset = gene_exons_final[gene_exons_final['ID'] == i]\n",
    "    #order by exon_number\n",
    "    df_subset = df_subset.sort_values(by=['exon_number'])\n",
    "    \n",
    "    df_subset['CDS_length_so_far'] = np.append(np.array(0),df_subset['exon_length'].cumsum())[:-1]\n",
    "    gene_exons_final[gene_exons_final['ID'] == i] = df_subset\n",
    "#add total cds length\n",
    "gene_exons_final['total_CDS_length'] = np.nan\n",
    "for i in np.unique(gene_exons_final['ID']):\n",
    "    df_subset = gene_exons_final[gene_exons_final['ID'] == i]\n",
    "    df_subset['total_CDS_length'] = sum(df_subset['exon_length'])\n",
    "    gene_exons_final[gene_exons_final['ID'] == i] = df_subset\n",
    "\n",
    "gene_exons_final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "#add to each row the unique number of positions that ID had\n",
    "#add a column called unique\n",
    "#remove duplicates of ID and position\n",
    "#keep only entries with \"object\" is CDS\n",
    "result = result[result['object'] == 'CDS']\n",
    "result_final = result.drop_duplicates(subset=['ID', 'position'])\n",
    "#reset index\n",
    "result_final = result_final.reset_index(drop=True)\n",
    "unique = np.array([])\n",
    "for i in range(len(result_final)):\n",
    "    #get the ID\n",
    "    ID = result_final['ID'][i]\n",
    "    #get the number of times ID appears in result\n",
    "    count = len(result_final[result_final['ID'] == ID])\n",
    "    #add count to unique\n",
    "    unique = np.append(unique, count)\n",
    "result_final['unique'] = unique\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "#show SPBC290.07 in result\n",
    "#gather positions for each ID, and put in a single entry as a list. Add that to the df. THen remove duplicated IDs\n",
    "result_final_ = result_final.groupby('ID')['position'].apply(list).reset_index(name='positions')\n",
    "result_final__ = pd.merge(result_final, result_final_, on='ID')\n",
    "\n",
    "#remove duplicates of ID\n",
    "result_final__ = result_final__.drop_duplicates(subset=['ID'])\n",
    "\n",
    "#reset index\n",
    "result_final__ = result_final__.reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "#iterate through result_final__. If ID is found in orthologs, then add the albicans_name(s)cerevisiae_name(s) to the df\n",
    "from numpy import dtype\n",
    "\n",
    "albicans_names = np.array([np.nan]*len(result_final__),dtype=object)\n",
    "cerevisiae_names = np.array([np.nan]*len(result_final__),dtype=object)\n",
    "for i in range(len(result_final__)):\n",
    "    for j in range(len(orthologs)):\n",
    "        if result_final__['ID'][i] in orthologs['pombe_name(s)'][j]:\n",
    "            albicans_names[i] = orthologs['albicans_name(s)'][j]\n",
    "            cerevisiae_names[i] = orthologs['cerevisiae_name(s)'][j]\n",
    "#add to df\n",
    "result_final__['albicans_name(s)'] = albicans_names\n",
    "result_final__['cerevisiae_name(s)'] = cerevisiae_names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>position</th>\n",
       "      <th>ID</th>\n",
       "      <th>Parent</th>\n",
       "      <th>object</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>detail</th>\n",
       "      <th>p_value</th>\n",
       "      <th>correlation</th>\n",
       "      <th>p_value_corr</th>\n",
       "      <th>gene</th>\n",
       "      <th>unique</th>\n",
       "      <th>positions</th>\n",
       "      <th>albicans_name(s)</th>\n",
       "      <th>cerevisiae_name(s)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>424</th>\n",
       "      <td>4473108</td>\n",
       "      <td>SPAC17C9.16c</td>\n",
       "      <td>SPAC17C9.16c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>4472611</td>\n",
       "      <td>4473271</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.678393</td>\n",
       "      <td>3.042498e-07</td>\n",
       "      <td>mfs1</td>\n",
       "      <td>6.0</td>\n",
       "      <td>[4473108, 4473996, 4474485, 4474680, 4474681, ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>195</th>\n",
       "      <td>3006775</td>\n",
       "      <td>SPAC31G5.11</td>\n",
       "      <td>SPAC31G5.11.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3006138</td>\n",
       "      <td>3006845</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.664822</td>\n",
       "      <td>6.329116e-07</td>\n",
       "      <td>pac2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[3006775]</td>\n",
       "      <td>['C1_10150W_A' 'nan']</td>\n",
       "      <td>['YEL007W' 'YHR177W']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>194</th>\n",
       "      <td>3002877</td>\n",
       "      <td>SPNCRNA.872</td>\n",
       "      <td>SPNCRNA.872.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3001750</td>\n",
       "      <td>3004335</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.618445</td>\n",
       "      <td>5.970721e-06</td>\n",
       "      <td>eta2-antisense-1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[3002877]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425</th>\n",
       "      <td>4473108</td>\n",
       "      <td>SPNCRNA.991</td>\n",
       "      <td>SPNCRNA.991.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>4472218</td>\n",
       "      <td>4473227</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-0.561792</td>\n",
       "      <td>5.931923e-05</td>\n",
       "      <td>mfs1-antisense-1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[4473108]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>276</th>\n",
       "      <td>3147678</td>\n",
       "      <td>SPAC3G9.14</td>\n",
       "      <td>SPAC3G9.14.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3147207</td>\n",
       "      <td>3149507</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.464559</td>\n",
       "      <td>1.305065e-03</td>\n",
       "      <td>sak1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>[3147678, 3149857]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>454</th>\n",
       "      <td>4531317</td>\n",
       "      <td>SPAC27D7.11c</td>\n",
       "      <td>SPAC27D7.11c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>4529462</td>\n",
       "      <td>4530853</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-0.452833</td>\n",
       "      <td>1.787387e-03</td>\n",
       "      <td>SPAC27D7.11c</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[4531317]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>253</th>\n",
       "      <td>3103928</td>\n",
       "      <td>SPNCRNA.560</td>\n",
       "      <td>SPNCRNA.560.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3104356</td>\n",
       "      <td>3105524</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.451910</td>\n",
       "      <td>1.831298e-03</td>\n",
       "      <td>atg1801-antisense-1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[3103928]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233</th>\n",
       "      <td>3074774</td>\n",
       "      <td>SPAC24C9.15c</td>\n",
       "      <td>SPAC24C9.15c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3074772</td>\n",
       "      <td>3075353</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-0.438076</td>\n",
       "      <td>2.614700e-03</td>\n",
       "      <td>spn5</td>\n",
       "      <td>2.0</td>\n",
       "      <td>[3074774, 3075383]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>2759901</td>\n",
       "      <td>SPAC6F6.19</td>\n",
       "      <td>SPAC6F6.19.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>2759479</td>\n",
       "      <td>2759709</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.415810</td>\n",
       "      <td>4.501092e-03</td>\n",
       "      <td>SPAC6F6.19</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[2759901]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>2797593</td>\n",
       "      <td>SPAC1805.14</td>\n",
       "      <td>SPAC1805.14.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>2796469</td>\n",
       "      <td>2797827</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.412780</td>\n",
       "      <td>4.833124e-03</td>\n",
       "      <td>SPAC1805.14</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[2797593]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>281</th>\n",
       "      <td>3151749</td>\n",
       "      <td>SPAC3G9.13c</td>\n",
       "      <td>SPAC3G9.13c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3151326</td>\n",
       "      <td>3151625</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.406324</td>\n",
       "      <td>5.612496e-03</td>\n",
       "      <td>msw1</td>\n",
       "      <td>4.0</td>\n",
       "      <td>[3151749, 3152754, 3152848, 3152996]</td>\n",
       "      <td>['C5_02770W_A']</td>\n",
       "      <td>['YDR268W']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>344</th>\n",
       "      <td>3274317</td>\n",
       "      <td>SPNCRNA.899</td>\n",
       "      <td>SPNCRNA.899.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3273276</td>\n",
       "      <td>3274078</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.399789</td>\n",
       "      <td>6.510566e-03</td>\n",
       "      <td>SPNCRNA.899</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[3274317]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>2622695</td>\n",
       "      <td>SPAC7D4.10</td>\n",
       "      <td>SPAC7D4.10.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>2622169</td>\n",
       "      <td>2622195</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-0.399736</td>\n",
       "      <td>6.518284e-03</td>\n",
       "      <td>vma13</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[2622695]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>2726497</td>\n",
       "      <td>SPNCRNA.204</td>\n",
       "      <td>SPNCRNA.204.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>2726058</td>\n",
       "      <td>2726682</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-0.397760</td>\n",
       "      <td>6.813634e-03</td>\n",
       "      <td>SPNCRNA.204</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[2726497]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>254</th>\n",
       "      <td>3109392</td>\n",
       "      <td>SPAC589.11</td>\n",
       "      <td>SPAC589.11.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3109174</td>\n",
       "      <td>3109312</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.395784</td>\n",
       "      <td>7.120510e-03</td>\n",
       "      <td>pth4</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[3109392]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>2797593</td>\n",
       "      <td>SPAC1805.15c</td>\n",
       "      <td>SPAC1805.15c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>2798023</td>\n",
       "      <td>2798122</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-0.383663</td>\n",
       "      <td>9.277232e-03</td>\n",
       "      <td>pub2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[2797593]</td>\n",
       "      <td>['C2_08500W_A']</td>\n",
       "      <td>['YER125W']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273</th>\n",
       "      <td>3139763</td>\n",
       "      <td>SPNCRNA.889</td>\n",
       "      <td>SPNCRNA.889.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3138602</td>\n",
       "      <td>3139458</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.375231</td>\n",
       "      <td>1.109004e-02</td>\n",
       "      <td>SPAC688.12c-antisense-1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[3139763]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>294</th>\n",
       "      <td>3175090</td>\n",
       "      <td>SPAC3G9.05</td>\n",
       "      <td>SPAC3G9.05.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3173946</td>\n",
       "      <td>3175729</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-0.372200</td>\n",
       "      <td>1.181179e-02</td>\n",
       "      <td>spa2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>[3175090, 3175428]</td>\n",
       "      <td>['C1_12620W_A' 'nan']</td>\n",
       "      <td>['YLL021W' 'YLR313C']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>325</th>\n",
       "      <td>3230578</td>\n",
       "      <td>SPAC6G10.07</td>\n",
       "      <td>SPAC6G10.07.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3228833</td>\n",
       "      <td>3230924</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-0.363109</td>\n",
       "      <td>1.422279e-02</td>\n",
       "      <td>cbc1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[3230578]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202</th>\n",
       "      <td>3021278</td>\n",
       "      <td>SPAC31G5.18c</td>\n",
       "      <td>SPAC31G5.18c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3020942</td>\n",
       "      <td>3021733</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.357839</td>\n",
       "      <td>1.580339e-02</td>\n",
       "      <td>sde2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[3021278]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>367</th>\n",
       "      <td>3327476</td>\n",
       "      <td>SPAC2E1P5.04c</td>\n",
       "      <td>SPAC2E1P5.04c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3327940</td>\n",
       "      <td>3329007</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-0.353491</td>\n",
       "      <td>1.721726e-02</td>\n",
       "      <td>cwg2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>[3327476, 3329415]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160</th>\n",
       "      <td>2937534</td>\n",
       "      <td>SPNCRNA.861</td>\n",
       "      <td>SPNCRNA.861.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>2937952</td>\n",
       "      <td>2939052</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.352042</td>\n",
       "      <td>1.771171e-02</td>\n",
       "      <td>SPNCRNA.861</td>\n",
       "      <td>2.0</td>\n",
       "      <td>[2937534, 2937800]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>256</th>\n",
       "      <td>3115792</td>\n",
       "      <td>SPAC688.03c</td>\n",
       "      <td>SPAC688.03c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3115103</td>\n",
       "      <td>3115326</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.351910</td>\n",
       "      <td>1.775725e-02</td>\n",
       "      <td>SPAC688.03c</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[3115792]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>401</th>\n",
       "      <td>3412265</td>\n",
       "      <td>SPAC3H5.13</td>\n",
       "      <td>SPAC3H5.13.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3412527</td>\n",
       "      <td>3412569</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.351647</td>\n",
       "      <td>1.784863e-02</td>\n",
       "      <td>new4</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[3412265]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>2658236</td>\n",
       "      <td>SPAC4F8.12c</td>\n",
       "      <td>SPAC4F8.12c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>2650738</td>\n",
       "      <td>2657829</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-0.347299</td>\n",
       "      <td>1.941465e-02</td>\n",
       "      <td>spp42</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[2658236]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>446</th>\n",
       "      <td>4510905</td>\n",
       "      <td>SPAC27D7.03c</td>\n",
       "      <td>SPAC27D7.03c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>4511040</td>\n",
       "      <td>4513292</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.342951</td>\n",
       "      <td>2.109488e-02</td>\n",
       "      <td>mei2</td>\n",
       "      <td>4.0</td>\n",
       "      <td>[4510905, 4513497, 4513710, 4513733]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126</th>\n",
       "      <td>2881073</td>\n",
       "      <td>SPACUNK4.15</td>\n",
       "      <td>SPACUNK4.15.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>2881105</td>\n",
       "      <td>2881731</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-0.342556</td>\n",
       "      <td>2.125351e-02</td>\n",
       "      <td>SPACUNK4.15</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[2881073]</td>\n",
       "      <td>['CR_06690C_A']</td>\n",
       "      <td>['YGR247W']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113</th>\n",
       "      <td>2804242</td>\n",
       "      <td>SPAC1805.17</td>\n",
       "      <td>SPAC1805.17.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>2804508</td>\n",
       "      <td>2807744</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-0.334124</td>\n",
       "      <td>2.488393e-02</td>\n",
       "      <td>crm1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>[2804242, 2807441]</td>\n",
       "      <td>['CR_00520C_A']</td>\n",
       "      <td>['YGR218W']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>312</th>\n",
       "      <td>3202214</td>\n",
       "      <td>SPNCRNA.894</td>\n",
       "      <td>SPNCRNA.894.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3202438</td>\n",
       "      <td>3203222</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.326219</td>\n",
       "      <td>2.874420e-02</td>\n",
       "      <td>fmc1-antisense-1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[3202214]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>352</th>\n",
       "      <td>3301454</td>\n",
       "      <td>SPNCRNA.900</td>\n",
       "      <td>SPNCRNA.900.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3300812</td>\n",
       "      <td>3302082</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.314756</td>\n",
       "      <td>3.521419e-02</td>\n",
       "      <td>SPNCRNA.900</td>\n",
       "      <td>2.0</td>\n",
       "      <td>[3301454, 3302072]</td>\n",
       "      <td>[nan]</td>\n",
       "      <td>[nan]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>2911919</td>\n",
       "      <td>SPAC513.02</td>\n",
       "      <td>SPAC513.02.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>2911719</td>\n",
       "      <td>2912369</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.314493</td>\n",
       "      <td>3.537597e-02</td>\n",
       "      <td>SPAC513.02</td>\n",
       "      <td>2.0</td>\n",
       "      <td>[2911919, 2911931]</td>\n",
       "      <td>['C3_06920W_A']</td>\n",
       "      <td>['YKL128C']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>183</th>\n",
       "      <td>2975371</td>\n",
       "      <td>SPNCRNA.31</td>\n",
       "      <td>SPNCRNA.31.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>2975608</td>\n",
       "      <td>2976007</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.311331</td>\n",
       "      <td>3.736541e-02</td>\n",
       "      <td>SPNCRNA.31</td>\n",
       "      <td>9.0</td>\n",
       "      <td>[2975371, 2975378, 2975424, 2975613, 2975812, ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>211</th>\n",
       "      <td>3322640</td>\n",
       "      <td>SPAC2E1P5.01c</td>\n",
       "      <td>SPAC2E1P5.01c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3322913</td>\n",
       "      <td>3324478</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-0.310145</td>\n",
       "      <td>3.813473e-02</td>\n",
       "      <td>mns1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[3322640]</td>\n",
       "      <td>['C1_03730C_A']</td>\n",
       "      <td>['YJR131W']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199</th>\n",
       "      <td>3013837</td>\n",
       "      <td>SPAC31G5.14</td>\n",
       "      <td>SPAC31G5.14.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3012593</td>\n",
       "      <td>3013756</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.307773</td>\n",
       "      <td>3.971237e-02</td>\n",
       "      <td>gcv1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[3013837]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>331</th>\n",
       "      <td>4515380</td>\n",
       "      <td>SPNCRNA.993</td>\n",
       "      <td>SPNCRNA.993.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>4515182</td>\n",
       "      <td>4515993</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.306983</td>\n",
       "      <td>4.024996e-02</td>\n",
       "      <td>SPNCRNA.993</td>\n",
       "      <td>2.0</td>\n",
       "      <td>[4515380, 4516287]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>282</th>\n",
       "      <td>3152754</td>\n",
       "      <td>SPAC3G9.12</td>\n",
       "      <td>SPAC3G9.12.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3152779</td>\n",
       "      <td>3157050</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-0.306324</td>\n",
       "      <td>4.070250e-02</td>\n",
       "      <td>peg1</td>\n",
       "      <td>6.0</td>\n",
       "      <td>[3152754, 3152848, 3152996, 3154972, 3155730, ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>2617567</td>\n",
       "      <td>SPAC7D4.11c</td>\n",
       "      <td>SPAC7D4.11c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>2617975</td>\n",
       "      <td>2620284</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-0.305929</td>\n",
       "      <td>4.097601e-02</td>\n",
       "      <td>sec39</td>\n",
       "      <td>2.0</td>\n",
       "      <td>[2617567, 2617646]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>168</th>\n",
       "      <td>2958937</td>\n",
       "      <td>SPAPB24D3.07c</td>\n",
       "      <td>SPAPB24D3.07c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>2958024</td>\n",
       "      <td>2958926</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.303426</td>\n",
       "      <td>4.274331e-02</td>\n",
       "      <td>SPAPB24D3.07c</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[2958937]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148</th>\n",
       "      <td>2924501</td>\n",
       "      <td>SPNCRNA.859</td>\n",
       "      <td>SPNCRNA.859.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>2924745</td>\n",
       "      <td>2926536</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.302108</td>\n",
       "      <td>4.369815e-02</td>\n",
       "      <td>amt3-antisense-1</td>\n",
       "      <td>9.0</td>\n",
       "      <td>[2924501, 2924596, 2924608, 2924675, 2924705, ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>2635318</td>\n",
       "      <td>SPAC7D4.03c</td>\n",
       "      <td>SPAC7D4.03c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>2634211</td>\n",
       "      <td>2636871</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-0.301054</td>\n",
       "      <td>4.447447e-02</td>\n",
       "      <td>ahk1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>[2635318, 2635612]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>3158351</td>\n",
       "      <td>SPNCRNA.217</td>\n",
       "      <td>SPNCRNA.217.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3158482</td>\n",
       "      <td>3159120</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-0.299249</td>\n",
       "      <td>4.583003e-02</td>\n",
       "      <td>SPNCRNA.217</td>\n",
       "      <td>10.0</td>\n",
       "      <td>[3158351, 3158357, 3158359, 3158363, 3158367, ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>457</th>\n",
       "      <td>4531317</td>\n",
       "      <td>SPNCRNA.996</td>\n",
       "      <td>SPNCRNA.996.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>4531144</td>\n",
       "      <td>4532754</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-0.292490</td>\n",
       "      <td>5.120673e-02</td>\n",
       "      <td>SPNCRNA.996</td>\n",
       "      <td>2.0</td>\n",
       "      <td>[4531317, 4532855]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>467</th>\n",
       "      <td>4544534</td>\n",
       "      <td>SPAC637.04</td>\n",
       "      <td>SPAC637.04.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>4544074</td>\n",
       "      <td>4544082</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-0.291963</td>\n",
       "      <td>5.164652e-02</td>\n",
       "      <td>ypp1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[4544534]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>2733748</td>\n",
       "      <td>SPAC6F6.02c</td>\n",
       "      <td>SPAC6F6.02c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>2732524</td>\n",
       "      <td>2733570</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.286034</td>\n",
       "      <td>5.680726e-02</td>\n",
       "      <td>pof5</td>\n",
       "      <td>2.0</td>\n",
       "      <td>[2733748, 2733863]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>2698054</td>\n",
       "      <td>SPAC644.13c</td>\n",
       "      <td>SPAC644.13c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>2697418</td>\n",
       "      <td>2698095</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-0.285771</td>\n",
       "      <td>5.704589e-02</td>\n",
       "      <td>SPAC644.13c</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[2698054]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>2722524</td>\n",
       "      <td>SPAPB2B4.06</td>\n",
       "      <td>SPAPB2B4.06.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>2722735</td>\n",
       "      <td>2723210</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-0.285112</td>\n",
       "      <td>5.764600e-02</td>\n",
       "      <td>SPAPB2B4.06</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[2722524]</td>\n",
       "      <td>['C2_05170W_A']</td>\n",
       "      <td>['YBL095W']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>2722524</td>\n",
       "      <td>SPAPB2B4.05</td>\n",
       "      <td>SPAPB2B4.05.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>2722060</td>\n",
       "      <td>2722291</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-0.284585</td>\n",
       "      <td>5.812972e-02</td>\n",
       "      <td>vma5</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[2722524]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>485</th>\n",
       "      <td>4579546</td>\n",
       "      <td>SPAC12B10.03</td>\n",
       "      <td>SPAC12B10.03.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>4578627</td>\n",
       "      <td>4580258</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-0.277470</td>\n",
       "      <td>6.498375e-02</td>\n",
       "      <td>bun62</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[4579546]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>456</th>\n",
       "      <td>4531317</td>\n",
       "      <td>SPNCRNA.995</td>\n",
       "      <td>SPNCRNA.995.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>4531036</td>\n",
       "      <td>4532486</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-0.275889</td>\n",
       "      <td>6.659088e-02</td>\n",
       "      <td>SPNCRNA.995</td>\n",
       "      <td>2.0</td>\n",
       "      <td>[4531317, 4532855]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133</th>\n",
       "      <td>2905076</td>\n",
       "      <td>SPAPYUK71.03c</td>\n",
       "      <td>SPAPYUK71.03c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>2903369</td>\n",
       "      <td>2907046</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-0.274835</td>\n",
       "      <td>6.767973e-02</td>\n",
       "      <td>tcb3</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[2905076]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     position             ID           Parent object chromosome    start  \\\n",
       "424   4473108   SPAC17C9.16c   SPAC17C9.16c.1    CDS          I  4472611   \n",
       "195   3006775    SPAC31G5.11    SPAC31G5.11.1    CDS          I  3006138   \n",
       "194   3002877    SPNCRNA.872    SPNCRNA.872.1    CDS          I  3001750   \n",
       "425   4473108    SPNCRNA.991    SPNCRNA.991.1    CDS          I  4472218   \n",
       "276   3147678     SPAC3G9.14     SPAC3G9.14.1    CDS          I  3147207   \n",
       "454   4531317   SPAC27D7.11c   SPAC27D7.11c.1    CDS          I  4529462   \n",
       "253   3103928    SPNCRNA.560    SPNCRNA.560.1    CDS          I  3104356   \n",
       "233   3074774   SPAC24C9.15c   SPAC24C9.15c.1    CDS          I  3074772   \n",
       "91    2759901     SPAC6F6.19     SPAC6F6.19.1    CDS          I  2759479   \n",
       "109   2797593    SPAC1805.14    SPAC1805.14.1    CDS          I  2796469   \n",
       "281   3151749    SPAC3G9.13c    SPAC3G9.13c.1    CDS          I  3151326   \n",
       "344   3274317    SPNCRNA.899    SPNCRNA.899.1    CDS          I  3273276   \n",
       "13    2622695     SPAC7D4.10     SPAC7D4.10.1    CDS          I  2622169   \n",
       "71    2726497    SPNCRNA.204    SPNCRNA.204.1    CDS          I  2726058   \n",
       "254   3109392     SPAC589.11     SPAC589.11.1    CDS          I  3109174   \n",
       "110   2797593   SPAC1805.15c   SPAC1805.15c.1    CDS          I  2798023   \n",
       "273   3139763    SPNCRNA.889    SPNCRNA.889.1    CDS          I  3138602   \n",
       "294   3175090     SPAC3G9.05     SPAC3G9.05.1    CDS          I  3173946   \n",
       "325   3230578    SPAC6G10.07    SPAC6G10.07.1    CDS          I  3228833   \n",
       "202   3021278   SPAC31G5.18c   SPAC31G5.18c.1    CDS          I  3020942   \n",
       "367   3327476  SPAC2E1P5.04c  SPAC2E1P5.04c.1    CDS          I  3327940   \n",
       "160   2937534    SPNCRNA.861    SPNCRNA.861.1    CDS          I  2937952   \n",
       "256   3115792    SPAC688.03c    SPAC688.03c.1    CDS          I  3115103   \n",
       "401   3412265     SPAC3H5.13     SPAC3H5.13.1    CDS          I  3412527   \n",
       "26    2658236    SPAC4F8.12c    SPAC4F8.12c.1    CDS          I  2650738   \n",
       "446   4510905   SPAC27D7.03c   SPAC27D7.03c.1    CDS          I  4511040   \n",
       "126   2881073    SPACUNK4.15    SPACUNK4.15.1    CDS          I  2881105   \n",
       "113   2804242    SPAC1805.17    SPAC1805.17.1    CDS          I  2804508   \n",
       "312   3202214    SPNCRNA.894    SPNCRNA.894.1    CDS          I  3202438   \n",
       "352   3301454    SPNCRNA.900    SPNCRNA.900.1    CDS          I  3300812   \n",
       "139   2911919     SPAC513.02     SPAC513.02.1    CDS          I  2911719   \n",
       "183   2975371     SPNCRNA.31     SPNCRNA.31.1    CDS          I  2975608   \n",
       "211   3322640  SPAC2E1P5.01c  SPAC2E1P5.01c.1    CDS          I  3322913   \n",
       "199   3013837    SPAC31G5.14    SPAC31G5.14.1    CDS          I  3012593   \n",
       "331   4515380    SPNCRNA.993    SPNCRNA.993.1    CDS          I  4515182   \n",
       "282   3152754     SPAC3G9.12     SPAC3G9.12.1    CDS          I  3152779   \n",
       "10    2617567    SPAC7D4.11c    SPAC7D4.11c.1    CDS          I  2617975   \n",
       "168   2958937  SPAPB24D3.07c  SPAPB24D3.07c.1    CDS          I  2958024   \n",
       "148   2924501    SPNCRNA.859    SPNCRNA.859.1    CDS          I  2924745   \n",
       "18    2635318    SPAC7D4.03c    SPAC7D4.03c.1    CDS          I  2634211   \n",
       "150   3158351    SPNCRNA.217    SPNCRNA.217.1    CDS          I  3158482   \n",
       "457   4531317    SPNCRNA.996    SPNCRNA.996.1    CDS          I  4531144   \n",
       "467   4544534     SPAC637.04     SPAC637.04.1    CDS          I  4544074   \n",
       "73    2733748    SPAC6F6.02c    SPAC6F6.02c.1    CDS          I  2732524   \n",
       "53    2698054    SPAC644.13c    SPAC644.13c.1    CDS          I  2697418   \n",
       "66    2722524    SPAPB2B4.06    SPAPB2B4.06.1    CDS          I  2722735   \n",
       "65    2722524    SPAPB2B4.05    SPAPB2B4.05.1    CDS          I  2722060   \n",
       "485   4579546   SPAC12B10.03   SPAC12B10.03.1    CDS          I  4578627   \n",
       "456   4531317    SPNCRNA.995    SPNCRNA.995.1    CDS          I  4531036   \n",
       "133   2905076  SPAPYUK71.03c  SPAPYUK71.03c.1    CDS          I  2903369   \n",
       "\n",
       "         end strand detail  p_value  correlation  p_value_corr  \\\n",
       "424  4473271      +   exon      1.0     0.678393  3.042498e-07   \n",
       "195  3006845      +   exon      1.0     0.664822  6.329116e-07   \n",
       "194  3004335      -   exon      1.0     0.618445  5.970721e-06   \n",
       "425  4473227      -   exon      1.0    -0.561792  5.931923e-05   \n",
       "276  3149507      -   exon      1.0     0.464559  1.305065e-03   \n",
       "454  4530853      -   exon      1.0    -0.452833  1.787387e-03   \n",
       "253  3105524      +   exon      1.0     0.451910  1.831298e-03   \n",
       "233  3075353      -   exon      1.0    -0.438076  2.614700e-03   \n",
       "91   2759709      -   exon      1.0     0.415810  4.501092e-03   \n",
       "109  2797827      +   exon      1.0     0.412780  4.833124e-03   \n",
       "281  3151625      +   exon      1.0     0.406324  5.612496e-03   \n",
       "344  3274078      -   exon      1.0     0.399789  6.510566e-03   \n",
       "13   2622195      -   exon      1.0    -0.399736  6.518284e-03   \n",
       "71   2726682      +   exon      1.0    -0.397760  6.813634e-03   \n",
       "254  3109312      +   exon      1.0     0.395784  7.120510e-03   \n",
       "110  2798122      -   exon      1.0    -0.383663  9.277232e-03   \n",
       "273  3139458      +   exon      1.0     0.375231  1.109004e-02   \n",
       "294  3175729      -   exon      1.0    -0.372200  1.181179e-02   \n",
       "325  3230924      +   exon      1.0    -0.363109  1.422279e-02   \n",
       "202  3021733      -   exon      1.0     0.357839  1.580339e-02   \n",
       "367  3329007      -   exon      1.0    -0.353491  1.721726e-02   \n",
       "160  2939052      -   exon      1.0     0.352042  1.771171e-02   \n",
       "256  3115326      -   exon      1.0     0.351910  1.775725e-02   \n",
       "401  3412569      +   exon      1.0     0.351647  1.784863e-02   \n",
       "26   2657829      +   exon      1.0    -0.347299  1.941465e-02   \n",
       "446  4513292      -   exon      1.0     0.342951  2.109488e-02   \n",
       "126  2881731      -   exon      1.0    -0.342556  2.125351e-02   \n",
       "113  2807744      +   exon      1.0    -0.334124  2.488393e-02   \n",
       "312  3203222      -   exon      1.0     0.326219  2.874420e-02   \n",
       "352  3302082      -   exon      1.0     0.314756  3.521419e-02   \n",
       "139  2912369      +   exon      1.0     0.314493  3.537597e-02   \n",
       "183  2976007      +   exon      1.0     0.311331  3.736541e-02   \n",
       "211  3324478      -   exon      1.0    -0.310145  3.813473e-02   \n",
       "199  3013756      +   exon      1.0     0.307773  3.971237e-02   \n",
       "331  4515993      +   exon      0.0    -0.306983  4.024996e-02   \n",
       "282  3157050      -   exon      1.0    -0.306324  4.070250e-02   \n",
       "10   2620284      +   exon      1.0    -0.305929  4.097601e-02   \n",
       "168  2958926      -   exon      NaN     0.303426  4.274331e-02   \n",
       "148  2926536      +   exon      1.0     0.302108  4.369815e-02   \n",
       "18   2636871      +   exon      1.0    -0.301054  4.447447e-02   \n",
       "150  3159120      +   exon      1.0    -0.299249  4.583003e-02   \n",
       "457  4532754      +   exon      1.0    -0.292490  5.120673e-02   \n",
       "467  4544082      +   exon      1.0    -0.291963  5.164652e-02   \n",
       "73   2733570      -   exon      1.0     0.286034  5.680726e-02   \n",
       "53   2698095      -   exon      1.0    -0.285771  5.704589e-02   \n",
       "66   2723210      +   exon      1.0    -0.285112  5.764600e-02   \n",
       "65   2722291      +   exon      1.0    -0.284585  5.812972e-02   \n",
       "485  4580258      +   exon      1.0    -0.277470  6.498375e-02   \n",
       "456  4532486      -   exon      1.0    -0.275889  6.659088e-02   \n",
       "133  2907046      -   exon      1.0    -0.274835  6.767973e-02   \n",
       "\n",
       "                        gene  unique  \\\n",
       "424                     mfs1     6.0   \n",
       "195                     pac2     1.0   \n",
       "194         eta2-antisense-1     1.0   \n",
       "425         mfs1-antisense-1     1.0   \n",
       "276                     sak1     2.0   \n",
       "454             SPAC27D7.11c     1.0   \n",
       "253      atg1801-antisense-1     1.0   \n",
       "233                     spn5     2.0   \n",
       "91                SPAC6F6.19     1.0   \n",
       "109              SPAC1805.14     1.0   \n",
       "281                     msw1     4.0   \n",
       "344              SPNCRNA.899     1.0   \n",
       "13                     vma13     1.0   \n",
       "71               SPNCRNA.204     1.0   \n",
       "254                     pth4     1.0   \n",
       "110                     pub2     1.0   \n",
       "273  SPAC688.12c-antisense-1     1.0   \n",
       "294                     spa2     2.0   \n",
       "325                     cbc1     1.0   \n",
       "202                     sde2     1.0   \n",
       "367                     cwg2     2.0   \n",
       "160              SPNCRNA.861     2.0   \n",
       "256              SPAC688.03c     1.0   \n",
       "401                     new4     1.0   \n",
       "26                     spp42     1.0   \n",
       "446                     mei2     4.0   \n",
       "126              SPACUNK4.15     1.0   \n",
       "113                     crm1     2.0   \n",
       "312         fmc1-antisense-1     1.0   \n",
       "352              SPNCRNA.900     2.0   \n",
       "139               SPAC513.02     2.0   \n",
       "183               SPNCRNA.31     9.0   \n",
       "211                     mns1     1.0   \n",
       "199                     gcv1     1.0   \n",
       "331              SPNCRNA.993     2.0   \n",
       "282                     peg1     6.0   \n",
       "10                     sec39     2.0   \n",
       "168            SPAPB24D3.07c     1.0   \n",
       "148         amt3-antisense-1     9.0   \n",
       "18                      ahk1     2.0   \n",
       "150              SPNCRNA.217    10.0   \n",
       "457              SPNCRNA.996     2.0   \n",
       "467                     ypp1     1.0   \n",
       "73                      pof5     2.0   \n",
       "53               SPAC644.13c     1.0   \n",
       "66               SPAPB2B4.06     1.0   \n",
       "65                      vma5     1.0   \n",
       "485                    bun62     1.0   \n",
       "456              SPNCRNA.995     2.0   \n",
       "133                     tcb3     1.0   \n",
       "\n",
       "                                             positions       albicans_name(s)  \\\n",
       "424  [4473108, 4473996, 4474485, 4474680, 4474681, ...                    NaN   \n",
       "195                                          [3006775]  ['C1_10150W_A' 'nan']   \n",
       "194                                          [3002877]                    NaN   \n",
       "425                                          [4473108]                    NaN   \n",
       "276                                 [3147678, 3149857]                    NaN   \n",
       "454                                          [4531317]                    NaN   \n",
       "253                                          [3103928]                    NaN   \n",
       "233                                 [3074774, 3075383]                    NaN   \n",
       "91                                           [2759901]                    NaN   \n",
       "109                                          [2797593]                    NaN   \n",
       "281               [3151749, 3152754, 3152848, 3152996]        ['C5_02770W_A']   \n",
       "344                                          [3274317]                    NaN   \n",
       "13                                           [2622695]                    NaN   \n",
       "71                                           [2726497]                    NaN   \n",
       "254                                          [3109392]                    NaN   \n",
       "110                                          [2797593]        ['C2_08500W_A']   \n",
       "273                                          [3139763]                    NaN   \n",
       "294                                 [3175090, 3175428]  ['C1_12620W_A' 'nan']   \n",
       "325                                          [3230578]                    NaN   \n",
       "202                                          [3021278]                    NaN   \n",
       "367                                 [3327476, 3329415]                    NaN   \n",
       "160                                 [2937534, 2937800]                    NaN   \n",
       "256                                          [3115792]                    NaN   \n",
       "401                                          [3412265]                    NaN   \n",
       "26                                           [2658236]                    NaN   \n",
       "446               [4510905, 4513497, 4513710, 4513733]                    NaN   \n",
       "126                                          [2881073]        ['CR_06690C_A']   \n",
       "113                                 [2804242, 2807441]        ['CR_00520C_A']   \n",
       "312                                          [3202214]                    NaN   \n",
       "352                                 [3301454, 3302072]                  [nan]   \n",
       "139                                 [2911919, 2911931]        ['C3_06920W_A']   \n",
       "183  [2975371, 2975378, 2975424, 2975613, 2975812, ...                    NaN   \n",
       "211                                          [3322640]        ['C1_03730C_A']   \n",
       "199                                          [3013837]                    NaN   \n",
       "331                                 [4515380, 4516287]                    NaN   \n",
       "282  [3152754, 3152848, 3152996, 3154972, 3155730, ...                    NaN   \n",
       "10                                  [2617567, 2617646]                    NaN   \n",
       "168                                          [2958937]                    NaN   \n",
       "148  [2924501, 2924596, 2924608, 2924675, 2924705, ...                    NaN   \n",
       "18                                  [2635318, 2635612]                    NaN   \n",
       "150  [3158351, 3158357, 3158359, 3158363, 3158367, ...                    NaN   \n",
       "457                                 [4531317, 4532855]                    NaN   \n",
       "467                                          [4544534]                    NaN   \n",
       "73                                  [2733748, 2733863]                    NaN   \n",
       "53                                           [2698054]                    NaN   \n",
       "66                                           [2722524]        ['C2_05170W_A']   \n",
       "65                                           [2722524]                    NaN   \n",
       "485                                          [4579546]                    NaN   \n",
       "456                                 [4531317, 4532855]                    NaN   \n",
       "133                                          [2905076]                    NaN   \n",
       "\n",
       "        cerevisiae_name(s)  \n",
       "424                    NaN  \n",
       "195  ['YEL007W' 'YHR177W']  \n",
       "194                    NaN  \n",
       "425                    NaN  \n",
       "276                    NaN  \n",
       "454                    NaN  \n",
       "253                    NaN  \n",
       "233                    NaN  \n",
       "91                     NaN  \n",
       "109                    NaN  \n",
       "281            ['YDR268W']  \n",
       "344                    NaN  \n",
       "13                     NaN  \n",
       "71                     NaN  \n",
       "254                    NaN  \n",
       "110            ['YER125W']  \n",
       "273                    NaN  \n",
       "294  ['YLL021W' 'YLR313C']  \n",
       "325                    NaN  \n",
       "202                    NaN  \n",
       "367                    NaN  \n",
       "160                    NaN  \n",
       "256                    NaN  \n",
       "401                    NaN  \n",
       "26                     NaN  \n",
       "446                    NaN  \n",
       "126            ['YGR247W']  \n",
       "113            ['YGR218W']  \n",
       "312                    NaN  \n",
       "352                  [nan]  \n",
       "139            ['YKL128C']  \n",
       "183                    NaN  \n",
       "211            ['YJR131W']  \n",
       "199                    NaN  \n",
       "331                    NaN  \n",
       "282                    NaN  \n",
       "10                     NaN  \n",
       "168                    NaN  \n",
       "148                    NaN  \n",
       "18                     NaN  \n",
       "150                    NaN  \n",
       "457                    NaN  \n",
       "467                    NaN  \n",
       "73                     NaN  \n",
       "53                     NaN  \n",
       "66             ['YBL095W']  \n",
       "65                     NaN  \n",
       "485                    NaN  \n",
       "456                    NaN  \n",
       "133                    NaN  "
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#sort by p_value\n",
    "result_final__ = result_final__.sort_values(by=['p_value_corr'])\n",
    "result_final__.head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "result_final__ = result.copy().reset_index(drop=True, inplace=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "#iterate through positions in result_final__  and find the corresponding alleles in snps. Then add the two values in X968_allele\tY0036_allele to the df in two new columns at the start \n",
    "#add two columns to result_final__\n",
    "result_final__['X968_allele'] = np.nan\n",
    "result_final__['Y0036_allele'] = np.nan\n",
    "for i in range(len(result_final__)):\n",
    "    for j in range(len(snps)):\n",
    "        if result_final__['position'][i] == snps['position'][j]:\n",
    "            result_final__['X968_allele'][i] = snps['X968_allele'][j]\n",
    "            result_final__['Y0036_allele'][i] = snps['Y0036_allele'][j]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "#iterate through positions in result_final__ and look up the chromosome and position in regions, and find the corresponding letter +/- 5 bp in regions. Add it to a new column\n",
    "result_final__['reference_seq']=np.nan\n",
    "for i in range(len(result_final__)):\n",
    "    for j in range(len(regions)):\n",
    "        if result_final__['chromosome'][i] == list(regions.keys())[j]:\n",
    "            #add | | characters left and right of tletter at position-1\n",
    "            \n",
    "            result_final__['reference_seq'][i] = regions[list(regions.keys())[j]][result_final__['position'][i]-4:result_final__['position'][i]-1] + ' |' + regions[list(regions.keys())[j]][result_final__['position'][i]-1] + '| ' + regions[list(regions.keys())[j]][result_final__['position'][i]:result_final__['position'][i]+3] \n",
    "\n",
    "#if strand is -, reverse reference_seq\n",
    "for i in range(len(result_final__)):\n",
    "    if result_final__['strand'][i] == '-':\n",
    "        result_final__['reference_seq'][i] = result_final__['reference_seq'][i][::-1]\n",
    "        #also flip the alleles\n",
    "        result_final__['X968_allele'][i] = result_final__['X968_allele'][i][::-1]\n",
    "        result_final__['Y0036_allele'][i] = result_final__['Y0036_allele'][i][::-1]\n",
    "#also create reference_seq_ without the | | characters\n",
    "result_final__['reference_seq_'] = result_final__['reference_seq'].str.replace('|', '')\n",
    "result_final__['reference_seq_'] = result_final__['reference_seq_'].str.replace(' ', '')\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SPNCRNA.3165\n",
      "SPAC823.14\n",
      "SPNCRNA.3165\n",
      "SPNCRNA.3166\n",
      "SPAC7D4.13c\n",
      "SPAC7D4.13c\n",
      "SPNCRNA.3169\n",
      "SPNCRNA.3171\n",
      "SPAC7D4.11c\n",
      "SPNCRNA.3171\n",
      "SPAC7D4.11c\n",
      "SPNCRNA.3171\n",
      "SPNCRNA.3172\n",
      "SPAC7D4.10\n",
      "SPNCRNA.3173\n",
      "SPNCRNA.842\n",
      "SPAC7D4.15c\n",
      "SPAC7D4.15c\n",
      "SPNCRNA.3178\n",
      "SPNCRNA.3178\n",
      "SPNCRNA.3184\n",
      "SPNCRNA.3184\n",
      "SPAC4F8.12c\n",
      "SPNCRNA.3189\n",
      "SPNCRNA.326\n",
      "SPNCRNA.3189\n",
      "SPNCRNA.200\n",
      "SPNCRNA.3196\n",
      "SPAC4F8.01\n",
      "SPNCRNA.3199\n",
      "SPAC644.05c\n",
      "SPNCRNA.3202\n",
      "SPNCRNA.844\n",
      "SPNCRNA.3207\n",
      "SPNCRNA.3207\n",
      "SPNCRNA.3208\n",
      "SPNCRNA.202\n",
      "SPNCRNA.3212\n",
      "SPNCRNA.3213\n",
      "SPNCRNA.3214\n",
      "SPNCRNA.846\n",
      "SPNCRNA.846\n",
      "SPNCRNA.3216\n",
      "SPNCRNA.3217\n",
      "SPNCRNA.3218\n",
      "SPAPB2B4.06\n",
      "SPNCRNA.3220\n",
      "SPAPB2B4.07\n",
      "SPNCRNA.3221\n",
      "SPNCRNA.204\n",
      "SPNCRNA.3222\n",
      "SPNCRNA.847\n",
      "SPNCRNA.847\n",
      "SPNCRNA.847\n",
      "SPNCRNA.848\n",
      "SPNCRNA.3227\n",
      "SPNCRNA.3227\n",
      "SPNCRNA.3227\n",
      "SPNCRNA.3227\n",
      "SPNCRNA.3228\n",
      "SPNCRNA.3230\n",
      "SPNCRNA.3230\n",
      "SPAC6F6.10c\n",
      "SPNCRNA.3235\n",
      "SPNCRNA.849\n",
      "SPNCRNA.849\n",
      "SPNCRNA.849\n",
      "SPNCRNA.849\n",
      "SPNCRNA.849\n",
      "SPNCRNA.850\n",
      "SPAC6F6.19\n",
      "SPNCRNA.3239\n",
      "SPNCRNA.851\n",
      "SPNCRNA.3239\n",
      "SPNCRNA.3240\n",
      "SPNCRNA.3240\n",
      "SPNCRNA.3242\n",
      "SPNCRNA.3242\n",
      "SPNCRNA.852\n",
      "SPNCRNA.3244\n",
      "SPNCRNA.3246\n",
      "SPAC1805.11c\n",
      "SPAC1805.15c\n",
      "SPNCRNA.3259\n",
      "SPNCRNA.3263\n",
      "SPNCRNA.3264\n",
      "SPAC8F11.04\n",
      "SPNCRNA.856\n",
      "SPNCRNA.3296\n",
      "SPNCRNA.3297\n",
      "SPNCRNA.3299\n",
      "SPNCRNA.858\n",
      "SPACUNK4.06c\n",
      "SPNCRNA.209\n",
      "SPNCRNA.3316\n",
      "SPNCRNA.209\n",
      "SPNCRNA.3316\n",
      "SPNCRNA.3317\n",
      "SPNCRNA.3317\n",
      "SPNCRNA.3318\n",
      "SPNCRNA.3321\n",
      "SPNCRNA.3321\n",
      "SPNCRNA.3321\n",
      "SPNCRNA.3321\n",
      "SPNCRNA.859\n",
      "SPNCRNA.859\n",
      "SPNCRNA.859\n",
      "SPNCRNA.859\n",
      "SPNCRNA.859\n",
      "SPNCRNA.859\n",
      "SPNCRNA.859\n",
      "SPNCRNA.859\n",
      "SPAC2E1P3.03c\n",
      "SPNCRNA.859\n",
      "SPNCRNA.217\n",
      "SPNCRNA.3406\n",
      "SPNCRNA.217\n",
      "SPNCRNA.3406\n",
      "SPNCRNA.217\n",
      "SPNCRNA.3406\n",
      "SPNCRNA.217\n",
      "SPNCRNA.3406\n",
      "SPNCRNA.217\n",
      "SPNCRNA.3406\n",
      "SPNCRNA.217\n",
      "SPNCRNA.3406\n",
      "SPNCRNA.217\n",
      "SPNCRNA.3406\n",
      "SPNCRNA.892\n",
      "SPNCRNA.3413\n",
      "SPNCRNA.3414\n",
      "SPNCRNA.217\n",
      "SPNCRNA.3406\n",
      "SPNCRNA.3327\n",
      "SPNCRNA.3328\n",
      "SPNCRNA.3327\n",
      "SPNCRNA.3328\n",
      "SPNCRNA.861\n",
      "SPNCRNA.3327\n",
      "SPNCRNA.3328\n",
      "SPNCRNA.861\n",
      "SPNCRNA.862\n",
      "SPAPB24D3.04c\n",
      "SPNCRNA.864\n",
      "SPNCRNA.865\n",
      "SPNCRNA.3337\n",
      "SPNCRNA.865\n",
      "SPNCRNA.3337\n",
      "SPNCRNA.3338\n",
      "SPNCRNA.3337\n",
      "SPNCRNA.3338\n",
      "SPNCRNA.3339\n",
      "SPNCRNA.3340\n",
      "SPNCRNA.3341\n",
      "SPNCRNA.866\n",
      "SPNCRNA.867\n",
      "SPNCRNA.3341\n",
      "SPNCRNA.866\n",
      "SPNCRNA.867\n",
      "SPNCRNA.866\n",
      "SPNCRNA.867\n",
      "SPNCRNA.866\n",
      "SPNCRNA.867\n",
      "SPAC6G10.11c\n",
      "SPNCRNA.3448\n",
      "SPNCRNA.31\n",
      "SPNCRNA.866\n",
      "SPNCRNA.867\n",
      "SPNCRNA.31\n",
      "SPNCRNA.866\n",
      "SPNCRNA.867\n",
      "SPNCRNA.31\n",
      "SPNCRNA.866\n",
      "SPNCRNA.867\n",
      "SPNCRNA.31\n",
      "SPNCRNA.866\n",
      "SPNCRNA.867\n",
      "SPNCRNA.31\n",
      "SPNCRNA.866\n",
      "SPNCRNA.867\n",
      "SPNCRNA.31\n",
      "SPNCRNA.866\n",
      "SPNCRNA.867\n",
      "SPNCRNA.31\n",
      "SPNCRNA.866\n",
      "SPNCRNA.867\n",
      "SPNCRNA.31\n",
      "SPNCRNA.866\n",
      "SPNCRNA.867\n",
      "SPNCRNA.31\n",
      "SPNCRNA.866\n",
      "SPNCRNA.867\n",
      "SPNCRNA.3342\n",
      "SPNCRNA.867\n",
      "SPNCRNA.3342\n",
      "SPNCRNA.867\n",
      "SPNCRNA.3342\n",
      "SPNCRNA.868\n",
      "SPAPB1A11.01\n",
      "SPNCRNA.3342\n",
      "SPNCRNA.868\n",
      "SPAPB1A11.01\n",
      "SPNCRNA.3343\n",
      "SPNCRNA.868\n",
      "SPNCRNA.869\n",
      "SPAC31G5.03\n",
      "SPNCRNA.870\n",
      "SPNCRNA.872\n",
      "SPNCRNA.3350\n",
      "SPNCRNA.3351\n",
      "SPNCRNA.3353\n",
      "SPNCRNA.875\n",
      "SPNCRNA.3356\n",
      "SPAC1786.01c\n",
      "SPAC1786.01c\n",
      "SPNCRNA.3358\n",
      "SPNCRNA.3359\n",
      "SPNCRNA.876\n",
      "SPNCRNA.3359\n",
      "SPNCRNA.876\n",
      "SPNCRNA.3359\n",
      "SPNCRNA.876\n",
      "SPNCRNA.3483\n",
      "SPNCRNA.3484\n",
      "SPAC2E1P5.01c\n",
      "SPNCRNA.3484\n",
      "SPNCRNA.3361\n",
      "SPNCRNA.3361\n",
      "SPNCRNA.3364\n",
      "SPAC24C9.16c\n",
      "SPNCRNA.3365\n",
      "SPNCRNA.3366\n",
      "SPNCRNA.3370\n",
      "SPNCRNA.879\n",
      "SPNCRNA.3372\n",
      "SPNCRNA.3376\n",
      "SPNCRNA.3376\n",
      "SPNCRNA.883\n",
      "SPNCRNA.883\n",
      "SPNCRNA.3380\n",
      "SPAC8E11.04c\n",
      "SPAC8E11.04c\n",
      "SPNCRNA.3507\n",
      "SPNCRNA.3380\n",
      "SPNCRNA.214\n",
      "SPNCRNA.214\n",
      "SPNCRNA.3382\n",
      "SPNCRNA.214\n",
      "SPNCRNA.215\n",
      "SPNCRNA.3383\n",
      "SPNCRNA.214\n",
      "SPNCRNA.215\n",
      "SPNCRNA.3383\n",
      "SPAC16A10.05c\n",
      "SPAC16A10.06c\n",
      "SPNCRNA.3384\n",
      "SPAC16A10.06c\n",
      "SPNCRNA.3384\n",
      "SPAC589.07c\n",
      "SPNCRNA.560\n",
      "SPNCRNA.3389\n",
      "SPAC688.03c\n",
      "SPNCRNA.3390\n",
      "SPNCRNA.887\n",
      "SPNCRNA.3393\n",
      "SPNCRNA.3394\n",
      "SPNCRNA.3395\n",
      "SPNCRNA.3396\n",
      "SPNCRNA.3397\n",
      "SPNCRNA.3399\n",
      "SPNCRNA.3399\n",
      "SPNCRNA.3399\n",
      "SPNCRNA.3399\n",
      "SPNCRNA.889\n",
      "SPNCRNA.891\n",
      "SPNCRNA.216\n",
      "SPNCRNA.216\n",
      "SPNCRNA.3405\n",
      "SPNCRNA.3405\n",
      "SPNCRNA.217\n",
      "SPNCRNA.3406\n",
      "SPNCRNA.3407\n",
      "SPNCRNA.217\n",
      "SPNCRNA.3406\n",
      "SPNCRNA.3407\n",
      "SPNCRNA.3408\n",
      "SPNCRNA.3409\n",
      "SPNCRNA.3411\n",
      "SPNCRNA.3412\n",
      "SPRRNA.14\n",
      "SPNCRNA.218\n",
      "SPNCRNA.3411\n",
      "SPNCRNA.3412\n",
      "SPRRNA.14\n",
      "SPNCRNA.218\n",
      "SPNCRNA.3411\n",
      "SPNCRNA.3412\n",
      "SPRRNA.14\n",
      "SPNCRNA.3415\n",
      "SPNCRNA.3415\n",
      "SPNCRNA.3417\n",
      "SPNCRNA.3418\n",
      "SPNCRNA.3419\n",
      "SPNCRNA.3419\n",
      "SPNCRNA.3420\n",
      "SPNCRNA.3419\n",
      "SPNCRNA.3420\n",
      "SPNCRNA.3421\n",
      "SPNCRNA.3424\n",
      "SPNCRNA.3425\n",
      "SPNCRNA.3426\n",
      "SPNCRNA.3426\n",
      "SPNCRNA.3426\n",
      "SPNCRNA.3429\n",
      "SPNCRNA.3429\n",
      "SPNCRNA.894\n",
      "SPNCRNA.3433\n",
      "SPNCRNA.3433\n",
      "SPNCRNA.3434\n",
      "SPNCRNA.3434\n",
      "SPNCRNA.3435\n",
      "SPNCRNA.3438\n",
      "SPNCRNA.992\n",
      "SPNCRNA.896\n",
      "SPNCRNA.3446\n",
      "SPAC6G10.08\n",
      "SPNCRNA.3447\n",
      "SPNCRNA.3447\n",
      "SPNCRNA.3447\n",
      "SPAC6G10.11c\n",
      "SPNCRNA.3448\n",
      "SPNCRNA.4033\n",
      "SPNCRNA.993\n",
      "SPAC6G10.11c\n",
      "SPNCRNA.3449\n",
      "SPNCRNA.3449\n",
      "SPNCRNA.897\n",
      "SPNCRNA.897\n",
      "SPNCRNA.898\n",
      "SPNCRNA.3455\n",
      "SPNCRNA.3456\n",
      "SPNCRNA.899\n",
      "SPNCRNA.3467\n",
      "SPNCRNA.3476\n",
      "SPNCRNA.900\n",
      "SPNCRNA.3476\n",
      "SPNCRNA.3477\n",
      "SPNCRNA.3478\n",
      "SPNCRNA.900\n",
      "SPNCRNA.3477\n",
      "SPNCRNA.3477\n",
      "SPNCRNA.3477\n",
      "SPNCRNA.3481\n",
      "SPNCRNA.3481\n",
      "SPNCRNA.3482\n",
      "SPNCRNA.3483\n",
      "SPAC2E1P5.02c\n",
      "SPNCRNA.3487\n",
      "SPNCRNA.3488\n",
      "SPNCRNA.3488\n",
      "SPAC2E1P5.04c\n",
      "SPAC2E1P5.04c\n",
      "SPNCRNA.3490\n",
      "SPNCRNA.3490\n",
      "SPAC26A3.11\n",
      "SPAC8E11.07c\n",
      "SPAC8E11.07c\n",
      "SPNCRNA.3507\n",
      "SPNCRNA.905\n",
      "SPNCRNA.905\n",
      "SPNCRNA.3512\n",
      "SPNCRNA.3512\n",
      "SPNCRNA.3513\n",
      "SPSNORNA.20\n",
      "SPNCRNA.3514\n",
      "SPAC959.06c\n",
      "SPNCRNA.3514\n",
      "SPNCRNA.3515\n",
      "SPNCRNA.3517\n",
      "SPNCRNA.3519\n",
      "SPNCRNA.3519\n",
      "SPNCRNA.3520\n",
      "SPNCRNA.907\n",
      "SPNCRNA.3520\n",
      "SPNCRNA.907\n",
      "SPAC3H5.12c\n",
      "SPAC3H5.13\n",
      "SPAC3H5.13\n",
      "SPNCRNA.3523\n",
      "SPNCRNA.3524\n",
      "SPNCRNA.3525\n",
      "SPNCRNA.3526\n",
      "SPNCRNA.3531\n",
      "SPNCRNA.3531\n",
      "SPNCRNA.3531\n",
      "SPNCRNA.3531\n",
      "SPNCRNA.3531\n",
      "SPNCRNA.3531\n",
      "SPAC3H5.07\n",
      "SPNCRNA.3532\n",
      "SPNCRNA.3532\n",
      "SPNCRNA.3532\n",
      "SPNCRNA.3535\n",
      "SPNCRNA.3535\n",
      "SPNCRNA.3535\n",
      "SPATRNAARG.01\n",
      "SPATRNAARG.01\n",
      "SPNCRNA.3537\n",
      "SPATRNAARG.01\n",
      "SPATRNAARG.01\n",
      "SPNCRNA.991\n",
      "SPNCRNA.4017\n",
      "SPAC17C9.09c\n",
      "SPAC17C9.08\n",
      "SPNCRNA.4018\n",
      "SPNCRNA.4019\n",
      "SPAC17C9.08\n",
      "SPNCRNA.4018\n",
      "SPNCRNA.4019\n",
      "SPNCRNA.4021\n",
      "SPNCRNA.4022\n",
      "SPNCRNA.4021\n",
      "SPNCRNA.4022\n",
      "SPNCRNA.4021\n",
      "SPNCRNA.4022\n",
      "SPNCRNA.4023\n",
      "SPNCRNA.4027\n",
      "SPNCRNA.4028\n",
      "SPNCRNA.4028\n",
      "SPNCRNA.4028\n",
      "SPNCRNA.4028\n",
      "SPNCRNA.992\n",
      "SPNCRNA.992\n",
      "SPNCRNA.4029\n",
      "SPNCRNA.4030\n",
      "SPNCRNA.4029\n",
      "SPNCRNA.4030\n",
      "SPNCRNA.4031\n",
      "SPNCRNA.4032\n",
      "SPNCRNA.4031\n",
      "SPNCRNA.4032\n",
      "SPNCRNA.4031\n",
      "SPNCRNA.4032\n",
      "SPNCRNA.4032\n",
      "SPNCRNA.130\n",
      "SPNCRNA.4034\n",
      "SPNCRNA.4035\n",
      "SPNCRNA.993\n",
      "SPNCRNA.4040\n",
      "SPAC27D7.11c\n",
      "SPNCRNA.4049\n",
      "SPNCRNA.995\n",
      "SPNCRNA.996\n",
      "SPNCRNA.995\n",
      "SPNCRNA.996\n",
      "SPNCRNA.4051\n",
      "SPAC27D7.13c\n",
      "SPNCRNA.4051\n",
      "SPNCRNA.4052\n",
      "SPNCRNA.4055\n",
      "SPAC637.05c\n",
      "SPAC637.05c\n",
      "SPNCRNA.4056\n",
      "SPNCRNA.4057\n",
      "SPAC637.06\n",
      "SPNCRNA.4057\n",
      "SPNCRNA.4057\n",
      "SPNCRNA.4057\n",
      "SPNCRNA.563\n",
      "SPNCRNA.998\n",
      "SPNCRNA.998\n",
      "SPAC637.13c\n",
      "SPNCRNA.249\n",
      "SPNCRNA.4059\n",
      "SPAC12B10.04\n",
      "SPAC12B10.04\n",
      "SPNCRNA.1001\n",
      "SPNCRNA.4064\n",
      "SPNCRNA.4064\n",
      "SPNCRNA.4066\n",
      "SPNCRNA.3165\n",
      "SPAC823.14\n",
      "SPNCRNA.3165\n",
      "SPNCRNA.3166\n",
      "SPAC7D4.13c\n",
      "SPAC7D4.13c\n",
      "SPNCRNA.3169\n",
      "SPNCRNA.3171\n",
      "SPAC7D4.11c\n",
      "SPNCRNA.3171\n",
      "SPAC7D4.11c\n",
      "SPNCRNA.3171\n",
      "SPNCRNA.3172\n",
      "SPAC7D4.10\n",
      "SPNCRNA.3173\n",
      "SPNCRNA.842\n",
      "SPAC7D4.15c\n",
      "SPAC7D4.15c\n",
      "SPNCRNA.3178\n",
      "SPNCRNA.3178\n",
      "SPNCRNA.3184\n",
      "SPNCRNA.3184\n",
      "SPAC4F8.12c\n",
      "SPNCRNA.3189\n",
      "SPNCRNA.326\n",
      "SPNCRNA.3189\n",
      "SPNCRNA.200\n",
      "SPNCRNA.3196\n",
      "SPAC4F8.01\n",
      "SPNCRNA.3199\n",
      "SPAC644.05c\n",
      "SPNCRNA.3202\n",
      "SPNCRNA.844\n",
      "SPNCRNA.3207\n",
      "SPNCRNA.3207\n",
      "SPNCRNA.3208\n",
      "SPNCRNA.202\n",
      "SPNCRNA.3212\n",
      "SPNCRNA.3213\n",
      "SPNCRNA.3214\n",
      "SPNCRNA.846\n",
      "SPNCRNA.846\n",
      "SPNCRNA.3216\n",
      "SPNCRNA.3217\n",
      "SPNCRNA.3218\n",
      "SPAPB2B4.06\n",
      "SPNCRNA.3220\n",
      "SPAPB2B4.07\n",
      "SPNCRNA.3221\n",
      "SPNCRNA.204\n",
      "SPNCRNA.3222\n",
      "SPNCRNA.847\n",
      "SPNCRNA.847\n",
      "SPNCRNA.847\n",
      "SPNCRNA.848\n",
      "SPNCRNA.3227\n",
      "SPNCRNA.3227\n",
      "SPNCRNA.3227\n",
      "SPNCRNA.3227\n",
      "SPNCRNA.3228\n",
      "SPNCRNA.3230\n",
      "SPNCRNA.3230\n",
      "SPAC6F6.10c\n",
      "SPNCRNA.3235\n",
      "SPNCRNA.849\n",
      "SPNCRNA.849\n",
      "SPNCRNA.849\n",
      "SPNCRNA.849\n",
      "SPNCRNA.849\n",
      "SPNCRNA.850\n",
      "SPAC6F6.19\n",
      "SPNCRNA.3239\n",
      "SPNCRNA.851\n",
      "SPNCRNA.3239\n",
      "SPNCRNA.3240\n",
      "SPNCRNA.3240\n",
      "SPNCRNA.3242\n",
      "SPNCRNA.3242\n",
      "SPNCRNA.852\n",
      "SPNCRNA.3244\n",
      "SPNCRNA.3246\n",
      "SPAC1805.11c\n",
      "SPAC1805.15c\n",
      "SPNCRNA.3259\n",
      "SPNCRNA.3263\n",
      "SPNCRNA.3264\n",
      "SPAC8F11.04\n",
      "SPNCRNA.856\n",
      "SPNCRNA.3296\n",
      "SPNCRNA.3297\n",
      "SPNCRNA.3299\n",
      "SPNCRNA.858\n",
      "SPACUNK4.06c\n",
      "SPNCRNA.209\n",
      "SPNCRNA.3316\n",
      "SPNCRNA.209\n",
      "SPNCRNA.3316\n",
      "SPNCRNA.3317\n",
      "SPNCRNA.3317\n",
      "SPNCRNA.3318\n",
      "SPNCRNA.3321\n",
      "SPNCRNA.3321\n",
      "SPNCRNA.3321\n",
      "SPNCRNA.3321\n",
      "SPNCRNA.859\n",
      "SPNCRNA.859\n",
      "SPNCRNA.859\n",
      "SPNCRNA.859\n",
      "SPNCRNA.859\n",
      "SPNCRNA.859\n",
      "SPNCRNA.859\n",
      "SPNCRNA.859\n",
      "SPAC2E1P3.03c\n",
      "SPNCRNA.859\n",
      "SPNCRNA.217\n",
      "SPNCRNA.3406\n",
      "SPNCRNA.217\n",
      "SPNCRNA.3406\n",
      "SPNCRNA.217\n",
      "SPNCRNA.3406\n",
      "SPNCRNA.217\n",
      "SPNCRNA.3406\n",
      "SPNCRNA.217\n",
      "SPNCRNA.3406\n",
      "SPNCRNA.217\n",
      "SPNCRNA.3406\n",
      "SPNCRNA.217\n",
      "SPNCRNA.3406\n",
      "SPNCRNA.892\n",
      "SPNCRNA.3413\n",
      "SPNCRNA.3414\n",
      "SPNCRNA.217\n",
      "SPNCRNA.3406\n",
      "SPNCRNA.3327\n",
      "SPNCRNA.3328\n",
      "SPNCRNA.3327\n",
      "SPNCRNA.3328\n",
      "SPNCRNA.861\n",
      "SPNCRNA.3327\n",
      "SPNCRNA.3328\n",
      "SPNCRNA.861\n",
      "SPNCRNA.862\n",
      "SPAPB24D3.04c\n",
      "SPNCRNA.864\n",
      "SPNCRNA.865\n",
      "SPNCRNA.3337\n",
      "SPNCRNA.865\n",
      "SPNCRNA.3337\n",
      "SPNCRNA.3338\n",
      "SPNCRNA.3337\n",
      "SPNCRNA.3338\n",
      "SPNCRNA.3339\n",
      "SPNCRNA.3340\n",
      "SPNCRNA.3341\n",
      "SPNCRNA.866\n",
      "SPNCRNA.867\n",
      "SPNCRNA.3341\n",
      "SPNCRNA.866\n",
      "SPNCRNA.867\n",
      "SPNCRNA.866\n",
      "SPNCRNA.867\n",
      "SPNCRNA.866\n",
      "SPNCRNA.867\n",
      "SPAC6G10.11c\n",
      "SPNCRNA.3448\n",
      "SPNCRNA.31\n",
      "SPNCRNA.866\n",
      "SPNCRNA.867\n",
      "SPNCRNA.31\n",
      "SPNCRNA.866\n",
      "SPNCRNA.867\n",
      "SPNCRNA.31\n",
      "SPNCRNA.866\n",
      "SPNCRNA.867\n",
      "SPNCRNA.31\n",
      "SPNCRNA.866\n",
      "SPNCRNA.867\n",
      "SPNCRNA.31\n",
      "SPNCRNA.866\n",
      "SPNCRNA.867\n",
      "SPNCRNA.31\n",
      "SPNCRNA.866\n",
      "SPNCRNA.867\n",
      "SPNCRNA.31\n",
      "SPNCRNA.866\n",
      "SPNCRNA.867\n",
      "SPNCRNA.31\n",
      "SPNCRNA.866\n",
      "SPNCRNA.867\n",
      "SPNCRNA.31\n",
      "SPNCRNA.866\n",
      "SPNCRNA.867\n",
      "SPNCRNA.3342\n",
      "SPNCRNA.867\n",
      "SPNCRNA.3342\n",
      "SPNCRNA.867\n",
      "SPNCRNA.3342\n",
      "SPNCRNA.868\n",
      "SPAPB1A11.01\n",
      "SPNCRNA.3342\n",
      "SPNCRNA.868\n",
      "SPAPB1A11.01\n",
      "SPNCRNA.3343\n",
      "SPNCRNA.868\n",
      "SPNCRNA.869\n",
      "SPAC31G5.03\n",
      "SPNCRNA.870\n",
      "SPNCRNA.872\n",
      "SPNCRNA.3350\n",
      "SPNCRNA.3351\n",
      "SPNCRNA.3353\n",
      "SPNCRNA.875\n",
      "SPNCRNA.3356\n",
      "SPAC1786.01c\n",
      "SPAC1786.01c\n",
      "SPNCRNA.3358\n",
      "SPNCRNA.3359\n",
      "SPNCRNA.876\n",
      "SPNCRNA.3359\n",
      "SPNCRNA.876\n",
      "SPNCRNA.3359\n",
      "SPNCRNA.876\n",
      "SPNCRNA.3483\n",
      "SPNCRNA.3484\n",
      "SPAC2E1P5.01c\n",
      "SPNCRNA.3484\n",
      "SPNCRNA.3361\n",
      "SPNCRNA.3361\n",
      "SPNCRNA.3364\n",
      "SPAC24C9.16c\n",
      "SPNCRNA.3365\n",
      "SPNCRNA.3366\n",
      "SPNCRNA.3370\n",
      "SPNCRNA.879\n",
      "SPNCRNA.3372\n",
      "SPNCRNA.3376\n",
      "SPNCRNA.3376\n",
      "SPNCRNA.883\n",
      "SPNCRNA.883\n",
      "SPNCRNA.3380\n",
      "SPAC8E11.04c\n",
      "SPAC8E11.04c\n",
      "SPNCRNA.3507\n",
      "SPNCRNA.3380\n",
      "SPNCRNA.214\n",
      "SPNCRNA.214\n",
      "SPNCRNA.3382\n",
      "SPNCRNA.214\n",
      "SPNCRNA.215\n",
      "SPNCRNA.3383\n",
      "SPNCRNA.214\n",
      "SPNCRNA.215\n",
      "SPNCRNA.3383\n",
      "SPAC16A10.05c\n",
      "SPAC16A10.06c\n",
      "SPNCRNA.3384\n",
      "SPAC16A10.06c\n",
      "SPNCRNA.3384\n",
      "SPAC589.07c\n",
      "SPNCRNA.560\n",
      "SPNCRNA.3389\n",
      "SPAC688.03c\n",
      "SPNCRNA.3390\n",
      "SPNCRNA.887\n",
      "SPNCRNA.3393\n",
      "SPNCRNA.3394\n",
      "SPNCRNA.3395\n",
      "SPNCRNA.3396\n",
      "SPNCRNA.3397\n",
      "SPNCRNA.3399\n",
      "SPNCRNA.3399\n",
      "SPNCRNA.3399\n",
      "SPNCRNA.3399\n",
      "SPNCRNA.889\n",
      "SPNCRNA.891\n",
      "SPNCRNA.216\n",
      "SPNCRNA.216\n",
      "SPNCRNA.3405\n",
      "SPNCRNA.3405\n",
      "SPNCRNA.217\n",
      "SPNCRNA.3406\n",
      "SPNCRNA.3407\n",
      "SPNCRNA.217\n",
      "SPNCRNA.3406\n",
      "SPNCRNA.3407\n",
      "SPNCRNA.3408\n",
      "SPNCRNA.3409\n",
      "SPNCRNA.3411\n",
      "SPNCRNA.3412\n",
      "SPRRNA.14\n",
      "SPNCRNA.218\n",
      "SPNCRNA.3411\n",
      "SPNCRNA.3412\n",
      "SPRRNA.14\n",
      "SPNCRNA.218\n",
      "SPNCRNA.3411\n",
      "SPNCRNA.3412\n",
      "SPRRNA.14\n",
      "SPNCRNA.3415\n",
      "SPNCRNA.3415\n",
      "SPNCRNA.3417\n",
      "SPNCRNA.3418\n",
      "SPNCRNA.3419\n",
      "SPNCRNA.3419\n",
      "SPNCRNA.3420\n",
      "SPNCRNA.3419\n",
      "SPNCRNA.3420\n",
      "SPNCRNA.3421\n",
      "SPNCRNA.3424\n",
      "SPNCRNA.3425\n",
      "SPNCRNA.3426\n",
      "SPNCRNA.3426\n",
      "SPNCRNA.3426\n",
      "SPNCRNA.3429\n",
      "SPNCRNA.3429\n",
      "SPNCRNA.894\n",
      "SPNCRNA.3433\n",
      "SPNCRNA.3433\n",
      "SPNCRNA.3434\n",
      "SPNCRNA.3434\n",
      "SPNCRNA.3435\n",
      "SPNCRNA.3438\n",
      "SPNCRNA.992\n",
      "SPNCRNA.896\n",
      "SPNCRNA.3446\n",
      "SPAC6G10.08\n",
      "SPNCRNA.3447\n",
      "SPNCRNA.3447\n",
      "SPNCRNA.3447\n",
      "SPAC6G10.11c\n",
      "SPNCRNA.3448\n",
      "SPNCRNA.4033\n",
      "SPNCRNA.993\n",
      "SPAC6G10.11c\n",
      "SPNCRNA.3449\n",
      "SPNCRNA.3449\n",
      "SPNCRNA.897\n",
      "SPNCRNA.897\n",
      "SPNCRNA.898\n",
      "SPNCRNA.3455\n",
      "SPNCRNA.3456\n",
      "SPNCRNA.899\n",
      "SPNCRNA.3467\n",
      "SPNCRNA.3476\n",
      "SPNCRNA.900\n",
      "SPNCRNA.3476\n",
      "SPNCRNA.3477\n",
      "SPNCRNA.3478\n",
      "SPNCRNA.900\n",
      "SPNCRNA.3477\n",
      "SPNCRNA.3477\n",
      "SPNCRNA.3477\n",
      "SPNCRNA.3481\n",
      "SPNCRNA.3481\n",
      "SPNCRNA.3482\n",
      "SPNCRNA.3483\n",
      "SPAC2E1P5.02c\n",
      "SPNCRNA.3487\n",
      "SPNCRNA.3488\n",
      "SPNCRNA.3488\n",
      "SPAC2E1P5.04c\n",
      "SPAC2E1P5.04c\n",
      "SPNCRNA.3490\n",
      "SPNCRNA.3490\n",
      "SPAC26A3.11\n",
      "SPAC8E11.07c\n",
      "SPAC8E11.07c\n",
      "SPNCRNA.3507\n",
      "SPNCRNA.905\n",
      "SPNCRNA.905\n",
      "SPNCRNA.3512\n",
      "SPNCRNA.3512\n",
      "SPNCRNA.3513\n",
      "SPSNORNA.20\n",
      "SPNCRNA.3514\n",
      "SPAC959.06c\n",
      "SPNCRNA.3514\n",
      "SPNCRNA.3515\n",
      "SPNCRNA.3517\n",
      "SPNCRNA.3519\n",
      "SPNCRNA.3519\n",
      "SPNCRNA.3520\n",
      "SPNCRNA.907\n",
      "SPNCRNA.3520\n",
      "SPNCRNA.907\n",
      "SPAC3H5.12c\n",
      "SPAC3H5.13\n",
      "SPAC3H5.13\n",
      "SPNCRNA.3523\n",
      "SPNCRNA.3524\n",
      "SPNCRNA.3525\n",
      "SPNCRNA.3526\n",
      "SPNCRNA.3531\n",
      "SPNCRNA.3531\n",
      "SPNCRNA.3531\n",
      "SPNCRNA.3531\n",
      "SPNCRNA.3531\n",
      "SPNCRNA.3531\n",
      "SPAC3H5.07\n",
      "SPNCRNA.3532\n",
      "SPNCRNA.3532\n",
      "SPNCRNA.3532\n",
      "SPNCRNA.3535\n",
      "SPNCRNA.3535\n",
      "SPNCRNA.3535\n",
      "SPATRNAARG.01\n",
      "SPATRNAARG.01\n",
      "SPNCRNA.3537\n",
      "SPATRNAARG.01\n",
      "SPATRNAARG.01\n",
      "SPNCRNA.991\n",
      "SPNCRNA.4017\n",
      "SPAC17C9.09c\n",
      "SPAC17C9.08\n",
      "SPNCRNA.4018\n",
      "SPNCRNA.4019\n",
      "SPAC17C9.08\n",
      "SPNCRNA.4018\n",
      "SPNCRNA.4019\n",
      "SPNCRNA.4021\n",
      "SPNCRNA.4022\n",
      "SPNCRNA.4021\n",
      "SPNCRNA.4022\n",
      "SPNCRNA.4021\n",
      "SPNCRNA.4022\n",
      "SPNCRNA.4023\n",
      "SPNCRNA.4027\n",
      "SPNCRNA.4028\n",
      "SPNCRNA.4028\n",
      "SPNCRNA.4028\n",
      "SPNCRNA.4028\n",
      "SPNCRNA.992\n",
      "SPNCRNA.992\n",
      "SPNCRNA.4029\n",
      "SPNCRNA.4030\n",
      "SPNCRNA.4029\n",
      "SPNCRNA.4030\n",
      "SPNCRNA.4031\n",
      "SPNCRNA.4032\n",
      "SPNCRNA.4031\n",
      "SPNCRNA.4032\n",
      "SPNCRNA.4031\n",
      "SPNCRNA.4032\n",
      "SPNCRNA.4032\n",
      "SPNCRNA.130\n",
      "SPNCRNA.4034\n",
      "SPNCRNA.4035\n",
      "SPNCRNA.993\n",
      "SPNCRNA.4040\n",
      "SPAC27D7.11c\n",
      "SPNCRNA.4049\n",
      "SPNCRNA.995\n",
      "SPNCRNA.996\n",
      "SPNCRNA.995\n",
      "SPNCRNA.996\n",
      "SPNCRNA.4051\n",
      "SPAC27D7.13c\n",
      "SPNCRNA.4051\n",
      "SPNCRNA.4052\n",
      "SPNCRNA.4055\n",
      "SPAC637.05c\n",
      "SPAC637.05c\n",
      "SPNCRNA.4056\n",
      "SPNCRNA.4057\n",
      "SPAC637.06\n",
      "SPNCRNA.4057\n",
      "SPNCRNA.4057\n",
      "SPNCRNA.4057\n",
      "SPNCRNA.563\n",
      "SPNCRNA.998\n",
      "SPNCRNA.998\n",
      "SPAC637.13c\n",
      "SPNCRNA.249\n",
      "SPNCRNA.4059\n",
      "SPAC12B10.04\n",
      "SPAC12B10.04\n",
      "SPNCRNA.1001\n",
      "SPNCRNA.4064\n",
      "SPNCRNA.4064\n",
      "SPNCRNA.4066\n"
     ]
    }
   ],
   "source": [
    "#create column, whole gene as codons\n",
    "result_final__['wt_gene'] = np.nan\n",
    "#iterate through result_final__ positions. If object is CDS, then look up the sequence from start to end in regions.\n",
    "#add the sequence to the df\n",
    "for i in range(len(result_final__)):\n",
    "    \n",
    "    if result_final__['object'][i] == 'CDS':\n",
    "    \n",
    "        start = result_final__['start'][i]\n",
    "        end = result_final__['end'][i]\n",
    "        chrom = result_final__['chromosome'][i]\n",
    "        pos = result_final__['position'][i]\n",
    "        gene = regions[chrom][start-1:end]\n",
    "        #gene_exons_final where start is the same as start in result_final__\n",
    "        exon = gene_exons_final[(gene_exons_final['start'] < start+20) & (gene_exons_final['end'] > end-20)] \n",
    "        #if exon is empty, then print gene name\n",
    "        if exon.empty:\n",
    "            print(result_final__['ID'][i])\n",
    "            continue\n",
    "        exon_len_so_far=0\n",
    "        exon_len_so_far = exon['CDS_length_so_far'].values[0]\n",
    "        exon_len_so_far=exon_len_so_far%3\n",
    "        if pos > start and pos < end:\n",
    "        \n",
    "            if result_final__['strand'][i] == '-':\n",
    "                \n",
    "                gene = gene[::-1]\n",
    "            \n",
    "                gene = ''.join(gene)\n",
    "            gene = 'A'*int(exon_len_so_far) + gene\n",
    "            #write the gene as triplets of letters like AAA, ATC, etc\n",
    "            #if divisible by 3\n",
    "            if len(gene)%3 != 0:\n",
    "                gene = gene[:-(len(gene)%3)]\n",
    "            gene = [gene[i:i+3] for i in range(0, len(gene), 3)]\n",
    "                #only triplets near the position\n",
    "            if result_final__['strand'][i] == '+':\n",
    "                gene = gene[(pos-(start-int(exon_len_so_far)))//3-2:(pos-end)//3+25]\n",
    "            else: \n",
    "                gene = gene[(end-pos)//3-2:(start-int(exon_len_so_far)-pos)//3+25]\n",
    "            \n",
    "                \n",
    "                \n",
    "\n",
    "\n",
    "            result_final__['wt_gene'][i] = gene\n",
    "\n",
    "result_final__['mutant allele'] = np.nan\n",
    "for i in range(len(result_final__)):\n",
    "    if result_final__['object'][i] == 'CDS':\n",
    "    \n",
    "        start = result_final__['start'][i]\n",
    "        end = result_final__['end'][i]\n",
    "        chrom = result_final__['chromosome'][i]\n",
    "        pos = result_final__['position'][i]\n",
    "        gene = regions[chrom][start-1:end]\n",
    "         #gene_exons_final where start is the same as start in result_final__\n",
    "        exon = gene_exons_final[(gene_exons_final['start'] < start+20) & (gene_exons_final['end'] > end-20)] \n",
    "        #if exon is empty, then print gene name\n",
    "        if exon.empty:\n",
    "            print(result_final__['ID'][i])\n",
    "            continue\n",
    "        exon_len_so_far=0\n",
    "        exon_len_so_far = exon['CDS_length_so_far'].values[0]\n",
    "        exon_len_so_far=exon_len_so_far%3\n",
    "        if pos > start and pos < end:\n",
    "        \n",
    "            if result_final__['strand'][i] == '+':\n",
    "                #decide which allele to use\n",
    "                #check \n",
    "                #check how long the y0036 allele is \n",
    "                x968_len = len(result_final__['X968_allele'][i])\n",
    "                y0036_len = len(result_final__['Y0036_allele'][i])\n",
    "                \n",
    "                mutation = result_final__['Y0036_allele'][i]\n",
    "               \n",
    "\n",
    "                gene = gene[:pos-start] + mutation + gene[pos-start+x968_len:]\n",
    "            elif result_final__['strand'][i] == '-':\n",
    "                x968_len = len(result_final__['X968_allele'][i])\n",
    "                y0036_len = len(result_final__['Y0036_allele'][i])\n",
    "                gene = gene[::-1]\n",
    "                gene = ''.join(gene)\n",
    "                mutation = result_final__['Y0036_allele'][i]\n",
    "                \n",
    "                gene = gene[:end-pos] + mutation + gene[end-pos+x968_len:]\n",
    "                \n",
    "            gene = 'A'*int(exon_len_so_far) + gene\n",
    "            if len(gene)%3 != 0:\n",
    "                gene = gene[:-(len(gene)%3)]\n",
    "            gene = [gene[k:k+3] for k in range(0, len(gene), 3)]\n",
    "                #only triplets near the position\n",
    "            if result_final__['strand'][i] == '+':\n",
    "                gene = gene[(pos-(start-int(exon_len_so_far)))//3-2:(pos-end)//3+25]\n",
    "            else:\n",
    "                gene = gene[(end-pos)//3-2:(start-int(exon_len_so_far)-pos)//3+25]\n",
    "            result_final__[\"mutant allele\"][i] = gene\n",
    "                \n",
    "           \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>position</th>\n",
       "      <th>ID</th>\n",
       "      <th>Parent</th>\n",
       "      <th>object</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>detail</th>\n",
       "      <th>p_value</th>\n",
       "      <th>correlation</th>\n",
       "      <th>p_value_corr</th>\n",
       "      <th>gene</th>\n",
       "      <th>X968_allele</th>\n",
       "      <th>Y0036_allele</th>\n",
       "      <th>reference_seq</th>\n",
       "      <th>reference_seq_</th>\n",
       "      <th>wt_gene</th>\n",
       "      <th>mutant allele</th>\n",
       "      <th>stop_codon_predicted</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2612379</td>\n",
       "      <td>SPNCRNA.3168</td>\n",
       "      <td>SPNCRNA.3168.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>2611788</td>\n",
       "      <td>2612409</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>TAC |G| GTA</td>\n",
       "      <td>TACGGTA</td>\n",
       "      <td>[ACT, TAC, GGT, AAT, ATT, CCT, CCT, TCG, CTA, ...</td>\n",
       "      <td>[ACT, TAC, AGT, AAT, ATT, CCT, CCT, TCG, CTA, ...</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>2648315</td>\n",
       "      <td>SPNCRNA.3183</td>\n",
       "      <td>SPNCRNA.3183.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>2647955</td>\n",
       "      <td>2648490</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>GTT |T| ACT</td>\n",
       "      <td>GTTTACT</td>\n",
       "      <td>[TTA, GGT, TTA, CTA, ACA, GCC, GAT, GAT, TTT, ...</td>\n",
       "      <td>[TTA, GGT, TCA, CTA, ACA, GCC, GAT, GAT, TTT, ...</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>2658731</td>\n",
       "      <td>SPNCRNA.3188</td>\n",
       "      <td>SPNCRNA.3188.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>2658334</td>\n",
       "      <td>2658989</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>TTC |G| TTA</td>\n",
       "      <td>TTCGTTA</td>\n",
       "      <td>[ACA, TTT, CGT, TAT, GTG, ACG, TAC, CAG, AAT, ...</td>\n",
       "      <td>[ACA, TTT, CAT, TAT, GTG, ACG, TAC, CAG, AAT, ...</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>2766755</td>\n",
       "      <td>SPNCRNA.3243</td>\n",
       "      <td>SPNCRNA.3243.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>2766609</td>\n",
       "      <td>2767122</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>ATC |T| AAA</td>\n",
       "      <td>ATCTAAA</td>\n",
       "      <td>[ACA, TAT, CTA, AAA, GTT, TTT, CCT, TCT, GTC, ...</td>\n",
       "      <td>[ACA, TAT, CCA, AAA, GTT, TTT, CCT, TCT, GTC, ...</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>2779113</td>\n",
       "      <td>SPNCRNA.3247</td>\n",
       "      <td>SPNCRNA.3247.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>2778732</td>\n",
       "      <td>2779145</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>GAT |T| GTC</td>\n",
       "      <td>GATTGTC</td>\n",
       "      <td>[TTT, ATA, GAT, TGT, CTT, AAC, GTA, AAA, AGC, ...</td>\n",
       "      <td>[TTT, ATA, GAT, CGT, CTT, AAC, GTA, AAA, AGC, ...</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>2905076</td>\n",
       "      <td>SPNCRNA.3311</td>\n",
       "      <td>SPNCRNA.3311.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>2903507</td>\n",
       "      <td>2905392</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>AAA |C| AAG</td>\n",
       "      <td>AAACAAG</td>\n",
       "      <td>[AGC, AAA, CAA, GTC, CTG, CAA, AGA, TAC, ATT, ...</td>\n",
       "      <td>[AGC, AAA, TAA, GTC, CTG, CAA, AGA, TAC, ATT, ...</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>3092232</td>\n",
       "      <td>SPNCRNA.3385</td>\n",
       "      <td>SPNCRNA.3385.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3091339</td>\n",
       "      <td>3092431</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>GAG |G| AGA</td>\n",
       "      <td>GAGGAGA</td>\n",
       "      <td>[CTG, TAG, AGG, AGA, ATT, AGA, AGT, ATT, ACC, ...</td>\n",
       "      <td>[CTG, TAG, AGA, AGA, ATT, AGA, AGT, ATT, ACC, ...</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>3202214</td>\n",
       "      <td>SPNCRNA.3430</td>\n",
       "      <td>SPNCRNA.3430.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3201501</td>\n",
       "      <td>3202270</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>GAT |T| TTC</td>\n",
       "      <td>GATTTTC</td>\n",
       "      <td>[ATG, GAA, GAT, TTT, CTA, GGT, TTA, TTA, GCC, ...</td>\n",
       "      <td>[ATG, GAA, GAT, CTT, CTA, GGT, TTA, TTA, GCC, ...</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122</th>\n",
       "      <td>3292079</td>\n",
       "      <td>SPNCRNA.3473</td>\n",
       "      <td>SPNCRNA.3473.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3291576</td>\n",
       "      <td>3292459</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>CAC |T| TCC</td>\n",
       "      <td>CACTTCC</td>\n",
       "      <td>[CTC, ACC, ACT, TCC, GAA, AAT, GTT, CCA, AAC, ...</td>\n",
       "      <td>[CTC, ACC, ACC, TCC, GAA, AAT, GTT, CCA, AAC, ...</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>3402585</td>\n",
       "      <td>SPNCRNA.3518</td>\n",
       "      <td>SPNCRNA.3518.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3402491</td>\n",
       "      <td>3402911</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>CGC |G| CAA</td>\n",
       "      <td>CGCGCAA</td>\n",
       "      <td>[TTA, CGC, GCA, AGA, TGT, TCT, AGT, ATT, TGA, ...</td>\n",
       "      <td>[TTA, CGC, ACA, AGA, TGT, TCT, AGT, ATT, TGA, ...</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156</th>\n",
       "      <td>3419862</td>\n",
       "      <td>SPNCRNA.3528</td>\n",
       "      <td>SPNCRNA.3528.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3419306</td>\n",
       "      <td>3420208</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>GCT |T| TGC</td>\n",
       "      <td>GCTTTGC</td>\n",
       "      <td>[GAT, GGC, TTT, GCA, AAA, AAT, CCT, TGC, TTC, ...</td>\n",
       "      <td>[GAT, GGC, TGT, GCA, AAA, AAT, CCT, TGC, TTC, ...</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>168</th>\n",
       "      <td>3440649</td>\n",
       "      <td>SPNCRNA.561</td>\n",
       "      <td>SPNCRNA.561.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3440092</td>\n",
       "      <td>3441229</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.187879</td>\n",
       "      <td>2.164940e-01</td>\n",
       "      <td>pkl1-antisense-1</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>TGA |G| CTA</td>\n",
       "      <td>TGAGCTA</td>\n",
       "      <td>[GTT, TCT, GAG, CTA, GTT, TCC, TTT, TAA, GTA, ...</td>\n",
       "      <td>[GTT, TCT, GAA, CTA, GTT, TCC, TTT, TAA, GTA, ...</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>171</th>\n",
       "      <td>4473996</td>\n",
       "      <td>SPAC17C9.16c</td>\n",
       "      <td>SPAC17C9.16c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>4473325</td>\n",
       "      <td>4474259</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.678393</td>\n",
       "      <td>3.042498e-07</td>\n",
       "      <td>mfs1</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>TTC |C| AAA</td>\n",
       "      <td>TTCCAAA</td>\n",
       "      <td>[TTG, TTC, CAA, AGT, ATG, TTT, CAA, TAT, TTG, ...</td>\n",
       "      <td>[TTG, TTC, TAA, AGT, ATG, TTT, CAA, TAT, TTG, ...</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184</th>\n",
       "      <td>4534144</td>\n",
       "      <td>SPNCRNA.4050</td>\n",
       "      <td>SPNCRNA.4050.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>4533502</td>\n",
       "      <td>4534252</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>GCTT</td>\n",
       "      <td>AGA |G| CTT</td>\n",
       "      <td>AGAGCTT</td>\n",
       "      <td>[CGT, AGA, GCT, TCT, TCT, TTT, TTT, TGT, AGA, ...</td>\n",
       "      <td>[CGT, AGA, GCT, TCT, TCT, TCT, TTT, TTT, TGT, ...</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>196</th>\n",
       "      <td>4573241</td>\n",
       "      <td>SPNCRNA.4060</td>\n",
       "      <td>SPNCRNA.4060.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>4572682</td>\n",
       "      <td>4574887</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>GAA |C| ATA</td>\n",
       "      <td>GAACATA</td>\n",
       "      <td>[AAT, TGA, ACA, TAA, CAT, TTT, CAA, TGC, CGT, ...</td>\n",
       "      <td>[AAT, TGA, ATA, TAA, CAT, TTT, CAA, TGC, CGT, ...</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198</th>\n",
       "      <td>4582389</td>\n",
       "      <td>SPAC12B10.05</td>\n",
       "      <td>SPAC12B10.05.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>4582378</td>\n",
       "      <td>4582549</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.194993</td>\n",
       "      <td>1.992787e-01</td>\n",
       "      <td>icp55</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>TTA |T| ATC</td>\n",
       "      <td>TTATATC</td>\n",
       "      <td>[AGC, GGT, TAT, ATC, CGG, ACT, CTC, TTT, ATT, ...</td>\n",
       "      <td>[AGC, GGT, TAA, ATC, CGG, ACT, CTC, TTT, ATT, ...</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>204</th>\n",
       "      <td>4602196</td>\n",
       "      <td>SPNCRNA.4071</td>\n",
       "      <td>SPNCRNA.4071.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>4602079</td>\n",
       "      <td>4602303</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>ACT |G| CGT</td>\n",
       "      <td>ACTGCGT</td>\n",
       "      <td>[AGC, ACT, GCG, TAT, TAA, AGA, TGC, ATT, AGC, ...</td>\n",
       "      <td>[AGC, ACT, TCG, TAT, TAA, AGA, TGC, ATT, AGC, ...</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     position            ID          Parent object chromosome    start  \\\n",
       "3     2612379  SPNCRNA.3168  SPNCRNA.3168.1    CDS          I  2611788   \n",
       "14    2648315  SPNCRNA.3183  SPNCRNA.3183.1    CDS          I  2647955   \n",
       "18    2658731  SPNCRNA.3188  SPNCRNA.3188.1    CDS          I  2658334   \n",
       "40    2766755  SPNCRNA.3243  SPNCRNA.3243.1    CDS          I  2766609   \n",
       "44    2779113  SPNCRNA.3247  SPNCRNA.3247.1    CDS          I  2778732   \n",
       "51    2905076  SPNCRNA.3311  SPNCRNA.3311.1    CDS          I  2903507   \n",
       "82    3092232  SPNCRNA.3385  SPNCRNA.3385.1    CDS          I  3091339   \n",
       "110   3202214  SPNCRNA.3430  SPNCRNA.3430.1    CDS          I  3201501   \n",
       "122   3292079  SPNCRNA.3473  SPNCRNA.3473.1    CDS          I  3291576   \n",
       "149   3402585  SPNCRNA.3518  SPNCRNA.3518.1    CDS          I  3402491   \n",
       "156   3419862  SPNCRNA.3528  SPNCRNA.3528.1    CDS          I  3419306   \n",
       "168   3440649   SPNCRNA.561   SPNCRNA.561.1    CDS          I  3440092   \n",
       "171   4473996  SPAC17C9.16c  SPAC17C9.16c.1    CDS          I  4473325   \n",
       "184   4534144  SPNCRNA.4050  SPNCRNA.4050.1    CDS          I  4533502   \n",
       "196   4573241  SPNCRNA.4060  SPNCRNA.4060.1    CDS          I  4572682   \n",
       "198   4582389  SPAC12B10.05  SPAC12B10.05.1    CDS          I  4582378   \n",
       "204   4602196  SPNCRNA.4071  SPNCRNA.4071.1    CDS          I  4602079   \n",
       "\n",
       "         end strand detail  p_value  correlation  p_value_corr  \\\n",
       "3    2612409      -   exon      1.0          NaN           NaN   \n",
       "14   2648490      -   exon      1.0          NaN           NaN   \n",
       "18   2658989      +   exon      1.0          NaN           NaN   \n",
       "40   2767122      -   exon      1.0          NaN           NaN   \n",
       "44   2779145      -   exon      1.0          NaN           NaN   \n",
       "51   2905392      +   exon      1.0          NaN           NaN   \n",
       "82   3092431      +   exon      1.0          NaN           NaN   \n",
       "110  3202270      -   exon      1.0          NaN           NaN   \n",
       "122  3292459      +   exon      1.0          NaN           NaN   \n",
       "149  3402911      +   exon      1.0          NaN           NaN   \n",
       "156  3420208      -   exon      1.0          NaN           NaN   \n",
       "168  3441229      -   exon      1.0     0.187879  2.164940e-01   \n",
       "171  4474259      +   exon      1.0     0.678393  3.042498e-07   \n",
       "184  4534252      +   exon      1.0          NaN           NaN   \n",
       "196  4574887      +   exon      1.0          NaN           NaN   \n",
       "198  4582549      +   exon      1.0     0.194993  1.992787e-01   \n",
       "204  4602303      +   exon      1.0          NaN           NaN   \n",
       "\n",
       "                 gene X968_allele Y0036_allele reference_seq reference_seq_  \\\n",
       "3                 NaN           G            A   TAC |G| GTA        TACGGTA   \n",
       "14                NaN           T            C   GTT |T| ACT        GTTTACT   \n",
       "18                NaN           G            A   TTC |G| TTA        TTCGTTA   \n",
       "40                NaN           T            C   ATC |T| AAA        ATCTAAA   \n",
       "44                NaN           T            C   GAT |T| GTC        GATTGTC   \n",
       "51                NaN           C            T   AAA |C| AAG        AAACAAG   \n",
       "82                NaN           G            A   GAG |G| AGA        GAGGAGA   \n",
       "110               NaN           T            C   GAT |T| TTC        GATTTTC   \n",
       "122               NaN           T            C   CAC |T| TCC        CACTTCC   \n",
       "149               NaN           G            A   CGC |G| CAA        CGCGCAA   \n",
       "156               NaN           T            G   GCT |T| TGC        GCTTTGC   \n",
       "168  pkl1-antisense-1           G            A   TGA |G| CTA        TGAGCTA   \n",
       "171              mfs1           C            T   TTC |C| AAA        TTCCAAA   \n",
       "184               NaN           G         GCTT   AGA |G| CTT        AGAGCTT   \n",
       "196               NaN           C            T   GAA |C| ATA        GAACATA   \n",
       "198             icp55           T            A   TTA |T| ATC        TTATATC   \n",
       "204               NaN           G            T   ACT |G| CGT        ACTGCGT   \n",
       "\n",
       "                                               wt_gene  \\\n",
       "3    [ACT, TAC, GGT, AAT, ATT, CCT, CCT, TCG, CTA, ...   \n",
       "14   [TTA, GGT, TTA, CTA, ACA, GCC, GAT, GAT, TTT, ...   \n",
       "18   [ACA, TTT, CGT, TAT, GTG, ACG, TAC, CAG, AAT, ...   \n",
       "40   [ACA, TAT, CTA, AAA, GTT, TTT, CCT, TCT, GTC, ...   \n",
       "44   [TTT, ATA, GAT, TGT, CTT, AAC, GTA, AAA, AGC, ...   \n",
       "51   [AGC, AAA, CAA, GTC, CTG, CAA, AGA, TAC, ATT, ...   \n",
       "82   [CTG, TAG, AGG, AGA, ATT, AGA, AGT, ATT, ACC, ...   \n",
       "110  [ATG, GAA, GAT, TTT, CTA, GGT, TTA, TTA, GCC, ...   \n",
       "122  [CTC, ACC, ACT, TCC, GAA, AAT, GTT, CCA, AAC, ...   \n",
       "149  [TTA, CGC, GCA, AGA, TGT, TCT, AGT, ATT, TGA, ...   \n",
       "156  [GAT, GGC, TTT, GCA, AAA, AAT, CCT, TGC, TTC, ...   \n",
       "168  [GTT, TCT, GAG, CTA, GTT, TCC, TTT, TAA, GTA, ...   \n",
       "171  [TTG, TTC, CAA, AGT, ATG, TTT, CAA, TAT, TTG, ...   \n",
       "184  [CGT, AGA, GCT, TCT, TCT, TTT, TTT, TGT, AGA, ...   \n",
       "196  [AAT, TGA, ACA, TAA, CAT, TTT, CAA, TGC, CGT, ...   \n",
       "198  [AGC, GGT, TAT, ATC, CGG, ACT, CTC, TTT, ATT, ...   \n",
       "204  [AGC, ACT, GCG, TAT, TAA, AGA, TGC, ATT, AGC, ...   \n",
       "\n",
       "                                         mutant allele stop_codon_predicted  \n",
       "3    [ACT, TAC, AGT, AAT, ATT, CCT, CCT, TCG, CTA, ...                  yes  \n",
       "14   [TTA, GGT, TCA, CTA, ACA, GCC, GAT, GAT, TTT, ...                  yes  \n",
       "18   [ACA, TTT, CAT, TAT, GTG, ACG, TAC, CAG, AAT, ...                  yes  \n",
       "40   [ACA, TAT, CCA, AAA, GTT, TTT, CCT, TCT, GTC, ...                  yes  \n",
       "44   [TTT, ATA, GAT, CGT, CTT, AAC, GTA, AAA, AGC, ...                  yes  \n",
       "51   [AGC, AAA, TAA, GTC, CTG, CAA, AGA, TAC, ATT, ...                  yes  \n",
       "82   [CTG, TAG, AGA, AGA, ATT, AGA, AGT, ATT, ACC, ...                  yes  \n",
       "110  [ATG, GAA, GAT, CTT, CTA, GGT, TTA, TTA, GCC, ...                  yes  \n",
       "122  [CTC, ACC, ACC, TCC, GAA, AAT, GTT, CCA, AAC, ...                  yes  \n",
       "149  [TTA, CGC, ACA, AGA, TGT, TCT, AGT, ATT, TGA, ...                  yes  \n",
       "156  [GAT, GGC, TGT, GCA, AAA, AAT, CCT, TGC, TTC, ...                  yes  \n",
       "168  [GTT, TCT, GAA, CTA, GTT, TCC, TTT, TAA, GTA, ...                  yes  \n",
       "171  [TTG, TTC, TAA, AGT, ATG, TTT, CAA, TAT, TTG, ...                  yes  \n",
       "184  [CGT, AGA, GCT, TCT, TCT, TCT, TTT, TTT, TGT, ...                  yes  \n",
       "196  [AAT, TGA, ATA, TAA, CAT, TTT, CAA, TGC, CGT, ...                  yes  \n",
       "198  [AGC, GGT, TAA, ATC, CGG, ACT, CTC, TTT, ATT, ...                  yes  \n",
       "204  [AGC, ACT, TCG, TAT, TAA, AGA, TGC, ATT, AGC, ...                  yes  "
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#CreATE A DF WHERE WHOLE GENE IS NOT NAN\n",
    "stop_codon_df = result_final__[result_final__['wt_gene'].notna()]\n",
    "stop_codon_df[\"stop_codon_predicted\"] = np.nan\n",
    "#iterate through mutant allele, and see whether it contains\n",
    "#reset index\n",
    "stop_codon_df = stop_codon_df.reset_index(drop=True)\n",
    "\n",
    "for i in range(len(stop_codon_df)):\n",
    "    for j in range(len(stop_codon_df['mutant allele'][i])):\n",
    "        if stop_codon_df[\"strand\"][i] == '+':\n",
    "            if stop_codon_df['mutant allele'][i][j] == 'TAA' or stop_codon_df['mutant allele'][i][j] == 'TAG' or stop_codon_df['mutant allele'][i][j] == 'TGA':\n",
    "                stop_codon_df['stop_codon_predicted'][i] = 'yes'\n",
    "        else:\n",
    "            if stop_codon_df['mutant allele'][i][j] == 'ATT' or stop_codon_df['mutant allele'][i][j] == 'ATC' or stop_codon_df['mutant allele'][i][j] == 'ACT':\n",
    "                stop_codon_df['stop_codon_predicted'][i] = 'yes'\n",
    "#show which entries have a stop codon predicted\n",
    "stop_codon_df[stop_codon_df['stop_codon_predicted'] == 'yes']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "#in stop_codon_df create a column \"possible frameshift\" and fill it with nan. Then loop through X96_allele and Y0036_allele and check if the length is different. If yes and the difference is not divisible by 3, fill the column with \"yes\". Then show the entries where the column is \"yes\"\n",
    "frameshift_df = stop_codon_df.copy()\n",
    "frameshift_df['possible frameshift'] = np.nan\n",
    "\n",
    "for i in range(len(frameshift_df)):\n",
    "    if len(frameshift_df['X968_allele'][i]) != len(frameshift_df['Y0036_allele'][i]):\n",
    "        if (len(frameshift_df['X968_allele'][i]) - len(frameshift_df['Y0036_allele'][i]))%3 != 0:\n",
    "            frameshift_df['possible frameshift'][i] = 'yes'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "#import a codon table as a dictionary\n",
    "codon_table = {\n",
    "    'TTT': 'F', 'TTC': 'F', 'TTA': 'L', 'TTG': 'L',\n",
    "    'TCT': 'S', 'TCC': 'S', 'TCA': 'S', 'TCG': 'S',\n",
    "    'TAT': 'Y', 'TAC': 'Y', 'TAA': '*', 'TAG': '*',\n",
    "    'TGT': 'C', 'TGC': 'C', 'TGA': '*', 'TGG': 'W',\n",
    "    'CTT': 'L', 'CTC': 'L', 'CTA': 'L', 'CTG': 'L',\n",
    "    'CCT': 'P', 'CCC': 'P', 'CCA': 'P', 'CCG': 'P',\n",
    "    'CAT': 'H', 'CAC': 'H', 'CAA': 'Q', 'CAG': 'Q', \n",
    "    'CGT': 'R', 'CGC': 'R', 'CGA': 'R', 'CGG': 'R',\n",
    "    'ATT': 'I', 'ATC': 'I', 'ATA': 'I', 'ATG': 'M',\n",
    "    'ACT': 'T', 'ACC': 'T', 'ACA': 'T', 'ACG': 'T',\n",
    "    'AAT': 'N', 'AAC': 'N', 'AAA': 'K', 'AAG': 'K',\n",
    "    'AGT': 'S', 'AGC': 'S', 'AGA': 'R', 'AGG': 'R',\n",
    "    'GTT': 'V', 'GTC': 'V', 'GTA': 'V', 'GTG': 'V',\n",
    "    'GCT': 'A', 'GCC': 'A', 'GCA': 'A', 'GCG': 'A',\n",
    "    'GAT': 'D', 'GAC': 'D', 'GAA': 'E', 'GAG': 'E',\n",
    "    'GGT': 'G', 'GGC': 'G', 'GGA': 'G', 'GGG': 'G',\n",
    "}\n",
    "#iterate through frameshift_df and trnslate the lists of triplets found in wt_gene and mutant allele\n",
    "frameshift_df['wt_gene_translated'] = np.nan\n",
    "frameshift_df['mutant allele_translated'] = np.nan\n",
    "for i in range(len(frameshift_df)):\n",
    "    #if strand is -, then change a to t and so on\n",
    "    if frameshift_df['strand'][i] == '-':\n",
    "        temp_wt_allele = frameshift_df['wt_gene'][i]\n",
    "        temp_mut_allele = frameshift_df['mutant allele'][i]\n",
    "        for j in range(len(temp_wt_allele)):\n",
    "            temp_wt_allele[j] = temp_wt_allele[j].replace('A','t')\n",
    "            temp_wt_allele[j] = temp_wt_allele[j].replace('T','a')\n",
    "            temp_wt_allele[j] = temp_wt_allele[j].replace('C','g')\n",
    "            temp_wt_allele[j] = temp_wt_allele[j].replace('G','c')\n",
    "            temp_wt_allele[j] = temp_wt_allele[j].upper()\n",
    "        for j in range(len(temp_mut_allele)):\n",
    "            temp_mut_allele[j] = temp_mut_allele[j].replace('A','t')\n",
    "            temp_mut_allele[j] = temp_mut_allele[j].replace('T','a')\n",
    "            temp_mut_allele[j] = temp_mut_allele[j].replace('C','g')\n",
    "            temp_mut_allele[j] = temp_mut_allele[j].replace('G','c')\n",
    "            temp_mut_allele[j] = temp_mut_allele[j].upper()\n",
    "        frameshift_df['wt_gene_translated'][i] = ''.join([codon_table[codon] for codon in temp_wt_allele])\n",
    "        frameshift_df['mutant allele_translated'][i] = ''.join([codon_table[codon] for codon in temp_mut_allele])\n",
    "    else:\n",
    "        frameshift_df['wt_gene_translated'][i] = ''.join([codon_table[codon] for codon in frameshift_df['wt_gene'][i]])\n",
    "        frameshift_df['mutant allele_translated'][i] = ''.join([codon_table[codon] for codon in frameshift_df['mutant allele'][i]])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "91"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#count how many times wt_gene_translated is not equal to mutant allele_translated when ID doesnt contain SPNC\n",
    "frameshift_df[(frameshift_df['ID'].str.contains('SPNC') == False) & (frameshift_df['wt_gene_translated'] != frameshift_df['mutant allele_translated'])].shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>position</th>\n",
       "      <th>ID</th>\n",
       "      <th>Parent</th>\n",
       "      <th>object</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>detail</th>\n",
       "      <th>p_value</th>\n",
       "      <th>correlation</th>\n",
       "      <th>p_value_corr</th>\n",
       "      <th>gene</th>\n",
       "      <th>X968_allele</th>\n",
       "      <th>Y0036_allele</th>\n",
       "      <th>reference_seq</th>\n",
       "      <th>reference_seq_</th>\n",
       "      <th>wt_gene</th>\n",
       "      <th>mutant allele</th>\n",
       "      <th>stop_codon_predicted</th>\n",
       "      <th>possible frameshift</th>\n",
       "      <th>wt_gene_translated</th>\n",
       "      <th>mutant allele_translated</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>171</th>\n",
       "      <td>4473996</td>\n",
       "      <td>SPAC17C9.16c</td>\n",
       "      <td>SPAC17C9.16c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>4473325</td>\n",
       "      <td>4474259</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.678393</td>\n",
       "      <td>3.042498e-07</td>\n",
       "      <td>mfs1</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>TTC |C| AAA</td>\n",
       "      <td>TTCCAAA</td>\n",
       "      <td>[TTG, TTC, CAA, AGT, ATG, TTT, CAA, TAT, TTG, ...</td>\n",
       "      <td>[TTG, TTC, TAA, AGT, ATG, TTT, CAA, TAT, TTG, ...</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LFQSMFQYLAAAYPKYVASVFAGNALF</td>\n",
       "      <td>LF*SMFQYLAAAYPKYVASVFAGNALF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198</th>\n",
       "      <td>4582389</td>\n",
       "      <td>SPAC12B10.05</td>\n",
       "      <td>SPAC12B10.05.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>4582378</td>\n",
       "      <td>4582549</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.194993</td>\n",
       "      <td>1.992787e-01</td>\n",
       "      <td>icp55</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>TTA |T| ATC</td>\n",
       "      <td>TTATATC</td>\n",
       "      <td>[AGC, GGT, TAT, ATC, CGG, ACT, CTC, TTT, ATT, ...</td>\n",
       "      <td>[AGC, GGT, TAA, ATC, CGG, ACT, CTC, TTT, ATT, ...</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SGYIRTLFIRNRFSNYRLRSQIIKYKY</td>\n",
       "      <td>SG*IRTLFIRNRFSNYRLRSQIIKYKY</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     position            ID          Parent object chromosome    start  \\\n",
       "171   4473996  SPAC17C9.16c  SPAC17C9.16c.1    CDS          I  4473325   \n",
       "198   4582389  SPAC12B10.05  SPAC12B10.05.1    CDS          I  4582378   \n",
       "\n",
       "         end strand detail  p_value  correlation  p_value_corr   gene  \\\n",
       "171  4474259      +   exon      1.0     0.678393  3.042498e-07   mfs1   \n",
       "198  4582549      +   exon      1.0     0.194993  1.992787e-01  icp55   \n",
       "\n",
       "    X968_allele Y0036_allele reference_seq reference_seq_  \\\n",
       "171           C            T   TTC |C| AAA        TTCCAAA   \n",
       "198           T            A   TTA |T| ATC        TTATATC   \n",
       "\n",
       "                                               wt_gene  \\\n",
       "171  [TTG, TTC, CAA, AGT, ATG, TTT, CAA, TAT, TTG, ...   \n",
       "198  [AGC, GGT, TAT, ATC, CGG, ACT, CTC, TTT, ATT, ...   \n",
       "\n",
       "                                         mutant allele stop_codon_predicted  \\\n",
       "171  [TTG, TTC, TAA, AGT, ATG, TTT, CAA, TAT, TTG, ...                  yes   \n",
       "198  [AGC, GGT, TAA, ATC, CGG, ACT, CTC, TTT, ATT, ...                  yes   \n",
       "\n",
       "     possible frameshift           wt_gene_translated  \\\n",
       "171                  NaN  LFQSMFQYLAAAYPKYVASVFAGNALF   \n",
       "198                  NaN  SGYIRTLFIRNRFSNYRLRSQIIKYKY   \n",
       "\n",
       "        mutant allele_translated  \n",
       "171  LF*SMFQYLAAAYPKYVASVFAGNALF  \n",
       "198  SG*IRTLFIRNRFSNYRLRSQIIKYKY  "
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#show where gene doesnt start with SPNC and possible frameshift\n",
    "#show all rows and stop_codon_predicted\n",
    "pd.set_option('display.max_rows', None)\n",
    "frameshift_df[(frameshift_df['ID'].str.contains('SPNC') == False) & (frameshift_df['stop_codon_predicted'] == 'yes')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>position</th>\n",
       "      <th>ID</th>\n",
       "      <th>Parent</th>\n",
       "      <th>object</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>detail</th>\n",
       "      <th>p_value</th>\n",
       "      <th>correlation</th>\n",
       "      <th>p_value_corr</th>\n",
       "      <th>gene</th>\n",
       "      <th>X968_allele</th>\n",
       "      <th>Y0036_allele</th>\n",
       "      <th>reference_seq</th>\n",
       "      <th>reference_seq_</th>\n",
       "      <th>wt_gene</th>\n",
       "      <th>mutant allele</th>\n",
       "      <th>stop_codon_predicted</th>\n",
       "      <th>possible frameshift</th>\n",
       "      <th>wt_gene_translated</th>\n",
       "      <th>mutant allele_translated</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [position, ID, Parent, object, chromosome, start, end, strand, detail, p_value, correlation, p_value_corr, gene, X968_allele, Y0036_allele, reference_seq, reference_seq_, wt_gene, mutant allele, stop_codon_predicted, possible frameshift, wt_gene_translated, mutant allele_translated]\n",
       "Index: []"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#where possible frameshift is yes\n",
    "frameshift_df[frameshift_df['possible frameshift'] == 'yes']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.9.12 ('base')",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.0"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "2d0398177d7bb10ea33b6a5cd9d022d63e20680fcfd84762562c72ff35bdf44b"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
