{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "#find snps\n",
    "#import data science libraries\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "#import /Users/bencekover/Library/CloudStorage/OneDrive-UniversityCollegeLondon/MSci Bahler lab/S.-Pombe-biofilm/Bence folder/QTL mapping/permutations/results.csv\n",
    "#read in data\n",
    "snps = pd.read_csv('/Users/bencekover/Library/CloudStorage/OneDrive-UniversityCollegeLondon/MSci Bahler lab/S.-Pombe-biofilm/Bence folder/GWAS/gwas_hits.csv')\n",
    "#washing phenotype /Users/bencekover/Library/CloudStorage/OneDrive-UniversityCollegeLondon/MSci Bahler lab/S.-Pombe-biofilm/Bence folder/QTL mapping/washing_phenotypes.csv\n",
    "wash_pheno = pd.read_csv('/Users/bencekover/Library/CloudStorage/OneDrive-UniversityCollegeLondon/MSci Bahler lab/S.-Pombe-biofilm/Bence folder/QTL mapping/washing_phenotypes.csv')\n",
    "#import rna seq corrs /Users/bencekover/Library/CloudStorage/OneDrive-UniversityCollegeLondon/MSci Bahler lab/S.-Pombe-biofilm/Bence folder/Analysis of Clement-Ziza RNA-seq/wash_corr.csv\n",
    "wash_corr = pd.read_csv('/Users/bencekover/Library/CloudStorage/OneDrive-UniversityCollegeLondon/MSci Bahler lab/S.-Pombe-biofilm/Bence folder/Analysis of Clement-Ziza RNA-seq/wash_corr.csv')\n",
    "#full snp dataset /Users/bencekover/Library/CloudStorage/OneDrive-UniversityCollegeLondon/MSci Bahler lab/S.-Pombe-biofilm/Bence folder/QTL mapping/SupplementaryDataset_S7_genotype.tsv\n",
    "full_snps = pd.read_csv('/Users/bencekover/Library/CloudStorage/OneDrive-UniversityCollegeLondon/MSci Bahler lab/S.-Pombe-biofilm/Bence folder/QTL mapping/SupplementaryDataset_S7_genotype.tsv', sep='\\t',header=15)\n",
    "#load orthologs database /Users/bencekover/Library/CloudStorage/OneDrive-UniversityCollegeLondon/MSci Bahler lab/S.-Pombe-biofilm/Bence folder/Orthologs/final_orthology_database.csv\n",
    "orthologs = pd.read_csv('/Users/bencekover/Library/CloudStorage/OneDrive-UniversityCollegeLondon/MSci Bahler lab/S.-Pombe-biofilm/Bence folder/Orthologs/final_orthology_database.csv')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/mb/vdj4y7t13zd3tzlk99b8ycd40000gn/T/ipykernel_44819/480895657.py:8: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.\n",
      "  snps['allele1'] = snps['allele1'].str.replace('[','')\n",
      "/var/folders/mb/vdj4y7t13zd3tzlk99b8ycd40000gn/T/ipykernel_44819/480895657.py:9: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.\n",
      "  snps['allele2'] = snps['allele2'].str.replace(']','')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>locus</th>\n",
       "      <th>alleles</th>\n",
       "      <th>n</th>\n",
       "      <th>sum_x</th>\n",
       "      <th>y_transpose_x</th>\n",
       "      <th>beta</th>\n",
       "      <th>standard_error</th>\n",
       "      <th>t_stat</th>\n",
       "      <th>p_value</th>\n",
       "      <th>type</th>\n",
       "      <th>allele1</th>\n",
       "      <th>allele2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>32612</td>\n",
       "      <td>I:3364859</td>\n",
       "      <td>['C', 'T']</td>\n",
       "      <td>55</td>\n",
       "      <td>3.235294</td>\n",
       "      <td>2.112613</td>\n",
       "      <td>0.546460</td>\n",
       "      <td>0.073554</td>\n",
       "      <td>7.429402</td>\n",
       "      <td>1.620139e-09</td>\n",
       "      <td>snp</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>34916</td>\n",
       "      <td>I:3585317</td>\n",
       "      <td>['A', 'C']</td>\n",
       "      <td>55</td>\n",
       "      <td>42.000000</td>\n",
       "      <td>5.746557</td>\n",
       "      <td>-0.366654</td>\n",
       "      <td>0.057374</td>\n",
       "      <td>-6.390640</td>\n",
       "      <td>6.328207e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>100775</td>\n",
       "      <td>II:4296875</td>\n",
       "      <td>['T', 'A']</td>\n",
       "      <td>55</td>\n",
       "      <td>4.400000</td>\n",
       "      <td>2.394839</td>\n",
       "      <td>0.443424</td>\n",
       "      <td>0.070286</td>\n",
       "      <td>6.308884</td>\n",
       "      <td>8.445186e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>100776</td>\n",
       "      <td>II:4296877</td>\n",
       "      <td>['G', 'A']</td>\n",
       "      <td>55</td>\n",
       "      <td>4.400000</td>\n",
       "      <td>2.394839</td>\n",
       "      <td>0.443424</td>\n",
       "      <td>0.070286</td>\n",
       "      <td>6.308884</td>\n",
       "      <td>8.445186e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>133051</td>\n",
       "      <td>III:2219006</td>\n",
       "      <td>['A', 'C']</td>\n",
       "      <td>55</td>\n",
       "      <td>5.000000</td>\n",
       "      <td>2.231398</td>\n",
       "      <td>0.471360</td>\n",
       "      <td>0.069086</td>\n",
       "      <td>6.822801</td>\n",
       "      <td>1.374737e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>133289</td>\n",
       "      <td>III:2227937</td>\n",
       "      <td>['T', 'G']</td>\n",
       "      <td>55</td>\n",
       "      <td>6.000000</td>\n",
       "      <td>2.267975</td>\n",
       "      <td>0.558370</td>\n",
       "      <td>0.069510</td>\n",
       "      <td>8.032999</td>\n",
       "      <td>1.963371e-10</td>\n",
       "      <td>snp</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>133295</td>\n",
       "      <td>III:2228740</td>\n",
       "      <td>['C', 'T']</td>\n",
       "      <td>55</td>\n",
       "      <td>6.000000</td>\n",
       "      <td>2.267975</td>\n",
       "      <td>0.558370</td>\n",
       "      <td>0.069510</td>\n",
       "      <td>8.032999</td>\n",
       "      <td>1.963371e-10</td>\n",
       "      <td>snp</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>133311</td>\n",
       "      <td>III:2230713</td>\n",
       "      <td>['G', 'A']</td>\n",
       "      <td>55</td>\n",
       "      <td>4.000000</td>\n",
       "      <td>2.348044</td>\n",
       "      <td>0.444547</td>\n",
       "      <td>0.069142</td>\n",
       "      <td>6.429509</td>\n",
       "      <td>5.516623e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1011</td>\n",
       "      <td>I:498798</td>\n",
       "      <td>['TG', 'T']</td>\n",
       "      <td>52</td>\n",
       "      <td>6.000000</td>\n",
       "      <td>2.530626</td>\n",
       "      <td>0.311554</td>\n",
       "      <td>0.072372</td>\n",
       "      <td>4.304919</td>\n",
       "      <td>8.916224e-05</td>\n",
       "      <td>indel</td>\n",
       "      <td>TG</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1028</td>\n",
       "      <td>I:519884</td>\n",
       "      <td>['A', 'AT']</td>\n",
       "      <td>52</td>\n",
       "      <td>7.000000</td>\n",
       "      <td>2.673649</td>\n",
       "      <td>0.349112</td>\n",
       "      <td>0.075066</td>\n",
       "      <td>4.650735</td>\n",
       "      <td>2.914204e-05</td>\n",
       "      <td>indel</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>1036</td>\n",
       "      <td>I:522785</td>\n",
       "      <td>['A', 'AT']</td>\n",
       "      <td>52</td>\n",
       "      <td>4.000000</td>\n",
       "      <td>1.777813</td>\n",
       "      <td>0.479528</td>\n",
       "      <td>0.096000</td>\n",
       "      <td>4.995075</td>\n",
       "      <td>9.351192e-06</td>\n",
       "      <td>indel</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>1201</td>\n",
       "      <td>I:723772</td>\n",
       "      <td>['T', 'TGTCTC']</td>\n",
       "      <td>52</td>\n",
       "      <td>3.000000</td>\n",
       "      <td>1.739226</td>\n",
       "      <td>0.417287</td>\n",
       "      <td>0.088883</td>\n",
       "      <td>4.694774</td>\n",
       "      <td>2.522852e-05</td>\n",
       "      <td>indel</td>\n",
       "      <td>T</td>\n",
       "      <td>TGTCTC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>9218</td>\n",
       "      <td>II:1721443</td>\n",
       "      <td>['A', 'ACT']</td>\n",
       "      <td>52</td>\n",
       "      <td>32.000000</td>\n",
       "      <td>4.428636</td>\n",
       "      <td>-0.285830</td>\n",
       "      <td>0.057186</td>\n",
       "      <td>-4.998236</td>\n",
       "      <td>9.253300e-06</td>\n",
       "      <td>indel</td>\n",
       "      <td>A</td>\n",
       "      <td>ACT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>10303</td>\n",
       "      <td>II:2693701</td>\n",
       "      <td>['GT', 'G', 'GTT']</td>\n",
       "      <td>52</td>\n",
       "      <td>42.000000</td>\n",
       "      <td>6.130268</td>\n",
       "      <td>-0.397447</td>\n",
       "      <td>0.076744</td>\n",
       "      <td>-5.178866</td>\n",
       "      <td>5.059658e-06</td>\n",
       "      <td>indel</td>\n",
       "      <td>GT</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>10305</td>\n",
       "      <td>II:2693728</td>\n",
       "      <td>['AC', 'AACA']</td>\n",
       "      <td>52</td>\n",
       "      <td>42.000000</td>\n",
       "      <td>6.130268</td>\n",
       "      <td>-0.397447</td>\n",
       "      <td>0.076744</td>\n",
       "      <td>-5.178866</td>\n",
       "      <td>5.059658e-06</td>\n",
       "      <td>indel</td>\n",
       "      <td>AC</td>\n",
       "      <td>AACA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>10306</td>\n",
       "      <td>II:2694024</td>\n",
       "      <td>['A', 'AT']</td>\n",
       "      <td>52</td>\n",
       "      <td>42.000000</td>\n",
       "      <td>6.130268</td>\n",
       "      <td>-0.397447</td>\n",
       "      <td>0.076744</td>\n",
       "      <td>-5.178866</td>\n",
       "      <td>5.059658e-06</td>\n",
       "      <td>indel</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>10307</td>\n",
       "      <td>II:2694637</td>\n",
       "      <td>['G', 'GT']</td>\n",
       "      <td>52</td>\n",
       "      <td>42.000000</td>\n",
       "      <td>6.130268</td>\n",
       "      <td>-0.397447</td>\n",
       "      <td>0.076744</td>\n",
       "      <td>-5.178866</td>\n",
       "      <td>5.059658e-06</td>\n",
       "      <td>indel</td>\n",
       "      <td>G</td>\n",
       "      <td>GT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>10308</td>\n",
       "      <td>II:2697960</td>\n",
       "      <td>['AT', 'A']</td>\n",
       "      <td>52</td>\n",
       "      <td>42.000000</td>\n",
       "      <td>6.130268</td>\n",
       "      <td>-0.397447</td>\n",
       "      <td>0.076744</td>\n",
       "      <td>-5.178866</td>\n",
       "      <td>5.059658e-06</td>\n",
       "      <td>indel</td>\n",
       "      <td>AT</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>15230</td>\n",
       "      <td>III:2204023</td>\n",
       "      <td>['TA', 'T']</td>\n",
       "      <td>52</td>\n",
       "      <td>6.000000</td>\n",
       "      <td>1.846209</td>\n",
       "      <td>0.392285</td>\n",
       "      <td>0.086302</td>\n",
       "      <td>4.545506</td>\n",
       "      <td>4.106708e-05</td>\n",
       "      <td>indel</td>\n",
       "      <td>TA</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>15249</td>\n",
       "      <td>III:2218509</td>\n",
       "      <td>['C', 'CGA']</td>\n",
       "      <td>52</td>\n",
       "      <td>6.117647</td>\n",
       "      <td>1.782102</td>\n",
       "      <td>0.437682</td>\n",
       "      <td>0.089806</td>\n",
       "      <td>4.873659</td>\n",
       "      <td>1.399329e-05</td>\n",
       "      <td>indel</td>\n",
       "      <td>C</td>\n",
       "      <td>CGA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>15309</td>\n",
       "      <td>III:2277085</td>\n",
       "      <td>['T', 'TC']</td>\n",
       "      <td>52</td>\n",
       "      <td>3.000000</td>\n",
       "      <td>1.640883</td>\n",
       "      <td>0.401600</td>\n",
       "      <td>0.092993</td>\n",
       "      <td>4.318593</td>\n",
       "      <td>8.534981e-05</td>\n",
       "      <td>indel</td>\n",
       "      <td>T</td>\n",
       "      <td>TC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>15327</td>\n",
       "      <td>III:2291604</td>\n",
       "      <td>['GCTTTTCTTAACAGATGCTTTGAGGGTA', 'G']</td>\n",
       "      <td>52</td>\n",
       "      <td>3.000000</td>\n",
       "      <td>1.640883</td>\n",
       "      <td>0.401600</td>\n",
       "      <td>0.092993</td>\n",
       "      <td>4.318593</td>\n",
       "      <td>8.534981e-05</td>\n",
       "      <td>indel</td>\n",
       "      <td>GCTTTTCTTAACAGATGCTTTGAGGGTA</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>15333</td>\n",
       "      <td>III:2294439</td>\n",
       "      <td>['G', 'GTCA']</td>\n",
       "      <td>52</td>\n",
       "      <td>5.000000</td>\n",
       "      <td>2.327398</td>\n",
       "      <td>0.312051</td>\n",
       "      <td>0.071491</td>\n",
       "      <td>4.364909</td>\n",
       "      <td>7.358208e-05</td>\n",
       "      <td>indel</td>\n",
       "      <td>G</td>\n",
       "      <td>GTCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>15412</td>\n",
       "      <td>III:2357290</td>\n",
       "      <td>['C', 'CTTTTGATCTGATAAT']</td>\n",
       "      <td>52</td>\n",
       "      <td>3.000000</td>\n",
       "      <td>1.670345</td>\n",
       "      <td>0.393597</td>\n",
       "      <td>0.092129</td>\n",
       "      <td>4.272242</td>\n",
       "      <td>9.895852e-05</td>\n",
       "      <td>indel</td>\n",
       "      <td>C</td>\n",
       "      <td>CTTTTGATCTGATAAT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Unnamed: 0        locus                                alleles   n  \\\n",
       "0        32612    I:3364859                             ['C', 'T']  55   \n",
       "1        34916    I:3585317                             ['A', 'C']  55   \n",
       "2       100775   II:4296875                             ['T', 'A']  55   \n",
       "3       100776   II:4296877                             ['G', 'A']  55   \n",
       "4       133051  III:2219006                             ['A', 'C']  55   \n",
       "5       133289  III:2227937                             ['T', 'G']  55   \n",
       "6       133295  III:2228740                             ['C', 'T']  55   \n",
       "7       133311  III:2230713                             ['G', 'A']  55   \n",
       "8         1011     I:498798                            ['TG', 'T']  52   \n",
       "9         1028     I:519884                            ['A', 'AT']  52   \n",
       "10        1036     I:522785                            ['A', 'AT']  52   \n",
       "11        1201     I:723772                        ['T', 'TGTCTC']  52   \n",
       "12        9218   II:1721443                           ['A', 'ACT']  52   \n",
       "13       10303   II:2693701                     ['GT', 'G', 'GTT']  52   \n",
       "14       10305   II:2693728                         ['AC', 'AACA']  52   \n",
       "15       10306   II:2694024                            ['A', 'AT']  52   \n",
       "16       10307   II:2694637                            ['G', 'GT']  52   \n",
       "17       10308   II:2697960                            ['AT', 'A']  52   \n",
       "18       15230  III:2204023                            ['TA', 'T']  52   \n",
       "19       15249  III:2218509                           ['C', 'CGA']  52   \n",
       "20       15309  III:2277085                            ['T', 'TC']  52   \n",
       "21       15327  III:2291604  ['GCTTTTCTTAACAGATGCTTTGAGGGTA', 'G']  52   \n",
       "22       15333  III:2294439                          ['G', 'GTCA']  52   \n",
       "23       15412  III:2357290              ['C', 'CTTTTGATCTGATAAT']  52   \n",
       "\n",
       "        sum_x  y_transpose_x      beta  standard_error    t_stat  \\\n",
       "0    3.235294       2.112613  0.546460        0.073554  7.429402   \n",
       "1   42.000000       5.746557 -0.366654        0.057374 -6.390640   \n",
       "2    4.400000       2.394839  0.443424        0.070286  6.308884   \n",
       "3    4.400000       2.394839  0.443424        0.070286  6.308884   \n",
       "4    5.000000       2.231398  0.471360        0.069086  6.822801   \n",
       "5    6.000000       2.267975  0.558370        0.069510  8.032999   \n",
       "6    6.000000       2.267975  0.558370        0.069510  8.032999   \n",
       "7    4.000000       2.348044  0.444547        0.069142  6.429509   \n",
       "8    6.000000       2.530626  0.311554        0.072372  4.304919   \n",
       "9    7.000000       2.673649  0.349112        0.075066  4.650735   \n",
       "10   4.000000       1.777813  0.479528        0.096000  4.995075   \n",
       "11   3.000000       1.739226  0.417287        0.088883  4.694774   \n",
       "12  32.000000       4.428636 -0.285830        0.057186 -4.998236   \n",
       "13  42.000000       6.130268 -0.397447        0.076744 -5.178866   \n",
       "14  42.000000       6.130268 -0.397447        0.076744 -5.178866   \n",
       "15  42.000000       6.130268 -0.397447        0.076744 -5.178866   \n",
       "16  42.000000       6.130268 -0.397447        0.076744 -5.178866   \n",
       "17  42.000000       6.130268 -0.397447        0.076744 -5.178866   \n",
       "18   6.000000       1.846209  0.392285        0.086302  4.545506   \n",
       "19   6.117647       1.782102  0.437682        0.089806  4.873659   \n",
       "20   3.000000       1.640883  0.401600        0.092993  4.318593   \n",
       "21   3.000000       1.640883  0.401600        0.092993  4.318593   \n",
       "22   5.000000       2.327398  0.312051        0.071491  4.364909   \n",
       "23   3.000000       1.670345  0.393597        0.092129  4.272242   \n",
       "\n",
       "         p_value   type                       allele1           allele2  \n",
       "0   1.620139e-09    snp                             C                 T  \n",
       "1   6.328207e-08    snp                             A                 C  \n",
       "2   8.445186e-08    snp                             T                 A  \n",
       "3   8.445186e-08    snp                             G                 A  \n",
       "4   1.374737e-08    snp                             A                 C  \n",
       "5   1.963371e-10    snp                             T                 G  \n",
       "6   1.963371e-10    snp                             C                 T  \n",
       "7   5.516623e-08    snp                             G                 A  \n",
       "8   8.916224e-05  indel                            TG                 T  \n",
       "9   2.914204e-05  indel                             A                AT  \n",
       "10  9.351192e-06  indel                             A                AT  \n",
       "11  2.522852e-05  indel                             T            TGTCTC  \n",
       "12  9.253300e-06  indel                             A               ACT  \n",
       "13  5.059658e-06  indel                            GT                 G  \n",
       "14  5.059658e-06  indel                            AC              AACA  \n",
       "15  5.059658e-06  indel                             A                AT  \n",
       "16  5.059658e-06  indel                             G                GT  \n",
       "17  5.059658e-06  indel                            AT                 A  \n",
       "18  4.106708e-05  indel                            TA                 T  \n",
       "19  1.399329e-05  indel                             C               CGA  \n",
       "20  8.534981e-05  indel                             T                TC  \n",
       "21  8.534981e-05  indel  GCTTTTCTTAACAGATGCTTTGAGGGTA                 G  \n",
       "22  7.358208e-05  indel                             G              GTCA  \n",
       "23  9.895852e-05  indel                             C  CTTTTGATCTGATAAT  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#for now remove 2nd and 13th row, and reset index\n",
    "snps = snps.drop([2,13])\n",
    "snps = snps.reset_index(drop=True)\n",
    "#create two columns for the two alleles found in alleles\n",
    "snps['allele1'] = snps['alleles'].str.split(',').str[0]\n",
    "snps['allele2'] = snps['alleles'].str.split(',').str[1]\n",
    "#remove sorrounding ['' and ''] from alleles\n",
    "snps['allele1'] = snps['allele1'].str.replace('[','')\n",
    "snps['allele2'] = snps['allele2'].str.replace(']','')\n",
    "#remove surrounding '' from alleles\n",
    "snps['allele1'] = snps['allele1'].str.replace('\\'','')\n",
    "snps['allele2'] = snps['allele2'].str.replace('\\'','')\n",
    "#remove extra space from allele2\n",
    "snps['allele2'] = snps['allele2'].str.replace(' ','')\n",
    "snps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "#remove Unnamed: 0 column from snps\n",
    "snps = snps.drop(columns=['Unnamed: 0'])\n",
    "#split locus intro two columns, Chromosome and Position\n",
    "snps[['chromosome','position']] = snps['locus'].str.split(':',expand=True)\n",
    "#remove locus column\n",
    "snps = snps.drop(columns=['locus'])\n",
    "\n",
    "#turn snps position into int\n",
    "snps['position'] = snps['position'].astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "#pandas read in \"/Users/bencekover/Library/CloudStorage/OneDrive-UniversityCollegeLondon/MSci Bahler lab/S.-Pombe-biofilm/external data/Pombe genome annotation.gff3\"\n",
    "import pandas as pd\n",
    "df = pd.read_csv('/Users/bencekover/Library/CloudStorage/OneDrive-UniversityCollegeLondon/MSci Bahler lab/S.-Pombe-biofilm/external data/Pombe genome annotation.gff3', sep='\\t', header=1)\n",
    "#break last column into 2 at ;\n",
    "df[['a','b']] = df[\"ID=SPAC1002.01;Name=mrx11\"].str.split(';',expand=True)\n",
    "#remove column ID=SPAC1002.01;Name=mrx11\n",
    "df.drop(columns =[\"ID=SPAC1002.01;Name=mrx11\"], inplace = True)\n",
    "#remove ID= from column a and rename it to ID\n",
    "df['ID'] = df['a'].str.replace('ID=', '')\n",
    "#remove column a\n",
    "df.drop(columns =['a','PomBase',\".\",\"..1\"], inplace = True)\n",
    "#rename b to Parent\n",
    "df.rename(columns={'b':'Parent','gene':'object','I':'chromosome','1798347':\"start\",'1798835':\"end\",\"+\":\"strand\"}, inplace=True)\n",
    "#In parent remove anything before =\n",
    "df['Parent'] = df['Parent'].str.replace('Parent=', '')\n",
    "#or Name=\n",
    "df['Parent'] = df['Parent'].str.replace('Name=', '')\n",
    "#create an aempty column called detail\n",
    "df['detail'] = \"\"\n",
    "#split ID column at :, and if there is something after put it in detail column\n",
    "df['detail'] = df['ID'].str.split(':', expand=True)[1]\n",
    "#split ID column at :, and if there is something before put it in ID column\n",
    "df['ID'] = df['ID'].str.split(':', expand=True)[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "#find a subset of df, where snps[position] is +- 1000 of df[start] or df[end]. Create a result df, and say which snps were there\n",
    "#suppress warnings\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "\n",
    "result = pd.DataFrame()\n",
    "for i in range(len(snps)):\n",
    "    for j in range(len(df)):\n",
    "        if snps['position'][i] >= df['start'][j] - 1500 and snps['position'][i] <= df['end'][j] + 1500 :\n",
    "            #and if chromsome is chromosome_2 for df chromsome II etc\n",
    "            if snps['chromosome'][i] == df['chromosome'][j]:\n",
    "                result = result.append({'position': snps['position'][i], 'ID': df['ID'][j], 'Parent': df['Parent'][j], 'object': df['object'][j], 'chromosome': df['chromosome'][j], 'start': df['start'][j], 'end': df['end'][j], 'strand': df['strand'][j], 'detail': df['detail'][j], \"beta\": snps[\"beta\"][i], \"p_value\": snps[\"p_value\"][i], \"type\":snps[\"type\"][i]}, ignore_index=True)\n",
    "result\n",
    "\n",
    "#if the last two characters are .1, remove them\n",
    "for i in range(len(result)):\n",
    "    if result['ID'][i][-2:] == \".1\":\n",
    "        result['ID'][i] = result['ID'][i][:-2]\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "#match ID with systematic name in wash_corr and add columns gene\tcorrelation\tp-value\n",
    "#create np.arrays with NAs\n",
    "corrs = np.array([np.nan]*len(result))\n",
    "pvals = np.array([np.nan]*len(result))\n",
    "genes = np.array([np.nan]*len(result), dtype=object)\n",
    "for i in range(len(result)):\n",
    "    for j in range(len(wash_corr)):\n",
    "        if result['ID'][i] == wash_corr['systematic_names'][j]:\n",
    "            corrs[i] = wash_corr['correlation'][j]\n",
    "            pvals[i] = wash_corr['p-value'][j]\n",
    "            genes[i] = wash_corr['gene'][j]\n",
    "\n",
    "            \n",
    "\n",
    "\n",
    "result['correlation'] = corrs\n",
    "result['p_value_corr'] = pvals\n",
    "result['gene'] = genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "#add to each row the unique number of positions that ID had\n",
    "#add a column called unique\n",
    "#remove duplicates of ID and position but keep the highest p_value\n",
    "\n",
    "result_final = result.drop_duplicates(subset=['ID', 'position'])\n",
    "#reset index\n",
    "result_final = result_final.reset_index(drop=True)\n",
    "unique = np.array([])\n",
    "for i in range(len(result_final)):\n",
    "    #get the ID\n",
    "    ID = result_final['ID'][i]\n",
    "    #get the number of times ID appears in result\n",
    "    count = len(result_final[result_final['ID'] == ID])\n",
    "    #add count to unique\n",
    "    unique = np.append(unique, count)\n",
    "result_final['unique'] = unique\n",
    "#show SPBC290.07 in result\n",
    "#gather positions for each ID, and put in a single entry as a list. Add that to the df. THen remove duplicated IDs\n",
    "result_final_ = result_final.groupby('ID')['position'].apply(list).reset_index(name='positions')\n",
    "result_final__ = pd.merge(result_final, result_final_, on='ID')\n",
    "\n",
    "#remove duplicates of ID\n",
    "result_final__ = result_final__.drop_duplicates(subset=['ID'])\n",
    "\n",
    "#reset index\n",
    "result_final__ = result_final__.reset_index(drop=True)\n",
    "\n",
    "\n",
    "#ignore the above \n",
    "result_final__ = result.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "#iterate through result_final__. If ID is found in orthologs, then add the albicans_name(s)cerevisiae_name(s) to the df\n",
    "from numpy import dtype\n",
    "\n",
    "albicans_names = np.array([np.nan]*len(result_final__),dtype=object)\n",
    "cerevisiae_names = np.array([np.nan]*len(result_final__),dtype=object)\n",
    "for i in range(len(result_final__)):\n",
    "    for j in range(len(orthologs)):\n",
    "        if result_final__['ID'][i] in orthologs['pombe_name(s)'][j]:\n",
    "            albicans_names[i] = orthologs['albicans_name(s)'][j]\n",
    "            cerevisiae_names[i] = orthologs['cerevisiae_name(s)'][j]\n",
    "#add to df\n",
    "result_final__['albicans_name(s)'] = albicans_names\n",
    "result_final__['cerevisiae_name(s)'] = cerevisiae_names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>position</th>\n",
       "      <th>ID</th>\n",
       "      <th>Parent</th>\n",
       "      <th>object</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>detail</th>\n",
       "      <th>beta</th>\n",
       "      <th>p_value</th>\n",
       "      <th>type</th>\n",
       "      <th>correlation</th>\n",
       "      <th>p_value_corr</th>\n",
       "      <th>gene</th>\n",
       "      <th>albicans_name(s)</th>\n",
       "      <th>cerevisiae_name(s)</th>\n",
       "      <th>allele_1</th>\n",
       "      <th>allele_2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3364859</td>\n",
       "      <td>SPAC26A3.13c</td>\n",
       "      <td>Tf2-4</td>\n",
       "      <td>gene</td>\n",
       "      <td>I</td>\n",
       "      <td>3361547</td>\n",
       "      <td>3365548</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>0.546460</td>\n",
       "      <td>1.620139e-09</td>\n",
       "      <td>snp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Tf2-4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3364859</td>\n",
       "      <td>SPAC26A3.13c</td>\n",
       "      <td>SPAC26A3.13c</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>I</td>\n",
       "      <td>3361547</td>\n",
       "      <td>3365548</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>0.546460</td>\n",
       "      <td>1.620139e-09</td>\n",
       "      <td>snp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Tf2-4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3364859</td>\n",
       "      <td>SPAC26A3.13c</td>\n",
       "      <td>SPAC26A3.13c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3361547</td>\n",
       "      <td>3365548</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.546460</td>\n",
       "      <td>1.620139e-09</td>\n",
       "      <td>snp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Tf2-4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3364859</td>\n",
       "      <td>SPAC26A3.14c</td>\n",
       "      <td>None</td>\n",
       "      <td>gene</td>\n",
       "      <td>I</td>\n",
       "      <td>3366317</td>\n",
       "      <td>3367205</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>0.546460</td>\n",
       "      <td>1.620139e-09</td>\n",
       "      <td>snp</td>\n",
       "      <td>-0.399422</td>\n",
       "      <td>0.001896</td>\n",
       "      <td>SPAC26A3.14c</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3364859</td>\n",
       "      <td>SPAC26A3.14c</td>\n",
       "      <td>SPAC26A3.14c</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>I</td>\n",
       "      <td>3366317</td>\n",
       "      <td>3367205</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>0.546460</td>\n",
       "      <td>1.620139e-09</td>\n",
       "      <td>snp</td>\n",
       "      <td>-0.399422</td>\n",
       "      <td>0.001896</td>\n",
       "      <td>SPAC26A3.14c</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>449</th>\n",
       "      <td>2357290</td>\n",
       "      <td>SPCC70.12c</td>\n",
       "      <td>SPCC70.12c</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>III</td>\n",
       "      <td>2354968</td>\n",
       "      <td>2356317</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>0.393597</td>\n",
       "      <td>9.895852e-05</td>\n",
       "      <td>indel</td>\n",
       "      <td>0.221542</td>\n",
       "      <td>0.094658</td>\n",
       "      <td>ecl1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>CTTTTGATCTGATAAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>450</th>\n",
       "      <td>2357290</td>\n",
       "      <td>SPCC70.12c</td>\n",
       "      <td>SPCC70.12c.1</td>\n",
       "      <td>five_prime_UTR</td>\n",
       "      <td>III</td>\n",
       "      <td>2355788</td>\n",
       "      <td>2356317</td>\n",
       "      <td>-</td>\n",
       "      <td>five_prime_UTR</td>\n",
       "      <td>0.393597</td>\n",
       "      <td>9.895852e-05</td>\n",
       "      <td>indel</td>\n",
       "      <td>0.221542</td>\n",
       "      <td>0.094658</td>\n",
       "      <td>ecl1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>CTTTTGATCTGATAAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>451</th>\n",
       "      <td>2357290</td>\n",
       "      <td>SPNCRNA.7727</td>\n",
       "      <td>None</td>\n",
       "      <td>gene</td>\n",
       "      <td>III</td>\n",
       "      <td>2354421</td>\n",
       "      <td>2358979</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>0.393597</td>\n",
       "      <td>9.895852e-05</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>CTTTTGATCTGATAAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>452</th>\n",
       "      <td>2357290</td>\n",
       "      <td>SPNCRNA.7727</td>\n",
       "      <td>SPNCRNA.7727</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>III</td>\n",
       "      <td>2354421</td>\n",
       "      <td>2358979</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>0.393597</td>\n",
       "      <td>9.895852e-05</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>CTTTTGATCTGATAAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>453</th>\n",
       "      <td>2357290</td>\n",
       "      <td>SPNCRNA.7727</td>\n",
       "      <td>SPNCRNA.7727.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>III</td>\n",
       "      <td>2354421</td>\n",
       "      <td>2358979</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.393597</td>\n",
       "      <td>9.895852e-05</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>CTTTTGATCTGATAAT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>454 rows × 19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     position            ID          Parent          object chromosome  \\\n",
       "0     3364859  SPAC26A3.13c           Tf2-4            gene          I   \n",
       "1     3364859  SPAC26A3.13c    SPAC26A3.13c            mRNA          I   \n",
       "2     3364859  SPAC26A3.13c  SPAC26A3.13c.1             CDS          I   \n",
       "3     3364859  SPAC26A3.14c            None            gene          I   \n",
       "4     3364859  SPAC26A3.14c    SPAC26A3.14c            mRNA          I   \n",
       "..        ...           ...             ...             ...        ...   \n",
       "449   2357290    SPCC70.12c      SPCC70.12c            mRNA        III   \n",
       "450   2357290    SPCC70.12c    SPCC70.12c.1  five_prime_UTR        III   \n",
       "451   2357290  SPNCRNA.7727            None            gene        III   \n",
       "452   2357290  SPNCRNA.7727    SPNCRNA.7727           ncRNA        III   \n",
       "453   2357290  SPNCRNA.7727  SPNCRNA.7727.1             CDS        III   \n",
       "\n",
       "       start      end strand          detail      beta       p_value   type  \\\n",
       "0    3361547  3365548      -            None  0.546460  1.620139e-09    snp   \n",
       "1    3361547  3365548      -            None  0.546460  1.620139e-09    snp   \n",
       "2    3361547  3365548      -            exon  0.546460  1.620139e-09    snp   \n",
       "3    3366317  3367205      -            None  0.546460  1.620139e-09    snp   \n",
       "4    3366317  3367205      -            None  0.546460  1.620139e-09    snp   \n",
       "..       ...      ...    ...             ...       ...           ...    ...   \n",
       "449  2354968  2356317      -            None  0.393597  9.895852e-05  indel   \n",
       "450  2355788  2356317      -  five_prime_UTR  0.393597  9.895852e-05  indel   \n",
       "451  2354421  2358979      +            None  0.393597  9.895852e-05  indel   \n",
       "452  2354421  2358979      +            None  0.393597  9.895852e-05  indel   \n",
       "453  2354421  2358979      +            exon  0.393597  9.895852e-05  indel   \n",
       "\n",
       "     correlation  p_value_corr          gene albicans_name(s)  \\\n",
       "0            NaN           NaN         Tf2-4              NaN   \n",
       "1            NaN           NaN         Tf2-4              NaN   \n",
       "2            NaN           NaN         Tf2-4              NaN   \n",
       "3      -0.399422      0.001896  SPAC26A3.14c              NaN   \n",
       "4      -0.399422      0.001896  SPAC26A3.14c              NaN   \n",
       "..           ...           ...           ...              ...   \n",
       "449     0.221542      0.094658          ecl1              NaN   \n",
       "450     0.221542      0.094658          ecl1              NaN   \n",
       "451          NaN           NaN           NaN              NaN   \n",
       "452          NaN           NaN           NaN              NaN   \n",
       "453          NaN           NaN           NaN              NaN   \n",
       "\n",
       "    cerevisiae_name(s) allele_1          allele_2  \n",
       "0                  NaN        C                 T  \n",
       "1                  NaN        C                 T  \n",
       "2                  NaN        C                 T  \n",
       "3                  NaN        C                 T  \n",
       "4                  NaN        C                 T  \n",
       "..                 ...      ...               ...  \n",
       "449                NaN        C  CTTTTGATCTGATAAT  \n",
       "450                NaN        C  CTTTTGATCTGATAAT  \n",
       "451                NaN        C  CTTTTGATCTGATAAT  \n",
       "452                NaN        C  CTTTTGATCTGATAAT  \n",
       "453                NaN        C  CTTTTGATCTGATAAT  \n",
       "\n",
       "[454 rows x 19 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#iterate through positions in result_final__  and find the corresponding alleles in snps. Then add the two values in X968_allele\tY0036_allele to the df in two new columns at the start \n",
    "#add two columns to result_final__\n",
    "result_final__['allele_1'] = np.nan\n",
    "result_final__['allele_2'] = np.nan\n",
    "for i in range(len(result_final__)):\n",
    "    for j in range(len(snps)):\n",
    "        if result_final__['position'][i] == snps['position'][j]:\n",
    "            result_final__['allele_1'][i] = snps['allele1'][j]\n",
    "            result_final__['allele_2'][i] = snps['allele2'][j]\n",
    "result_final__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys, re\n",
    "\n",
    "f=open('/Users/bencekover/Library/CloudStorage/OneDrive-UniversityCollegeLondon/MSci Bahler lab/S.-Pombe-biofilm/external data/Schizosaccharomyces_pombe_all_chromosomes.fa','r')\n",
    "lines=f.readlines()\n",
    "\n",
    "hre=re.compile('>(\\S+)')\n",
    "lre=re.compile('^(\\S+)$')\n",
    "\n",
    "regions={}\n",
    "\n",
    "for line in lines:\n",
    "        outh = hre.search(line)\n",
    "        if outh:\n",
    "                id=outh.group(1)\n",
    "        else:\n",
    "                outl=lre.search(line)\n",
    "                if(id in regions.keys()):\n",
    "                        regions[id] += outl.group(1)\n",
    "                else:\n",
    "                        regions[id]  =outl.group(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "#iterate through positions in result_final__ and look up the chromosome and position in regions, and find the corresponding letter +/- 5 bp in regions. Add it to a new column\n",
    "result_final__['reference_seq']=np.nan\n",
    "for i in range(len(result_final__)):\n",
    "    for j in range(len(regions)):\n",
    "        if result_final__['chromosome'][i] == list(regions.keys())[j]:\n",
    "            #add | | characters left and right of tletter at position-1\n",
    "            \n",
    "            result_final__['reference_seq'][i] = regions[list(regions.keys())[j]][result_final__['position'][i]-4:result_final__['position'][i]-1] + ' |' + regions[list(regions.keys())[j]][result_final__['position'][i]-1] + '| ' + regions[list(regions.keys())[j]][result_final__['position'][i]:result_final__['position'][i]+3] \n",
    "\n",
    "#if strand is -, reverse reference_seq\n",
    "for i in range(len(result_final__)):\n",
    "    if result_final__['strand'][i] == '-':\n",
    "        result_final__['reference_seq'][i] = result_final__['reference_seq'][i][::-1]\n",
    "        #also flip the alleles\n",
    "        result_final__['allele_1'][i] = result_final__['allele_1'][i][::-1]\n",
    "        result_final__['allele_2'][i] = result_final__['allele_2'][i][::-1]\n",
    "#also create reference_seq_ without the | | characters\n",
    "result_final__['reference_seq_'] = result_final__['reference_seq'].str.replace('|', '')\n",
    "result_final__['reference_seq_'] = result_final__['reference_seq_'].str.replace(' ', '')\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2\n",
      "32\n",
      "98\n",
      "115\n",
      "175\n",
      "193\n",
      "216\n",
      "242\n",
      "257\n",
      "280\n",
      "297\n",
      "314\n",
      "331\n",
      "412\n",
      "416\n"
     ]
    }
   ],
   "source": [
    "######\n",
    "# reate column, whole gene as codons\n",
    "result_final__['wt_gene'] = np.nan\n",
    "#iterate through result_final__ positions. If object is CDS, then look up the sequence from start to end in regions.\n",
    "#add the sequence to the df\n",
    "for i in range(len(result_final__)):\n",
    "    \n",
    "    if result_final__['object'][i] == 'CDS':\n",
    "    \n",
    "        start = result_final__['start'][i]\n",
    "        end = result_final__['end'][i]\n",
    "        chrom = result_final__['chromosome'][i]\n",
    "        pos = result_final__['position'][i]\n",
    "        gene = regions[chrom][start-1:end]\n",
    "        if pos > start and pos < end:\n",
    "        \n",
    "            if result_final__['strand'][i] == '-':\n",
    "                \n",
    "                gene = gene[::-1]\n",
    "            \n",
    "                gene = ''.join(gene)\n",
    "            #write the gene as triplets of letters like AAA, ATC, etc\n",
    "            #if divisible by 3\n",
    "            if len(gene)%3 != 0:\n",
    "                gene = gene[:-(len(gene)%3)]\n",
    "            gene = [gene[i:i+3] for i in range(0, len(gene), 3)]\n",
    "                #only triplets near the position\n",
    "            if result_final__['strand'][i] == '+':\n",
    "                gene = gene[(pos-start)//3-4:(pos-end)//3+4]\n",
    "            else: \n",
    "                gene = gene[(end-pos)//3-4:(start-pos)//3+4]\n",
    "            \n",
    "                \n",
    "                \n",
    "\n",
    "\n",
    "            result_final__['wt_gene'][i] = gene\n",
    "\n",
    "result_final__['mutant allele'] = np.nan\n",
    "for i in range(len(result_final__)):\n",
    "    if result_final__['object'][i] == 'CDS':\n",
    "    \n",
    "        start = result_final__['start'][i]\n",
    "        end = result_final__['end'][i]\n",
    "        chrom = result_final__['chromosome'][i]\n",
    "        pos = result_final__['position'][i]\n",
    "        gene = regions[chrom][start-1:end]\n",
    "        if pos > start and pos < end:\n",
    "        \n",
    "            if result_final__['strand'][i] == '+':\n",
    "                #decide which allele to use\n",
    "                #check \n",
    "                #check how long the y0036 allele is \n",
    "                all1_len = len(result_final__['allele_1'][i])\n",
    "                all2_len = len(result_final__['allele_2'][i])\n",
    "                #check if y0036 allele matches the reference allele\n",
    "                \n",
    "                if result_final__['reference_seq_'][i][3:3+all2_len] == result_final__['allele_2'][i]:\n",
    "                    mutation = result_final__['allele_1'][i]\n",
    "\n",
    "                elif result_final__['reference_seq_'][i][3:3+all1_len] == result_final__['allele_1'][i]:\n",
    "                    #if yes, use x968 allele\n",
    "                    mutation = result_final__['allele_2'][i]\n",
    "                else:\n",
    "                    print(i)\n",
    "\n",
    "                gene = gene[:pos-start] + mutation + gene[pos-start+1:]\n",
    "            elif result_final__['strand'][i] == '-':\n",
    "                \n",
    "                gene = gene[::-1]\n",
    "            \n",
    "                gene = ''.join(gene)\n",
    "\n",
    "                all1_len = len(result_final__['allele_1'][i])\n",
    "                all2_len = len(result_final__['allele_2'][i])\n",
    "                \n",
    "                if result_final__['reference_seq_'][i][3:3-all2_len] == result_final__['allele_2'][i]:\n",
    "                    mutation = result_final__['allele_1'][i]\n",
    "                elif result_final__['reference_seq_'][i][3:3-all1_len] == result_final__['allele_1'][i]:\n",
    "                    mutation = result_final__['allele_2'][i]\n",
    "                \n",
    "                else:\n",
    "                    print(i)\n",
    "                \n",
    "                gene = gene[:end-pos] + mutation + gene[end-pos+1:]\n",
    "\n",
    "                \n",
    "            if len(gene)%3 != 0:\n",
    "                gene = gene[:-(len(gene)%3)]\n",
    "            gene = [gene[i:i+3] for i in range(0, len(gene), 3)]\n",
    "                #only triplets near the position\n",
    "            if result_final__['strand'][i] == '+':\n",
    "                gene = gene[(pos-start)//3-4:(pos-end)//3+4]\n",
    "            else:\n",
    "                gene = gene[(end-pos)//3-4:(start-pos)//3+4]\n",
    "            result_final__[\"mutant allele\"][i] = gene\n",
    "######"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "if2416\n"
     ]
    }
   ],
   "source": [
    "result_final__['wt_gene'] = np.nan\n",
    "#iterate through result_final__ positions. If object is CDS, then look up the sequence from start to end in regions.\n",
    "#add the sequence to the df\n",
    "for i in range(len(result_final__)):\n",
    "    \n",
    "    if result_final__['object'][i] == 'CDS':\n",
    "    \n",
    "        start = result_final__['start'][i]\n",
    "        end = result_final__['end'][i]\n",
    "        chrom = result_final__['chromosome'][i]\n",
    "        pos = result_final__['position'][i]\n",
    "        gene = regions[chrom][start-1:end]\n",
    "        if pos > start and pos < end:\n",
    "        \n",
    "            if result_final__['strand'][i] == '-':\n",
    "                \n",
    "                gene = gene[::-1]\n",
    "            \n",
    "                gene = ''.join(gene)\n",
    "            #write the gene as triplets of letters like AAA, ATC, etc\n",
    "            #if divisible by 3\n",
    "            if len(gene)%3 != 0:\n",
    "                gene = gene[:-(len(gene)%3)]\n",
    "            gene = [gene[i:i+3] for i in range(0, len(gene), 3)]\n",
    "                #only triplets near the position\n",
    "            if result_final__['strand'][i] == '+':\n",
    "                gene = gene[(pos-start)//3-3:(pos-end)//3+3]\n",
    "            else: \n",
    "                gene = gene[(end-pos)//3-3:(start-pos)//3+3]\n",
    "            \n",
    "                \n",
    "                \n",
    "\n",
    "\n",
    "            result_final__['wt_gene'][i] = gene\n",
    "\n",
    "result_final__['mutant allele'] = np.nan\n",
    "for i in range(len(result_final__)):\n",
    "    if result_final__['object'][i] == 'CDS':\n",
    "    \n",
    "        start = result_final__['start'][i]\n",
    "        end = result_final__['end'][i]\n",
    "        chrom = result_final__['chromosome'][i]\n",
    "        pos = result_final__['position'][i]\n",
    "        gene = regions[chrom][start-1:end]\n",
    "        if pos > start and pos < end:\n",
    "            mutation = \"\"\n",
    "            if result_final__['strand'][i] == '+':\n",
    "                #decide which allele to use\n",
    "                #check \n",
    "                #check how long the y0036 allele is \n",
    "                all1_len = len(result_final__['allele_1'][i])\n",
    "                #check if y0036 allele matches the reference allele\n",
    "                \n",
    "                if result_final__['reference_seq_'][i][3:(3+all1_len)] == result_final__['allele_1'][i]:\n",
    "                    mutation = result_final__['allele_2'][i]\n",
    "                else:\n",
    "                    print(\"if1\" + str(i))\n",
    "\n",
    "                gene = gene[:pos-start] + mutation + gene[pos-start+1:]\n",
    "            elif result_final__['strand'][i] == '-':\n",
    "                \n",
    "                gene = gene[::-1]\n",
    "            \n",
    "                gene = ''.join(gene)\n",
    "\n",
    "                all1_len = len(result_final__['allele_1'][i])\n",
    "                \n",
    "                \n",
    "                if result_final__['reference_seq_'][i][(4-all1_len):4] == result_final__['allele_1'][i]:\n",
    "                    mutation = result_final__['allele_2'][i]\n",
    "                \n",
    "                else:\n",
    "                    print(\"if2\" + str(i))\n",
    "                \n",
    "                gene = gene[:end-pos] + mutation + gene[end-pos+1:]\n",
    "\n",
    "                \n",
    "            if len(gene)%3 != 0:\n",
    "                gene = gene[:-(len(gene)%3)]\n",
    "            gene = [gene[i:i+3] for i in range(0, len(gene), 3)]\n",
    "                #only triplets near the position\n",
    "            if result_final__['strand'][i] == '+':\n",
    "                gene = gene[(pos-start)//3-3:(pos-end)//3+3]\n",
    "            else:\n",
    "                gene = gene[(end-pos)//3-3:(start-pos)//3+3]\n",
    "            result_final__[\"mutant allele\"][i] = gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>position</th>\n",
       "      <th>ID</th>\n",
       "      <th>Parent</th>\n",
       "      <th>object</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>detail</th>\n",
       "      <th>beta</th>\n",
       "      <th>p_value</th>\n",
       "      <th>type</th>\n",
       "      <th>correlation</th>\n",
       "      <th>p_value_corr</th>\n",
       "      <th>gene</th>\n",
       "      <th>albicans_name(s)</th>\n",
       "      <th>cerevisiae_name(s)</th>\n",
       "      <th>allele_1</th>\n",
       "      <th>allele_2</th>\n",
       "      <th>reference_seq</th>\n",
       "      <th>reference_seq_</th>\n",
       "      <th>wt_gene</th>\n",
       "      <th>mutant allele</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3364859</td>\n",
       "      <td>SPAC26A3.13c</td>\n",
       "      <td>Tf2-4</td>\n",
       "      <td>gene</td>\n",
       "      <td>I</td>\n",
       "      <td>3361547</td>\n",
       "      <td>3365548</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>0.546460</td>\n",
       "      <td>1.620139e-09</td>\n",
       "      <td>snp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Tf2-4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>CTT |C| TGT</td>\n",
       "      <td>CTTCTGT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3364859</td>\n",
       "      <td>SPAC26A3.13c</td>\n",
       "      <td>SPAC26A3.13c</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>I</td>\n",
       "      <td>3361547</td>\n",
       "      <td>3365548</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>0.546460</td>\n",
       "      <td>1.620139e-09</td>\n",
       "      <td>snp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Tf2-4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>CTT |C| TGT</td>\n",
       "      <td>CTTCTGT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3364859</td>\n",
       "      <td>SPAC26A3.13c</td>\n",
       "      <td>SPAC26A3.13c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3361547</td>\n",
       "      <td>3365548</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.546460</td>\n",
       "      <td>1.620139e-09</td>\n",
       "      <td>snp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Tf2-4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>CTT |C| TGT</td>\n",
       "      <td>CTTCTGT</td>\n",
       "      <td>[TTG, GAC, TCC, TTC, TGT, TTA, CAA]</td>\n",
       "      <td>[TTG, GAC, TCC, TTT, TGT, TTA, CAA]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3364859</td>\n",
       "      <td>SPAC26A3.14c</td>\n",
       "      <td>None</td>\n",
       "      <td>gene</td>\n",
       "      <td>I</td>\n",
       "      <td>3366317</td>\n",
       "      <td>3367205</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>0.546460</td>\n",
       "      <td>1.620139e-09</td>\n",
       "      <td>snp</td>\n",
       "      <td>-0.399422</td>\n",
       "      <td>0.001896</td>\n",
       "      <td>SPAC26A3.14c</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>CTT |C| TGT</td>\n",
       "      <td>CTTCTGT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3364859</td>\n",
       "      <td>SPAC26A3.14c</td>\n",
       "      <td>SPAC26A3.14c</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>I</td>\n",
       "      <td>3366317</td>\n",
       "      <td>3367205</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>0.546460</td>\n",
       "      <td>1.620139e-09</td>\n",
       "      <td>snp</td>\n",
       "      <td>-0.399422</td>\n",
       "      <td>0.001896</td>\n",
       "      <td>SPAC26A3.14c</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>CTT |C| TGT</td>\n",
       "      <td>CTTCTGT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>3364859</td>\n",
       "      <td>SPAC26A3.14c</td>\n",
       "      <td>SPAC26A3.14c.1</td>\n",
       "      <td>three_prime_UTR</td>\n",
       "      <td>I</td>\n",
       "      <td>3366317</td>\n",
       "      <td>3366742</td>\n",
       "      <td>-</td>\n",
       "      <td>three_prime_UTR</td>\n",
       "      <td>0.546460</td>\n",
       "      <td>1.620139e-09</td>\n",
       "      <td>snp</td>\n",
       "      <td>-0.399422</td>\n",
       "      <td>0.001896</td>\n",
       "      <td>SPAC26A3.14c</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>CTT |C| TGT</td>\n",
       "      <td>CTTCTGT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>3364859</td>\n",
       "      <td>SPNCRNA.3503</td>\n",
       "      <td>None</td>\n",
       "      <td>gene</td>\n",
       "      <td>I</td>\n",
       "      <td>3364029</td>\n",
       "      <td>3364812</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>0.546460</td>\n",
       "      <td>1.620139e-09</td>\n",
       "      <td>snp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>TGT |C| TTC</td>\n",
       "      <td>TGTCTTC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>3364859</td>\n",
       "      <td>SPNCRNA.3503</td>\n",
       "      <td>SPNCRNA.3503</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>I</td>\n",
       "      <td>3364029</td>\n",
       "      <td>3364812</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>0.546460</td>\n",
       "      <td>1.620139e-09</td>\n",
       "      <td>snp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>TGT |C| TTC</td>\n",
       "      <td>TGTCTTC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>3364859</td>\n",
       "      <td>SPNCRNA.3503</td>\n",
       "      <td>SPNCRNA.3503.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3364029</td>\n",
       "      <td>3364812</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.546460</td>\n",
       "      <td>1.620139e-09</td>\n",
       "      <td>snp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>TGT |C| TTC</td>\n",
       "      <td>TGTCTTC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>3364859</td>\n",
       "      <td>SPNCRNA.3504</td>\n",
       "      <td>None</td>\n",
       "      <td>gene</td>\n",
       "      <td>I</td>\n",
       "      <td>3365076</td>\n",
       "      <td>3365865</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>0.546460</td>\n",
       "      <td>1.620139e-09</td>\n",
       "      <td>snp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>TGT |C| TTC</td>\n",
       "      <td>TGTCTTC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>3364859</td>\n",
       "      <td>SPNCRNA.3504</td>\n",
       "      <td>SPNCRNA.3504</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>I</td>\n",
       "      <td>3365076</td>\n",
       "      <td>3365865</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>0.546460</td>\n",
       "      <td>1.620139e-09</td>\n",
       "      <td>snp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>TGT |C| TTC</td>\n",
       "      <td>TGTCTTC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>3364859</td>\n",
       "      <td>SPNCRNA.3504</td>\n",
       "      <td>SPNCRNA.3504.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3365076</td>\n",
       "      <td>3365865</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.546460</td>\n",
       "      <td>1.620139e-09</td>\n",
       "      <td>snp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>TGT |C| TTC</td>\n",
       "      <td>TGTCTTC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>3364859</td>\n",
       "      <td>SPNCRNA.3505</td>\n",
       "      <td>None</td>\n",
       "      <td>gene</td>\n",
       "      <td>I</td>\n",
       "      <td>3366144</td>\n",
       "      <td>3367500</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>0.546460</td>\n",
       "      <td>1.620139e-09</td>\n",
       "      <td>snp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>TGT |C| TTC</td>\n",
       "      <td>TGTCTTC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>3364859</td>\n",
       "      <td>SPNCRNA.3505</td>\n",
       "      <td>SPNCRNA.3505</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>I</td>\n",
       "      <td>3366144</td>\n",
       "      <td>3367500</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>0.546460</td>\n",
       "      <td>1.620139e-09</td>\n",
       "      <td>snp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>TGT |C| TTC</td>\n",
       "      <td>TGTCTTC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>3364859</td>\n",
       "      <td>SPNCRNA.3505</td>\n",
       "      <td>SPNCRNA.3505.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3366144</td>\n",
       "      <td>3367500</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.546460</td>\n",
       "      <td>1.620139e-09</td>\n",
       "      <td>snp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>TGT |C| TTC</td>\n",
       "      <td>TGTCTTC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>3364859</td>\n",
       "      <td>SPTF.5</td>\n",
       "      <td>None</td>\n",
       "      <td>LTR_retrotransposon</td>\n",
       "      <td>I</td>\n",
       "      <td>3361143</td>\n",
       "      <td>3366062</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>0.546460</td>\n",
       "      <td>1.620139e-09</td>\n",
       "      <td>snp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>CTT |C| TGT</td>\n",
       "      <td>CTTCTGT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>3364859</td>\n",
       "      <td>SPLTRA.55</td>\n",
       "      <td>None</td>\n",
       "      <td>long_terminal_repeat</td>\n",
       "      <td>I</td>\n",
       "      <td>3365714</td>\n",
       "      <td>3366062</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>0.546460</td>\n",
       "      <td>1.620139e-09</td>\n",
       "      <td>snp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>CTT |C| TGT</td>\n",
       "      <td>CTTCTGT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>3585317</td>\n",
       "      <td>SPAC17A2.13c</td>\n",
       "      <td>rad25</td>\n",
       "      <td>gene</td>\n",
       "      <td>I</td>\n",
       "      <td>3583128</td>\n",
       "      <td>3584899</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>-0.366654</td>\n",
       "      <td>6.328207e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>0.022739</td>\n",
       "      <td>0.865462</td>\n",
       "      <td>rad25</td>\n",
       "      <td>['C1_03220C_A']</td>\n",
       "      <td>['YDR099W' 'YER177W']</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>TTA |A| AGC</td>\n",
       "      <td>TTAAAGC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>3585317</td>\n",
       "      <td>SPAC17A2.13c</td>\n",
       "      <td>SPAC17A2.13c</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>I</td>\n",
       "      <td>3583128</td>\n",
       "      <td>3584899</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>-0.366654</td>\n",
       "      <td>6.328207e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>0.022739</td>\n",
       "      <td>0.865462</td>\n",
       "      <td>rad25</td>\n",
       "      <td>['C1_03220C_A']</td>\n",
       "      <td>['YDR099W' 'YER177W']</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>TTA |A| AGC</td>\n",
       "      <td>TTAAAGC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>3585317</td>\n",
       "      <td>SPAC17A2.13c</td>\n",
       "      <td>SPAC17A2.13c.1</td>\n",
       "      <td>five_prime_UTR</td>\n",
       "      <td>I</td>\n",
       "      <td>3584780</td>\n",
       "      <td>3584899</td>\n",
       "      <td>-</td>\n",
       "      <td>five_prime_UTR</td>\n",
       "      <td>-0.366654</td>\n",
       "      <td>6.328207e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>0.022739</td>\n",
       "      <td>0.865462</td>\n",
       "      <td>rad25</td>\n",
       "      <td>['C1_03220C_A']</td>\n",
       "      <td>['YDR099W' 'YER177W']</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>TTA |A| AGC</td>\n",
       "      <td>TTAAAGC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>3585317</td>\n",
       "      <td>SPAC17A2.13c</td>\n",
       "      <td>SPAC17A2.13c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3584713</td>\n",
       "      <td>3584779</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>-0.366654</td>\n",
       "      <td>6.328207e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>0.022739</td>\n",
       "      <td>0.865462</td>\n",
       "      <td>rad25</td>\n",
       "      <td>['C1_03220C_A']</td>\n",
       "      <td>['YDR099W' 'YER177W']</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>TTA |A| AGC</td>\n",
       "      <td>TTAAAGC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>3585317</td>\n",
       "      <td>SPAC17A2.13c</td>\n",
       "      <td>SPAC17A2.13c.1</td>\n",
       "      <td>intron</td>\n",
       "      <td>I</td>\n",
       "      <td>3583996</td>\n",
       "      <td>3584712</td>\n",
       "      <td>-</td>\n",
       "      <td>intron</td>\n",
       "      <td>-0.366654</td>\n",
       "      <td>6.328207e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>0.022739</td>\n",
       "      <td>0.865462</td>\n",
       "      <td>rad25</td>\n",
       "      <td>['C1_03220C_A']</td>\n",
       "      <td>['YDR099W' 'YER177W']</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>TTA |A| AGC</td>\n",
       "      <td>TTAAAGC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>3585317</td>\n",
       "      <td>SPAC17A2.13c</td>\n",
       "      <td>SPAC17A2.13c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3583250</td>\n",
       "      <td>3583995</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>-0.366654</td>\n",
       "      <td>6.328207e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>0.022739</td>\n",
       "      <td>0.865462</td>\n",
       "      <td>rad25</td>\n",
       "      <td>['C1_03220C_A']</td>\n",
       "      <td>['YDR099W' 'YER177W']</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>TTA |A| AGC</td>\n",
       "      <td>TTAAAGC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>3585317</td>\n",
       "      <td>SPAC17A2.14</td>\n",
       "      <td>mnr2</td>\n",
       "      <td>gene</td>\n",
       "      <td>I</td>\n",
       "      <td>3586167</td>\n",
       "      <td>3588777</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>-0.366654</td>\n",
       "      <td>6.328207e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>0.198895</td>\n",
       "      <td>0.134454</td>\n",
       "      <td>mnr2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>CGA |A| ATT</td>\n",
       "      <td>CGAAATT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>3585317</td>\n",
       "      <td>SPAC17A2.14</td>\n",
       "      <td>SPAC17A2.14</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>I</td>\n",
       "      <td>3586167</td>\n",
       "      <td>3588777</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>-0.366654</td>\n",
       "      <td>6.328207e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>0.198895</td>\n",
       "      <td>0.134454</td>\n",
       "      <td>mnr2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>CGA |A| ATT</td>\n",
       "      <td>CGAAATT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>3585317</td>\n",
       "      <td>SPAC17A2.14</td>\n",
       "      <td>SPAC17A2.14.1</td>\n",
       "      <td>five_prime_UTR</td>\n",
       "      <td>I</td>\n",
       "      <td>3586167</td>\n",
       "      <td>3586434</td>\n",
       "      <td>+</td>\n",
       "      <td>five_prime_UTR</td>\n",
       "      <td>-0.366654</td>\n",
       "      <td>6.328207e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>0.198895</td>\n",
       "      <td>0.134454</td>\n",
       "      <td>mnr2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>CGA |A| ATT</td>\n",
       "      <td>CGAAATT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>3585317</td>\n",
       "      <td>SPAC17A2.14</td>\n",
       "      <td>SPAC17A2.14.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3586435</td>\n",
       "      <td>3588288</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>-0.366654</td>\n",
       "      <td>6.328207e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>0.198895</td>\n",
       "      <td>0.134454</td>\n",
       "      <td>mnr2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>CGA |A| ATT</td>\n",
       "      <td>CGAAATT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>3585317</td>\n",
       "      <td>SPNCRNA.3602</td>\n",
       "      <td>None</td>\n",
       "      <td>gene</td>\n",
       "      <td>I</td>\n",
       "      <td>3586123</td>\n",
       "      <td>3588320</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>-0.366654</td>\n",
       "      <td>6.328207e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>TTA |A| AGC</td>\n",
       "      <td>TTAAAGC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>3585317</td>\n",
       "      <td>SPNCRNA.3602</td>\n",
       "      <td>SPNCRNA.3602</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>I</td>\n",
       "      <td>3586123</td>\n",
       "      <td>3588320</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>-0.366654</td>\n",
       "      <td>6.328207e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>TTA |A| AGC</td>\n",
       "      <td>TTAAAGC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>3585317</td>\n",
       "      <td>SPNCRNA.3602</td>\n",
       "      <td>SPNCRNA.3602.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3586123</td>\n",
       "      <td>3588320</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>-0.366654</td>\n",
       "      <td>6.328207e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>TTA |A| AGC</td>\n",
       "      <td>TTAAAGC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>3585317</td>\n",
       "      <td>SPNCRNA.919</td>\n",
       "      <td>None</td>\n",
       "      <td>gene</td>\n",
       "      <td>I</td>\n",
       "      <td>3584833</td>\n",
       "      <td>3585787</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>-0.366654</td>\n",
       "      <td>6.328207e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>0.147888</td>\n",
       "      <td>0.267917</td>\n",
       "      <td>SPNCRNA.919</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>TTA |A| AGC</td>\n",
       "      <td>TTAAAGC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>3585317</td>\n",
       "      <td>SPNCRNA.919</td>\n",
       "      <td>SPNCRNA.919</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>I</td>\n",
       "      <td>3584833</td>\n",
       "      <td>3585787</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>-0.366654</td>\n",
       "      <td>6.328207e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>0.147888</td>\n",
       "      <td>0.267917</td>\n",
       "      <td>SPNCRNA.919</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>TTA |A| AGC</td>\n",
       "      <td>TTAAAGC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>3585317</td>\n",
       "      <td>SPNCRNA.919</td>\n",
       "      <td>SPNCRNA.919.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3584833</td>\n",
       "      <td>3585787</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>-0.366654</td>\n",
       "      <td>6.328207e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>0.147888</td>\n",
       "      <td>0.267917</td>\n",
       "      <td>SPNCRNA.919</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>TTA |A| AGC</td>\n",
       "      <td>TTAAAGC</td>\n",
       "      <td>[ATA, TTA, CCT, TAA, AGC, GGC]</td>\n",
       "      <td>[ATA, TTA, CCT, TAC, AGC, GGC]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>3585317</td>\n",
       "      <td>SPNCRNA.920</td>\n",
       "      <td>None</td>\n",
       "      <td>gene</td>\n",
       "      <td>I</td>\n",
       "      <td>3584979</td>\n",
       "      <td>3586233</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>-0.366654</td>\n",
       "      <td>6.328207e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>-0.140556</td>\n",
       "      <td>0.292627</td>\n",
       "      <td>SPNCRNA.920</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>CGA |A| ATT</td>\n",
       "      <td>CGAAATT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>3585317</td>\n",
       "      <td>SPNCRNA.920</td>\n",
       "      <td>SPNCRNA.920</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>I</td>\n",
       "      <td>3584979</td>\n",
       "      <td>3586233</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>-0.366654</td>\n",
       "      <td>6.328207e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>-0.140556</td>\n",
       "      <td>0.292627</td>\n",
       "      <td>SPNCRNA.920</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>CGA |A| ATT</td>\n",
       "      <td>CGAAATT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>3585317</td>\n",
       "      <td>SPNCRNA.920</td>\n",
       "      <td>SPNCRNA.920.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3584979</td>\n",
       "      <td>3586233</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>-0.366654</td>\n",
       "      <td>6.328207e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>-0.140556</td>\n",
       "      <td>0.292627</td>\n",
       "      <td>SPNCRNA.920</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>CGA |A| ATT</td>\n",
       "      <td>CGAAATT</td>\n",
       "      <td>[GGA, GGC, GGC, GAA, ATT, CCA]</td>\n",
       "      <td>[GGA, GGC, GGC, GAC, ATT, CCA]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>3585317</td>\n",
       "      <td>SPLTRA.58</td>\n",
       "      <td>None</td>\n",
       "      <td>long_terminal_repeat</td>\n",
       "      <td>I</td>\n",
       "      <td>3586218</td>\n",
       "      <td>3586423</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>-0.366654</td>\n",
       "      <td>6.328207e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>CGA |A| ATT</td>\n",
       "      <td>CGAAATT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>4296875</td>\n",
       "      <td>SPBC16A3.02c</td>\n",
       "      <td>None</td>\n",
       "      <td>gene</td>\n",
       "      <td>II</td>\n",
       "      <td>4296979</td>\n",
       "      <td>4298874</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>0.443424</td>\n",
       "      <td>8.445186e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>0.229019</td>\n",
       "      <td>0.083764</td>\n",
       "      <td>SPBC16A3.02c</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>TAT |T| AGT</td>\n",
       "      <td>TATTAGT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>4296875</td>\n",
       "      <td>SPBC16A3.02c</td>\n",
       "      <td>SPBC16A3.02c</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>II</td>\n",
       "      <td>4296979</td>\n",
       "      <td>4298874</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>0.443424</td>\n",
       "      <td>8.445186e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>0.229019</td>\n",
       "      <td>0.083764</td>\n",
       "      <td>SPBC16A3.02c</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>TAT |T| AGT</td>\n",
       "      <td>TATTAGT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>4296875</td>\n",
       "      <td>SPBC16A3.02c</td>\n",
       "      <td>SPBC16A3.02c.1</td>\n",
       "      <td>five_prime_UTR</td>\n",
       "      <td>II</td>\n",
       "      <td>4296979</td>\n",
       "      <td>4297583</td>\n",
       "      <td>+</td>\n",
       "      <td>five_prime_UTR</td>\n",
       "      <td>0.443424</td>\n",
       "      <td>8.445186e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>0.229019</td>\n",
       "      <td>0.083764</td>\n",
       "      <td>SPBC16A3.02c</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>TAT |T| AGT</td>\n",
       "      <td>TATTAGT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>4296875</td>\n",
       "      <td>SPBC16A3.02c</td>\n",
       "      <td>SPBC16A3.02c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>4297584</td>\n",
       "      <td>4298627</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.443424</td>\n",
       "      <td>8.445186e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>0.229019</td>\n",
       "      <td>0.083764</td>\n",
       "      <td>SPBC16A3.02c</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>TAT |T| AGT</td>\n",
       "      <td>TATTAGT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>4296875</td>\n",
       "      <td>SPBC16A3.03c</td>\n",
       "      <td>ppr7</td>\n",
       "      <td>gene</td>\n",
       "      <td>II</td>\n",
       "      <td>4293219</td>\n",
       "      <td>4296025</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>0.443424</td>\n",
       "      <td>8.445186e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>-0.192218</td>\n",
       "      <td>0.148305</td>\n",
       "      <td>ppr7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>TAT |T| AGT</td>\n",
       "      <td>TATTAGT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>4296875</td>\n",
       "      <td>SPBC16A3.03c</td>\n",
       "      <td>SPBC16A3.03c</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>II</td>\n",
       "      <td>4293219</td>\n",
       "      <td>4296025</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>0.443424</td>\n",
       "      <td>8.445186e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>-0.192218</td>\n",
       "      <td>0.148305</td>\n",
       "      <td>ppr7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>TAT |T| AGT</td>\n",
       "      <td>TATTAGT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>4296875</td>\n",
       "      <td>SPBC16A3.03c</td>\n",
       "      <td>SPBC16A3.03c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>4293597</td>\n",
       "      <td>4295573</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.443424</td>\n",
       "      <td>8.445186e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>-0.192218</td>\n",
       "      <td>0.148305</td>\n",
       "      <td>ppr7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>TAT |T| AGT</td>\n",
       "      <td>TATTAGT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>4296875</td>\n",
       "      <td>SPBC16A3.03c</td>\n",
       "      <td>SPBC16A3.03c.1</td>\n",
       "      <td>three_prime_UTR</td>\n",
       "      <td>II</td>\n",
       "      <td>4295574</td>\n",
       "      <td>4296025</td>\n",
       "      <td>+</td>\n",
       "      <td>three_prime_UTR</td>\n",
       "      <td>0.443424</td>\n",
       "      <td>8.445186e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>-0.192218</td>\n",
       "      <td>0.148305</td>\n",
       "      <td>ppr7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>TAT |T| AGT</td>\n",
       "      <td>TATTAGT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>4296875</td>\n",
       "      <td>SPNCRNA.6532</td>\n",
       "      <td>None</td>\n",
       "      <td>gene</td>\n",
       "      <td>II</td>\n",
       "      <td>4294696</td>\n",
       "      <td>4296412</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>0.443424</td>\n",
       "      <td>8.445186e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>TGA |T| TAT</td>\n",
       "      <td>TGATTAT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>4296875</td>\n",
       "      <td>SPNCRNA.6532</td>\n",
       "      <td>SPNCRNA.6532</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>II</td>\n",
       "      <td>4294696</td>\n",
       "      <td>4296412</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>0.443424</td>\n",
       "      <td>8.445186e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>TGA |T| TAT</td>\n",
       "      <td>TGATTAT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>4296875</td>\n",
       "      <td>SPNCRNA.6532</td>\n",
       "      <td>SPNCRNA.6532.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>4294696</td>\n",
       "      <td>4296412</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.443424</td>\n",
       "      <td>8.445186e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>TGA |T| TAT</td>\n",
       "      <td>TGATTAT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>4296875</td>\n",
       "      <td>SPNCRNA.6533</td>\n",
       "      <td>None</td>\n",
       "      <td>gene</td>\n",
       "      <td>II</td>\n",
       "      <td>4296590</td>\n",
       "      <td>4296958</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>0.443424</td>\n",
       "      <td>8.445186e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>TAT |T| AGT</td>\n",
       "      <td>TATTAGT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>4296875</td>\n",
       "      <td>SPNCRNA.6533</td>\n",
       "      <td>SPNCRNA.6533</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>II</td>\n",
       "      <td>4296590</td>\n",
       "      <td>4296958</td>\n",
       "      <td>+</td>\n",
       "      <td>None</td>\n",
       "      <td>0.443424</td>\n",
       "      <td>8.445186e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>TAT |T| AGT</td>\n",
       "      <td>TATTAGT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    position            ID          Parent                object chromosome  \\\n",
       "0    3364859  SPAC26A3.13c           Tf2-4                  gene          I   \n",
       "1    3364859  SPAC26A3.13c    SPAC26A3.13c                  mRNA          I   \n",
       "2    3364859  SPAC26A3.13c  SPAC26A3.13c.1                   CDS          I   \n",
       "3    3364859  SPAC26A3.14c            None                  gene          I   \n",
       "4    3364859  SPAC26A3.14c    SPAC26A3.14c                  mRNA          I   \n",
       "5    3364859  SPAC26A3.14c  SPAC26A3.14c.1       three_prime_UTR          I   \n",
       "6    3364859  SPNCRNA.3503            None                  gene          I   \n",
       "7    3364859  SPNCRNA.3503    SPNCRNA.3503                 ncRNA          I   \n",
       "8    3364859  SPNCRNA.3503  SPNCRNA.3503.1                   CDS          I   \n",
       "9    3364859  SPNCRNA.3504            None                  gene          I   \n",
       "10   3364859  SPNCRNA.3504    SPNCRNA.3504                 ncRNA          I   \n",
       "11   3364859  SPNCRNA.3504  SPNCRNA.3504.1                   CDS          I   \n",
       "12   3364859  SPNCRNA.3505            None                  gene          I   \n",
       "13   3364859  SPNCRNA.3505    SPNCRNA.3505                 ncRNA          I   \n",
       "14   3364859  SPNCRNA.3505  SPNCRNA.3505.1                   CDS          I   \n",
       "15   3364859        SPTF.5            None   LTR_retrotransposon          I   \n",
       "16   3364859     SPLTRA.55            None  long_terminal_repeat          I   \n",
       "17   3585317  SPAC17A2.13c           rad25                  gene          I   \n",
       "18   3585317  SPAC17A2.13c    SPAC17A2.13c                  mRNA          I   \n",
       "19   3585317  SPAC17A2.13c  SPAC17A2.13c.1        five_prime_UTR          I   \n",
       "20   3585317  SPAC17A2.13c  SPAC17A2.13c.1                   CDS          I   \n",
       "21   3585317  SPAC17A2.13c  SPAC17A2.13c.1                intron          I   \n",
       "22   3585317  SPAC17A2.13c  SPAC17A2.13c.1                   CDS          I   \n",
       "23   3585317   SPAC17A2.14            mnr2                  gene          I   \n",
       "24   3585317   SPAC17A2.14     SPAC17A2.14                  mRNA          I   \n",
       "25   3585317   SPAC17A2.14   SPAC17A2.14.1        five_prime_UTR          I   \n",
       "26   3585317   SPAC17A2.14   SPAC17A2.14.1                   CDS          I   \n",
       "27   3585317  SPNCRNA.3602            None                  gene          I   \n",
       "28   3585317  SPNCRNA.3602    SPNCRNA.3602                 ncRNA          I   \n",
       "29   3585317  SPNCRNA.3602  SPNCRNA.3602.1                   CDS          I   \n",
       "30   3585317   SPNCRNA.919            None                  gene          I   \n",
       "31   3585317   SPNCRNA.919     SPNCRNA.919                 ncRNA          I   \n",
       "32   3585317   SPNCRNA.919   SPNCRNA.919.1                   CDS          I   \n",
       "33   3585317   SPNCRNA.920            None                  gene          I   \n",
       "34   3585317   SPNCRNA.920     SPNCRNA.920                 ncRNA          I   \n",
       "35   3585317   SPNCRNA.920   SPNCRNA.920.1                   CDS          I   \n",
       "36   3585317     SPLTRA.58            None  long_terminal_repeat          I   \n",
       "37   4296875  SPBC16A3.02c            None                  gene         II   \n",
       "38   4296875  SPBC16A3.02c    SPBC16A3.02c                  mRNA         II   \n",
       "39   4296875  SPBC16A3.02c  SPBC16A3.02c.1        five_prime_UTR         II   \n",
       "40   4296875  SPBC16A3.02c  SPBC16A3.02c.1                   CDS         II   \n",
       "41   4296875  SPBC16A3.03c            ppr7                  gene         II   \n",
       "42   4296875  SPBC16A3.03c    SPBC16A3.03c                  mRNA         II   \n",
       "43   4296875  SPBC16A3.03c  SPBC16A3.03c.1                   CDS         II   \n",
       "44   4296875  SPBC16A3.03c  SPBC16A3.03c.1       three_prime_UTR         II   \n",
       "45   4296875  SPNCRNA.6532            None                  gene         II   \n",
       "46   4296875  SPNCRNA.6532    SPNCRNA.6532                 ncRNA         II   \n",
       "47   4296875  SPNCRNA.6532  SPNCRNA.6532.1                   CDS         II   \n",
       "48   4296875  SPNCRNA.6533            None                  gene         II   \n",
       "49   4296875  SPNCRNA.6533    SPNCRNA.6533                 ncRNA         II   \n",
       "\n",
       "      start      end strand           detail      beta       p_value type  \\\n",
       "0   3361547  3365548      -             None  0.546460  1.620139e-09  snp   \n",
       "1   3361547  3365548      -             None  0.546460  1.620139e-09  snp   \n",
       "2   3361547  3365548      -             exon  0.546460  1.620139e-09  snp   \n",
       "3   3366317  3367205      -             None  0.546460  1.620139e-09  snp   \n",
       "4   3366317  3367205      -             None  0.546460  1.620139e-09  snp   \n",
       "5   3366317  3366742      -  three_prime_UTR  0.546460  1.620139e-09  snp   \n",
       "6   3364029  3364812      +             None  0.546460  1.620139e-09  snp   \n",
       "7   3364029  3364812      +             None  0.546460  1.620139e-09  snp   \n",
       "8   3364029  3364812      +             exon  0.546460  1.620139e-09  snp   \n",
       "9   3365076  3365865      +             None  0.546460  1.620139e-09  snp   \n",
       "10  3365076  3365865      +             None  0.546460  1.620139e-09  snp   \n",
       "11  3365076  3365865      +             exon  0.546460  1.620139e-09  snp   \n",
       "12  3366144  3367500      +             None  0.546460  1.620139e-09  snp   \n",
       "13  3366144  3367500      +             None  0.546460  1.620139e-09  snp   \n",
       "14  3366144  3367500      +             exon  0.546460  1.620139e-09  snp   \n",
       "15  3361143  3366062      -             None  0.546460  1.620139e-09  snp   \n",
       "16  3365714  3366062      -             None  0.546460  1.620139e-09  snp   \n",
       "17  3583128  3584899      -             None -0.366654  6.328207e-08  snp   \n",
       "18  3583128  3584899      -             None -0.366654  6.328207e-08  snp   \n",
       "19  3584780  3584899      -   five_prime_UTR -0.366654  6.328207e-08  snp   \n",
       "20  3584713  3584779      -             exon -0.366654  6.328207e-08  snp   \n",
       "21  3583996  3584712      -           intron -0.366654  6.328207e-08  snp   \n",
       "22  3583250  3583995      -             exon -0.366654  6.328207e-08  snp   \n",
       "23  3586167  3588777      +             None -0.366654  6.328207e-08  snp   \n",
       "24  3586167  3588777      +             None -0.366654  6.328207e-08  snp   \n",
       "25  3586167  3586434      +   five_prime_UTR -0.366654  6.328207e-08  snp   \n",
       "26  3586435  3588288      +             exon -0.366654  6.328207e-08  snp   \n",
       "27  3586123  3588320      -             None -0.366654  6.328207e-08  snp   \n",
       "28  3586123  3588320      -             None -0.366654  6.328207e-08  snp   \n",
       "29  3586123  3588320      -             exon -0.366654  6.328207e-08  snp   \n",
       "30  3584833  3585787      -             None -0.366654  6.328207e-08  snp   \n",
       "31  3584833  3585787      -             None -0.366654  6.328207e-08  snp   \n",
       "32  3584833  3585787      -             exon -0.366654  6.328207e-08  snp   \n",
       "33  3584979  3586233      +             None -0.366654  6.328207e-08  snp   \n",
       "34  3584979  3586233      +             None -0.366654  6.328207e-08  snp   \n",
       "35  3584979  3586233      +             exon -0.366654  6.328207e-08  snp   \n",
       "36  3586218  3586423      +             None -0.366654  6.328207e-08  snp   \n",
       "37  4296979  4298874      +             None  0.443424  8.445186e-08  snp   \n",
       "38  4296979  4298874      +             None  0.443424  8.445186e-08  snp   \n",
       "39  4296979  4297583      +   five_prime_UTR  0.443424  8.445186e-08  snp   \n",
       "40  4297584  4298627      +             exon  0.443424  8.445186e-08  snp   \n",
       "41  4293219  4296025      +             None  0.443424  8.445186e-08  snp   \n",
       "42  4293219  4296025      +             None  0.443424  8.445186e-08  snp   \n",
       "43  4293597  4295573      +             exon  0.443424  8.445186e-08  snp   \n",
       "44  4295574  4296025      +  three_prime_UTR  0.443424  8.445186e-08  snp   \n",
       "45  4294696  4296412      -             None  0.443424  8.445186e-08  snp   \n",
       "46  4294696  4296412      -             None  0.443424  8.445186e-08  snp   \n",
       "47  4294696  4296412      -             exon  0.443424  8.445186e-08  snp   \n",
       "48  4296590  4296958      +             None  0.443424  8.445186e-08  snp   \n",
       "49  4296590  4296958      +             None  0.443424  8.445186e-08  snp   \n",
       "\n",
       "    correlation  p_value_corr          gene albicans_name(s)  \\\n",
       "0           NaN           NaN         Tf2-4              NaN   \n",
       "1           NaN           NaN         Tf2-4              NaN   \n",
       "2           NaN           NaN         Tf2-4              NaN   \n",
       "3     -0.399422      0.001896  SPAC26A3.14c              NaN   \n",
       "4     -0.399422      0.001896  SPAC26A3.14c              NaN   \n",
       "5     -0.399422      0.001896  SPAC26A3.14c              NaN   \n",
       "6           NaN           NaN           NaN              NaN   \n",
       "7           NaN           NaN           NaN              NaN   \n",
       "8           NaN           NaN           NaN              NaN   \n",
       "9           NaN           NaN           NaN              NaN   \n",
       "10          NaN           NaN           NaN              NaN   \n",
       "11          NaN           NaN           NaN              NaN   \n",
       "12          NaN           NaN           NaN              NaN   \n",
       "13          NaN           NaN           NaN              NaN   \n",
       "14          NaN           NaN           NaN              NaN   \n",
       "15          NaN           NaN           NaN              NaN   \n",
       "16          NaN           NaN           NaN              NaN   \n",
       "17     0.022739      0.865462         rad25  ['C1_03220C_A']   \n",
       "18     0.022739      0.865462         rad25  ['C1_03220C_A']   \n",
       "19     0.022739      0.865462         rad25  ['C1_03220C_A']   \n",
       "20     0.022739      0.865462         rad25  ['C1_03220C_A']   \n",
       "21     0.022739      0.865462         rad25  ['C1_03220C_A']   \n",
       "22     0.022739      0.865462         rad25  ['C1_03220C_A']   \n",
       "23     0.198895      0.134454          mnr2              NaN   \n",
       "24     0.198895      0.134454          mnr2              NaN   \n",
       "25     0.198895      0.134454          mnr2              NaN   \n",
       "26     0.198895      0.134454          mnr2              NaN   \n",
       "27          NaN           NaN           NaN              NaN   \n",
       "28          NaN           NaN           NaN              NaN   \n",
       "29          NaN           NaN           NaN              NaN   \n",
       "30     0.147888      0.267917   SPNCRNA.919              NaN   \n",
       "31     0.147888      0.267917   SPNCRNA.919              NaN   \n",
       "32     0.147888      0.267917   SPNCRNA.919              NaN   \n",
       "33    -0.140556      0.292627   SPNCRNA.920              NaN   \n",
       "34    -0.140556      0.292627   SPNCRNA.920              NaN   \n",
       "35    -0.140556      0.292627   SPNCRNA.920              NaN   \n",
       "36          NaN           NaN           NaN              NaN   \n",
       "37     0.229019      0.083764  SPBC16A3.02c              NaN   \n",
       "38     0.229019      0.083764  SPBC16A3.02c              NaN   \n",
       "39     0.229019      0.083764  SPBC16A3.02c              NaN   \n",
       "40     0.229019      0.083764  SPBC16A3.02c              NaN   \n",
       "41    -0.192218      0.148305          ppr7              NaN   \n",
       "42    -0.192218      0.148305          ppr7              NaN   \n",
       "43    -0.192218      0.148305          ppr7              NaN   \n",
       "44    -0.192218      0.148305          ppr7              NaN   \n",
       "45          NaN           NaN           NaN              NaN   \n",
       "46          NaN           NaN           NaN              NaN   \n",
       "47          NaN           NaN           NaN              NaN   \n",
       "48          NaN           NaN           NaN              NaN   \n",
       "49          NaN           NaN           NaN              NaN   \n",
       "\n",
       "       cerevisiae_name(s) allele_1 allele_2 reference_seq reference_seq_  \\\n",
       "0                     NaN        C        T   CTT |C| TGT        CTTCTGT   \n",
       "1                     NaN        C        T   CTT |C| TGT        CTTCTGT   \n",
       "2                     NaN        C        T   CTT |C| TGT        CTTCTGT   \n",
       "3                     NaN        C        T   CTT |C| TGT        CTTCTGT   \n",
       "4                     NaN        C        T   CTT |C| TGT        CTTCTGT   \n",
       "5                     NaN        C        T   CTT |C| TGT        CTTCTGT   \n",
       "6                     NaN        C        T   TGT |C| TTC        TGTCTTC   \n",
       "7                     NaN        C        T   TGT |C| TTC        TGTCTTC   \n",
       "8                     NaN        C        T   TGT |C| TTC        TGTCTTC   \n",
       "9                     NaN        C        T   TGT |C| TTC        TGTCTTC   \n",
       "10                    NaN        C        T   TGT |C| TTC        TGTCTTC   \n",
       "11                    NaN        C        T   TGT |C| TTC        TGTCTTC   \n",
       "12                    NaN        C        T   TGT |C| TTC        TGTCTTC   \n",
       "13                    NaN        C        T   TGT |C| TTC        TGTCTTC   \n",
       "14                    NaN        C        T   TGT |C| TTC        TGTCTTC   \n",
       "15                    NaN        C        T   CTT |C| TGT        CTTCTGT   \n",
       "16                    NaN        C        T   CTT |C| TGT        CTTCTGT   \n",
       "17  ['YDR099W' 'YER177W']        A        C   TTA |A| AGC        TTAAAGC   \n",
       "18  ['YDR099W' 'YER177W']        A        C   TTA |A| AGC        TTAAAGC   \n",
       "19  ['YDR099W' 'YER177W']        A        C   TTA |A| AGC        TTAAAGC   \n",
       "20  ['YDR099W' 'YER177W']        A        C   TTA |A| AGC        TTAAAGC   \n",
       "21  ['YDR099W' 'YER177W']        A        C   TTA |A| AGC        TTAAAGC   \n",
       "22  ['YDR099W' 'YER177W']        A        C   TTA |A| AGC        TTAAAGC   \n",
       "23                    NaN        A        C   CGA |A| ATT        CGAAATT   \n",
       "24                    NaN        A        C   CGA |A| ATT        CGAAATT   \n",
       "25                    NaN        A        C   CGA |A| ATT        CGAAATT   \n",
       "26                    NaN        A        C   CGA |A| ATT        CGAAATT   \n",
       "27                    NaN        A        C   TTA |A| AGC        TTAAAGC   \n",
       "28                    NaN        A        C   TTA |A| AGC        TTAAAGC   \n",
       "29                    NaN        A        C   TTA |A| AGC        TTAAAGC   \n",
       "30                    NaN        A        C   TTA |A| AGC        TTAAAGC   \n",
       "31                    NaN        A        C   TTA |A| AGC        TTAAAGC   \n",
       "32                    NaN        A        C   TTA |A| AGC        TTAAAGC   \n",
       "33                    NaN        A        C   CGA |A| ATT        CGAAATT   \n",
       "34                    NaN        A        C   CGA |A| ATT        CGAAATT   \n",
       "35                    NaN        A        C   CGA |A| ATT        CGAAATT   \n",
       "36                    NaN        A        C   CGA |A| ATT        CGAAATT   \n",
       "37                    NaN        T        A   TAT |T| AGT        TATTAGT   \n",
       "38                    NaN        T        A   TAT |T| AGT        TATTAGT   \n",
       "39                    NaN        T        A   TAT |T| AGT        TATTAGT   \n",
       "40                    NaN        T        A   TAT |T| AGT        TATTAGT   \n",
       "41                    NaN        T        A   TAT |T| AGT        TATTAGT   \n",
       "42                    NaN        T        A   TAT |T| AGT        TATTAGT   \n",
       "43                    NaN        T        A   TAT |T| AGT        TATTAGT   \n",
       "44                    NaN        T        A   TAT |T| AGT        TATTAGT   \n",
       "45                    NaN        T        A   TGA |T| TAT        TGATTAT   \n",
       "46                    NaN        T        A   TGA |T| TAT        TGATTAT   \n",
       "47                    NaN        T        A   TGA |T| TAT        TGATTAT   \n",
       "48                    NaN        T        A   TAT |T| AGT        TATTAGT   \n",
       "49                    NaN        T        A   TAT |T| AGT        TATTAGT   \n",
       "\n",
       "                                wt_gene                        mutant allele  \n",
       "0                                   NaN                                  NaN  \n",
       "1                                   NaN                                  NaN  \n",
       "2   [TTG, GAC, TCC, TTC, TGT, TTA, CAA]  [TTG, GAC, TCC, TTT, TGT, TTA, CAA]  \n",
       "3                                   NaN                                  NaN  \n",
       "4                                   NaN                                  NaN  \n",
       "5                                   NaN                                  NaN  \n",
       "6                                   NaN                                  NaN  \n",
       "7                                   NaN                                  NaN  \n",
       "8                                   NaN                                  NaN  \n",
       "9                                   NaN                                  NaN  \n",
       "10                                  NaN                                  NaN  \n",
       "11                                  NaN                                  NaN  \n",
       "12                                  NaN                                  NaN  \n",
       "13                                  NaN                                  NaN  \n",
       "14                                  NaN                                  NaN  \n",
       "15                                  NaN                                  NaN  \n",
       "16                                  NaN                                  NaN  \n",
       "17                                  NaN                                  NaN  \n",
       "18                                  NaN                                  NaN  \n",
       "19                                  NaN                                  NaN  \n",
       "20                                  NaN                                  NaN  \n",
       "21                                  NaN                                  NaN  \n",
       "22                                  NaN                                  NaN  \n",
       "23                                  NaN                                  NaN  \n",
       "24                                  NaN                                  NaN  \n",
       "25                                  NaN                                  NaN  \n",
       "26                                  NaN                                  NaN  \n",
       "27                                  NaN                                  NaN  \n",
       "28                                  NaN                                  NaN  \n",
       "29                                  NaN                                  NaN  \n",
       "30                                  NaN                                  NaN  \n",
       "31                                  NaN                                  NaN  \n",
       "32       [ATA, TTA, CCT, TAA, AGC, GGC]       [ATA, TTA, CCT, TAC, AGC, GGC]  \n",
       "33                                  NaN                                  NaN  \n",
       "34                                  NaN                                  NaN  \n",
       "35       [GGA, GGC, GGC, GAA, ATT, CCA]       [GGA, GGC, GGC, GAC, ATT, CCA]  \n",
       "36                                  NaN                                  NaN  \n",
       "37                                  NaN                                  NaN  \n",
       "38                                  NaN                                  NaN  \n",
       "39                                  NaN                                  NaN  \n",
       "40                                  NaN                                  NaN  \n",
       "41                                  NaN                                  NaN  \n",
       "42                                  NaN                                  NaN  \n",
       "43                                  NaN                                  NaN  \n",
       "44                                  NaN                                  NaN  \n",
       "45                                  NaN                                  NaN  \n",
       "46                                  NaN                                  NaN  \n",
       "47                                  NaN                                  NaN  \n",
       "48                                  NaN                                  NaN  \n",
       "49                                  NaN                                  NaN  "
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "result_final__.head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>position</th>\n",
       "      <th>ID</th>\n",
       "      <th>Parent</th>\n",
       "      <th>object</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>detail</th>\n",
       "      <th>beta</th>\n",
       "      <th>...</th>\n",
       "      <th>gene</th>\n",
       "      <th>albicans_name(s)</th>\n",
       "      <th>cerevisiae_name(s)</th>\n",
       "      <th>allele_1</th>\n",
       "      <th>allele_2</th>\n",
       "      <th>reference_seq</th>\n",
       "      <th>reference_seq_</th>\n",
       "      <th>wt_gene</th>\n",
       "      <th>mutant allele</th>\n",
       "      <th>stop_codon_predicted</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4296875</td>\n",
       "      <td>SPNCRNA.6533</td>\n",
       "      <td>SPNCRNA.6533.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>4296590</td>\n",
       "      <td>4296958</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.443424</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>TAT |T| AGT</td>\n",
       "      <td>TATTAGT</td>\n",
       "      <td>[GAG, CTA, TAT, TAG, TGA, TAG]</td>\n",
       "      <td>[GAG, CTA, TAT, AAG, TGA, TAG]</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4296877</td>\n",
       "      <td>SPNCRNA.6533</td>\n",
       "      <td>SPNCRNA.6533.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>4296590</td>\n",
       "      <td>4296958</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.443424</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>TTA |G| TGA</td>\n",
       "      <td>TTAGTGA</td>\n",
       "      <td>[GAG, CTA, TAT, TAG, TGA, TAG, GTA]</td>\n",
       "      <td>[GAG, CTA, TAT, TAA, TGA, TAG, GTA]</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>2227937</td>\n",
       "      <td>SPNCRNA.7677</td>\n",
       "      <td>SPNCRNA.7677.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>III</td>\n",
       "      <td>2227612</td>\n",
       "      <td>2228044</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.558370</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>TTC |T| GGC</td>\n",
       "      <td>TTCTGGC</td>\n",
       "      <td>[TTT, TGA, ATT, CTG, GCT, TGG]</td>\n",
       "      <td>[TTT, TGA, ATT, CGG, GCT, TGG]</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>498798</td>\n",
       "      <td>SPNCRNA.2204</td>\n",
       "      <td>SPNCRNA.2204.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>497961</td>\n",
       "      <td>498854</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.311554</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GT</td>\n",
       "      <td>T</td>\n",
       "      <td>GGG |T| TTA</td>\n",
       "      <td>GGGTTTA</td>\n",
       "      <td>[CTT, TTG, GGG, GGT, TTA, ATT, AGT]</td>\n",
       "      <td>[CTT, TTG, GGG, GGT, TTA, ATT, AGT]</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>2693728</td>\n",
       "      <td>SPNCRNA.5773</td>\n",
       "      <td>SPNCRNA.5773.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2693622</td>\n",
       "      <td>2693887</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>-0.397447</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AC</td>\n",
       "      <td>AACA</td>\n",
       "      <td>AAA |A| CAA</td>\n",
       "      <td>AAAACAA</td>\n",
       "      <td>[GAC, TAA, AAA, AAC, AAA, AAA]</td>\n",
       "      <td>[GAC, TAA, AAA, AAA, CAC, AAA, AAA]</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>2693728</td>\n",
       "      <td>SPNCRNA.5774</td>\n",
       "      <td>SPNCRNA.5774.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2693641</td>\n",
       "      <td>2694958</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>-0.397447</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CA</td>\n",
       "      <td>ACAA</td>\n",
       "      <td>AAC |A| AAA</td>\n",
       "      <td>AACAAAA</td>\n",
       "      <td>[CAA, AAA, AAC, AAA, AAA, ATC]</td>\n",
       "      <td>[CAA, AAA, AAC, ACA, AAA, AAA, ATC]</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>2694024</td>\n",
       "      <td>SPNCRNA.5774</td>\n",
       "      <td>SPNCRNA.5774.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2693641</td>\n",
       "      <td>2694958</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>-0.397447</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>TA</td>\n",
       "      <td>TTT |A| GAA</td>\n",
       "      <td>TTTAGAA</td>\n",
       "      <td>[ATA, TGA, ATT, TAG, AAA, ATT]</td>\n",
       "      <td>[ATA, TGA, ATT, TTA, GAA, AAT]</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>2277085</td>\n",
       "      <td>SPNCRNA.516</td>\n",
       "      <td>SPNCRNA.516.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>III</td>\n",
       "      <td>2275648</td>\n",
       "      <td>2277744</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.401600</td>\n",
       "      <td>...</td>\n",
       "      <td>SPNCRNA.516</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>TC</td>\n",
       "      <td>TTA |T| AAT</td>\n",
       "      <td>TTATAAT</td>\n",
       "      <td>[CTT, GTC, TTA, TAA, TGT, AAA]</td>\n",
       "      <td>[CTT, GTC, TTA, TCA, ATG, TAA]</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    position            ID          Parent object chromosome    start  \\\n",
       "3    4296875  SPNCRNA.6533  SPNCRNA.6533.1    CDS         II  4296590   \n",
       "4    4296877  SPNCRNA.6533  SPNCRNA.6533.1    CDS         II  4296590   \n",
       "7    2227937  SPNCRNA.7677  SPNCRNA.7677.1    CDS        III  2227612   \n",
       "9     498798  SPNCRNA.2204  SPNCRNA.2204.1    CDS          I   497961   \n",
       "18   2693728  SPNCRNA.5773  SPNCRNA.5773.1    CDS         II  2693622   \n",
       "19   2693728  SPNCRNA.5774  SPNCRNA.5774.1    CDS         II  2693641   \n",
       "20   2694024  SPNCRNA.5774  SPNCRNA.5774.1    CDS         II  2693641   \n",
       "22   2277085   SPNCRNA.516   SPNCRNA.516.1    CDS        III  2275648   \n",
       "\n",
       "        end strand detail      beta  ...         gene albicans_name(s)  \\\n",
       "3   4296958      +   exon  0.443424  ...          NaN              NaN   \n",
       "4   4296958      +   exon  0.443424  ...          NaN              NaN   \n",
       "7   2228044      +   exon  0.558370  ...          NaN              NaN   \n",
       "9    498854      -   exon  0.311554  ...          NaN              NaN   \n",
       "18  2693887      +   exon -0.397447  ...          NaN              NaN   \n",
       "19  2694958      -   exon -0.397447  ...          NaN              NaN   \n",
       "20  2694958      -   exon -0.397447  ...          NaN              NaN   \n",
       "22  2277744      +   exon  0.401600  ...  SPNCRNA.516              NaN   \n",
       "\n",
       "    cerevisiae_name(s)  allele_1 allele_2 reference_seq reference_seq_  \\\n",
       "3                  NaN         T        A   TAT |T| AGT        TATTAGT   \n",
       "4                  NaN         G        A   TTA |G| TGA        TTAGTGA   \n",
       "7                  NaN         T        G   TTC |T| GGC        TTCTGGC   \n",
       "9                  NaN        GT        T   GGG |T| TTA        GGGTTTA   \n",
       "18                 NaN        AC     AACA   AAA |A| CAA        AAAACAA   \n",
       "19                 NaN        CA     ACAA   AAC |A| AAA        AACAAAA   \n",
       "20                 NaN         A       TA   TTT |A| GAA        TTTAGAA   \n",
       "22                 NaN         T       TC   TTA |T| AAT        TTATAAT   \n",
       "\n",
       "                                wt_gene                        mutant allele  \\\n",
       "3        [GAG, CTA, TAT, TAG, TGA, TAG]       [GAG, CTA, TAT, AAG, TGA, TAG]   \n",
       "4   [GAG, CTA, TAT, TAG, TGA, TAG, GTA]  [GAG, CTA, TAT, TAA, TGA, TAG, GTA]   \n",
       "7        [TTT, TGA, ATT, CTG, GCT, TGG]       [TTT, TGA, ATT, CGG, GCT, TGG]   \n",
       "9   [CTT, TTG, GGG, GGT, TTA, ATT, AGT]  [CTT, TTG, GGG, GGT, TTA, ATT, AGT]   \n",
       "18       [GAC, TAA, AAA, AAC, AAA, AAA]  [GAC, TAA, AAA, AAA, CAC, AAA, AAA]   \n",
       "19       [CAA, AAA, AAC, AAA, AAA, ATC]  [CAA, AAA, AAC, ACA, AAA, AAA, ATC]   \n",
       "20       [ATA, TGA, ATT, TAG, AAA, ATT]       [ATA, TGA, ATT, TTA, GAA, AAT]   \n",
       "22       [CTT, GTC, TTA, TAA, TGT, AAA]       [CTT, GTC, TTA, TCA, ATG, TAA]   \n",
       "\n",
       "   stop_codon_predicted  \n",
       "3                   yes  \n",
       "4                   yes  \n",
       "7                   yes  \n",
       "9                   yes  \n",
       "18                  yes  \n",
       "19                  yes  \n",
       "20                  yes  \n",
       "22                  yes  \n",
       "\n",
       "[8 rows x 24 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#CreATE A DF WHERE WHOLE GENE IS NOT NAN\n",
    "stop_codon_df = result_final__[result_final__['wt_gene'].notna()]\n",
    "stop_codon_df[\"stop_codon_predicted\"] = np.nan\n",
    "#iterate through mutant allele, and see whether it contains\n",
    "#reset index\n",
    "stop_codon_df = stop_codon_df.reset_index(drop=True)\n",
    "\n",
    "for i in range(len(stop_codon_df)):\n",
    "    for j in range(len(stop_codon_df['mutant allele'][i])):\n",
    "        if stop_codon_df[\"strand\"][i] == '+':\n",
    "            if stop_codon_df['mutant allele'][i][j] == 'TAA' or stop_codon_df['mutant allele'][i][j] == 'TAG' or stop_codon_df['mutant allele'][i][j] == 'TGA':\n",
    "                stop_codon_df['stop_codon_predicted'][i] = 'yes'\n",
    "        else:\n",
    "            if stop_codon_df['mutant allele'][i][j] == 'ATT' or stop_codon_df['mutant allele'][i][j] == 'ATC' or stop_codon_df['mutant allele'][i][j] == 'ACT':\n",
    "                stop_codon_df['stop_codon_predicted'][i] = 'yes'\n",
    "#show which entries have a stop codon predicted\n",
    "stop_codon_df[stop_codon_df['stop_codon_predicted'] == 'yes']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>position</th>\n",
       "      <th>ID</th>\n",
       "      <th>Parent</th>\n",
       "      <th>object</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>detail</th>\n",
       "      <th>beta</th>\n",
       "      <th>...</th>\n",
       "      <th>p_value_corr</th>\n",
       "      <th>gene</th>\n",
       "      <th>albicans_name(s)</th>\n",
       "      <th>cerevisiae_name(s)</th>\n",
       "      <th>allele_1</th>\n",
       "      <th>allele_2</th>\n",
       "      <th>reference_seq</th>\n",
       "      <th>reference_seq_</th>\n",
       "      <th>wt_gene</th>\n",
       "      <th>mutant allele</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3364859</td>\n",
       "      <td>SPAC26A3.13c</td>\n",
       "      <td>SPAC26A3.13c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3361547</td>\n",
       "      <td>3365548</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.54646</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Tf2-4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>CTT |C| TGT</td>\n",
       "      <td>CTTCTGT</td>\n",
       "      <td>[TTG, GAC, TCC, TTC, TGT, TTA, CAA]</td>\n",
       "      <td>[TTG, GAC, TCC, TTT, TGT, TTA, CAA]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>2227937</td>\n",
       "      <td>SPCC1919.10c</td>\n",
       "      <td>SPCC1919.10c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>III</td>\n",
       "      <td>2225048</td>\n",
       "      <td>2229460</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.55837</td>\n",
       "      <td>...</td>\n",
       "      <td>0.014482</td>\n",
       "      <td>myo52</td>\n",
       "      <td>['C1_13780W_A']</td>\n",
       "      <td>['YAL029C' 'YOR326W']</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>CGG |T| CTT</td>\n",
       "      <td>CGGTCTT</td>\n",
       "      <td>[AAG, AGG, TTC, GGT, CTT, AAG, TTT]</td>\n",
       "      <td>[AAG, AGG, TTC, GGG, CTT, AAG, TTT]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>2228740</td>\n",
       "      <td>SPCC1919.10c</td>\n",
       "      <td>SPCC1919.10c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>III</td>\n",
       "      <td>2225048</td>\n",
       "      <td>2229460</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.55837</td>\n",
       "      <td>...</td>\n",
       "      <td>0.014482</td>\n",
       "      <td>myo52</td>\n",
       "      <td>['C1_13780W_A']</td>\n",
       "      <td>['YAL029C' 'YOR326W']</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>AGC |C| TTG</td>\n",
       "      <td>AGCCTTG</td>\n",
       "      <td>[AGT, AGG, AGC, CTT, GTT, GAA]</td>\n",
       "      <td>[AGT, AGG, AGC, TTT, GTT, GAA]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>215</th>\n",
       "      <td>2291604</td>\n",
       "      <td>SPCC965.05c</td>\n",
       "      <td>SPCC965.05c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>III</td>\n",
       "      <td>2291004</td>\n",
       "      <td>2291981</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.40160</td>\n",
       "      <td>...</td>\n",
       "      <td>0.695544</td>\n",
       "      <td>thp1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATGGGAGTTTCGTAGACAATTCTTTTCG</td>\n",
       "      <td>G</td>\n",
       "      <td>TTC |G| TGC</td>\n",
       "      <td>TTCGTGC</td>\n",
       "      <td>[CAA, TTC, TTT, TCG, TGC, GAA, TTT]</td>\n",
       "      <td>[CAA, TTC, TTT, TCT, GCG, AAT]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     position            ID          Parent object chromosome    start  \\\n",
       "2     3364859  SPAC26A3.13c  SPAC26A3.13c.1    CDS          I  3361547   \n",
       "55    2227937  SPCC1919.10c  SPCC1919.10c.1    CDS        III  2225048   \n",
       "64    2228740  SPCC1919.10c  SPCC1919.10c.1    CDS        III  2225048   \n",
       "215   2291604   SPCC965.05c   SPCC965.05c.1    CDS        III  2291004   \n",
       "\n",
       "         end strand detail     beta  ...  p_value_corr   gene  \\\n",
       "2    3365548      -   exon  0.54646  ...           NaN  Tf2-4   \n",
       "55   2229460      -   exon  0.55837  ...      0.014482  myo52   \n",
       "64   2229460      -   exon  0.55837  ...      0.014482  myo52   \n",
       "215  2291981      -   exon  0.40160  ...      0.695544   thp1   \n",
       "\n",
       "     albicans_name(s)     cerevisiae_name(s)                      allele_1  \\\n",
       "2                 NaN                    NaN                             C   \n",
       "55    ['C1_13780W_A']  ['YAL029C' 'YOR326W']                             T   \n",
       "64    ['C1_13780W_A']  ['YAL029C' 'YOR326W']                             C   \n",
       "215               NaN                    NaN  ATGGGAGTTTCGTAGACAATTCTTTTCG   \n",
       "\n",
       "    allele_2 reference_seq reference_seq_  \\\n",
       "2          T   CTT |C| TGT        CTTCTGT   \n",
       "55         G   CGG |T| CTT        CGGTCTT   \n",
       "64         T   AGC |C| TTG        AGCCTTG   \n",
       "215        G   TTC |G| TGC        TTCGTGC   \n",
       "\n",
       "                                 wt_gene                        mutant allele  \n",
       "2    [TTG, GAC, TCC, TTC, TGT, TTA, CAA]  [TTG, GAC, TCC, TTT, TGT, TTA, CAA]  \n",
       "55   [AAG, AGG, TTC, GGT, CTT, AAG, TTT]  [AAG, AGG, TTC, GGG, CTT, AAG, TTT]  \n",
       "64        [AGT, AGG, AGC, CTT, GTT, GAA]       [AGT, AGG, AGC, TTT, GTT, GAA]  \n",
       "215  [CAA, TTC, TTT, TCG, TGC, GAA, TTT]       [CAA, TTC, TTT, TCT, GCG, AAT]  \n",
       "\n",
       "[4 rows x 23 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#show subset of result_final that are CDS and position is within the CDS and name doesnt start with SPNCRNA\n",
    "result_final__[(result_final__['object'] == 'CDS') & (result_final__['position'] > result_final__['start']) & (result_final__['position'] < result_final__['end'] ) & (result_final__['ID'].str.startswith('SPNCRNA') == False)]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>position</th>\n",
       "      <th>ID</th>\n",
       "      <th>Parent</th>\n",
       "      <th>object</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>detail</th>\n",
       "      <th>beta</th>\n",
       "      <th>p_value</th>\n",
       "      <th>type</th>\n",
       "      <th>correlation</th>\n",
       "      <th>p_value_corr</th>\n",
       "      <th>gene</th>\n",
       "      <th>albicans_name(s)</th>\n",
       "      <th>cerevisiae_name(s)</th>\n",
       "      <th>allele_1</th>\n",
       "      <th>allele_2</th>\n",
       "      <th>reference_seq</th>\n",
       "      <th>reference_seq_</th>\n",
       "      <th>wt_gene</th>\n",
       "      <th>mutant allele</th>\n",
       "      <th>stop_codon_predicted</th>\n",
       "      <th>possible frameshift</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>498798</td>\n",
       "      <td>SPNCRNA.2204</td>\n",
       "      <td>SPNCRNA.2204.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>497961</td>\n",
       "      <td>498854</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.311554</td>\n",
       "      <td>0.000089</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GT</td>\n",
       "      <td>T</td>\n",
       "      <td>GGG |T| TTA</td>\n",
       "      <td>GGGTTTA</td>\n",
       "      <td>[CTT, TTG, GGG, GGT, TTA, ATT, AGT]</td>\n",
       "      <td>[CTT, TTG, GGG, GGT, TTA, ATT, AGT]</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>519884</td>\n",
       "      <td>SPNCRNA.2211</td>\n",
       "      <td>SPNCRNA.2211.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>519783</td>\n",
       "      <td>520113</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.349112</td>\n",
       "      <td>0.000029</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>TA</td>\n",
       "      <td>GAT |A| CCT</td>\n",
       "      <td>GATACCT</td>\n",
       "      <td>[GGT, CGT, TGA, TAC, CTG, AGC]</td>\n",
       "      <td>[GGT, CGT, TGA, TTA, CCT, GAG]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>519884</td>\n",
       "      <td>SPNCRNA.658</td>\n",
       "      <td>SPNCRNA.658.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>519257</td>\n",
       "      <td>520194</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.349112</td>\n",
       "      <td>0.000029</td>\n",
       "      <td>indel</td>\n",
       "      <td>0.191849</td>\n",
       "      <td>0.149101</td>\n",
       "      <td>SPNCRNA.658</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>TCC |A| TAG</td>\n",
       "      <td>TCCATAG</td>\n",
       "      <td>[AGC, GAG, TCC, ATA, GTT]</td>\n",
       "      <td>[AGC, GAG, TCC, ATT, AGT, TGC]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>522785</td>\n",
       "      <td>SPNCRNA.2214</td>\n",
       "      <td>SPNCRNA.2214.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>521928</td>\n",
       "      <td>524210</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.479528</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>TA</td>\n",
       "      <td>TTT |A| TCG</td>\n",
       "      <td>TTTATCG</td>\n",
       "      <td>[AAA, CTT, TTT, ATC, GGG, GAT]</td>\n",
       "      <td>[AAA, CTT, TTT, TAT, CGG, GGA]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>723772</td>\n",
       "      <td>SPNCRNA.2311</td>\n",
       "      <td>SPNCRNA.2311.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>723712</td>\n",
       "      <td>724313</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.417287</td>\n",
       "      <td>0.000025</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>CTCTGT</td>\n",
       "      <td>CTG |T| TTT</td>\n",
       "      <td>CTGTTTT</td>\n",
       "      <td>[AAT, TTC, TCT, GTT, TTT, ATG]</td>\n",
       "      <td>[AAT, TTC, TCT, GCT, CTG, TTT, TTA, TGC]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>1721443</td>\n",
       "      <td>SPNCRNA.1457</td>\n",
       "      <td>SPNCRNA.1457.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>1721083</td>\n",
       "      <td>1724641</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>-0.285830</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>indel</td>\n",
       "      <td>0.092063</td>\n",
       "      <td>0.491875</td>\n",
       "      <td>atf21-antisense-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>TCA</td>\n",
       "      <td>CTC |A| TTG</td>\n",
       "      <td>CTCATTG</td>\n",
       "      <td>[TTC, GTT, CTC, ATT, GTA, CAA]</td>\n",
       "      <td>[TTC, GTT, CTC, TCA, TTG, TAC, AAT]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>1721443</td>\n",
       "      <td>SPNCRNA.376</td>\n",
       "      <td>SPNCRNA.376.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>1721073</td>\n",
       "      <td>1721574</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>-0.285830</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>indel</td>\n",
       "      <td>0.065749</td>\n",
       "      <td>0.623881</td>\n",
       "      <td>SPNCRNA.376</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>ACT</td>\n",
       "      <td>GTT |A| CTC</td>\n",
       "      <td>GTTACTC</td>\n",
       "      <td>[ATA, ACA, TGT, TAC, TCT, TGC]</td>\n",
       "      <td>[ATA, ACA, TGT, TAC, TCT, CTT, GCT]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>2693701</td>\n",
       "      <td>SPNCRNA.5773</td>\n",
       "      <td>SPNCRNA.5773.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2693622</td>\n",
       "      <td>2693887</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>-0.397447</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GT</td>\n",
       "      <td>G</td>\n",
       "      <td>TTT |G| TTT</td>\n",
       "      <td>TTTGTTT</td>\n",
       "      <td>[ACC, AGT, TTT, TGT, TTT, TTT]</td>\n",
       "      <td>[ACC, AGT, TTT, TGT, TTT, TTT]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>2693701</td>\n",
       "      <td>SPNCRNA.5774</td>\n",
       "      <td>SPNCRNA.5774.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2693641</td>\n",
       "      <td>2694958</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>-0.397447</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TG</td>\n",
       "      <td>G</td>\n",
       "      <td>TTT |G| TTT</td>\n",
       "      <td>TTTGTTT</td>\n",
       "      <td>[TTT, TTT, TTT, GTT, TTT, GAC]</td>\n",
       "      <td>[TTT, TTT, TTT, GTT, TTT, GAC]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>2693728</td>\n",
       "      <td>SPNCRNA.5773</td>\n",
       "      <td>SPNCRNA.5773.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2693622</td>\n",
       "      <td>2693887</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>-0.397447</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AC</td>\n",
       "      <td>AACA</td>\n",
       "      <td>AAA |A| CAA</td>\n",
       "      <td>AAAACAA</td>\n",
       "      <td>[GAC, TAA, AAA, AAC, AAA, AAA]</td>\n",
       "      <td>[GAC, TAA, AAA, AAA, CAC, AAA, AAA]</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>2693728</td>\n",
       "      <td>SPNCRNA.5774</td>\n",
       "      <td>SPNCRNA.5774.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2693641</td>\n",
       "      <td>2694958</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>-0.397447</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CA</td>\n",
       "      <td>ACAA</td>\n",
       "      <td>AAC |A| AAA</td>\n",
       "      <td>AACAAAA</td>\n",
       "      <td>[CAA, AAA, AAC, AAA, AAA, ATC]</td>\n",
       "      <td>[CAA, AAA, AAC, ACA, AAA, AAA, ATC]</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>2694024</td>\n",
       "      <td>SPNCRNA.5774</td>\n",
       "      <td>SPNCRNA.5774.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2693641</td>\n",
       "      <td>2694958</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>-0.397447</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>TA</td>\n",
       "      <td>TTT |A| GAA</td>\n",
       "      <td>TTTAGAA</td>\n",
       "      <td>[ATA, TGA, ATT, TAG, AAA, ATT]</td>\n",
       "      <td>[ATA, TGA, ATT, TTA, GAA, AAT]</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>2694637</td>\n",
       "      <td>SPNCRNA.5774</td>\n",
       "      <td>SPNCRNA.5774.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2693641</td>\n",
       "      <td>2694958</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>-0.397447</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>TG</td>\n",
       "      <td>CTG |G| TTT</td>\n",
       "      <td>CTGGTTT</td>\n",
       "      <td>[TAT, TTG, CTG, GTT, TTT, TAT]</td>\n",
       "      <td>[TAT, TTG, CTG, TGT, TTT, TTA]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>2277085</td>\n",
       "      <td>SPNCRNA.516</td>\n",
       "      <td>SPNCRNA.516.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>III</td>\n",
       "      <td>2275648</td>\n",
       "      <td>2277744</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.401600</td>\n",
       "      <td>0.000085</td>\n",
       "      <td>indel</td>\n",
       "      <td>-0.237347</td>\n",
       "      <td>0.072819</td>\n",
       "      <td>SPNCRNA.516</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>TC</td>\n",
       "      <td>TTA |T| AAT</td>\n",
       "      <td>TTATAAT</td>\n",
       "      <td>[CTT, GTC, TTA, TAA, TGT, AAA]</td>\n",
       "      <td>[CTT, GTC, TTA, TCA, ATG, TAA]</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>2277085</td>\n",
       "      <td>SPNCRNA.7699</td>\n",
       "      <td>SPNCRNA.7699.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>III</td>\n",
       "      <td>2276821</td>\n",
       "      <td>2277672</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.401600</td>\n",
       "      <td>0.000085</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>CT</td>\n",
       "      <td>TAA |T| ATT</td>\n",
       "      <td>TAATATT</td>\n",
       "      <td>[CAA, AAA, TGT, AAT, ATT, CTG, TTC]</td>\n",
       "      <td>[CAA, AAA, TGT, AAC, TAT, TCT, GTT]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    position            ID          Parent object chromosome    start  \\\n",
       "9     498798  SPNCRNA.2204  SPNCRNA.2204.1    CDS          I   497961   \n",
       "10    519884  SPNCRNA.2211  SPNCRNA.2211.1    CDS          I   519783   \n",
       "11    519884   SPNCRNA.658   SPNCRNA.658.1    CDS          I   519257   \n",
       "12    522785  SPNCRNA.2214  SPNCRNA.2214.1    CDS          I   521928   \n",
       "13    723772  SPNCRNA.2311  SPNCRNA.2311.1    CDS          I   723712   \n",
       "14   1721443  SPNCRNA.1457  SPNCRNA.1457.1    CDS         II  1721083   \n",
       "15   1721443   SPNCRNA.376   SPNCRNA.376.1    CDS         II  1721073   \n",
       "16   2693701  SPNCRNA.5773  SPNCRNA.5773.1    CDS         II  2693622   \n",
       "17   2693701  SPNCRNA.5774  SPNCRNA.5774.1    CDS         II  2693641   \n",
       "18   2693728  SPNCRNA.5773  SPNCRNA.5773.1    CDS         II  2693622   \n",
       "19   2693728  SPNCRNA.5774  SPNCRNA.5774.1    CDS         II  2693641   \n",
       "20   2694024  SPNCRNA.5774  SPNCRNA.5774.1    CDS         II  2693641   \n",
       "21   2694637  SPNCRNA.5774  SPNCRNA.5774.1    CDS         II  2693641   \n",
       "22   2277085   SPNCRNA.516   SPNCRNA.516.1    CDS        III  2275648   \n",
       "23   2277085  SPNCRNA.7699  SPNCRNA.7699.1    CDS        III  2276821   \n",
       "\n",
       "        end strand detail      beta   p_value   type  correlation  \\\n",
       "9    498854      -   exon  0.311554  0.000089  indel          NaN   \n",
       "10   520113      -   exon  0.349112  0.000029  indel          NaN   \n",
       "11   520194      +   exon  0.349112  0.000029  indel     0.191849   \n",
       "12   524210      -   exon  0.479528  0.000009  indel          NaN   \n",
       "13   724313      -   exon  0.417287  0.000025  indel          NaN   \n",
       "14  1724641      -   exon -0.285830  0.000009  indel     0.092063   \n",
       "15  1721574      +   exon -0.285830  0.000009  indel     0.065749   \n",
       "16  2693887      +   exon -0.397447  0.000005  indel          NaN   \n",
       "17  2694958      -   exon -0.397447  0.000005  indel          NaN   \n",
       "18  2693887      +   exon -0.397447  0.000005  indel          NaN   \n",
       "19  2694958      -   exon -0.397447  0.000005  indel          NaN   \n",
       "20  2694958      -   exon -0.397447  0.000005  indel          NaN   \n",
       "21  2694958      -   exon -0.397447  0.000005  indel          NaN   \n",
       "22  2277744      +   exon  0.401600  0.000085  indel    -0.237347   \n",
       "23  2277672      -   exon  0.401600  0.000085  indel          NaN   \n",
       "\n",
       "    p_value_corr               gene albicans_name(s) cerevisiae_name(s)  \\\n",
       "9            NaN                NaN              NaN                NaN   \n",
       "10           NaN                NaN              NaN                NaN   \n",
       "11      0.149101        SPNCRNA.658              NaN                NaN   \n",
       "12           NaN                NaN              NaN                NaN   \n",
       "13           NaN                NaN              NaN                NaN   \n",
       "14      0.491875  atf21-antisense-1              NaN                NaN   \n",
       "15      0.623881        SPNCRNA.376              NaN                NaN   \n",
       "16           NaN                NaN              NaN                NaN   \n",
       "17           NaN                NaN              NaN                NaN   \n",
       "18           NaN                NaN              NaN                NaN   \n",
       "19           NaN                NaN              NaN                NaN   \n",
       "20           NaN                NaN              NaN                NaN   \n",
       "21           NaN                NaN              NaN                NaN   \n",
       "22      0.072819        SPNCRNA.516              NaN                NaN   \n",
       "23           NaN                NaN              NaN                NaN   \n",
       "\n",
       "   allele_1 allele_2 reference_seq reference_seq_  \\\n",
       "9        GT        T   GGG |T| TTA        GGGTTTA   \n",
       "10        A       TA   GAT |A| CCT        GATACCT   \n",
       "11        A       AT   TCC |A| TAG        TCCATAG   \n",
       "12        A       TA   TTT |A| TCG        TTTATCG   \n",
       "13        T   CTCTGT   CTG |T| TTT        CTGTTTT   \n",
       "14        A      TCA   CTC |A| TTG        CTCATTG   \n",
       "15        A      ACT   GTT |A| CTC        GTTACTC   \n",
       "16       GT        G   TTT |G| TTT        TTTGTTT   \n",
       "17       TG        G   TTT |G| TTT        TTTGTTT   \n",
       "18       AC     AACA   AAA |A| CAA        AAAACAA   \n",
       "19       CA     ACAA   AAC |A| AAA        AACAAAA   \n",
       "20        A       TA   TTT |A| GAA        TTTAGAA   \n",
       "21        G       TG   CTG |G| TTT        CTGGTTT   \n",
       "22        T       TC   TTA |T| AAT        TTATAAT   \n",
       "23        T       CT   TAA |T| ATT        TAATATT   \n",
       "\n",
       "                                wt_gene  \\\n",
       "9   [CTT, TTG, GGG, GGT, TTA, ATT, AGT]   \n",
       "10       [GGT, CGT, TGA, TAC, CTG, AGC]   \n",
       "11            [AGC, GAG, TCC, ATA, GTT]   \n",
       "12       [AAA, CTT, TTT, ATC, GGG, GAT]   \n",
       "13       [AAT, TTC, TCT, GTT, TTT, ATG]   \n",
       "14       [TTC, GTT, CTC, ATT, GTA, CAA]   \n",
       "15       [ATA, ACA, TGT, TAC, TCT, TGC]   \n",
       "16       [ACC, AGT, TTT, TGT, TTT, TTT]   \n",
       "17       [TTT, TTT, TTT, GTT, TTT, GAC]   \n",
       "18       [GAC, TAA, AAA, AAC, AAA, AAA]   \n",
       "19       [CAA, AAA, AAC, AAA, AAA, ATC]   \n",
       "20       [ATA, TGA, ATT, TAG, AAA, ATT]   \n",
       "21       [TAT, TTG, CTG, GTT, TTT, TAT]   \n",
       "22       [CTT, GTC, TTA, TAA, TGT, AAA]   \n",
       "23  [CAA, AAA, TGT, AAT, ATT, CTG, TTC]   \n",
       "\n",
       "                               mutant allele stop_codon_predicted  \\\n",
       "9        [CTT, TTG, GGG, GGT, TTA, ATT, AGT]                  yes   \n",
       "10            [GGT, CGT, TGA, TTA, CCT, GAG]                  NaN   \n",
       "11            [AGC, GAG, TCC, ATT, AGT, TGC]                  NaN   \n",
       "12            [AAA, CTT, TTT, TAT, CGG, GGA]                  NaN   \n",
       "13  [AAT, TTC, TCT, GCT, CTG, TTT, TTA, TGC]                  NaN   \n",
       "14       [TTC, GTT, CTC, TCA, TTG, TAC, AAT]                  NaN   \n",
       "15       [ATA, ACA, TGT, TAC, TCT, CTT, GCT]                  NaN   \n",
       "16            [ACC, AGT, TTT, TGT, TTT, TTT]                  NaN   \n",
       "17            [TTT, TTT, TTT, GTT, TTT, GAC]                  NaN   \n",
       "18       [GAC, TAA, AAA, AAA, CAC, AAA, AAA]                  yes   \n",
       "19       [CAA, AAA, AAC, ACA, AAA, AAA, ATC]                  yes   \n",
       "20            [ATA, TGA, ATT, TTA, GAA, AAT]                  yes   \n",
       "21            [TAT, TTG, CTG, TGT, TTT, TTA]                  NaN   \n",
       "22            [CTT, GTC, TTA, TCA, ATG, TAA]                  yes   \n",
       "23       [CAA, AAA, TGT, AAC, TAT, TCT, GTT]                  NaN   \n",
       "\n",
       "   possible frameshift  \n",
       "9                  yes  \n",
       "10                 yes  \n",
       "11                 yes  \n",
       "12                 yes  \n",
       "13                 yes  \n",
       "14                 yes  \n",
       "15                 yes  \n",
       "16                 yes  \n",
       "17                 yes  \n",
       "18                 yes  \n",
       "19                 yes  \n",
       "20                 yes  \n",
       "21                 yes  \n",
       "22                 yes  \n",
       "23                 yes  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#in stop_codon_df create a column \"possible frameshift\" and fill it with nan. Then loop through X96_allele and Y0036_allele and check if the length is different. If yes and the difference is not divisible by 3, fill the column with \"yes\". Then show the entries where the column is \"yes\"\n",
    "frameshift_df = stop_codon_df.copy()\n",
    "frameshift_df['possible frameshift'] = np.nan\n",
    "for i in range(len(frameshift_df)):\n",
    "    if len(frameshift_df['allele_1'][i]) != len(frameshift_df['allele_2'][i]):\n",
    "        if (len(frameshift_df['allele_1'][i]) - len(frameshift_df['allele_2'][i]))%3 != 0:\n",
    "            frameshift_df['possible frameshift'][i] = 'yes'\n",
    "frameshift_df[frameshift_df['possible frameshift'] == 'yes']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "#import a codon table as a dictionary\n",
    "codon_table = {\n",
    "    'TTT': 'F', 'TTC': 'F', 'TTA': 'L', 'TTG': 'L',\n",
    "    'TCT': 'S', 'TCC': 'S', 'TCA': 'S', 'TCG': 'S',\n",
    "    'TAT': 'Y', 'TAC': 'Y', 'TAA': '*', 'TAG': '*',\n",
    "    'TGT': 'C', 'TGC': 'C', 'TGA': '*', 'TGG': 'W',\n",
    "    'CTT': 'L', 'CTC': 'L', 'CTA': 'L', 'CTG': 'L',\n",
    "    'CCT': 'P', 'CCC': 'P', 'CCA': 'P', 'CCG': 'P',\n",
    "    'CAT': 'H', 'CAC': 'H', 'CAA': 'Q', 'CAG': 'Q', \n",
    "    'CGT': 'R', 'CGC': 'R', 'CGA': 'R', 'CGG': 'R',\n",
    "    'ATT': 'I', 'ATC': 'I', 'ATA': 'I', 'ATG': 'M',\n",
    "    'ACT': 'T', 'ACC': 'T', 'ACA': 'T', 'ACG': 'T',\n",
    "    'AAT': 'N', 'AAC': 'N', 'AAA': 'K', 'AAG': 'K',\n",
    "    'AGT': 'S', 'AGC': 'S', 'AGA': 'R', 'AGG': 'R',\n",
    "    'GTT': 'V', 'GTC': 'V', 'GTA': 'V', 'GTG': 'V',\n",
    "    'GCT': 'A', 'GCC': 'A', 'GCA': 'A', 'GCG': 'A',\n",
    "    'GAT': 'D', 'GAC': 'D', 'GAA': 'E', 'GAG': 'E',\n",
    "    'GGT': 'G', 'GGC': 'G', 'GGA': 'G', 'GGG': 'G',\n",
    "}\n",
    "#iterate through frameshift_df and trnslate the lists of triplets found in wt_gene and mutant allele\n",
    "frameshift_df['wt_gene_translated'] = np.nan\n",
    "frameshift_df['mutant allele_translated'] = np.nan\n",
    "for i in range(len(frameshift_df)):\n",
    "    #if strand is -, then change a to t and so on\n",
    "    if frameshift_df['strand'][i] == '-':\n",
    "        temp_wt_allele = frameshift_df['wt_gene'][i]\n",
    "        temp_mut_allele = frameshift_df['mutant allele'][i]\n",
    "        for j in range(len(temp_wt_allele)):\n",
    "            temp_wt_allele[j] = temp_wt_allele[j].replace('A','t')\n",
    "            temp_wt_allele[j] = temp_wt_allele[j].replace('T','a')\n",
    "            temp_wt_allele[j] = temp_wt_allele[j].replace('C','g')\n",
    "            temp_wt_allele[j] = temp_wt_allele[j].replace('G','c')\n",
    "            temp_wt_allele[j] = temp_wt_allele[j].upper()\n",
    "        for j in range(len(temp_mut_allele)):\n",
    "            temp_mut_allele[j] = temp_mut_allele[j].replace('A','t')\n",
    "            temp_mut_allele[j] = temp_mut_allele[j].replace('T','a')\n",
    "            temp_mut_allele[j] = temp_mut_allele[j].replace('C','g')\n",
    "            temp_mut_allele[j] = temp_mut_allele[j].replace('G','c')\n",
    "            temp_mut_allele[j] = temp_mut_allele[j].upper()\n",
    "        frameshift_df['wt_gene_translated'][i] = ''.join([codon_table[codon] for codon in temp_wt_allele])\n",
    "        frameshift_df['mutant allele_translated'][i] = ''.join([codon_table[codon] for codon in temp_mut_allele])\n",
    "    else:\n",
    "        frameshift_df['wt_gene_translated'][i] = ''.join([codon_table[codon] for codon in frameshift_df['wt_gene'][i]])\n",
    "        frameshift_df['mutant allele_translated'][i] = ''.join([codon_table[codon] for codon in frameshift_df['mutant allele'][i]])\n",
    "\n",
    "        \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "#DISPLAY all columns\n",
    "pd.set_option('display.max_columns', None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>position</th>\n",
       "      <th>ID</th>\n",
       "      <th>Parent</th>\n",
       "      <th>object</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>detail</th>\n",
       "      <th>beta</th>\n",
       "      <th>p_value</th>\n",
       "      <th>type</th>\n",
       "      <th>correlation</th>\n",
       "      <th>p_value_corr</th>\n",
       "      <th>gene</th>\n",
       "      <th>albicans_name(s)</th>\n",
       "      <th>cerevisiae_name(s)</th>\n",
       "      <th>allele_1</th>\n",
       "      <th>allele_2</th>\n",
       "      <th>reference_seq</th>\n",
       "      <th>reference_seq_</th>\n",
       "      <th>wt_gene</th>\n",
       "      <th>mutant allele</th>\n",
       "      <th>stop_codon_predicted</th>\n",
       "      <th>possible frameshift</th>\n",
       "      <th>wt_gene_translated</th>\n",
       "      <th>mutant allele_translated</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3364859</td>\n",
       "      <td>SPAC26A3.13c</td>\n",
       "      <td>SPAC26A3.13c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3361547</td>\n",
       "      <td>3365548</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.546460</td>\n",
       "      <td>1.620139e-09</td>\n",
       "      <td>snp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Tf2-4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>CTT |C| TGT</td>\n",
       "      <td>CTTCTGT</td>\n",
       "      <td>[AAC, CTG, AGG, AAG, ACA, AAT, GTT]</td>\n",
       "      <td>[AAC, CTG, AGG, AAA, ACA, AAT, GTT]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NLRKTNV</td>\n",
       "      <td>NLRKTNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3585317</td>\n",
       "      <td>SPNCRNA.919</td>\n",
       "      <td>SPNCRNA.919.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3584833</td>\n",
       "      <td>3585787</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>-0.366654</td>\n",
       "      <td>6.328207e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>0.147888</td>\n",
       "      <td>0.267917</td>\n",
       "      <td>SPNCRNA.919</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>TTA |A| AGC</td>\n",
       "      <td>TTAAAGC</td>\n",
       "      <td>[TAT, AAT, GGA, ATT, TCG, CCG]</td>\n",
       "      <td>[TAT, AAT, GGA, ATG, TCG, CCG]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>YNGISP</td>\n",
       "      <td>YNGMSP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3585317</td>\n",
       "      <td>SPNCRNA.920</td>\n",
       "      <td>SPNCRNA.920.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3584979</td>\n",
       "      <td>3586233</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>-0.366654</td>\n",
       "      <td>6.328207e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>-0.140556</td>\n",
       "      <td>0.292627</td>\n",
       "      <td>SPNCRNA.920</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>CGA |A| ATT</td>\n",
       "      <td>CGAAATT</td>\n",
       "      <td>[GGA, GGC, GGC, GAA, ATT, CCA]</td>\n",
       "      <td>[GGA, GGC, GGC, GAC, ATT, CCA]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GGGEIP</td>\n",
       "      <td>GGGDIP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4296875</td>\n",
       "      <td>SPNCRNA.6533</td>\n",
       "      <td>SPNCRNA.6533.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>4296590</td>\n",
       "      <td>4296958</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.443424</td>\n",
       "      <td>8.445186e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>TAT |T| AGT</td>\n",
       "      <td>TATTAGT</td>\n",
       "      <td>[GAG, CTA, TAT, TAG, TGA, TAG]</td>\n",
       "      <td>[GAG, CTA, TAT, AAG, TGA, TAG]</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ELY***</td>\n",
       "      <td>ELYK**</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4296877</td>\n",
       "      <td>SPNCRNA.6533</td>\n",
       "      <td>SPNCRNA.6533.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>4296590</td>\n",
       "      <td>4296958</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.443424</td>\n",
       "      <td>8.445186e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>TTA |G| TGA</td>\n",
       "      <td>TTAGTGA</td>\n",
       "      <td>[GAG, CTA, TAT, TAG, TGA, TAG, GTA]</td>\n",
       "      <td>[GAG, CTA, TAT, TAA, TGA, TAG, GTA]</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ELY***V</td>\n",
       "      <td>ELY***V</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2219006</td>\n",
       "      <td>SPNCRNA.7673</td>\n",
       "      <td>SPNCRNA.7673.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>III</td>\n",
       "      <td>2218876</td>\n",
       "      <td>2220006</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.471360</td>\n",
       "      <td>1.374737e-08</td>\n",
       "      <td>snp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>ACA |A| ATT</td>\n",
       "      <td>ACAAATT</td>\n",
       "      <td>[ACT, CTA, AAC, AAA, TTC, ATT]</td>\n",
       "      <td>[ACT, CTA, AAC, ACA, TTC, ATT]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TLNKFI</td>\n",
       "      <td>TLNTFI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2227937</td>\n",
       "      <td>SPCC1919.10c</td>\n",
       "      <td>SPCC1919.10c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>III</td>\n",
       "      <td>2225048</td>\n",
       "      <td>2229460</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.558370</td>\n",
       "      <td>1.963371e-10</td>\n",
       "      <td>snp</td>\n",
       "      <td>-0.319543</td>\n",
       "      <td>0.014482</td>\n",
       "      <td>myo52</td>\n",
       "      <td>['C1_13780W_A']</td>\n",
       "      <td>['YAL029C' 'YOR326W']</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>CGG |T| CTT</td>\n",
       "      <td>CGGTCTT</td>\n",
       "      <td>[TTC, TCC, AAG, CCA, GAA, TTC, AAA]</td>\n",
       "      <td>[TTC, TCC, AAG, CCC, GAA, TTC, AAA]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FSKPEFK</td>\n",
       "      <td>FSKPEFK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>2227937</td>\n",
       "      <td>SPNCRNA.7677</td>\n",
       "      <td>SPNCRNA.7677.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>III</td>\n",
       "      <td>2227612</td>\n",
       "      <td>2228044</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.558370</td>\n",
       "      <td>1.963371e-10</td>\n",
       "      <td>snp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>TTC |T| GGC</td>\n",
       "      <td>TTCTGGC</td>\n",
       "      <td>[TTT, TGA, ATT, CTG, GCT, TGG]</td>\n",
       "      <td>[TTT, TGA, ATT, CGG, GCT, TGG]</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>F*ILAW</td>\n",
       "      <td>F*IRAW</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>2228740</td>\n",
       "      <td>SPCC1919.10c</td>\n",
       "      <td>SPCC1919.10c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>III</td>\n",
       "      <td>2225048</td>\n",
       "      <td>2229460</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.558370</td>\n",
       "      <td>1.963371e-10</td>\n",
       "      <td>snp</td>\n",
       "      <td>-0.319543</td>\n",
       "      <td>0.014482</td>\n",
       "      <td>myo52</td>\n",
       "      <td>['C1_13780W_A']</td>\n",
       "      <td>['YAL029C' 'YOR326W']</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>AGC |C| TTG</td>\n",
       "      <td>AGCCTTG</td>\n",
       "      <td>[TCA, TCC, TCG, GAA, CAA, CTT]</td>\n",
       "      <td>[TCA, TCC, TCG, AAA, CAA, CTT]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SSSEQL</td>\n",
       "      <td>SSSKQL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>498798</td>\n",
       "      <td>SPNCRNA.2204</td>\n",
       "      <td>SPNCRNA.2204.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>497961</td>\n",
       "      <td>498854</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.311554</td>\n",
       "      <td>8.916224e-05</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GT</td>\n",
       "      <td>T</td>\n",
       "      <td>GGG |T| TTA</td>\n",
       "      <td>GGGTTTA</td>\n",
       "      <td>[GAA, AAC, CCC, CCA, AAT, TAA, TCA]</td>\n",
       "      <td>[GAA, AAC, CCC, CCA, AAT, TAA, TCA]</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>ENPPN*S</td>\n",
       "      <td>ENPPN*S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>519884</td>\n",
       "      <td>SPNCRNA.2211</td>\n",
       "      <td>SPNCRNA.2211.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>519783</td>\n",
       "      <td>520113</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.349112</td>\n",
       "      <td>2.914204e-05</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>TA</td>\n",
       "      <td>GAT |A| CCT</td>\n",
       "      <td>GATACCT</td>\n",
       "      <td>[CCA, GCA, ACT, ATG, GAC, TCG]</td>\n",
       "      <td>[CCA, GCA, ACT, AAT, GGA, CTC]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>PATMDS</td>\n",
       "      <td>PATNGL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>519884</td>\n",
       "      <td>SPNCRNA.658</td>\n",
       "      <td>SPNCRNA.658.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>519257</td>\n",
       "      <td>520194</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.349112</td>\n",
       "      <td>2.914204e-05</td>\n",
       "      <td>indel</td>\n",
       "      <td>0.191849</td>\n",
       "      <td>0.149101</td>\n",
       "      <td>SPNCRNA.658</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>TCC |A| TAG</td>\n",
       "      <td>TCCATAG</td>\n",
       "      <td>[AGC, GAG, TCC, ATA, GTT]</td>\n",
       "      <td>[AGC, GAG, TCC, ATT, AGT, TGC]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>SESIV</td>\n",
       "      <td>SESISC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>522785</td>\n",
       "      <td>SPNCRNA.2214</td>\n",
       "      <td>SPNCRNA.2214.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>521928</td>\n",
       "      <td>524210</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.479528</td>\n",
       "      <td>9.351192e-06</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>TA</td>\n",
       "      <td>TTT |A| TCG</td>\n",
       "      <td>TTTATCG</td>\n",
       "      <td>[TTT, GAA, AAA, TAG, CCC, CTA]</td>\n",
       "      <td>[TTT, GAA, AAA, ATA, GCC, CCT]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>FEK*PL</td>\n",
       "      <td>FEKIAP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>723772</td>\n",
       "      <td>SPNCRNA.2311</td>\n",
       "      <td>SPNCRNA.2311.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>723712</td>\n",
       "      <td>724313</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.417287</td>\n",
       "      <td>2.522852e-05</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>CTCTGT</td>\n",
       "      <td>CTG |T| TTT</td>\n",
       "      <td>CTGTTTT</td>\n",
       "      <td>[TTA, AAG, AGA, CAA, AAA, TAC]</td>\n",
       "      <td>[TTA, AAG, AGA, CGA, GAC, AAA, AAT, ACG]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>LKRQKY</td>\n",
       "      <td>LKRRDKNT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>1721443</td>\n",
       "      <td>SPNCRNA.1457</td>\n",
       "      <td>SPNCRNA.1457.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>1721083</td>\n",
       "      <td>1724641</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>-0.285830</td>\n",
       "      <td>9.253300e-06</td>\n",
       "      <td>indel</td>\n",
       "      <td>0.092063</td>\n",
       "      <td>0.491875</td>\n",
       "      <td>atf21-antisense-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>TCA</td>\n",
       "      <td>CTC |A| TTG</td>\n",
       "      <td>CTCATTG</td>\n",
       "      <td>[AAG, CAA, GAG, TAA, CAT, GTT]</td>\n",
       "      <td>[AAG, CAA, GAG, AGT, AAC, ATG, TTA]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>KQE*HV</td>\n",
       "      <td>KQESNML</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>1721443</td>\n",
       "      <td>SPNCRNA.376</td>\n",
       "      <td>SPNCRNA.376.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>1721073</td>\n",
       "      <td>1721574</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>-0.285830</td>\n",
       "      <td>9.253300e-06</td>\n",
       "      <td>indel</td>\n",
       "      <td>0.065749</td>\n",
       "      <td>0.623881</td>\n",
       "      <td>SPNCRNA.376</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>ACT</td>\n",
       "      <td>GTT |A| CTC</td>\n",
       "      <td>GTTACTC</td>\n",
       "      <td>[ATA, ACA, TGT, TAC, TCT, TGC]</td>\n",
       "      <td>[ATA, ACA, TGT, TAC, TCT, CTT, GCT]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>ITCYSC</td>\n",
       "      <td>ITCYSLA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>2693701</td>\n",
       "      <td>SPNCRNA.5773</td>\n",
       "      <td>SPNCRNA.5773.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2693622</td>\n",
       "      <td>2693887</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>-0.397447</td>\n",
       "      <td>5.059658e-06</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GT</td>\n",
       "      <td>G</td>\n",
       "      <td>TTT |G| TTT</td>\n",
       "      <td>TTTGTTT</td>\n",
       "      <td>[ACC, AGT, TTT, TGT, TTT, TTT]</td>\n",
       "      <td>[ACC, AGT, TTT, TGT, TTT, TTT]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>TSFCFF</td>\n",
       "      <td>TSFCFF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>2693701</td>\n",
       "      <td>SPNCRNA.5774</td>\n",
       "      <td>SPNCRNA.5774.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2693641</td>\n",
       "      <td>2694958</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>-0.397447</td>\n",
       "      <td>5.059658e-06</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TG</td>\n",
       "      <td>G</td>\n",
       "      <td>TTT |G| TTT</td>\n",
       "      <td>TTTGTTT</td>\n",
       "      <td>[AAA, AAA, AAA, CAA, AAA, CTG]</td>\n",
       "      <td>[AAA, AAA, AAA, CAA, AAA, CTG]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>KKKQKL</td>\n",
       "      <td>KKKQKL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>2693728</td>\n",
       "      <td>SPNCRNA.5773</td>\n",
       "      <td>SPNCRNA.5773.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2693622</td>\n",
       "      <td>2693887</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>-0.397447</td>\n",
       "      <td>5.059658e-06</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AC</td>\n",
       "      <td>AACA</td>\n",
       "      <td>AAA |A| CAA</td>\n",
       "      <td>AAAACAA</td>\n",
       "      <td>[GAC, TAA, AAA, AAC, AAA, AAA]</td>\n",
       "      <td>[GAC, TAA, AAA, AAA, CAC, AAA, AAA]</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>D*KNKK</td>\n",
       "      <td>D*KKHKK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>2693728</td>\n",
       "      <td>SPNCRNA.5774</td>\n",
       "      <td>SPNCRNA.5774.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2693641</td>\n",
       "      <td>2694958</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>-0.397447</td>\n",
       "      <td>5.059658e-06</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CA</td>\n",
       "      <td>ACAA</td>\n",
       "      <td>AAC |A| AAA</td>\n",
       "      <td>AACAAAA</td>\n",
       "      <td>[GTT, TTT, TTG, TTT, TTT, TAG]</td>\n",
       "      <td>[GTT, TTT, TTG, TGT, TTT, TTT, TAG]</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>VFLFF*</td>\n",
       "      <td>VFLCFF*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>2694024</td>\n",
       "      <td>SPNCRNA.5774</td>\n",
       "      <td>SPNCRNA.5774.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2693641</td>\n",
       "      <td>2694958</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>-0.397447</td>\n",
       "      <td>5.059658e-06</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>TA</td>\n",
       "      <td>TTT |A| GAA</td>\n",
       "      <td>TTTAGAA</td>\n",
       "      <td>[TAT, ACT, TAA, ATC, TTT, TAA]</td>\n",
       "      <td>[TAT, ACT, TAA, AAT, CTT, TTA]</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>YT*IF*</td>\n",
       "      <td>YT*NLL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>2694637</td>\n",
       "      <td>SPNCRNA.5774</td>\n",
       "      <td>SPNCRNA.5774.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>II</td>\n",
       "      <td>2693641</td>\n",
       "      <td>2694958</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>-0.397447</td>\n",
       "      <td>5.059658e-06</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>TG</td>\n",
       "      <td>CTG |G| TTT</td>\n",
       "      <td>CTGGTTT</td>\n",
       "      <td>[ATA, AAC, GAC, CAA, AAA, ATA]</td>\n",
       "      <td>[ATA, AAC, GAC, ACA, AAA, AAT]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>INDQKI</td>\n",
       "      <td>INDTKN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>2277085</td>\n",
       "      <td>SPNCRNA.516</td>\n",
       "      <td>SPNCRNA.516.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>III</td>\n",
       "      <td>2275648</td>\n",
       "      <td>2277744</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.401600</td>\n",
       "      <td>8.534981e-05</td>\n",
       "      <td>indel</td>\n",
       "      <td>-0.237347</td>\n",
       "      <td>0.072819</td>\n",
       "      <td>SPNCRNA.516</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>TC</td>\n",
       "      <td>TTA |T| AAT</td>\n",
       "      <td>TTATAAT</td>\n",
       "      <td>[CTT, GTC, TTA, TAA, TGT, AAA]</td>\n",
       "      <td>[CTT, GTC, TTA, TCA, ATG, TAA]</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>LVL*CK</td>\n",
       "      <td>LVLSM*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>2277085</td>\n",
       "      <td>SPNCRNA.7699</td>\n",
       "      <td>SPNCRNA.7699.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>III</td>\n",
       "      <td>2276821</td>\n",
       "      <td>2277672</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.401600</td>\n",
       "      <td>8.534981e-05</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>CT</td>\n",
       "      <td>TAA |T| ATT</td>\n",
       "      <td>TAATATT</td>\n",
       "      <td>[GTT, TTT, ACA, TTA, TAA, GAC, AAG]</td>\n",
       "      <td>[GTT, TTT, ACA, TTG, ATA, AGA, CAA]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>VFTL*DK</td>\n",
       "      <td>VFTLIRQ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>2291604</td>\n",
       "      <td>SPCC965.05c</td>\n",
       "      <td>SPCC965.05c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>III</td>\n",
       "      <td>2291004</td>\n",
       "      <td>2291981</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.401600</td>\n",
       "      <td>8.534981e-05</td>\n",
       "      <td>indel</td>\n",
       "      <td>0.052493</td>\n",
       "      <td>0.695544</td>\n",
       "      <td>thp1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATGGGAGTTTCGTAGACAATTCTTTTCG</td>\n",
       "      <td>G</td>\n",
       "      <td>TTC |G| TGC</td>\n",
       "      <td>TTCGTGC</td>\n",
       "      <td>[GTT, AAG, AAA, AGC, ACG, CTT, AAA]</td>\n",
       "      <td>[GTT, AAG, AAA, AGA, CGC, TTA]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VKKSTLK</td>\n",
       "      <td>VKKRRL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>2357290</td>\n",
       "      <td>SPNCRNA.7727</td>\n",
       "      <td>SPNCRNA.7727.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>III</td>\n",
       "      <td>2354421</td>\n",
       "      <td>2358979</td>\n",
       "      <td>+</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.393597</td>\n",
       "      <td>9.895852e-05</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>CTTTTGATCTGATAAT</td>\n",
       "      <td>TAA |C| TTT</td>\n",
       "      <td>TAACTTT</td>\n",
       "      <td>[TCA, CGC, TTA, ACT, TTT, GAT]</td>\n",
       "      <td>[TCA, CGC, TTA, ACT, TTT, GAT, CTG, ATA, ATT, ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SRLTFD</td>\n",
       "      <td>SRLTFDLIIFD</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    position            ID          Parent object chromosome    start  \\\n",
       "0    3364859  SPAC26A3.13c  SPAC26A3.13c.1    CDS          I  3361547   \n",
       "1    3585317   SPNCRNA.919   SPNCRNA.919.1    CDS          I  3584833   \n",
       "2    3585317   SPNCRNA.920   SPNCRNA.920.1    CDS          I  3584979   \n",
       "3    4296875  SPNCRNA.6533  SPNCRNA.6533.1    CDS         II  4296590   \n",
       "4    4296877  SPNCRNA.6533  SPNCRNA.6533.1    CDS         II  4296590   \n",
       "5    2219006  SPNCRNA.7673  SPNCRNA.7673.1    CDS        III  2218876   \n",
       "6    2227937  SPCC1919.10c  SPCC1919.10c.1    CDS        III  2225048   \n",
       "7    2227937  SPNCRNA.7677  SPNCRNA.7677.1    CDS        III  2227612   \n",
       "8    2228740  SPCC1919.10c  SPCC1919.10c.1    CDS        III  2225048   \n",
       "9     498798  SPNCRNA.2204  SPNCRNA.2204.1    CDS          I   497961   \n",
       "10    519884  SPNCRNA.2211  SPNCRNA.2211.1    CDS          I   519783   \n",
       "11    519884   SPNCRNA.658   SPNCRNA.658.1    CDS          I   519257   \n",
       "12    522785  SPNCRNA.2214  SPNCRNA.2214.1    CDS          I   521928   \n",
       "13    723772  SPNCRNA.2311  SPNCRNA.2311.1    CDS          I   723712   \n",
       "14   1721443  SPNCRNA.1457  SPNCRNA.1457.1    CDS         II  1721083   \n",
       "15   1721443   SPNCRNA.376   SPNCRNA.376.1    CDS         II  1721073   \n",
       "16   2693701  SPNCRNA.5773  SPNCRNA.5773.1    CDS         II  2693622   \n",
       "17   2693701  SPNCRNA.5774  SPNCRNA.5774.1    CDS         II  2693641   \n",
       "18   2693728  SPNCRNA.5773  SPNCRNA.5773.1    CDS         II  2693622   \n",
       "19   2693728  SPNCRNA.5774  SPNCRNA.5774.1    CDS         II  2693641   \n",
       "20   2694024  SPNCRNA.5774  SPNCRNA.5774.1    CDS         II  2693641   \n",
       "21   2694637  SPNCRNA.5774  SPNCRNA.5774.1    CDS         II  2693641   \n",
       "22   2277085   SPNCRNA.516   SPNCRNA.516.1    CDS        III  2275648   \n",
       "23   2277085  SPNCRNA.7699  SPNCRNA.7699.1    CDS        III  2276821   \n",
       "24   2291604   SPCC965.05c   SPCC965.05c.1    CDS        III  2291004   \n",
       "25   2357290  SPNCRNA.7727  SPNCRNA.7727.1    CDS        III  2354421   \n",
       "\n",
       "        end strand detail      beta       p_value   type  correlation  \\\n",
       "0   3365548      -   exon  0.546460  1.620139e-09    snp          NaN   \n",
       "1   3585787      -   exon -0.366654  6.328207e-08    snp     0.147888   \n",
       "2   3586233      +   exon -0.366654  6.328207e-08    snp    -0.140556   \n",
       "3   4296958      +   exon  0.443424  8.445186e-08    snp          NaN   \n",
       "4   4296958      +   exon  0.443424  8.445186e-08    snp          NaN   \n",
       "5   2220006      +   exon  0.471360  1.374737e-08    snp          NaN   \n",
       "6   2229460      -   exon  0.558370  1.963371e-10    snp    -0.319543   \n",
       "7   2228044      +   exon  0.558370  1.963371e-10    snp          NaN   \n",
       "8   2229460      -   exon  0.558370  1.963371e-10    snp    -0.319543   \n",
       "9    498854      -   exon  0.311554  8.916224e-05  indel          NaN   \n",
       "10   520113      -   exon  0.349112  2.914204e-05  indel          NaN   \n",
       "11   520194      +   exon  0.349112  2.914204e-05  indel     0.191849   \n",
       "12   524210      -   exon  0.479528  9.351192e-06  indel          NaN   \n",
       "13   724313      -   exon  0.417287  2.522852e-05  indel          NaN   \n",
       "14  1724641      -   exon -0.285830  9.253300e-06  indel     0.092063   \n",
       "15  1721574      +   exon -0.285830  9.253300e-06  indel     0.065749   \n",
       "16  2693887      +   exon -0.397447  5.059658e-06  indel          NaN   \n",
       "17  2694958      -   exon -0.397447  5.059658e-06  indel          NaN   \n",
       "18  2693887      +   exon -0.397447  5.059658e-06  indel          NaN   \n",
       "19  2694958      -   exon -0.397447  5.059658e-06  indel          NaN   \n",
       "20  2694958      -   exon -0.397447  5.059658e-06  indel          NaN   \n",
       "21  2694958      -   exon -0.397447  5.059658e-06  indel          NaN   \n",
       "22  2277744      +   exon  0.401600  8.534981e-05  indel    -0.237347   \n",
       "23  2277672      -   exon  0.401600  8.534981e-05  indel          NaN   \n",
       "24  2291981      -   exon  0.401600  8.534981e-05  indel     0.052493   \n",
       "25  2358979      +   exon  0.393597  9.895852e-05  indel          NaN   \n",
       "\n",
       "    p_value_corr               gene albicans_name(s)     cerevisiae_name(s)  \\\n",
       "0            NaN              Tf2-4              NaN                    NaN   \n",
       "1       0.267917        SPNCRNA.919              NaN                    NaN   \n",
       "2       0.292627        SPNCRNA.920              NaN                    NaN   \n",
       "3            NaN                NaN              NaN                    NaN   \n",
       "4            NaN                NaN              NaN                    NaN   \n",
       "5            NaN                NaN              NaN                    NaN   \n",
       "6       0.014482              myo52  ['C1_13780W_A']  ['YAL029C' 'YOR326W']   \n",
       "7            NaN                NaN              NaN                    NaN   \n",
       "8       0.014482              myo52  ['C1_13780W_A']  ['YAL029C' 'YOR326W']   \n",
       "9            NaN                NaN              NaN                    NaN   \n",
       "10           NaN                NaN              NaN                    NaN   \n",
       "11      0.149101        SPNCRNA.658              NaN                    NaN   \n",
       "12           NaN                NaN              NaN                    NaN   \n",
       "13           NaN                NaN              NaN                    NaN   \n",
       "14      0.491875  atf21-antisense-1              NaN                    NaN   \n",
       "15      0.623881        SPNCRNA.376              NaN                    NaN   \n",
       "16           NaN                NaN              NaN                    NaN   \n",
       "17           NaN                NaN              NaN                    NaN   \n",
       "18           NaN                NaN              NaN                    NaN   \n",
       "19           NaN                NaN              NaN                    NaN   \n",
       "20           NaN                NaN              NaN                    NaN   \n",
       "21           NaN                NaN              NaN                    NaN   \n",
       "22      0.072819        SPNCRNA.516              NaN                    NaN   \n",
       "23           NaN                NaN              NaN                    NaN   \n",
       "24      0.695544               thp1              NaN                    NaN   \n",
       "25           NaN                NaN              NaN                    NaN   \n",
       "\n",
       "                        allele_1          allele_2 reference_seq  \\\n",
       "0                              C                 T   CTT |C| TGT   \n",
       "1                              A                 C   TTA |A| AGC   \n",
       "2                              A                 C   CGA |A| ATT   \n",
       "3                              T                 A   TAT |T| AGT   \n",
       "4                              G                 A   TTA |G| TGA   \n",
       "5                              A                 C   ACA |A| ATT   \n",
       "6                              T                 G   CGG |T| CTT   \n",
       "7                              T                 G   TTC |T| GGC   \n",
       "8                              C                 T   AGC |C| TTG   \n",
       "9                             GT                 T   GGG |T| TTA   \n",
       "10                             A                TA   GAT |A| CCT   \n",
       "11                             A                AT   TCC |A| TAG   \n",
       "12                             A                TA   TTT |A| TCG   \n",
       "13                             T            CTCTGT   CTG |T| TTT   \n",
       "14                             A               TCA   CTC |A| TTG   \n",
       "15                             A               ACT   GTT |A| CTC   \n",
       "16                            GT                 G   TTT |G| TTT   \n",
       "17                            TG                 G   TTT |G| TTT   \n",
       "18                            AC              AACA   AAA |A| CAA   \n",
       "19                            CA              ACAA   AAC |A| AAA   \n",
       "20                             A                TA   TTT |A| GAA   \n",
       "21                             G                TG   CTG |G| TTT   \n",
       "22                             T                TC   TTA |T| AAT   \n",
       "23                             T                CT   TAA |T| ATT   \n",
       "24  ATGGGAGTTTCGTAGACAATTCTTTTCG                 G   TTC |G| TGC   \n",
       "25                             C  CTTTTGATCTGATAAT   TAA |C| TTT   \n",
       "\n",
       "   reference_seq_                              wt_gene  \\\n",
       "0         CTTCTGT  [AAC, CTG, AGG, AAG, ACA, AAT, GTT]   \n",
       "1         TTAAAGC       [TAT, AAT, GGA, ATT, TCG, CCG]   \n",
       "2         CGAAATT       [GGA, GGC, GGC, GAA, ATT, CCA]   \n",
       "3         TATTAGT       [GAG, CTA, TAT, TAG, TGA, TAG]   \n",
       "4         TTAGTGA  [GAG, CTA, TAT, TAG, TGA, TAG, GTA]   \n",
       "5         ACAAATT       [ACT, CTA, AAC, AAA, TTC, ATT]   \n",
       "6         CGGTCTT  [TTC, TCC, AAG, CCA, GAA, TTC, AAA]   \n",
       "7         TTCTGGC       [TTT, TGA, ATT, CTG, GCT, TGG]   \n",
       "8         AGCCTTG       [TCA, TCC, TCG, GAA, CAA, CTT]   \n",
       "9         GGGTTTA  [GAA, AAC, CCC, CCA, AAT, TAA, TCA]   \n",
       "10        GATACCT       [CCA, GCA, ACT, ATG, GAC, TCG]   \n",
       "11        TCCATAG            [AGC, GAG, TCC, ATA, GTT]   \n",
       "12        TTTATCG       [TTT, GAA, AAA, TAG, CCC, CTA]   \n",
       "13        CTGTTTT       [TTA, AAG, AGA, CAA, AAA, TAC]   \n",
       "14        CTCATTG       [AAG, CAA, GAG, TAA, CAT, GTT]   \n",
       "15        GTTACTC       [ATA, ACA, TGT, TAC, TCT, TGC]   \n",
       "16        TTTGTTT       [ACC, AGT, TTT, TGT, TTT, TTT]   \n",
       "17        TTTGTTT       [AAA, AAA, AAA, CAA, AAA, CTG]   \n",
       "18        AAAACAA       [GAC, TAA, AAA, AAC, AAA, AAA]   \n",
       "19        AACAAAA       [GTT, TTT, TTG, TTT, TTT, TAG]   \n",
       "20        TTTAGAA       [TAT, ACT, TAA, ATC, TTT, TAA]   \n",
       "21        CTGGTTT       [ATA, AAC, GAC, CAA, AAA, ATA]   \n",
       "22        TTATAAT       [CTT, GTC, TTA, TAA, TGT, AAA]   \n",
       "23        TAATATT  [GTT, TTT, ACA, TTA, TAA, GAC, AAG]   \n",
       "24        TTCGTGC  [GTT, AAG, AAA, AGC, ACG, CTT, AAA]   \n",
       "25        TAACTTT       [TCA, CGC, TTA, ACT, TTT, GAT]   \n",
       "\n",
       "                                        mutant allele stop_codon_predicted  \\\n",
       "0                 [AAC, CTG, AGG, AAA, ACA, AAT, GTT]                  NaN   \n",
       "1                      [TAT, AAT, GGA, ATG, TCG, CCG]                  NaN   \n",
       "2                      [GGA, GGC, GGC, GAC, ATT, CCA]                  NaN   \n",
       "3                      [GAG, CTA, TAT, AAG, TGA, TAG]                  yes   \n",
       "4                 [GAG, CTA, TAT, TAA, TGA, TAG, GTA]                  yes   \n",
       "5                      [ACT, CTA, AAC, ACA, TTC, ATT]                  NaN   \n",
       "6                 [TTC, TCC, AAG, CCC, GAA, TTC, AAA]                  NaN   \n",
       "7                      [TTT, TGA, ATT, CGG, GCT, TGG]                  yes   \n",
       "8                      [TCA, TCC, TCG, AAA, CAA, CTT]                  NaN   \n",
       "9                 [GAA, AAC, CCC, CCA, AAT, TAA, TCA]                  yes   \n",
       "10                     [CCA, GCA, ACT, AAT, GGA, CTC]                  NaN   \n",
       "11                     [AGC, GAG, TCC, ATT, AGT, TGC]                  NaN   \n",
       "12                     [TTT, GAA, AAA, ATA, GCC, CCT]                  NaN   \n",
       "13           [TTA, AAG, AGA, CGA, GAC, AAA, AAT, ACG]                  NaN   \n",
       "14                [AAG, CAA, GAG, AGT, AAC, ATG, TTA]                  NaN   \n",
       "15                [ATA, ACA, TGT, TAC, TCT, CTT, GCT]                  NaN   \n",
       "16                     [ACC, AGT, TTT, TGT, TTT, TTT]                  NaN   \n",
       "17                     [AAA, AAA, AAA, CAA, AAA, CTG]                  NaN   \n",
       "18                [GAC, TAA, AAA, AAA, CAC, AAA, AAA]                  yes   \n",
       "19                [GTT, TTT, TTG, TGT, TTT, TTT, TAG]                  yes   \n",
       "20                     [TAT, ACT, TAA, AAT, CTT, TTA]                  yes   \n",
       "21                     [ATA, AAC, GAC, ACA, AAA, AAT]                  NaN   \n",
       "22                     [CTT, GTC, TTA, TCA, ATG, TAA]                  yes   \n",
       "23                [GTT, TTT, ACA, TTG, ATA, AGA, CAA]                  NaN   \n",
       "24                     [GTT, AAG, AAA, AGA, CGC, TTA]                  NaN   \n",
       "25  [TCA, CGC, TTA, ACT, TTT, GAT, CTG, ATA, ATT, ...                  NaN   \n",
       "\n",
       "   possible frameshift wt_gene_translated mutant allele_translated  \n",
       "0                  NaN            NLRKTNV                  NLRKTNV  \n",
       "1                  NaN             YNGISP                   YNGMSP  \n",
       "2                  NaN             GGGEIP                   GGGDIP  \n",
       "3                  NaN             ELY***                   ELYK**  \n",
       "4                  NaN            ELY***V                  ELY***V  \n",
       "5                  NaN             TLNKFI                   TLNTFI  \n",
       "6                  NaN            FSKPEFK                  FSKPEFK  \n",
       "7                  NaN             F*ILAW                   F*IRAW  \n",
       "8                  NaN             SSSEQL                   SSSKQL  \n",
       "9                  yes            ENPPN*S                  ENPPN*S  \n",
       "10                 yes             PATMDS                   PATNGL  \n",
       "11                 yes              SESIV                   SESISC  \n",
       "12                 yes             FEK*PL                   FEKIAP  \n",
       "13                 yes             LKRQKY                 LKRRDKNT  \n",
       "14                 yes             KQE*HV                  KQESNML  \n",
       "15                 yes             ITCYSC                  ITCYSLA  \n",
       "16                 yes             TSFCFF                   TSFCFF  \n",
       "17                 yes             KKKQKL                   KKKQKL  \n",
       "18                 yes             D*KNKK                  D*KKHKK  \n",
       "19                 yes             VFLFF*                  VFLCFF*  \n",
       "20                 yes             YT*IF*                   YT*NLL  \n",
       "21                 yes             INDQKI                   INDTKN  \n",
       "22                 yes             LVL*CK                   LVLSM*  \n",
       "23                 yes            VFTL*DK                  VFTLIRQ  \n",
       "24                 NaN            VKKSTLK                   VKKRRL  \n",
       "25                 NaN             SRLTFD              SRLTFDLIIFD  "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "frameshift_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>position</th>\n",
       "      <th>ID</th>\n",
       "      <th>Parent</th>\n",
       "      <th>object</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>detail</th>\n",
       "      <th>beta</th>\n",
       "      <th>...</th>\n",
       "      <th>allele_1</th>\n",
       "      <th>allele_2</th>\n",
       "      <th>reference_seq</th>\n",
       "      <th>reference_seq_</th>\n",
       "      <th>wt_gene</th>\n",
       "      <th>mutant allele</th>\n",
       "      <th>stop_codon_predicted</th>\n",
       "      <th>possible frameshift</th>\n",
       "      <th>wt_gene_translated</th>\n",
       "      <th>mutant allele_translated</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3364859</td>\n",
       "      <td>SPAC26A3.13c</td>\n",
       "      <td>SPAC26A3.13c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>I</td>\n",
       "      <td>3361547</td>\n",
       "      <td>3365548</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.54646</td>\n",
       "      <td>...</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>CTT |C| TGT</td>\n",
       "      <td>CTTCTGT</td>\n",
       "      <td>[CTG, AGG, AAG, ACA, AAT, GTT, TCG, AGA]</td>\n",
       "      <td>[CTG, AGG, AAA, ACA, AAT, GTT, TCG, AGA]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LRKTNVSR</td>\n",
       "      <td>LRKTNVSR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2227937</td>\n",
       "      <td>SPCC1919.10c</td>\n",
       "      <td>SPCC1919.10c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>III</td>\n",
       "      <td>2225048</td>\n",
       "      <td>2229460</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.55837</td>\n",
       "      <td>...</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>CGG |T| CTT</td>\n",
       "      <td>CGGTCTT</td>\n",
       "      <td>[TCC, AAG, CCA, GAA, TTC, AAA, AAT, AGT]</td>\n",
       "      <td>[TCC, AAG, CCC, GAA, TTC, AAA, AAT, AGT]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SKPEFKNS</td>\n",
       "      <td>SKPEFKNS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>2228740</td>\n",
       "      <td>SPCC1919.10c</td>\n",
       "      <td>SPCC1919.10c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>III</td>\n",
       "      <td>2225048</td>\n",
       "      <td>2229460</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.55837</td>\n",
       "      <td>...</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>AGC |C| TTG</td>\n",
       "      <td>AGCCTTG</td>\n",
       "      <td>[TCC, TCG, GAA, CAA, CTT, GAA, AAG]</td>\n",
       "      <td>[TCC, TCG, AAA, CAA, CTT, GAA, AAG]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SSEQLEK</td>\n",
       "      <td>SSKQLEK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>2291604</td>\n",
       "      <td>SPCC965.05c</td>\n",
       "      <td>SPCC965.05c.1</td>\n",
       "      <td>CDS</td>\n",
       "      <td>III</td>\n",
       "      <td>2291004</td>\n",
       "      <td>2291981</td>\n",
       "      <td>-</td>\n",
       "      <td>exon</td>\n",
       "      <td>0.40160</td>\n",
       "      <td>...</td>\n",
       "      <td>ATGGGAGTTTCGTAGACAATTCTTTTCG</td>\n",
       "      <td>G</td>\n",
       "      <td>TTC |G| TGC</td>\n",
       "      <td>TTCGTGC</td>\n",
       "      <td>[AAG, AAA, AGC, ACG, CTT, AAA, AAG, AAG]</td>\n",
       "      <td>[AAG, AAA, AGA, CGC, TTA, AAA, AGA]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>KKSTLKKK</td>\n",
       "      <td>KKRRLKR</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4 rows × 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    position            ID          Parent object chromosome    start  \\\n",
       "0    3364859  SPAC26A3.13c  SPAC26A3.13c.1    CDS          I  3361547   \n",
       "6    2227937  SPCC1919.10c  SPCC1919.10c.1    CDS        III  2225048   \n",
       "8    2228740  SPCC1919.10c  SPCC1919.10c.1    CDS        III  2225048   \n",
       "24   2291604   SPCC965.05c   SPCC965.05c.1    CDS        III  2291004   \n",
       "\n",
       "        end strand detail     beta  ...                      allele_1  \\\n",
       "0   3365548      -   exon  0.54646  ...                             C   \n",
       "6   2229460      -   exon  0.55837  ...                             T   \n",
       "8   2229460      -   exon  0.55837  ...                             C   \n",
       "24  2291981      -   exon  0.40160  ...  ATGGGAGTTTCGTAGACAATTCTTTTCG   \n",
       "\n",
       "   allele_2  reference_seq  reference_seq_  \\\n",
       "0         T    CTT |C| TGT         CTTCTGT   \n",
       "6         G    CGG |T| CTT         CGGTCTT   \n",
       "8         T    AGC |C| TTG         AGCCTTG   \n",
       "24        G    TTC |G| TGC         TTCGTGC   \n",
       "\n",
       "                                     wt_gene  \\\n",
       "0   [CTG, AGG, AAG, ACA, AAT, GTT, TCG, AGA]   \n",
       "6   [TCC, AAG, CCA, GAA, TTC, AAA, AAT, AGT]   \n",
       "8        [TCC, TCG, GAA, CAA, CTT, GAA, AAG]   \n",
       "24  [AAG, AAA, AGC, ACG, CTT, AAA, AAG, AAG]   \n",
       "\n",
       "                               mutant allele stop_codon_predicted  \\\n",
       "0   [CTG, AGG, AAA, ACA, AAT, GTT, TCG, AGA]                  NaN   \n",
       "6   [TCC, AAG, CCC, GAA, TTC, AAA, AAT, AGT]                  NaN   \n",
       "8        [TCC, TCG, AAA, CAA, CTT, GAA, AAG]                  NaN   \n",
       "24       [AAG, AAA, AGA, CGC, TTA, AAA, AGA]                  NaN   \n",
       "\n",
       "   possible frameshift wt_gene_translated mutant allele_translated  \n",
       "0                  NaN           LRKTNVSR                 LRKTNVSR  \n",
       "6                  NaN           SKPEFKNS                 SKPEFKNS  \n",
       "8                  NaN            SSEQLEK                  SSKQLEK  \n",
       "24                 NaN           KKSTLKKK                  KKRRLKR  \n",
       "\n",
       "[4 rows x 27 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#show for entries where ID doesnt start wiht SPNCRNA\n",
    "frameshift_df[frameshift_df['ID'].str.startswith('SPNCRNA') == False]"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.9.12 ('base')",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.0"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "2d0398177d7bb10ea33b6a5cd9d022d63e20680fcfd84762562c72ff35bdf44b"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
